// @domg-wc/components/block/search-filter/vl-search-filter.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, webComponent } from '@domg-wc/common';
import { html, nothing } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import { searchFilterStyles } from './vl-search-filter.css';
import { searchFilterDefaults } from './vl-search-filter.defaults';
import { searchFilterGlobalStyles } from './vl-search-filter.global.css';
let VlSearchFilterComponent = class VlSearchFilterComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.filterTitle = searchFilterDefaults.filterTitle;
        this.mobileModal = searchFilterDefaults.mobileModal;
        this.resizeObserver = null;
        this.handleSubmitModal = (event) => {
            event.preventDefault();
            if (this.mobileModal) {
                this.setAttribute('hidden', 'true');
            }
        };
    }
    static get styles() {
        return [searchFilterStyles];
    }
    static get properties() {
        return {
            filterTitle: { type: String, attribute: 'filter-title' },
            alt: { type: Boolean },
            mobileModal: { type: Boolean, attribute: 'mobile-modal', reflect: true },
            mobileModalTitle: { type: String, attribute: 'mobile-modal-title' },
        };
    }
    get form() {
        return this.querySelector('form');
    }
    get submitButton() {
        return this.form?.querySelector('vl-button[type="submit"]');
    }
    get buttonContainer() {
        return this.form?.querySelector('*:has(vl-button[type="submit"],button[type="submit"])');
    }
    get formData() {
        return this.form ? new FormData(this.form) : undefined;
    }
    async firstUpdated(changedProperties) {
        super.firstUpdated(changedProperties);
        this.form?.classList.add('vl-search-filter--form');
        this.form?.addEventListener('submit', this.handleSubmitModal);
        const buttonContainer = this.form?.querySelector('div:has(vl-button[type="submit"],button[type="submit"])');
        buttonContainer?.classList.add('vl-search-filter--submit');
        // Deze stylesheet moet toegevoegd worden aan de adoptedStyleSheets van het document
        // omdat deze styles betrekking hebben op de slotted content en dus niet op de shadow dom
        document.adoptedStyleSheets = [
            ...document.adoptedStyleSheets,
            searchFilterGlobalStyles.styleSheet,
        ];
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        if (changedProperties.has('alt')) {
            if (this.alt) {
                this.form?.classList.add('vl-search-filter--form__alt');
            }
            else {
                this.form?.classList.remove('vl-search-filter--form__alt');
            }
        }
        if (changedProperties.has('mobileModal')) {
            if (this.mobileModal) {
                this.buttonContainer?.classList.add('vl-search-filter--footer-modal');
                this.form?.classList.add('vl-search-filter--form__mobile-modal');
            }
            else {
                this.buttonContainer?.classList.remove('vl-search-filter--footer-modal');
                this.form?.classList.remove('vl-search-filter--form__mobile-modal');
            }
        }
    }
    render() {
        const classes = {
            'vl-search-filter': true,
            'vl-search-filter--mobile-modal': this.mobileModal,
        };
        return html `
            <div class=${classMap(classes)}>
                ${!this.mobileModal ? this.renderTitle() : this.renderMobileModal()}
                <div class="form-container">
                    <slot></slot>
                </div>
            </div>
        `;
    }
    connectedCallback() {
        super.connectedCallback();
        this.resizeObserver = new ResizeObserver((entries) => {
            for (const entry of entries) {
                this.mobileModal = entry.contentRect.width < 767;
            }
        });
        this.resizeObserver.observe(document.body);
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.form?.removeEventListener('submit', this.handleSubmitModal);
        if (this.resizeObserver) {
            this.resizeObserver.disconnect();
            this.resizeObserver = null;
        }
    }
    renderTitle() {
        return this.filterTitle ? html `<p class="vl-search-filter--intro">${this.filterTitle}</p>` : nothing;
    }
    renderMobileModal() {
        return html `
            <div class="vl-search-filter--header-modal">
                <h2>${this.mobileModalTitle || this.filterTitle || 'Filter'}</h2>
            </div>
        `;
    }
};
VlSearchFilterComponent = __decorate([
    webComponent('vl-search-filter')
], VlSearchFilterComponent);
export { VlSearchFilterComponent };
