// @domg-wc/map/components/action/layer-action/select-action/vl-map-select-action.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { webComponent } from '@domg-wc/common';
import { VlSelectAction } from '../../../../actions/select/select-action';
import { VlMapLayerStyle } from '../../../layer-style/vl-map-layer-style';
import { VlMapLayerAction } from '../vl-map-layer-action';
let VlMapSelectAction = class VlMapSelectAction extends VlMapLayerAction {
    // @ts-ignore: Negeer override van de property "style" van de native Element klasse die van een ander type is.
    get style() {
        return this._style;
    }
    // @ts-ignore: Negeer override van de property "style" van de native Element klasse die van een ander type is.
    set style(style) {
        if (style instanceof VlMapLayerStyle) {
            this._style = style.style;
        }
        else {
            this._style = style;
        }
        this._processAction();
    }
    get _cluster() {
        return this.getAttribute('cluster');
    }
    mark(id, layer) {
        if (this._action && id) {
            this._action.markFeatureWithId(id, layer || this.layer);
        }
    }
    removeMarks() {
        if (this._action) {
            this._action.demarkAllFeatures();
        }
    }
    select(feature) {
        if (this.action && feature) {
            this._action.selectFeature(feature);
        }
    }
    onSelect(callback) {
        this.__callback = callback;
    }
    reset() {
        if (this.action) {
            this._action.clearFeatures();
        }
    }
    /**
     * Specifies if the action is allowed to be performed on a feature and/or a layer. Returns true by default.
     *
     * @param {Object} feature Openlayers feature
     * @param {Object} layer Openlayers layer
     *
     * @Return {boolean} true if the action is allowed to be performed, false if the action may not be performed for the supplied feature and/or layer
     */
    appliesTo(feature, layer) {
        return true;
    }
    _createAction(layer) {
        const options = {
            style: this.style,
            cluster: this._cluster !== undefined,
            filter: this.appliesTo.bind(this),
        };
        return new VlSelectAction(layer, this._callback, options);
    }
};
VlMapSelectAction = __decorate([
    webComponent('vl-map-select-action')
], VlMapSelectAction);
export { VlMapSelectAction };
