// @domg-wc/components/form/input-field-masked/validators.js
export const maskValidator = {
    key: 'patternMismatch',
    message: 'Value does not meet the given mask pattern.',
    isValid(instance, value) {
        const { disableMaskValidation, cleaveInstance } = instance;
        const regex = instance.regex || instance.maskOptions?.regex;
        if (!value || disableMaskValidation || !regex || !cleaveInstance) {
            return true;
        }
        const rawValue = cleaveInstance.getRawValue();
        const regExp = new RegExp(regex);
        return regExp.test(rawValue);
    },
};
export const minValueValidator = {
    attribute: 'min',
    key: 'rangeUnderflow',
    message(instance) {
        if (instance.minExclusive) {
            return `Value must be greater than ${instance.min}.`;
        }
        return `Value must be greater than or equal to ${instance.min}.`;
    },
    isValid(instance, value) {
        const { min, minExclusive, cleaveInstance } = instance;
        if (!value || (!min && min !== 0) || !cleaveInstance) {
            return true;
        }
        const rawValue = cleaveInstance.getRawValue();
        if (parseFloat(rawValue) < min) {
            return false;
        }
        if (minExclusive && parseFloat(rawValue) === min) {
            return false;
        }
        return true;
    },
};
export const maxValueValidator = {
    attribute: 'max',
    key: 'rangeOverflow',
    message(instance) {
        if (instance.maxExclusive) {
            return `Value must be lesser than ${instance.max}.`;
        }
        return `Value must be lesser than or equal to ${instance.max}.`;
    },
    isValid(instance, value) {
        const { max, maxExclusive, cleaveInstance } = instance;
        if (!value || (!max && max !== 0) || !cleaveInstance) {
            return true;
        }
        const rawValue = cleaveInstance.getRawValue();
        if (parseFloat(rawValue) > max) {
            return false;
        }
        if (maxExclusive && parseFloat(rawValue) === max) {
            return false;
        }
        return true;
    },
};
