// @domg-wc/components/form/input-field-masked/vl-input-field-masked.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var VlInputFieldMaskedComponent_1;
import { webComponent } from '@domg-wc/common';
import Cleave from 'cleave.js';
import { VlInputFieldComponent } from '../input-field';
import { DEFAULT_MASKS } from './default-masks';
import { maskValidator, maxValueValidator, minValueValidator } from './validators';
import { inputFieldMaskedDefaults } from './vl-input-field-masked.defaults';
let VlInputFieldMaskedComponent = VlInputFieldMaskedComponent_1 = class VlInputFieldMaskedComponent extends VlInputFieldComponent {
    constructor() {
        super(...arguments);
        // Attributes
        this.mask = inputFieldMaskedDefaults.mask;
        this.maskPrefix = inputFieldMaskedDefaults.maskPrefix;
        this.rawValue = inputFieldMaskedDefaults.rawValue;
        this.disableMaskValidation = inputFieldMaskedDefaults.disableMaskValidation; // Wordt enkel gebruikt in de mask validator
        this.maskOptions = null;
        this.cleaveInstance = null;
    }
    static get properties() {
        return {
            mask: { type: String },
            maskPrefix: { type: String, attribute: 'mask-prefix' },
            rawValue: { type: Boolean, attribute: 'raw-value' },
            disableMaskValidation: { type: Boolean, attribute: 'disable-mask-validation' },
        };
    }
    static setMasks(masks) {
        VlInputFieldMaskedComponent_1.masks = { ...DEFAULT_MASKS, ...masks };
    }
    static getMask(key) {
        return VlInputFieldMaskedComponent_1.masks[key];
    }
    connectedCallback() {
        super.connectedCallback();
        this.maskOptions = {
            ...VlInputFieldMaskedComponent_1.getMask(this.mask),
            onValueChanged: (event) => {
                this.handleValueChanged(event?.target?.value);
            },
        };
        if (!this.maskOptions) {
            return;
        }
        else {
            if (this.maskOptions.numericOnly) {
                this.inputMode = 'numeric';
            }
            else {
                switch (this.mask) {
                    case 'date':
                        this.inputMode = 'decimal';
                        break;
                    default:
                        this.inputMode = 'text';
                }
            }
        }
        if (this.maskPrefix) {
            this.maskOptions.prefix = this.maskPrefix;
        }
        const value = this.value ?? '';
        if (!value.startsWith(this.maskOptions.prefix || '')) {
            this.value = this.maskOptions.prefix + value;
        }
    }
    firstUpdated(changedProperties) {
        super.firstUpdated(changedProperties);
        if (!this.maskOptions) {
            return;
        }
        this.cleaveInstance = new Cleave(this.validationTarget, this.maskOptions);
    }
    resetFormControl() {
        super.resetFormControl();
        const initialValue = this.initialValue ?? '';
        if (!initialValue.startsWith(this.maskOptions?.prefix || '')) {
            this.value = this.maskOptions?.prefix + initialValue;
        }
        else {
            this.value = this.initialValue;
        }
    }
    getRawValue() {
        return this.cleaveInstance?.getRawValue();
    }
    onInput() {
        // we definiëren hier een lege functie om de standaard onInput() van VlInputFieldComponent te overschrijven
        // we updaten de transformeerde value alreeds in een cleave.js callback (handleValueChanged()) daarom moet deze functie leeg zijn
        this.dispatchInput = true;
    }
    onUpdated(changedProperties) {
        if (changedProperties.has('value') || changedProperties.has('rawValue')) {
            let value = this.rawValue ? this.getRawValue() || '' : this.value;
            if (this.value === this.maskOptions?.prefix || this.getRawValue() === this.maskOptions?.prefix) {
                // Zet de ElementInternals value op een lege string indien enkel de prefix overschiet.
                // Dit zorgt ervoor dat de required validator kan inkicken als de gebruiker geen waarde heeft ingevuld.
                value = '';
            }
            const detail = { value };
            this.setValue(value);
            this.dispatchEvent(new CustomEvent('vl-change', { composed: true, bubbles: true, detail }));
            if (this.dispatchInput) {
                this.dispatchEvent(new CustomEvent('vl-input', { composed: true, bubbles: true, detail }));
                this.dispatchInput = false;
            }
            this.dispatchEventIfValid(detail);
        }
    }
    handleValueChanged(value) {
        const customTransformFn = this.maskOptions?.customTransformFn;
        const transformedValue = customTransformFn ? customTransformFn(value) : value;
        this.value = transformedValue;
    }
};
// Variables
VlInputFieldMaskedComponent.masks = { ...DEFAULT_MASKS };
VlInputFieldMaskedComponent.formControlValidators = [
    ...VlInputFieldComponent.formControlValidators,
    maskValidator,
    minValueValidator,
    maxValueValidator,
];
VlInputFieldMaskedComponent = VlInputFieldMaskedComponent_1 = __decorate([
    webComponent('vl-input-field-masked')
], VlInputFieldMaskedComponent);
export { VlInputFieldMaskedComponent };
