// @domg-wc/components/block/toaster/vl-toaster.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, registerWebComponents, webComponent } from '@domg-wc/common';
import { html } from 'lit';
import { VlAlert } from '../alert';
import { toasterStyles } from './vl-toaster.css';
import { toasterDefaults } from './vl-toaster.defaults';
registerWebComponents([VlAlert]);
let VlToasterComponent = class VlToasterComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        // properties
        this.fadeOut = toasterDefaults.fadeOut;
        this.placement = toasterDefaults.placement;
        // state
        this.abortController = new AbortController();
        this.initialized = false;
        this.showToast = async (toast) => {
            // wacht tot <output> element is gerenderd
            if (!this.initialized) {
                this.initialized = true;
                await this.updateComplete;
            }
            if (this.placement === 'bottom-right' || this.placement === 'bottom-left') {
                this.outputElement?.appendChild(toast);
            }
            else {
                this.outputElement?.prepend(toast);
            }
            if (this.fadeOut) {
                toast.addEventListener('animationend', (animationEvent) => {
                    if (animationEvent?.animationName === 'fade-out') {
                        toast.remove();
                    }
                }, { signal: this.abortController.signal });
            }
        };
    }
    static get styles() {
        return [toasterStyles];
    }
    static get properties() {
        return {
            fadeOut: { type: Boolean, attribute: 'fade-out' },
            placement: { type: String },
        };
    }
    render() {
        return html `
            <output class="vl-toaster">
                <slot></slot>
            </output>
        `;
    }
    disconnectedCallback() {
        this.abortController.abort('animationend');
    }
    /**
     * Wanneer aangeroepen zonder argument, wordt het eerste element in de toaster getoond.
     * Je kan ook een specifiek element tonen door een selector mee te geven of een HTMLElement.
     * @param target
     */
    show(target) {
        let toast;
        const firstChild = this.children[0]?.cloneNode(true);
        if (!target && firstChild) {
            toast = firstChild;
            // we verwijderen het id attribuut op de gekloonde node om dubbele id's te voorkomen
            toast?.removeAttribute('id');
            toast?.removeAttribute('slot');
        }
        else if (target && typeof target === 'string') {
            const foundElement = document.querySelector(target);
            if (foundElement instanceof HTMLTemplateElement) {
                const documentFragment = foundElement.content.cloneNode(true);
                toast = documentFragment?.firstElementChild;
            }
            else {
                toast = foundElement?.cloneNode(true);
            }
            // we verwijderen het id attribuut op de gekloonde node om dubbele id's te voorkomen
            toast?.removeAttribute('id');
            toast?.removeAttribute('slot');
        }
        else if (target instanceof HTMLElement) {
            toast = target;
        }
        if (!toast) {
            return;
        }
        this.showToast(toast);
    }
    /**
     * Toont een vl-alert in de toaster met de opgegeven attributen/properties.
     * @param alert
     */
    showAlert(alert) {
        const vlAlert = document.createElement('vl-alert');
        Object.entries(alert).forEach(([key, value]) => {
            if (value) {
                vlAlert.setAttribute(`${key}`, value);
            }
        });
        this.showToast(vlAlert);
    }
    get outputElement() {
        return this.shadowRoot?.querySelector('output');
    }
};
VlToasterComponent = __decorate([
    webComponent('vl-toaster')
], VlToasterComponent);
export { VlToasterComponent };
