// @domg-wc/map/components/controls/vl-map-action-controls.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { html } from 'lit';
import { customElement } from 'lit/decorators.js';
import { vlMapControlsFluxStyles } from './vl-map-controls.flux-css';
import { BaseLitElement } from '@domg-wc/common';
let VlMapActionControls = class VlMapActionControls extends BaseLitElement {
    static get styles() {
        return [vlMapControlsFluxStyles];
    }
    connectedCallback() {
        super.connectedCallback();
        this.actionControls.forEach((control) => {
            control.addEventListener('change-control', (event) => {
                const actionId = control.getAttribute('action-id');
                this.handleActionControlChange(actionId, event);
            });
        });
    }
    render() {
        return html `
            <div>
                <slot></slot>
            </div>
        `;
    }
    get actionControls() {
        return this.querySelectorAll('vl-map-action-control');
    }
    handleActionControlChange(actionId, event) {
        if (event.detail?.isActive) {
            [...this.actionControls]
                .filter((control) => control.getAttribute('action-id') !== actionId)
                .filter((control) => control.active)
                .forEach((control) => control.deactivate());
        }
    }
};
VlMapActionControls = __decorate([
    customElement('vl-map-action-controls')
], VlMapActionControls);
export { VlMapActionControls };
