// @domg/govflanders-style/component/doormat/vl-doormat.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

/**
 * Webcomponents design refresh
 * This is still in alpha phase, don't touch & don't use!
*/
.vl-doormat {
  display: flex;
  position: relative;
  padding: 1.5rem 2rem;
  background-color: #fff;
  text-decoration: none;
  border: 1px #cbd2da solid;
}
.vl-doormat::before, .vl-doormat--graphic .vl-doormat__graphic-wrapper::before {
  content: "";
  position: absolute;
  top: -1px;
  left: -1px;
  bottom: -1px;
  background-color: var(--vl-theme-primary-color);
  width: 0.5rem;
  height: 100%;
  margin-left: 1px;
  margin-top: 1px;
}
.vl-doormat--alt {
  background-color: #f7f9fc;
}
.vl-doormat--graphic {
  flex-direction: column;
  border-top: none;
}
.vl-doormat--graphic .vl-doormat__graphic-wrapper {
  display: block;
  position: relative;
  margin-top: -1.5rem;
  margin-right: -2.1rem;
  margin-bottom: 1.6rem;
  margin-left: -2.1rem;
  overflow: hidden;
}
.vl-doormat--graphic .vl-doormat__graphic-wrapper::before {
  background-color: rgba(51, 51, 50, 0.2);
  width: 0.6rem;
  z-index: 2;
}
.vl-doormat--graphic .vl-doormat__graphic-wrapper .vl-doormat__graphic {
  width: 100%;
  display: block;
  transition: transform 0.2s;
}
.vl-doormat--graphic:hover .vl-doormat__graphic, .vl-doormat--graphic:focus .vl-doormat__graphic {
  transform: scale(1.05);
}
.vl-doormat[href]::before {
  background-color: var(--vl-theme-primary-color);
  transition: background-color 0.2s;
}
.vl-doormat[href]:hover .vl-doormat__title, .vl-doormat[href]:focus .vl-doormat__title {
  text-decoration: underline;
}
.vl-doormat[href]:hover::before {
  background-color: var(--vl-theme-primary-color);
}
.vl-doormat[href]:focus::before {
  background-color: var(--vl-theme-primary-color);
}
.vl-doormat[href]:visited::before {
  background: var(--vl-theme-primary-color);
}
.vl-doormat__title {
  font-size: 2rem;
  font-family: "Flanders Art Sans", sans-serif;
  font-weight: 500;
  margin-bottom: 1.6rem;
  line-height: 1.4;
  margin-bottom: 0;
}
@media screen and (max-width: 767px) {
  .vl-doormat__title {
    font-size: 1.8rem;
    margin-bottom: 1.2rem;
  }
}
@media screen and (max-width: 767px) {
  .vl-doormat__title {
    margin-bottom: 0;
  }
}
.vl-doormat__text, .vl-doormat__content {
  font-size: 1.8rem;
  line-height: 1.5;
}
@media screen and (max-width: 767px) {
  .vl-doormat__text, .vl-doormat__content {
    font-size: 1.6rem;
    line-height: 1.33;
  }
}
.vl-doormat__text {
  color: #333332;
  margin-top: 1rem;
}
.vl-doormat__image {
  max-width: 15rem;
  float: right;
  flex-shrink: 0;
  margin-left: 1.5rem;
}
`;
export default style;
