// @domg-wc/common/styles/global-styles.js
import { vlLegacyStyles, vlResetStyles, vlGlobalStyles } from '@domg-wc/styles';
import { FluxConfig } from '../config/flux-config';
export class GlobalStyles {
    constructor() {
        this.defaultStyles = [];
        this.registered = false;
        this.customCss = [];
        this.registeredCustomCss = false;
    }
    static getInstance(defaultStyles = [vlResetStyles, ...vlLegacyStyles, ...vlGlobalStyles]) {
        if (this.instance) {
            return this.instance;
        }
        this.instance = new GlobalStyles();
        this.instance.defaultStyles = defaultStyles;
        return this.instance;
    }
    register() {
        if (!this.registered) {
            if (FluxConfig.getPreferences().autoRegisterStyles) {
                document.adoptedStyleSheets = [
                    ...document.adoptedStyleSheets,
                    ...this.defaultStyles.map((style) => style.styleSheet),
                ];
                console.info('GlobalStyles: global styling toegevoegd aan het document');
            }
            else {
                console.info('GlobalStyles: geen global styling toegevoegd aan het document - autoRegisterStyles is uitgeschakeld');
            }
            this.registered = true;
            this.registerCustomCSS();
        }
    }
    addCustomCss(customCSS) {
        this.customCss = [...this.customCss, ...customCSS];
        this.registeredCustomCss = false;
        this.registerCustomCSS();
    }
    registerCustomCSS() {
        if (!this.registered)
            return;
        if (this.customCss.length > 0 && !this.registeredCustomCss) {
            document.adoptedStyleSheets = [
                ...document.adoptedStyleSheets,
                ...this.customCss.map((style) => style.styleSheet),
            ];
            this.customCss = [];
            this.registeredCustomCss = true;
            console.info('GlobalStyles: custom global styling toegevoegd aan het document');
        }
    }
}
GlobalStyles.instance = null;
