// @domg-wc/components/compliance/footer/vl-footer.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { awaitScript, BaseLitElement, legacyBreakpoint, legacyCore, registerWebComponents, webComponentCustom, } from '@domg-wc/common';
registerWebComponents([legacyCore, legacyBreakpoint]);
const customRegistration = () => awaitScript('vl-footer-polyfill', 'https://prod.widgets.burgerprofiel.vlaanderen.be/api/v1/node_modules/@govflanders/vl-widget-polyfill/dist/index.js')
    .then(() => {
    awaitScript('vl-footer-client', 'https://prod.widgets.burgerprofiel.vlaanderen.be/api/v1/node_modules/@govflanders/vl-widget-client/dist/index.js').finally(() => {
        customElements.define('vl-footer', VlFooter);
    });
})
    .catch(() => {
    customElements.define('vl-footer', VlFooter);
});
let VlFooter = class VlFooter extends BaseLitElement {
    static get properties() {
        return {
            development: {
                type: Boolean,
                attribute: 'development',
                reflect: true,
            },
            identifier: {
                type: String,
                attribute: 'identifier',
                reflect: true,
            },
        };
    }
    constructor() {
        super();
        this.development = false;
        this.identifier = '';
        this.observer = null;
        this.allowCustomCSS = false;
    }
    get footerContainer() {
        return document.querySelector('#footer__container');
    }
    injectFooterContainer() {
        const vlBody = document.querySelector('body');
        (vlBody || document.body).insertAdjacentHTML('beforeend', '<div id="footer__container"><div id="footer"></div></div>');
    }
    observeWidgetIsAdded() {
        const isFooter = (node) => {
            return (node.tagName === 'FOOTER' || (node.childNodes && [...node.childNodes].some(isFooter)));
        };
        this.observer = new MutationObserver((mutations, observer) => {
            const nodes = mutations.flatMap((mutation) => [...mutation.addedNodes]);
            if (nodes.some(isFooter)) {
                observer.disconnect();
                this.dispatchEvent(new CustomEvent('ready'));
            }
        });
        if (this.footerContainer)
            this.observer.observe(this.footerContainer, { childList: true });
    }
    loadWidget() {
        const widgetUrl = this.development
            ? `https://tni.widgets.burgerprofiel.dev-vlaanderen.be/api/v1/widget/${this.identifier}`
            : `https://prod.widgets.burgerprofiel.vlaanderen.be/api/v1/widget/${this.identifier}`;
        const client = window.vl.widget.client;
        const bootstrapResponse = client.bootstrap(widgetUrl);
        bootstrapResponse
            .then((widget) => {
            widget.setMountElement(document.getElementById('footer'));
            widget.mount().catch((e) => console.error(e));
        })
            .catch((e) => {
            console.error(e);
        });
    }
    render() {
        this.footerContainer?.remove();
        this.injectFooterContainer();
        this.observer?.disconnect();
        this.observeWidgetIsAdded();
        this.loadWidget();
        // setTimeout(() => {
        //     this.loadWidget();
        // }, 2000);
    }
    createRenderRoot() {
        return this;
    }
    disconnectedCallback() {
        this.observer?.disconnect();
    }
    /**
     * @deprecated Deze methode mag niet gebruikt worden.
     */
    injectFooter() {
        console.warn('VlFooter - injectFooter() - deze methode is deprecated en mag niet gebruikt worden.');
        this.injectFooterContainer();
    }
    /**
     * @deprecated Deze methode mag niet gebruikt worden.
     */
    vlwFooter() {
        console.warn('VlFooter - vlwFooter() - deze methode is deprecated en mag niet gebruikt worden.');
        return document.querySelector('footer[class=vlw__footer]');
    }
    /**
     * @deprecated Deze methode mag niet gebruikt worden.
     */
    footer() {
        console.warn('VlFooter - footer() - deze methode is deprecated en mag niet gebruikt worden.');
        return this.footerContainer;
    }
};
VlFooter = __decorate([
    webComponentCustom(customRegistration),
    __metadata("design:paramtypes", [])
], VlFooter);
export { VlFooter };
