// @domg-wc/components/block/steps/vl-step.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, registerWebComponents } from '@domg-wc/common';
import { vlLegacyStyles } from '@domg-wc/styles';
import { resetStyle } from '@domg/govflanders-style/common';
import { stepsStyle } from '@domg/govflanders-style/component';
import { html } from 'lit';
import { customElement } from 'lit/decorators.js';
import { classMap } from 'lit/directives/class-map.js';
import { VlAccordionComponent } from '../accordion/vl-accordion.component';
import { vlStepFluxStyles } from './vl-step.flux-css';
let VlStepComponent = class VlStepComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        // Attributen
        this.type = null;
        this.toggleable = false;
        this.defaultOpen = false;
        // Private properties
        this.isTitleAnnotationSlotAssigned = true;
        this.customCSSStyleSheet = new CSSStyleSheet();
    }
    static get styles() {
        return [resetStyle, vlLegacyStyles, stepsStyle, vlStepFluxStyles];
    }
    static get properties() {
        return {
            type: { type: String, reflect: true },
            toggleable: { type: Boolean, reflect: true },
            defaultOpen: { type: Boolean, attribute: 'default-open', reflect: true },
            isTitleAnnotationSlotAssigned: { attribute: false },
        };
    }
    connectedCallback() {
        super.connectedCallback();
        if (this.shadowRoot) {
            this.shadowRoot.adoptedStyleSheets = [...this.shadowRoot.adoptedStyleSheets, this.customCSSStyleSheet];
        }
    }
    render() {
        const stepType = this.type;
        const stepHeaderTemplate = !this.toggleable
            ? this.getStepHeaderTemplate()
            : this.getAccordionStepHeaderTemplate();
        const classes = {
            'vl-step': true,
            [`vl-step--${stepType}`]: !!stepType,
            'vl-step--accordion js-vl-accordion': this.toggleable,
        };
        return html `
            <li role="listitem" class=${classMap(classes)}>
                <div class="vl-step__container">
                    <div class="vl-step__icon">
                        <slot name="icon"></slot>
                        <span class="vl-step__icon__sub">
                            <slot name="sub-icon"></slot>
                        </span>
                    </div>
                    <div class="vl-step__wrapper">
                        ${stepHeaderTemplate}
                        <div class="vl-step__content-wrapper">
                            <div class="vl-step__content">
                                <slot name="content"></slot>
                            </div>
                        </div>
                    </div>
                </div>
            </li>
        `;
    }
    setCustomStyles(customCSS) {
        this.customCSSStyleSheet.replaceSync(customCSS);
    }
    firstUpdated(changedProperties) {
        super.firstUpdated(changedProperties);
        if (this.toggleable) {
            const accordionToggle = this.shadowRoot?.querySelector('.js-vl-accordion__toggle');
            const isAccordionDressed = accordionToggle?.hasAttribute('accordion-dressed');
            if (!isAccordionDressed) {
                vl.accordion.dress(accordionToggle);
                this.shadowRoot?.querySelector('slot[name="title"]')?.addEventListener('click', (event) => {
                    event.stopPropagation();
                    this.shadowRoot?.querySelector('button.js-vl-accordion__toggle')?.click();
                });
            }
            if (this.defaultOpen) {
                if (!(this.shadowRoot?.querySelector('li.js-vl-accordion--open'))) {
                    accordionToggle?.click();
                }
            }
        }
        const titleAnnotationSlot = this.shadowRoot?.querySelector('slot[name="title-annotation"]');
        this.isTitleAnnotationSlotAssigned =
            (titleAnnotationSlot && titleAnnotationSlot.assignedNodes().length > 0) || false;
    }
    getStepHeaderTemplate() {
        const stepHeaderTitleTemplate = this.getStepHeaderTitleTemplate();
        return html ` <div class="vl-step__header">${stepHeaderTitleTemplate}</div>`;
    }
    getAccordionStepHeaderTemplate() {
        const stepHeaderTitleTemplate = this.getStepHeaderTitleTemplate();
        return html `
            <button class="vl-step__header js-vl-accordion__toggle">
                ${stepHeaderTitleTemplate}
                <div class="vl-step__header__info" aria-hidden="true">
                    <em class="vl-step__accordion-toggle"></em>
                </div>
            </button>
        `;
    }
    getStepHeaderTitleTemplate() {
        return html `
            <div class="vl-step__header__titles">
                <h3 class="vl-step__title">
                    <slot name="title"></slot>
                    ${this.isTitleAnnotationSlotAssigned
            ? html `
                              <span class="vl-step__title__annotation">
                                  <slot name="title-annotation"></slot>
                              </span>
                          `
            : ''}
                </h3>
                <p class="vl-step__subtitle">
                    <slot name="subtitle"></slot>
                </p>
            </div>
        `;
    }
};
(() => {
    registerWebComponents([VlAccordionComponent]);
})();
VlStepComponent = __decorate([
    customElement('vl-step')
], VlStepComponent);
export { VlStepComponent };
