// @domg-wc/components/block/input-slider/vl-input-slider.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement } from '@domg-wc/common';
import { resetStyle } from '@domg/govflanders-style/common';
import { html } from 'lit';
import { customElement } from 'lit/decorators.js';
import 'reflect-metadata';
import { inputFieldStyles } from '../../form/input-field';
import { vlInputSliderFluxStyles } from './vl-input-slider.flux-css';
let VlInputSliderComponent = class VlInputSliderComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.maxValue = 100;
        this.minValue = 0;
        this.value = 0;
    }
    static get styles() {
        return [resetStyle, inputFieldStyles, vlInputSliderFluxStyles];
    }
    static get properties() {
        return {
            maxValue: {
                type: Number,
                attribute: 'max-value',
                reflect: true,
            },
            minValue: {
                type: Number,
                attribute: 'min-value',
                reflect: true,
            },
            value: {
                type: Number,
                attribute: 'value',
                reflect: true,
            },
        };
    }
    updated(changedProperties) {
        if (changedProperties.has('value') || changedProperties.has('minValue') || changedProperties.has('maxValue')) {
            const validatedValue = this.validateValue(this.value);
            if (this.value !== validatedValue) {
                this.value = validatedValue;
                return;
            }
            if (changedProperties.has('value') && !isNaN(this.value)) {
                this.dispatchEvent(new CustomEvent('vl-change-value', {
                    detail: {
                        value: this.value,
                    },
                }));
            }
        }
    }
    render() {
        return html `
            <div class="vl-input-slider">
                <input
                    id="input-range"
                    type="range"
                    aria-label="schuifregelaar"
                    class="vl-input-slider__input-range"
                    min=${this.minValue}
                    max=${this.maxValue}
                    .value=${this.value}
                    @input=${this.handleInput}
                />
                <input
                    id="input-number"
                    type="number"
                    aria-label="nummerinvoer"
                    class="vl-input-slider__input-number vl-input-field"
                    min=${this.minValue}
                    max=${this.maxValue}
                    .value=${this.value}
                    @input=${this.handleInput}
                />
            </div>
        `;
    }
    handleInput(event) {
        const target = event.target;
        const newValue = parseInt(target?.value ?? '');
        this.value = newValue;
    }
    validateValue(value) {
        let validatedValue = value;
        if (validatedValue < this.minValue) {
            validatedValue = this.minValue;
        }
        if (validatedValue > this.maxValue) {
            validatedValue = this.maxValue;
        }
        return validatedValue;
    }
};
VlInputSliderComponent = __decorate([
    customElement('vl-input-slider')
], VlInputSliderComponent);
export { VlInputSliderComponent };
