// @domg-wc/components/block/progress-indicator/vl-progress-indicator.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { BaseLitElement, registerWebComponents } from '@domg-wc/common';
import { vlLegacyStyles } from '@domg-wc/styles';
import { accessibilityStyle, resetStyle } from '@domg/govflanders-style/common';
import { progressBarStyle } from '@domg/govflanders-style/component';
import { html, nothing } from 'lit';
import { customElement } from 'lit/decorators.js';
import { classMap } from 'lit/directives/class-map.js';
import 'reflect-metadata';
import { VlTooltipComponent } from '../tooltip';
import { vlProgressIndicatorFluxStyles } from './vl-progress-indicator.flux-css';
class ProgressIndicator {
    updateStep(shadowRoot, activeStep, focusOnChange) {
        if (shadowRoot) {
            const segments = shadowRoot.querySelectorAll('.vl-progress-indicator__segment');
            segments.forEach((segment, index) => {
                segment.classList.toggle('vl-progress-indicator__segment--active', index + 1 === activeStep);
                if (focusOnChange && index + 1 === activeStep) {
                    segment.querySelector('button.vl-progress-indicator__step')?.focus();
                }
            });
        }
    }
}
let VlProgressIndicatorComponent = class VlProgressIndicatorComponent extends BaseLitElement {
    constructor() {
        super();
        this.activeStep = 1;
        this.focusOnChange = false;
        this.numeric = false;
        this.progressIndicator = new ProgressIndicator();
        this.steps = [];
        this.showLabels = false;
        this.staticSteps = false;
        this.enableFutureSteps = false;
        this.updatePadding = async () => {
            this.style.setProperty('--vl-progress-indicator--padding-right', `${this.showLabels ? this.lastLabelWidth / 2 : 0}px`);
        };
        this.numeric = false;
        this.focusOnChange = false;
        this.activeStep = 1;
        this.progressIndicator = new ProgressIndicator();
        this.steps = [];
        this.showLabels = false;
        this.staticSteps = false;
        this.enableFutureSteps = false;
    }
    static get styles() {
        return [resetStyle, vlLegacyStyles, progressBarStyle, vlProgressIndicatorFluxStyles, accessibilityStyle];
    }
    static get properties() {
        return {
            activeStep: {
                type: Number,
                attribute: 'active-step',
                reflect: true,
            },
            focusOnChange: {
                type: Boolean,
                attribute: 'focus-on-change',
                reflect: true,
            },
            numeric: { type: Boolean, attribute: 'numeric', reflect: true },
            steps: { type: Array },
            showLabels: { type: Boolean, attribute: 'show-labels', reflect: true },
            staticSteps: { type: Boolean, attribute: 'static-steps' },
            enableFutureSteps: { type: Boolean, attribute: 'enable-future-steps' },
        };
    }
    async updated(changedProps) {
        this.progressIndicator.updateStep(this.shadowRoot, this.activeStep, this.focusOnChange);
        if (changedProps.has('showLabels') || changedProps.has('steps')) {
            this.updatePadding();
        }
    }
    connectedCallback() {
        super.connectedCallback();
        window.addEventListener('resize', this.updatePadding);
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        window.removeEventListener('resize', this.updatePadding);
    }
    render() {
        const progressIndicatorClasses = {
            'vl-progress-indicator': true,
            'vl-progress-indicator--numeric': this.numeric,
            'vl-progress-indicator--padded': this.showLabels,
        };
        return this.staticSteps
            ? html `
                  <div class=${classMap(progressIndicatorClasses)}>
                      ${this.steps.map((step, index) => this.renderSegment(step, index))}
                  </div>
              `
            : html ` <nav class=${classMap(progressIndicatorClasses)}>
                  ${this.steps.map((step, index) => this.renderSegment(step, index))}
              </nav>`;
    }
    get lastLabelWidth() {
        return (this.shadowRoot
            ?.querySelector('.vl-progress-indicator__segment:last-child .vl-progress-indicator__label')
            ?.getBoundingClientRect().width || 0);
    }
    renderBullet(step, index) {
        return html `<span class="vl-progress-indicator__bullet"></span>
            ${this.showLabels
            ? html `<span class="vl-progress-indicator__label" title=${step}>${step}</span>`
            : nothing} `;
    }
    renderStep(step, index) {
        return html `${this.staticSteps
            ? html `<div class="vl-progress-indicator__step" tabindex="0" id="step-${index + 1}">
                  ${this.renderBullet(step, index)}
              </div>`
            : html ` <button
                  ?disabled=${index > this.activeStep - 1 && !this.enableFutureSteps}
                  @click=${() => this.handleStepClick(step, index + 1)}
                  class="vl-progress-indicator__step"
                  type="button"
                  id="step-${index + 1}"
              >
                  ${this.renderBullet(step, index)}
              </button>`}
        ${this.showLabels
            ? nothing
            : html `
                  <vl-tooltip
                      id="tooltip-step-${index + 1}"
                      for="step-${index + 1}"
                      placement="top"
                      distance="15"
                      type="label"
                  >
                      ${step}
                  </vl-tooltip>
              `}`;
    }
    renderSegment(step, index) {
        const segmentClasses = {
            'vl-progress-indicator__segment': true,
            'vl-progress-indicator__segment--active': this.activeStep === index + 1,
        };
        return html `<div class=${classMap(segmentClasses)}>${this.renderStep(step, index)}</div>`;
    }
    handleStepClick(step, stepNumber) {
        this.dispatchEvent(new CustomEvent('vl-click-step', {
            bubbles: true,
            composed: true,
            detail: { step, number: stepNumber },
        }));
    }
};
(() => {
    registerWebComponents([VlTooltipComponent]);
})();
VlProgressIndicatorComponent = __decorate([
    customElement('vl-progress-indicator'),
    __metadata("design:paramtypes", [])
], VlProgressIndicatorComponent);
export { VlProgressIndicatorComponent };
