// @domg-wc/components/block/table/vl-table.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { registerWebComponents, webComponent } from '@domg-wc/common';
import { LitElement } from 'lit';
import { VlButtonComponent } from '../../atom/button';
import { tableStyles } from './vl-table.css';
registerWebComponents([VlButtonComponent]);
let VlTableComponent = class VlTableComponent extends LitElement {
    constructor() {
        super();
        this.table?.classList.add('vl-table');
    }
    static get properties() {
        return {
            hover: { type: Boolean, reflect: true },
            matrix: { type: Boolean, reflect: true },
            grid: { type: Boolean, reflect: true },
            zebra: { type: Boolean, reflect: true },
            fluxZebra: { type: Boolean, reflect: true, attribute: 'flux-zebra' },
            collapsedM: { type: Boolean, reflect: true, attribute: 'collapsed-m' },
            collapsedS: { type: Boolean, reflect: true, attribute: 'collapsed-s' },
            collapsedXs: { type: Boolean, reflect: true, attribute: 'collapsed-xs' },
        };
    }
    get table() {
        return this.querySelector('table');
    }
    get caption() {
        return this.querySelector('caption');
    }
    get headHeaderElements() {
        return Array.from(this.querySelectorAll('thead tr th'));
    }
    get bodyHeaderElements() {
        return Array.from(this.querySelectorAll('tbody tr th'));
    }
    get bodyRowElements() {
        return Array.from(this.querySelectorAll('tbody tr'));
    }
    connectedCallback() {
        super.connectedCallback();
        this.processScopeAttributes();
        this.processRowElements();
        // scrollable element moet focusbaar zijn voor a11y
        this.setAttribute('tabindex', '0');
        this.observer = this.observeHeaderElements(() => this.processScopeAttributes());
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.observer?.disconnect();
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        if (!this.table)
            return;
        // Map van properties naar class namen
        const classMap = {
            hover: 'vl-table--hover',
            matrix: 'vl-table--matrix',
            grid: 'vl-table--grid',
            zebra: 'vl-table--zebra',
            fluxZebra: 'vl-table--flux-zebra',
            collapsedM: 'vl-table--collapsed-m',
            collapsedS: 'vl-table--collapsed-s',
            collapsedXs: 'vl-table--collapsed-xs',
        };
        // voeg of verwijder klassen op basis van property waarden
        Object.entries(classMap).forEach(([property, className]) => {
            if (this[property]) {
                this.table.classList.add(className);
            }
            else {
                this.table.classList.remove(className);
            }
        });
        this.processEvenOdd();
    }
    collapseDetails(id) {
        this.showDetails(id, false);
    }
    expandDetails(id) {
        this.showDetails(id, true);
    }
    toggleDetails(id) {
        const details = this.detailsTableRowElements(id);
        const detailsVisible = details ? details[0].style.display !== 'none' : false;
        details?.forEach((detail) => (detail.style.display = detailsVisible ? 'none' : 'table-row'));
        this.showDetails(id, !detailsVisible);
    }
    createRenderRoot() {
        return this;
    }
    firstUpdated(changedProperties) {
        super.firstUpdated(changedProperties);
        // Gezien dit component geen shadow dom heeft, moeten de styles van de children toegevoegd worden
        // aan de adoptedStyleSheets van de omliggende shadow dom, anders zullen de styles niet toegepast worden
        const shadowRoot = this.shadowRoot || this.getRootNode();
        if (shadowRoot instanceof ShadowRoot) {
            shadowRoot.adoptedStyleSheets = [...shadowRoot.adoptedStyleSheets, tableStyles.styleSheet];
        }
        else {
            document.adoptedStyleSheets = [...document.adoptedStyleSheets, tableStyles.styleSheet];
        }
        this.caption?.classList.add('vl-table__caption');
    }
    detailsToggleButtonElement(id) {
        return this.querySelector(`tbody tr td vl-button[id="details-toggle-${id}"]`);
    }
    detailsTableRowElements(id) {
        return this.table?.querySelectorAll(`tbody tr[data-details-id="${id}"]`);
    }
    processScopeAttributes() {
        this.headHeaderElements
            .filter((header) => !header.hasAttribute('scope'))
            .forEach((header) => header.setAttribute('scope', 'col'));
        this.bodyHeaderElements
            .filter((header) => !header.hasAttribute('scope'))
            .forEach((header) => header.setAttribute('scope', 'row'));
    }
    expandCollapseTemplate(id) {
        const button = document.createElement('vl-button');
        button.id = `details-toggle-${id}`;
        button.setAttribute('type', 'button');
        button.setAttribute('narrow', '');
        button.setAttribute('secondary', '');
        button.setAttribute('icon', 'arrow-down-fat');
        button.addEventListener('vl-click', (e) => {
            e.preventDefault();
            this.toggleDetails(id);
        });
        return button;
    }
    showDetails(id, show) {
        const details = this.detailsTableRowElements(id);
        const vlButton = this.detailsToggleButtonElement(id);
        if (show) {
            details?.forEach((detail) => detail.style.removeProperty('display'));
            vlButton?.shadowRoot?.querySelector('button')?.setAttribute('aria-expanded', 'true');
            vlButton?.setAttribute('icon', 'arrow-up-fat');
        }
        else {
            details?.forEach((detail) => (detail.style.display = 'none'));
            vlButton?.shadowRoot?.querySelector('button')?.setAttribute('aria-expanded', 'false');
            vlButton?.setAttribute('icon', 'arrow-down-fat');
        }
    }
    processRowElements() {
        const rows = this.bodyRowElements;
        rows.forEach((row, i) => {
            if (!this.isDataRow(row)) {
                const id = row.getAttribute('data-details-id');
                row.style.display = 'none';
                const dataRow = rows[i - 1];
                if (dataRow.querySelectorAll('td[data-with-expand-details]').length === 0 && id && this.isDataRow(dataRow)) {
                    const cell = document.createElement('td');
                    const vlButton = this.expandCollapseTemplate(id);
                    cell.appendChild(vlButton);
                    dataRow.appendChild(cell);
                    vlButton.updateComplete.then(() => {
                        vlButton.shadowRoot?.querySelector('button')?.setAttribute('aria-expanded', 'false');
                        vlButton.shadowRoot
                            ?.querySelector('button')
                            ?.setAttribute('aria-label', 'toggle details for ' + id);
                    });
                }
                const detailsCellCount = row?.querySelectorAll('td')?.length;
                if (detailsCellCount === 1) {
                    const dataCellCount = dataRow.querySelectorAll('td').length;
                    const detailsCell = row.querySelector('td');
                    if (detailsCell)
                        detailsCell.colSpan = dataCellCount;
                }
            }
        });
    }
    processEvenOdd() {
        const rows = this.bodyRowElements;
        let dataRowIndex = 0;
        rows.forEach((row) => {
            if (this.isDataRow(row)) {
                dataRowIndex++;
            }
            row.classList.add(dataRowIndex % 2 === 0 ? 'even' : 'odd');
        });
    }
    observeHeaderElements(callback) {
        const observer = new MutationObserver(callback);
        observer.observe(this, { childList: true });
        return observer;
    }
    isDataRow(rowValue) {
        return !rowValue.hasAttribute('data-details-id');
    }
};
VlTableComponent = __decorate([
    webComponent('vl-table'),
    __metadata("design:paramtypes", [])
], VlTableComponent);
export { VlTableComponent };
