// @domg-wc/map/components/legend/vl-map-legend.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { BaseLitElement } from '@domg-wc/common';
import { css, html, unsafeCSS } from 'lit';
import { customElement } from 'lit/decorators.js';
import { VlMapLayerCircleStyle } from '../layer-style/vl-map-layer-circle-style/vl-map-layer-circle-style';
import { VlMapVectorLayer } from '../layer/vector-layer/vl-map-vector-layer';
import { VlMapLegendItem } from '../legend-item/vl-map-legend-item';
import { LEGEND_PLACEMENT } from './vl-map-legend.defaults';
import { vlMapLegendFluxStyles } from './vl-map-legend.flux-css';
let VlMapLegend = class VlMapLegend extends BaseLitElement {
    constructor() {
        super();
        this.items = [];
        this.styledItems = [];
        this.urlItems = [];
        this.customItems = [];
        this.placement = LEGEND_PLACEMENT.BOTTOM_RIGHT;
    }
    static get styles() {
        return [
            css `
                ${unsafeCSS(vlMapLegendFluxStyles)}
            `,
        ];
    }
    static get properties() {
        // TODO: Aligneer deze attributen (missen  prefix)
        return {
            left: { type: String, reflect: true },
            top: { type: String, reflect: true },
            right: { type: String, reflect: true },
            bottom: { type: String, reflect: true },
            layoutVertical: {
                type: Boolean,
                attribute: 'layout-vertical',
                reflect: true,
            },
            placement: {
                type: String,
                attribute: 'placement',
            },
            hideTitle: {
                type: Boolean,
                attribute: 'hide-title',
            },
            version: {
                type: String,
                attribute: 'version',
            },
        };
    }
    connectedCallback() {
        super.connectedCallback();
        this.mapElement = this.closest('vl-map');
        const imageLayers = [].concat(this.mapElement.wmsLayers);
        const geometryLayers = [].concat(this.mapElement.featuresLayers, this.mapElement.wfsLayers);
        imageLayers.forEach((wmsLayer) => {
            // eslint-disable-next-line @typescript-eslint/no-unused-expressions
            wmsLayer &&
                this.urlItems.push({
                    type: 'url',
                    url: this.legendUrl(wmsLayer),
                    name: wmsLayer.getAttribute('name'),
                });
        });
        this.customItems = this.customLegendItems();
        this.updateItems();
        geometryLayers.forEach((layer) => {
            layer.addEventListener(VlMapVectorLayer.EVENTS.styleChanged, () => {
                this.updateLegendGeometryItems(geometryLayers);
            });
        });
        this.initializeCustomLegendObserver();
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.observer.disconnect();
    }
    render() {
        const hasIconText = !!this.customItems?.find((item) => item.iconText);
        if (!this.items) {
            return null;
        }
        return html ` <div
            class=${`flux-map-legend ${this.layoutVertical ? 'flux-map-legend--vertical' : ''}`}
            style="${this.generateItemStyle()}"
        >
            ${this.hideTitle
            ? ''
            : html ` <div>
                      <span class="flux-map-legend-text flux-map-legend-title">Legende: </span>
                  </div>`}
            ${this.items.map((item) => {
            if (!item)
                return '';
            switch (item.type) {
                case 'custom':
                    return html ` ${item.styleElement} `;
                case 'styled':
                    return html ` <div class="flux-map-legend-item">
                            <div class="flux-map-legend-icon-container">
                                <div
                                    class="flux-map-legend-icon ${hasIconText ? 'flux-map-legend-icon-large' : ''}"
                                    style="${this.generateIconStyle(item.style)}"
                                >
                                    <div class="flux-map-legend-icon-text">${item.iconText}</div>
                                </div>
                            </div>
                            <span class="flux-map-legend-text">${item.name}</span>
                        </div>`;
                case 'url':
                    return html ` <div class="flux-map-legend-item flux-map-legend-image">
                            <img alt="map legend image" class="flux-map-legend-icon" src="${item.url}" />
                        </div>`;
                default:
                    return '';
            }
        })}
        </div>`;
    }
    getPosition() {
        const position = {};
        switch (this.placement) {
            case LEGEND_PLACEMENT.TOP_LEFT:
                position.top = '10px';
                position.left = '10px';
                position.right = undefined;
                position.bottom = undefined;
                break;
            case LEGEND_PLACEMENT.TOP_RIGHT:
                position.top = '10px';
                position.left = undefined;
                position.right = '10px';
                position.bottom = undefined;
                break;
            case LEGEND_PLACEMENT.BOTTOM_LEFT:
                position.top = undefined;
                position.left = '8px';
                position.right = undefined;
                position.bottom = '40px';
                break;
            case LEGEND_PLACEMENT.BOTTOM_RIGHT:
                position.top = undefined;
                position.left = undefined;
                position.right = '58px';
                position.bottom = '10px';
                break;
            default:
                break;
        }
        return {
            top: this.top ?? position.top,
            left: this.left ?? position.left,
            right: this.right ?? position.right,
            bottom: this.bottom ?? position.bottom,
        };
    }
    initializeCustomLegendObserver() {
        this.observer = new MutationObserver(() => {
            this.customItems = this.customLegendItems();
            this.updateItems();
        });
        const config = {
            childList: true,
            subtree: true,
            attributes: true,
        };
        this.observer.observe(this, config);
    }
    legendUrl(wmsLayer) {
        const layerUrl = new URL(wmsLayer.getAttribute('url'));
        const layout = this.layoutVertical ? 'layout:vertical' : 'layout:horizontal';
        const legendSearchParams = new URLSearchParams({
            service: 'WMS',
            request: 'GetLegendGraphic',
            format: 'image/png',
            layer: wmsLayer.getAttribute('layers'),
            legend_options: layout,
        });
        if (this.version)
            legendSearchParams.set('version', this.version);
        return new URL(`?${legendSearchParams}`, layerUrl);
    }
    customLegendItems() {
        const childNodeList = Array.from(this.childNodes);
        const legendItemsList = childNodeList.filter((child) => child instanceof VlMapLegendItem);
        const legendItems = legendItemsList.map((item) => {
            return {
                type: 'custom',
                name: item.layer,
                iconText: item.iconText,
                styleElement: item.cloneNode(true),
            };
        });
        return legendItems;
    }
    updateLegendGeometryItems(layers) {
        let items = [];
        layers.forEach((layer) => {
            if (layer._styles.length === 1) {
                const style = layer._styles[0];
                if (style.name || (layer.name !== undefined && layer.name != null)) {
                    items.push({ type: 'styled', style: style, name: layer.name });
                }
            }
            else {
                items = items.concat(...(layer._styles
                    ?.filter((style) => style.name)
                    ?.map((style) => ({ type: 'styled', style: style, name: style.name })) || []));
            }
        });
        this.styledItems = items;
        this.updateItems();
    }
    updateItems() {
        const defaultItems = [...this.styledItems, ...this.urlItems];
        const hasCustomLegendItems = this.customItems.length > 0;
        if (hasCustomLegendItems) {
            // build a custom map legend from the vl-map-legend-item elements in the vl-map-legend
            this.items = [];
            this.customItems.forEach((customItem) => {
                if (customItem.styleElement.children.length > 0) {
                    this.items.push(customItem);
                }
                else {
                    const layer = customItem.name;
                    const defaultItemForLayer = defaultItems.find((item) => item.name === layer);
                    if (defaultItemForLayer && defaultItemForLayer.type === 'styled') {
                        defaultItemForLayer.iconText = customItem.iconText;
                    }
                    this.items.push(defaultItemForLayer);
                }
            });
        }
        else {
            // use the default generated legend items
            this.items = defaultItems;
        }
        this.requestUpdate();
    }
    generateItemStyle() {
        const position = this.getPosition();
        return ((position.left ? `;left:${position.left}` : '') +
            (position.top ? `;top:${position.top}` : '') +
            (position.right ? `;right:${position.right}` : '') +
            (position.bottom ? `;bottom:${position.bottom}` : ''));
    }
    generateIconStyle(item) {
        let borderRadius = ``;
        if (item instanceof VlMapLayerCircleStyle) {
            borderRadius = 'border-radius: 50%;';
        }
        return `border: ${item.borderSize}px solid ${item.borderColor}; color:${item.textColor}; background-color:${item.color}; ${borderRadius}`;
    }
};
VlMapLegend = __decorate([
    customElement('vl-map-legend'),
    __metadata("design:paramtypes", [])
], VlMapLegend);
export { VlMapLegend };
