// @domg-wc/components/compliance/header/vl-header.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { awaitScript, BaseLitElement, legacyBreakpoint, legacyCore, registerWebComponents, webComponentCustom, } from '@domg-wc/common';
import { headerContainerStyles, headerSkeletonStyles } from './vl-header.component.flux-css';
import { headerDefaults } from './vl-header.defaults';
registerWebComponents([legacyCore, legacyBreakpoint]);
const customRegistration = () => awaitScript('vl-header-polyfill', 'https://prod.widgets.burgerprofiel.vlaanderen.be/api/v1/node_modules/@govflanders/vl-widget-polyfill/dist/index.js')
    .then(() => {
    awaitScript('vl-header-client', 'https://prod.widgets.burgerprofiel.vlaanderen.be/api/v1/node_modules/@govflanders/vl-widget-client/dist/index.js').finally(() => {
        customElements.define('vl-header', VlHeader);
    });
})
    .catch(() => {
    customElements.define('vl-header', VlHeader);
});
let VlHeader = class VlHeader extends BaseLitElement {
    constructor() {
        super();
        // Properties
        this.logoutCallback = headerDefaults.logoutCallback;
        this.applicationLinks = headerDefaults.applicationLinks;
        // Attributes
        this.authenticatedUserUrl = headerDefaults.authenticatedUserUrl;
        this.development = headerDefaults.development;
        this.identifier = headerDefaults.identifier;
        this.loginRedirectUrl = headerDefaults.loginRedirectUrl;
        this.loginUrl = headerDefaults.loginUrl;
        this.logoutUrl = headerDefaults.logoutUrl;
        this.switchCapacityUrl = headerDefaults.switchCapacityUrl;
        this.simple = headerDefaults.simple;
        this.skeleton = headerDefaults.skeleton;
        this.rejectLogout = headerDefaults.rejectLogout;
        // Variables
        this.observer = null;
        this.session = null;
        this.authenticated = false;
        this.allowCustomCSS = false;
    }
    static get properties() {
        return {
            authenticatedUserUrl: { type: String, attribute: 'authenticated-user-url' },
            development: { type: Boolean, attribute: 'development' },
            identifier: { type: String, attribute: 'identifier' },
            loginRedirectUrl: { type: String, attribute: 'login-redirect-url' },
            loginUrl: { type: String, attribute: 'login-url' },
            logoutUrl: { type: String, attribute: 'logout-url' },
            simple: { type: Boolean, attribute: 'simple' },
            skeleton: { type: Boolean, attribute: 'skeleton' },
            switchCapacityUrl: { type: String, attribute: 'switch-capacity-url' },
            rejectLogout: { type: Boolean, attribute: 'reject-logout' },
            logoutCallback: { type: Function },
            applicationLinks: { type: Array },
        };
    }
    get headerContainer() {
        return document.querySelector('#header__container');
    }
    get headerContainerSkeleton() {
        return document.querySelector('#header__container__skeleton');
    }
    connectedCallback() {
        super.connectedCallback();
        this.injectHeaderContainer();
        if (this.skeleton) {
            this.injectHeaderContainerSkeleton();
        }
        this.observeWidgetIsAdded();
        this.loadWidget();
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.headerContainer?.remove();
        if (this.skeleton) {
            this.headerContainerSkeleton?.remove();
        }
        this.observer?.disconnect();
    }
    /**
     * @deprecated Deze methode mag niet gebruikt worden.
     */
    injectHeader() {
        console.warn('VlHeader - injectHeader() - deze methode is deprecated en mag niet gebruikt worden.');
        this.injectHeaderContainer();
    }
    /**
     * @deprecated Deze methode mag niet gebruikt worden.
     */
    vlwHeader() {
        console.warn('VlHeader - vlwHeader() - deze methode is deprecated en mag niet gebruikt worden.');
        return document.querySelector('div[class=vlw__header]');
    }
    /**
     * @deprecated Deze methode nag niet gebruikt worden.
     */
    header() {
        console.warn('VlHeader - header() - deze methode is deprecated en mag niet gebruikt worden.');
        return this.headerContainer;
    }
    willUpdate(changedProperties) {
        const sessionProperties = ['loginUrl', 'loginRedirectUrl', 'logoutUrl', 'switchCapacityUrl'];
        if (sessionProperties.some((property) => changedProperties.has(property))) {
            this.configureSession();
        }
    }
    createRenderRoot() {
        return this;
    }
    injectHeaderContainer() {
        (document.querySelector('body') || document.body).insertAdjacentHTML('afterbegin', '<div id="header__container"><div id="header"></div></div>');
        document.adoptedStyleSheets = [...document.adoptedStyleSheets, headerContainerStyles.styleSheet];
    }
    injectHeaderContainerSkeleton() {
        this.headerContainer?.insertAdjacentHTML('afterend', '<div id="header__skeleton"></div>');
        document.adoptedStyleSheets = [...document.adoptedStyleSheets, headerSkeletonStyles.styleSheet];
    }
    observeWidgetIsAdded() {
        const isHeader = (node) => {
            return (node.tagName === 'HEADER' || (node.childNodes && [...node.childNodes].some(isHeader)));
        };
        this.observer = new MutationObserver((mutations, observer) => {
            const nodes = mutations.flatMap((mutation) => [...mutation.addedNodes]);
            if (nodes.some(isHeader)) {
                observer.disconnect();
                this.dispatchEvent(new CustomEvent('ready'));
            }
        });
        if (this.headerContainer)
            this.observer.observe(this.headerContainer, { childList: true });
    }
    async isUserAuthenticated() {
        const response = await fetch(this.authenticatedUserUrl);
        return response.status === 200;
    }
    loadWidget() {
        const widgetUrl = this.development
            ? `https://tni.widgets.burgerprofiel.dev-vlaanderen.be/api/v1/widget/${this.identifier}`
            : `https://prod.widgets.burgerprofiel.vlaanderen.be/api/v1/widget/${this.identifier}`;
        window.vl.widget.client
            .bootstrap(widgetUrl)
            .then((widget) => {
            widget.setMountElement(document.getElementById('header'));
            widget.mount();
            return widget;
        })
            .then((widget) => {
            if (this.applicationLinks.length > 0) {
                const links = this.applicationLinks.map((link) => {
                    return {
                        type: 'link',
                        ...link,
                    };
                });
                widget.getExtension('citizen_profile').then((extension) => {
                    extension.getMenu().getGroup('application').addMultiple(links);
                });
            }
            if (this.simple) {
                return;
            }
            widget.getExtension('citizen_profile.session').then(async (session) => {
                this.session = session;
                this.authenticated = await this.isUserAuthenticated();
                this.configureSession();
            });
            widget.on('citizen_profile.session.logout.request', async (logoutRequest) => {
                // Acknowledge het logout request om te voorkomen dat de sessie extensie de default actie uitvoert door response timeout (5 seconden).
                logoutRequest.acknowledge();
                const logoutReason = logoutRequest.getRequest().getReason();
                if (logoutReason === 'user') {
                    //  Logout request door de gebruiker. Dit request mag nooit afgewezen worden in normale omstandigheden.
                    logoutRequest.accept();
                    return;
                }
                if (this.rejectLogout) {
                    // Wijs het logout request af.
                    logoutRequest.reject();
                    return;
                }
                if (this.logoutCallback && !(await this.logoutCallback(logoutReason))) {
                    // Wijs het logout request af als de logoutCallback een Promise<boolean> teruggeeft die false is.
                    logoutRequest.reject();
                    return;
                }
                // Accepteer het logout request in alle andere gevallen.
                logoutRequest.accept();
            });
        })
            .catch((e) => {
            console.error(e);
        });
    }
    async configureSession() {
        const config = {
            active: this.authenticated,
            endpoints: {
                loginUrl: this.loginUrl,
                loginRedirectUrl: this.loginRedirectUrl,
                logoutUrl: this.logoutUrl,
                switchCapacityUrl: this.switchCapacityUrl,
            },
        };
        this.session?.configure(config);
    }
    get height() {
        return this.headerContainer?.getBoundingClientRect().height || 0;
    }
};
VlHeader = __decorate([
    webComponentCustom(customRegistration),
    __metadata("design:paramtypes", [])
], VlHeader);
export { VlHeader };
