// @domg-wc/components/form/select/vl-select.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { webComponent } from '@domg-wc/common';
import { baseStyle, resetStyle } from '@domg/govflanders-style/common';
import { iconStyle } from '@domg/govflanders-style/component';
import { html, nothing } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import { live } from 'lit/directives/live.js';
import { FormControl } from '../form-control';
import selectStyle from './styles/vl-select.dv-css';
import { vlSelectFluxStyles } from './styles/vl-select.flux-css';
import { selectDefaults } from './vl-select.defaults';
let VlSelectComponent = class VlSelectComponent extends FormControl {
    constructor() {
        super(...arguments);
        // Properties
        this.options = selectDefaults.options;
        this.initialOptions = selectDefaults.initialOptions;
        // State
        this.value = '';
        // Attributes
        this.block = selectDefaults.block;
        this.placeholder = selectDefaults.placeholder;
        this.autocomplete = selectDefaults.autocomplete;
        this.notDeletable = selectDefaults.notDeletable;
        // Variables
        this.DEFAULT_GROUP_LABEL = 'Overig';
        this.dispatchInput = false;
        this.parsedOptions = [];
    }
    static get styles() {
        return [resetStyle, baseStyle, selectStyle, iconStyle, vlSelectFluxStyles];
    }
    static get properties() {
        return {
            options: {
                type: Array,
                attribute: false,
            },
            initialOptions: { type: Array, attribute: 'initial-options' },
            block: { type: Boolean },
            readonly: { type: Boolean },
            placeholder: { type: String },
            autocomplete: { type: String },
            notDeletable: { type: Boolean, attribute: 'not-deletable' },
            value: { type: String },
        };
    }
    get validationTarget() {
        return this.shadowRoot?.querySelector('select');
    }
    connectedCallback() {
        super.connectedCallback();
        this.parseSlottedOptions();
        const selectedOption = this.getSelectedOption();
        this.value = selectedOption?.value || '';
        this.initialOptions = structuredClone(this.getAllOptions());
        this.setupSlotObserver();
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.slotObserver?.disconnect();
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        if (changedProperties.has('options')) {
            const selectedOption = this.getSelectedOption();
            this.value = selectedOption?.value || '';
        }
        if (changedProperties.has('value')) {
            const detail = { value: this.value };
            this.setValue(this.value);
            this.dispatchEvent(new CustomEvent('vl-change', { composed: true, bubbles: true, detail }));
            if (this.dispatchInput) {
                this.dispatchEvent(new CustomEvent('vl-input', { bubbles: true, composed: true, detail }));
                this.dispatchInput = false;
            }
            this.dispatchEventIfValid(detail);
        }
    }
    render() {
        const containerClasses = {
            'vl-select__container': true,
            'vl-select__container--block': this.block,
        };
        const selectClasses = {
            'vl-select': true,
            'vl-select--disabled': this.disabled,
            'vl-select--error': this.isInvalid || this.error,
            'vl-select--success': this.success,
            'vl-select--block': this.block,
        };
        const hasValue = !!this.value;
        const hasGroups = this.getAllOptions().some((option) => option.group);
        return html `
            <div class=${classMap(containerClasses)}>
                <select
                    id=${this.id || nothing}
                    name=${this.name || nothing}
                    class=${classMap(selectClasses)}
                    aria-label=${this.label || nothing}
                    aria-invalid=${this.isInvalid || nothing}
                    ?required=${this.required}
                    ?disabled=${this.disabled}
                    ?aria-disabled=${this.disabled}
                    ?error=${this.error}
                    .value=${live(this.value)}
                    autocomplete=${this.autocomplete || nothing}
                    @change=${this.onChange}
                    @input=${this.onSelect}
                >
                    ${this.placeholder ? this.renderPlaceholder(hasValue) : nothing}
                    ${hasGroups ? this.renderGroupedOptions() : this.renderSelectOptions(this.getAllOptions())}
                </select>
                ${hasValue && !this.notDeletable ? this.renderClearButton() : nothing}
                <span class="vl-icon vl-vi vl-vi-nav-down" aria-hidden="true"></span>
            </div>
            <div class="slot-container">
                <slot @slotchange=${this.onSlotChange}></slot>
            </div>
        `;
    }
    renderPlaceholder(hasValue) {
        return html ` <option class="vl-select__placeholder" value="" ?selected=${!hasValue} disabled>
            ${this.placeholder}
        </option>`;
    }
    renderClearButton() {
        return html `
            <button
                type="button"
                class="vl-select__button"
                aria-label=${`Verwijder ${this.label} keuze ${this.getSelectedOption()?.label || this.value || ''}`}
                @click=${this.clearValue}
            >
                <span class="vl-icon vl-vi vl-vi-close" aria-hidden="true"></span>
            </button>
        `;
    }
    renderGroupedOptions() {
        const groupedOptions = this.getGroupedOptions();
        return Object.entries(groupedOptions).map(([group, options]) => {
            return html ` <optgroup label=${group}>${this.renderSelectOptions(options)}</optgroup>`;
        });
    }
    renderSelectOptions(options) {
        return options.map((option) => {
            return html ` <option
                value=${option.value}
                ?selected=${this.value === option.value}
                ?disabled=${option.disabled}
            >
                ${option.label || option.value}
            </option>`;
        });
    }
    resetFormControl() {
        super.resetFormControl();
        if (this.options.length > 0) {
            this.options = structuredClone(this.initialOptions);
        }
        else {
            this.parseSlottedOptions();
        }
        const selectedOption = this.getSelectedOption();
        this.value = selectedOption?.value || '';
    }
    onChange(event) {
        this.value = event?.target?.value;
    }
    onSelect() {
        this.dispatchInput = true;
    }
    clearValue() {
        this.dispatchInput = true;
        this.value = '';
    }
    onSlotChange() {
        this.parseSlottedOptions();
        this.requestUpdate();
    }
    getSelectedOption() {
        // Zoek de laatste optie met selected true, dit is dezelfde werking als een native select element.
        return [...this.getAllOptions()].reverse().find((option) => option.selected);
    }
    getGroupedOptions() {
        const groupedOptions = this.getAllOptions().reduce((groups, option) => {
            const group = option.group || this.DEFAULT_GROUP_LABEL;
            if (!groups[group]) {
                groups[group] = [option];
            }
            else {
                groups[group].push(option);
            }
            return groups;
        }, {});
        return groupedOptions;
    }
    setupSlotObserver() {
        this.slotObserver = new MutationObserver(() => {
            this.parseSlottedOptions();
            this.requestUpdate();
        });
        this.slotObserver.observe(this, {
            childList: true,
            subtree: true,
            attributes: true,
            attributeFilter: ['value', 'label', 'selected', 'disabled'],
        });
    }
    parseSlottedOptions() {
        const slottedElements = Array.from(this.children);
        this.parsedOptions = slottedElements.reduce((options, element) => {
            if (element.tagName.toLowerCase() === 'option') {
                options.push(this.parseOptionElement(element));
            }
            else if (element.tagName.toLowerCase() === 'optgroup') {
                const groupLabel = element.label || this.DEFAULT_GROUP_LABEL;
                const optionElements = Array.from(element.querySelectorAll('option'));
                const groupOptions = optionElements.map((optionElement) => {
                    const option = this.parseOptionElement(optionElement);
                    option.group = groupLabel;
                    return option;
                });
                options.push(...groupOptions);
            }
            return options;
        }, []);
    }
    parseOptionElement(optionElement) {
        return {
            value: optionElement.value || optionElement.textContent?.trim() || '',
            label: optionElement.textContent?.trim() || optionElement.value || '',
            selected: optionElement.selected,
            disabled: optionElement.disabled,
        };
    }
    getAllOptions() {
        // Als we programmatische options hebben, gebruik die; anders gebruik parsed options van slot
        return this.options.length > 0 ? this.options : this.parsedOptions;
    }
};
VlSelectComponent = __decorate([
    webComponent('vl-select')
], VlSelectComponent);
export { VlSelectComponent };
