// @domg-wc/map/components/layer/wmts-layer/vl-map-wmts-layer.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { webComponent } from '@domg-wc/common';
import * as OlExtent from 'ol/extent';
import OlTileLayer from 'ol/layer/Tile';
import OlWMTSSource from 'ol/source/WMTS';
import OlWMTSTileGrid from 'ol/tilegrid/WMTS';
import { getLambert2008Code, getLambert2008Extent, getLambert72Code } from '../../../utils/capabilities';
import { VlMap } from '../../../vl-map';
import { VlMapLayer } from '../vl-map-layer';
let VlMapWmtsLayer = class VlMapWmtsLayer extends VlMapLayer {
    connectedCallback() {
        this._source = this.__createSource();
        this._layer = this._createLayer();
        return super.connectedCallback();
    }
    get _projection() {
        if (this.parentNode && this.parentNode instanceof VlMap) {
            return this.parentNode._projection;
        }
    }
    get url() {
        const url = this.getAttribute('url');
        if (!url) {
            throw new Error('URL not defined');
        }
        return url;
    }
    get _wmtsLayer() {
        const layer = this.getAttribute('layer');
        if (!layer) {
            throw new Error('Layer not defined');
        }
        return layer;
    }
    _createLayer() {
        const layer = new OlTileLayer({
            title: this._name,
            source: this._source,
            minResolution: this._minResolution,
            maxResolution: this._maxResolution,
            visible: this._visible,
            opacity: this._opacity,
        });
        layer.set('id', VlMapLayer._counter);
        return layer;
    }
    __createSource() {
        const tileLimits = this.__grbTileLimits;
        return new OlWMTSSource({
            url: this.url,
            layer: this._wmtsLayer,
            matrixSet: this.__grbMatrixSet,
            format: this.__wmtsFormat,
            projection: this._projection,
            tileGrid: new OlWMTSTileGrid({
                extent: this.__extent,
                origin: OlExtent.getTopLeft(this.__extent),
                resolutions: tileLimits.resolutions,
                matrixIds: tileLimits.matrixIds,
            }),
            style: '',
        });
    }
    get __wmtsFormat() {
        return 'image/png';
    }
    get __extent() {
        return this._projection?.getExtent();
    }
    get __grbMatrixSet() {
        if (this.hasAttribute('matrix-set')) {
            return this.getAttribute('matrix-set');
        }
        switch (this._projection?.getCode()) {
            case getLambert2008Code():
                return 'BPL2008VL';
            case getLambert72Code():
                return 'BPL72VL';
            default:
                return 'WGS84VL';
        }
    }
    get __prefixMatrix() {
        return this.hasAttribute('matrix-prefix');
    }
    get __grbTileLimits() {
        const size = OlExtent.getWidth(getLambert2008Extent()) / 256;
        const resolutions = new Array(16);
        const matrixIds = new Array(16);
        for (let z = 0; z < 16; ++z) {
            resolutions[z] = size / Math.pow(2, z);
            matrixIds[z] = this.__prefixMatrix ? this.__grbMatrixSet + ':' + z : z;
        }
        return { matrixIds, resolutions };
    }
};
VlMapWmtsLayer = __decorate([
    webComponent('vl-map-wmts-layer')
], VlMapWmtsLayer);
export { VlMapWmtsLayer };
