// @domg/govflanders-style/component/pager/vl-pager.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-pager {
  font-size: 1.8rem;
}
@media screen and (max-width: 767px) {
  .vl-pager {
    font-size: 1.6rem;
  }
}
.vl-pager__list {
  text-align: left;
}
.vl-pager--align-center .vl-pager__list {
  text-align: center;
}
.vl-pager--align-right .vl-pager__list {
  text-align: right;
}
@media screen and (min-width: 1023px) {
  .vl-pager--align-left--l .vl-pager__list {
    text-align: left;
  }
  .vl-pager--align-center--l .vl-pager__list {
    text-align: center;
  }
  .vl-pager--align-right--l .vl-pager__list {
    text-align: right;
  }
}
@media screen and (max-width: 1023px) {
  .vl-pager--align-left--m .vl-pager__list {
    text-align: left;
  }
  .vl-pager--align-center--m .vl-pager__list {
    text-align: center;
  }
  .vl-pager--align-right--m .vl-pager__list {
    text-align: right;
  }
}
@media screen and (max-width: 767px) {
  .vl-pager--align-left--s .vl-pager__list {
    text-align: left;
  }
  .vl-pager--align-center--s .vl-pager__list {
    text-align: center;
  }
  .vl-pager--align-right--s .vl-pager__list {
    text-align: right;
  }
}
@media screen and (max-width: 500px) {
  .vl-pager--align-left--xs .vl-pager__list {
    text-align: left;
  }
  .vl-pager--align-center--xs .vl-pager__list {
    text-align: center;
  }
  .vl-pager--align-right--xs .vl-pager__list {
    text-align: right;
  }
}
.vl-pager__element {
  display: inline-block;
  vertical-align: middle;
  padding: 0 1.5rem;
  position: relative;
}
.vl-pager__element:first-child {
  padding: 0;
  padding-right: 1.5rem;
}
.vl-pager__element:nth-child(2) {
  padding-left: 0.2rem;
}
.vl-pager__element:last-child {
  padding-right: 0;
}
.vl-pager__element:first-child::after, .vl-pager__element:last-child::after {
  display: none;
}
.vl-pager__element--active {
  font-weight: 700;
}
.vl-pager__element--active .vl-pager__element__cta {
  color: #333332;
  text-decoration: none;
}
.vl-pager__element--previous {
  margin-right: 1rem;
}
.vl-pager__element--next {
  margin-left: 1rem;
}
.vl-pager__element__cta,
.vl-pager__element a {
  box-shadow: none;
}
.vl-pager__element__cta:hover, .vl-pager__element__cta:focus,
.vl-pager__element a:hover,
.vl-pager__element a:focus {
  box-shadow: none;
}
.vl-pager__element__cta .vl-link__icon,
.vl-pager__element a .vl-link__icon {
  color: inherit;
  display: inline-block;
  margin: 0;
  align-self: inherit;
}
.vl-pager__element__cta .vl-link__icon.vl-vi-external,
.vl-pager__element a .vl-link__icon.vl-vi-external {
  transform: translateY(1px);
}
`;
export default style;
