// @domg/govflanders-style/component/alert/vl-alert.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-alert {
  display: flex;
  position: relative;
  background-color: #e8ebee;
  border: 0.1rem solid #cfd5dd;
  padding: 2.5rem 3rem;
  border-radius: 0.3rem;
}
@media screen and (max-width: 767px) {
  .vl-alert {
    padding: 1.5rem;
  }
}
.vl-alert__close {
  display: flex;
  position: absolute;
  border: 0;
  cursor: pointer;
  top: 0.9rem;
  right: 0.9rem;
  padding: 0.5rem;
  font-size: 1.6rem;
  overflow: hidden;
}
.vl-alert__close:hover, .vl-alert__close:focus {
  color: #003bb0;
}
.vl-alert__close > .vl-vi {
  display: inline-flex;
}
.vl-alert__content-wrapper {
  display: flex;
  flex: 1;
  overflow: hidden;
}
@media screen and (max-width: 767px) {
  .vl-alert__content-wrapper {
    flex-wrap: wrap;
  }
}
.vl-alert__content {
  display: flex;
  justify-content: center;
  flex-direction: column;
  flex-basis: 100%;
}
@media screen and (max-width: 767px) {
  .vl-alert__content {
    flex-basis: auto;
    flex-grow: 1;
  }
}
.vl-alert__icon {
  flex: 0 0 auto;
  margin-right: 2.5rem;
  font-size: 2.4rem;
  line-height: 1;
}
@media screen and (max-width: 767px) {
  .vl-alert__icon {
    margin-right: 1.5rem;
    font-size: 1.6rem;
    margin-top: 3px;
  }
}
.vl-alert__title {
  font-size: 1.8rem;
  font-family: "Flanders Art Sans", sans-serif;
  font-weight: 500;
  margin-bottom: 1.4rem;
  line-height: 1.44;
  margin-bottom: 0.3rem;
}
@media screen and (max-width: 767px) {
  .vl-alert__title {
    font-size: 1.8rem;
    margin-bottom: 1rem;
  }
}
.vl-alert__actions {
  margin-top: 1.5rem;
}
.vl-alert__message {
  color: #333332;
}
.vl-alert--cta::before {
  content: "";
  position: absolute;
  top: -0.1rem;
  left: -0.1rem;
  width: calc(100% + 0.2rem);
  height: 0.3rem;
  background: #05c;
}
.vl-alert--accent {
  background-color: rgba(var(--vl-theme-primary-color-rgb, 255, 230, 21), 0.1);
  border-color: rgba(var(--vl-theme-primary-color-rgb, 255, 230, 21), 0.3);
}
.vl-alert--accent .vl-alert__icon {
  color: #000;
}
.vl-alert--error {
  background-color: #fbebec;
  border-color: #edafb1;
}
.vl-alert--error .vl-alert__icon {
  color: #aa2729;
}
.vl-alert--warning {
  background-color: #fff6e7;
  border-color: #ffd99d;
}
.vl-alert--warning .vl-alert__icon {
  color: #d07b06;
}
.vl-alert--success {
  background-color: #e6f5ed;
  border-color: #99d8b5;
}
.vl-alert--success .vl-alert__icon {
  color: #007a37;
}
.vl-alert--info {
  background-color: #f7f9fc;
  border-color: #e8ebee;
}
.vl-alert--naked {
  background: transparent;
  padding: 0;
  border: 0;
  border-radius: 0;
}
.vl-alert--naked .vl-alert__icon {
  margin-right: 1.5rem;
}
.vl-alert--naked .vl-alert__content {
  display: block;
}
.vl-alert--naked .vl-alert__title {
  margin-bottom: 0;
}
.vl-alert--naked .vl-alert__title > * {
  display: inline;
}
.vl-alert--naked.vl-alert--error .vl-alert__icon {
  color: #d2373c;
}
.vl-alert--naked p {
  display: inline;
}
.vl-alert--small {
  font-size: 1.4rem;
  line-height: 1.2;
  padding: 1.5rem;
}
.vl-alert--small .vl-alert__title {
  font-size: 1.6rem;
}
.vl-alert--small .vl-alert__icon {
  width: 2rem;
  height: 2rem;
  font-size: 2rem;
  margin-right: 1.5rem;
}
.vl-alert--small .vl-alert__actions {
  margin-top: 1rem;
}
.vl-alert--large {
  padding: 5rem 5.5rem;
}
@media screen and (max-width: 767px) {
  .vl-alert--large {
    padding: 1.5rem;
  }
}
.vl-alert--large .vl-alert__title {
  font-size: 2.2rem;
}
@media screen and (max-width: 767px) {
  .vl-alert--large .vl-alert__title {
    font-size: 2rem;
  }
}
.vl-alert--center {
  align-items: center;
}

.vl-alert__image {
  width: 8rem;
  height: auto;
  margin-right: 4rem;
}
@media screen and (max-width: 767px) {
  .vl-alert__image {
    width: 7rem;
    margin-right: 1rem;
    margin-bottom: 1rem;
  }
  .vl-alert__image + .vl-alert__content {
    max-width: calc(100% - 8rem);
  }
}
@media screen and (max-width: 500px) {
  .vl-alert__image + .vl-alert__content {
    max-width: none;
  }
}
`;
export default style;
