// @domg-wc/components/form/textarea-rich/vl-textarea-rich.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { webComponent } from '@domg-wc/common';
import tinymce from 'tinymce';
import { VlTextareaComponent } from '../textarea/vl-textarea.component';
import { vlTextareaRichComponentFluxStyles } from './vl-textarea-rich.component.flux-css';
import { textareaRichDefaults } from './vl-textarea-rich.defaults';
let VlTextareaRichComponent = class VlTextareaRichComponent extends VlTextareaComponent {
    constructor() {
        super();
        // Attributes
        this.toolbar = textareaRichDefaults.toolbar;
        this.plugins = textareaRichDefaults.plugins;
        this.preview = textareaRichDefaults.preview;
        // Properties
        this.customConfig = textareaRichDefaults.customConfig;
        // Variables
        this.editor = null;
        if (!this.id) {
            // Zet id als deze nog niet bestaat zodat TinyMCE een uniek id heeft
            this.id = tinymce.DOM.uniqueId();
        }
    }
    static get styles() {
        return [...VlTextareaComponent.styles, vlTextareaRichComponentFluxStyles];
    }
    static get properties() {
        return {
            toolbar: { type: String },
            plugins: { type: String },
            preview: { type: Boolean },
            customConfig: { type: Object },
        };
    }
    firstUpdated(changedProperties) {
        super.firstUpdated(changedProperties);
        this.initTinyMCE().then(() => {
            // TinyMCE stelt aria-hidden in op true, maar dit is problematisch voor screenreaders
            this.removeAttribute('aria-hidden');
        });
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        if (changedProperties.has('success')) {
            this.setEditorSuccess();
        }
        if (changedProperties.has('error')) {
            this.setEditorError();
        }
        if (changedProperties.has('readonly')) {
            this.setEditorReadonly();
        }
        if (changedProperties.has('disabled')) {
            this.setEditorDisabled();
        }
        if (changedProperties.has('preview')) {
            this.setEditorPreview();
        }
        if (changedProperties.has('value')) {
            if (this.editor?.getContent() !== this.value) {
                this.editor?.setContent(this.value);
            }
        }
        if (changedProperties.has('isInvalid')) {
            this.editor?.getBody()?.classList?.toggle('error', this.isInvalid || this.error);
        }
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.editor?.mode?.set('readonly');
        this.editor?.destroy();
    }
    resetFormControl() {
        super.resetFormControl();
        this.editor?.setContent(this.initialValue);
    }
    focus() {
        this.editor?.focus();
    }
    async initTinyMCE() {
        await tinymce.init({
            target: this.validationTarget,
            suffix: '.min',
            language: 'nl_BE',
            base_url: 'https://cdn.omgeving.vlaanderen.be/domg/tinymce/6.8.3',
            toolbar: this.toolbar,
            menubar: false,
            elementpath: false,
            branding: false,
            content_css: 'https://cdn.omgeving.vlaanderen.be/domg/govflanders-style/14.1.0/custom/tinymce.css',
            // Niet zeker of dit nodig is, behouden voor de zekerheid.
            verify_html: false,
            body_class: 'vl-typography',
            plugins: this.plugins,
            formats: {
                bold: { inline: 'b' },
                italic: { inline: 'i' },
                underline: { inline: 'u' },
                strikethrough: { inline: 's' },
            },
            // Link plugin configuratie
            link_title: false,
            link_target_list: false,
            link_default_target: '_blank',
            // Custom TinyMCE configuratie
            ...this.customConfig,
        });
        this.editor = tinymce.get(this.id);
        this.editor?.on('input change redo undo', () => {
            this.value = this.editor?.getContent() || '';
        });
        // Fix voor Link plugin die niet het eerste input veld focus geeft
        this.editor?.on('ExecCommand', (e) => {
            if (e.command === 'mceLink') {
                setTimeout(() => {
                    const firstInputField = this.shadowRoot?.querySelector('.tox-dialog input.tox-textfield');
                    firstInputField?.focus();
                }, 0);
            }
        });
        this.setEditorSuccess();
        this.setEditorError();
        this.setEditorReadonly();
        this.setEditorDisabled();
        this.setEditorPreview();
    }
    setEditorSuccess() {
        this.editor?.getBody()?.classList?.toggle('success', this.success);
    }
    setEditorError() {
        this.editor?.getBody()?.classList?.toggle('error', this.error);
    }
    setEditorReadonly() {
        const shouldDisableEditor = this.shouldDisableEditor();
        this.editor?.mode?.set(shouldDisableEditor ? 'readonly' : 'design');
    }
    setEditorDisabled() {
        const shouldDisableEditor = this.shouldDisableEditor();
        this.editor?.mode?.set(shouldDisableEditor ? 'readonly' : 'design');
        this.editor?.getBody()?.classList?.toggle('vl-textarea--disabled', this.disabled);
    }
    setEditorPreview() {
        const shouldDisableEditor = this.shouldDisableEditor();
        this.editor?.mode.set(shouldDisableEditor ? 'readonly' : 'design');
        this.editor?.getContainer()?.querySelector('.tox-editor-header')?.classList.toggle('hidden', this.preview);
    }
    shouldDisableEditor() {
        return this.readonly || this.disabled || this.preview;
    }
};
VlTextareaRichComponent = __decorate([
    webComponent('vl-textarea-rich'),
    __metadata("design:paramtypes", [])
], VlTextareaRichComponent);
export { VlTextareaRichComponent };
