// @domg/govflanders-style/component/code-preview/vl-code-preview.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-code-preview__button-copy {
  position: absolute;
  bottom: 1rem;
  right: 1rem;
  border: 0;
  font-family: "Flanders Art Sans", sans-serif;
  display: inline-flex;
  cursor: pointer;
  z-index: 1;
  padding: 0.5rem;
  margin: 0.5em 0;
  font-size: 1.2rem;
}
.vl-code-preview__button-copy:hover {
  background: #05c;
  color: #fff;
}
.vl-code-preview code {
  position: relative;
}
.vl-code-preview--condensed {
  overflow: scroll;
}
.vl-code-preview__max-height-toggle-button {
  position: absolute;
  display: block;
  left: 50%;
  bottom: 0;
  background-color: #fff;
  transform: translateX(-50%) translateY(50%);
  z-index: 1;
}
.vl-code-preview__max-height-toggle-button.vl-button {
  background-color: #fff;
}
.vl-code-preview__max-height-toggle-button.vl-button:hover, .vl-code-preview__max-height-toggle-button.vl-button:focus {
  background-color: #fff;
}
.vl-code-preview__max-height-toggle-button:hover, .vl-code-preview__max-height-toggle-button:focus {
  background-color: #fff;
}

/**
 * prism.js Coy theme for JavaScript, CoffeeScript, CSS and HTML
 * Based on https://github.com/tshedor/workshop-wp-theme (Example: http://workshop.kansan.com/category/sessions/basics or http://workshop.timshedor.com/category/sessions/basics);
 * @author Tim  Shedor
 */
code[class*=language-],
pre[class*=language-] {
  color: #000;
  background: none;
  font-family: Consolas, Monaco, "Andale Mono", "Ubuntu Mono", monospace;
  text-align: left;
  white-space: pre;
  word-spacing: normal;
  word-break: normal;
  word-wrap: normal;
  line-height: 1.5;
  font-size: 1.4rem;
  tab-size: 4;
  -webkit-hyphens: none;
  -moz-hyphens: none;
  -ms-hyphens: none;
  hyphens: none;
}

/* Code blocks */
pre[class*=language-] {
  position: relative;
  margin: 0;
  padding: 0;
  overflow: visible;
}

pre[class*=language-] > code {
  position: relative;
  background-color: #f7f7f8;
  background-size: 3em 3em;
  background-origin: content-box;
  background-attachment: local;
}

code[class*=language] {
  max-height: inherit;
  padding: 1.5rem 2em;
  display: block;
  overflow: auto;
}

/* Inline code */
:not(pre) > code[class*=language-] {
  position: relative;
  padding: 0.2em;
  border-radius: 0.3em;
  color: #d2373c;
  border: 1px solid rgba(0, 0, 0, 0.1);
  display: inline;
  white-space: normal;
}

:not(pre) > code[class*=language-]:after,
pre[class*=language-]:after {
  right: 0.75em;
  left: auto;
  -webkit-transform: rotate(2deg);
  -moz-transform: rotate(2deg);
  -ms-transform: rotate(2deg);
  -o-transform: rotate(2deg);
  transform: rotate(2deg);
}

.token.comment,
.token.block-comment,
.token.prolog,
.token.doctype,
.token.cdata {
  color: #687483;
}

.token.punctuation {
  color: #5f6364;
}

.token.property,
.token.tag,
.token.boolean,
.token.number,
.token.function-name,
.token.constant,
.token.symbol,
.token.deleted {
  color: #d2373c;
}

.token.selector,
.token.attr-name,
.token.string,
.token.char,
.token.function,
.token.builtin,
.token.inserted {
  color: #238000;
}

.token.operator,
.token.entity,
.token.url,
.token.variable {
  color: #966729;
}

.token.atrule,
.token.attr-value,
.token.keyword,
.token.class-name {
  color: #00789b;
}

.token.regex,
.token.important {
  color: #966729;
}

.language-css .token.string,
.style .token.string {
  color: #a67f59;
  background: rgba(255, 255, 255, 0.5);
}

.token.important {
  font-weight: normal;
}

.token.bold {
  font-weight: bold;
}

.token.italic {
  font-style: italic;
}

.token.entity {
  cursor: help;
}

.namespace {
  opacity: 0.7;
}

@media screen and (max-width: 767px) {
  pre[class*=language-]:before,
  pre[class*=language-]:after {
    bottom: 14px;
    box-shadow: none;
  }
}
/* Plugin styles */
.token.tab:not(:empty):before,
.token.cr:before,
.token.lf:before {
  color: #e0d7d1;
}

/* Plugin styles: Line Numbers */
pre[class*=language-].line-numbers {
  padding-left: 0;
}

pre[class*=language-].line-numbers code {
  padding-left: 4.8em;
}

pre[class*=language-].line-numbers .line-numbers-rows {
  left: 0;
}

/* Plugin styles: Line Highlight */
pre[class*=language-][data-line] {
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 0;
}

pre[data-line] code {
  position: relative;
  padding-left: 4em;
}

pre .line-highlight {
  margin-top: 0;
}

pre.line-numbers {
  position: relative;
  padding-left: 3.8em;
  counter-reset: linenumber;
}

pre.line-numbers > code {
  position: relative;
  white-space: inherit;
}

.line-numbers .line-numbers-rows {
  position: absolute;
  pointer-events: none;
  top: 0;
  font-size: 100%;
  left: -3.8em;
  padding: 1.5rem 0;
  width: 3em; /* works for line-numbers below 1000 lines */
  letter-spacing: -1px;
  background-color: #e8ebee;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.line-numbers-rows > span {
  pointer-events: none;
  display: block;
  counter-increment: linenumber;
}

.line-numbers-rows > span:before {
  content: counter(linenumber);
  color: #333332;
  display: block;
  padding-right: 0.8em;
  text-align: right;
}
`;
export default style;
