// @domg-wc/components/form/form-control/form-control.js
import { formControlDefaults } from './form-control.defaults';
import { FormControlMixin, programmaticValidator, requiredValidator } from '@open-wc/form-control';
import { LitElement } from 'lit';
import { submit } from '@open-wc/form-helpers';
import { FORM_MESSAGE_CUSTOM_TAG } from '../form-message/vl-form-message.component';
import { BaseLitElement } from '@domg-wc/common';
import 'reflect-metadata';
export class FormControl extends FormControlMixin(BaseLitElement) {
    constructor() {
        super(...arguments);
        // Attributes
        this.id = formControlDefaults.id;
        this.name = formControlDefaults.name;
        this.label = formControlDefaults.label;
        this.required = formControlDefaults.required;
        this.disabled = formControlDefaults.disabled;
        this.error = formControlDefaults.error;
        this.success = formControlDefaults.success;
        // State
        this.isInvalid = false;
        // Variables
        this.submitFormOnEnter = true;
    }
    static get properties() {
        return {
            id: { type: String },
            name: { type: String },
            label: { type: String },
            required: { type: Boolean },
            disabled: { type: Boolean },
            error: { type: Boolean },
            success: { type: Boolean },
            isInvalid: { type: Boolean, state: true },
        };
    }
    connectedCallback() {
        super.connectedCallback();
        this.addEventListener('keydown', this.onKeydown);
        this.addEventListener('invalid', this.onInvalid);
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.removeEventListener('keydown', this.onKeydown);
        this.removeEventListener('invalid', this.onInvalid);
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        if (!changedProperties.has('isInvalid')) {
            this.isInvalid = false;
            this.hideFormMessages();
        }
    }
    resetFormControl() {
        this.isInvalid = false;
        this.hideFormMessages();
        this.dispatchEvent(new Event('vl-reset', { bubbles: true, composed: true }));
    }
    dispatchEventIfValid(detail) {
        if (this.validity.valid) {
            this.dispatchEvent(new CustomEvent('vl-valid', { composed: true, bubbles: true, detail }));
        }
    }
    onKeydown(event) {
        if (event.code === 'Enter') {
            if (this.form && this.submitFormOnEnter) {
                submit(this.form);
            }
        }
    }
    onInvalid(event) {
        event.preventDefault();
        this.isInvalid = true;
        this.focusFirstInvalidInput();
        this.showFormMessage();
    }
    focusFirstInvalidInput() {
        const firstInvalidInput = this.form?.querySelector(':invalid');
        if (this === firstInvalidInput) {
            firstInvalidInput?.focus();
            firstInvalidInput?.scrollIntoView({ behavior: 'smooth', block: 'center' });
        }
    }
    showFormMessage() {
        let errorState = '';
        for (const key in this.validity) {
            if (this.validity[key]) {
                errorState = key;
                break;
            }
        }
        // Zoek de error message die bij de huidige error state hoort
        let errorMessage = this.form?.querySelector(`${FORM_MESSAGE_CUSTOM_TAG}[for="${this.id}"][state="${errorState}"]`);
        // Als er geen error message is voor de huidige error state, zoek dan de algemene error message zonder state attribuut
        if (!errorMessage) {
            errorMessage = this.form?.querySelector(`${FORM_MESSAGE_CUSTOM_TAG}[for="${this.id}"]:not([state])`);
        }
        this.formMessageText = errorMessage?.textContent;
        if (this.formMessageText) {
            this.validationTarget?.setAttribute('aria-description', this.formMessageText);
        }
        else {
            this.validationTarget?.removeAttribute('aria-description');
        }
        errorMessage?.setAttribute('validation-message', this.validationMessage);
        errorMessage?.setAttribute('show', '');
    }
    hideFormMessages() {
        const formMessages = this.form?.querySelectorAll(`${FORM_MESSAGE_CUSTOM_TAG}[for="${this.id}"]`);
        this.formMessageText = null;
        formMessages?.forEach((formMessage) => {
            formMessage.removeAttribute('show');
        });
    }
}
FormControl.formControlValidators = [requiredValidator, programmaticValidator];
FormControl.shadowRootOptions = { ...LitElement.shadowRootOptions, delegatesFocus: true };
