// @domg/govflanders-style/component/search-filter/vl-search-filter.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-search-filter {
  padding: 1.3rem 0 2rem;
}
.vl-search-filter .vl-search-filter__field__label {
  color: #687483;
  font-weight: 400;
  font-size: 1.6rem;
  line-height: 1;
}
@media screen and (max-width: 767px) {
  .vl-search-filter .vl-search-filter__field__label {
    font-size: 1.6rem;
  }
}
.vl-search-filter--alt {
  padding: 0;
}
@media screen and (max-width: 767px) {
  .vl-search-filter--mobile-modal {
    display: flex;
    background-color: #fff;
    flex-direction: column;
    position: fixed;
    padding: 0;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    height: 100vh;
    z-index: 100020;
  }
}

.vl-search-filter__intro {
  text-transform: uppercase;
  font-weight: 500;
  border-bottom: 3px solid #e8ebee;
  font-size: 1.8rem;
  padding: 0 0 0.7rem;
  margin: 0 0 2rem;
}

.vl-search-filter__section {
  padding-bottom: 2rem;
  margin-bottom: 2rem;
  border-bottom: 1px solid #cbd2da;
}
.vl-search-filter__section:last-child {
  border-bottom: 0;
  padding-bottom: 0;
  margin-bottom: 0;
}
.vl-search-filter__section--no-border {
  border-bottom: 0;
}
.vl-search-filter__section--border {
  border-bottom: 1px solid #cbd2da;
}

.vl-search-filter__section-title {
  font-size: 1.5rem;
  display: block;
  text-transform: uppercase;
  margin-bottom: 1.2rem;
  letter-spacing: 0.1rem;
  font-weight: 500;
}
@media screen and (max-width: 767px) {
  .vl-search-filter__section-title {
    font-size: 1.4rem;
  }
}

.vl-search-filter__form {
  padding: 2rem;
  background-color: #e8ebee;
}
.vl-region--alt .vl-search-filter__form {
  background-color: #fff;
}
.vl-search-filter--alt .vl-search-filter__form {
  padding: 0rem;
  background-color: transparent;
}
@media screen and (max-width: 767px) {
  .vl-search-filter--alt .vl-search-filter__form {
    padding: 0 2rem 55px;
  }
}
@media screen and (max-width: 767px) {
  .vl-search-filter--alt.vl-search-filter--mobile-modal .vl-search-filter__form {
    padding: 15px 3rem 9rem;
  }
}
@media screen and (max-width: 767px) {
  .vl-search-filter__form {
    margin-left: -1.5rem;
    margin-right: -1.5rem;
    padding: 1.5rem;
  }
}
@media screen and (max-width: 767px) {
  .vl-search-filter--mobile-modal .vl-search-filter__form {
    padding: 15px 3rem;
    flex: 1;
    overflow-y: auto;
  }
}

.vl-search-filter__field {
  margin-bottom: 0.75rem;
}
.vl-search-filter__field:last-child {
  margin-bottom: 0;
}

.vl-search-filter__field__label {
  margin-bottom: 0.7rem;
  display: inline-block;
}

.vl-search-filter__footer {
  margin-top: 1.5rem;
}

.vl-search-filter__header-modal {
  display: none;
}
@media screen and (max-width: 767px) {
  .vl-search-filter__header-modal {
    padding: 10px 15px;
    border-bottom: 1px solid #8695a8;
    display: flex;
    align-items: center;
  }
  .vl-search-filter__header-modal .vl-title {
    margin-bottom: 0;
  }
}

.vl-search-filter__header__clear {
  margin: 0 15px;
}

.vl-search-filter__footer-modal {
  display: none;
}
@media screen and (max-width: 767px) {
  .vl-search-filter__footer-modal {
    display: block;
    position: fixed;
    padding: 10px 15px;
    width: 100%;
    bottom: 0;
    left: 0;
    right: 0;
    border-top: 1px solid #8695a8;
    background-color: #fff;
    z-index: 2;
    box-shadow: 0 0 20px 0 rgba(0, 0, 0, 0.1);
  }
}

.vl-search-filter-section__collapse-toggle {
  margin-top: 1.5rem;
}
.vl-search-filter-section__collapse-toggle .vl-link__icon {
  transform: translateY(-1px);
}
`;
export default style;
