// @domg-wc/components/block/progress-bar/vl-progress-bar.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement } from '@domg-wc/common';
import { html } from 'lit';
import { customElement } from 'lit/decorators.js';
import { classMap } from 'lit/directives/class-map.js';
import { ifDefined } from 'lit/directives/if-defined.js';
import { progressBarDefaults } from './vl-progress-bar.defaults';
import { vlProgressBarFluxStyles } from './vl-progress-bar.flux-css';
let VlProgressBarComponent = class VlProgressBarComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.value = progressBarDefaults.value;
        this.label = progressBarDefaults.label;
        this.labelledby = progressBarDefaults.labelledby;
        this.indeterminate = progressBarDefaults.indeterminate;
        this.error = progressBarDefaults.error;
    }
    static get styles() {
        return [vlProgressBarFluxStyles];
    }
    static get properties() {
        return {
            value: {
                type: Number,
                attribute: 'value',
                reflect: true,
            },
            label: {
                type: String,
                attribute: 'label',
            },
            labelledby: {
                type: String,
                attribute: 'labelledby',
            },
            indeterminate: {
                type: Boolean,
                attribute: 'indeterminate',
                reflect: true,
            },
            error: {
                type: Boolean,
                attribute: 'error',
                reflect: true,
            },
        };
    }
    connectedCallback() {
        super.connectedCallback();
        if (!this.label && !this.labelledby) {
            console.warn('Het invullen van `label` of `labelledby` is verplicht voor de toegankelijkheid.');
        }
        if (this.label && this.labelledby) {
            console.warn('De attributen `label` en `labelledby` mogen niet beiden ingevuld zijn.');
        }
    }
    get classes() {
        return {
            'vl-progress-bar': true,
            'vl-progress-bar--indeterminate': this.indeterminate,
            'vl-progress-bar--error': this.error,
        };
    }
    get progressValue() {
        if (this.indeterminate)
            return undefined;
        return Math.max(0, Math.min(100, this.value)).toFixed(0);
    }
    render() {
        return html `
            <div
                class=${classMap(this.classes)}
                role="progressbar"
                aria-valuenow=${ifDefined(this.progressValue)}
                aria-label=${ifDefined(this.label || (!this.labelledby ? 'Voortgang' : undefined))}
                aria-labelledby=${ifDefined(this.labelledby ? this.labelledby : undefined)}
            >
                <div class="vl-progress-bar__track">
                    <div
                        class="vl-progress-bar__progress"
                        style="${this.indeterminate ? '' : `width: ${this.progressValue}%`}"
                    ></div>
                </div>
            </div>
        `;
    }
};
VlProgressBarComponent = __decorate([
    customElement('vl-progress-bar')
], VlProgressBarComponent);
export { VlProgressBarComponent };
