// @domg/govflanders-style/component/textarea/vl-textarea.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-textarea {
  display: inline-block;
  background: #fff;
  font-family: "Flanders Art Sans", sans-serif;
  font-size: 1.6rem;
  color: #333332;
  max-width: 100%;
  height: auto;
  line-height: normal;
  border-radius: 0.3rem;
  border: 0.1rem solid #8695a8;
  -webkit-appearance: none;
  padding: 1rem;
  transition: background-color 0.2s;
  margin: 0.5rem 0;
}
@media screen and (max-width: 767px) {
  .vl-textarea {
    font-size: 1.6rem;
  }
}
.vl-textarea:hover {
  border: 0.2rem solid rgba(0, 85, 204, 0.65);
  padding: 0.9rem;
}
.vl-textarea:hover.vl-textarea--error, .vl-textarea:hover.invalid.validated {
  border-color: #d2373c;
}
.vl-textarea:hover.vl-textarea--success, .vl-textarea:hover.valid.validated {
  border-color: #009e47;
}
.vl-textarea:focus, .vl-textarea--focus {
  box-shadow: 0 0 0 2px #fff, 0 0 0 5px rgba(0, 85, 204, 0.65);
  outline: transparent solid 0.2rem;
  border: 0.1rem solid #687483;
  padding: 1rem;
}
@supports (outline-offset: 2px) {
  .vl-textarea:focus, .vl-textarea--focus {
    box-shadow: none;
    outline: 3px solid rgba(0, 85, 204, 0.65);
    outline-offset: 2px;
  }
}
.vl-textarea:focus.vl-textarea--error, .vl-textarea:focus.invalid.validated, .vl-textarea--focus.vl-textarea--error, .vl-textarea--focus.invalid.validated {
  border-color: #d2373c;
}
.vl-textarea:focus.vl-textarea--success, .vl-textarea:focus.valid.validated, .vl-textarea--focus.vl-textarea--success, .vl-textarea--focus.valid.validated {
  border-color: #009e47;
}
.vl-textarea::placeholder {
  color: #687483;
}
.vl-textarea--block {
  display: block;
  width: 100%;
  box-sizing: border-box;
}
.vl-textarea--error, .vl-textarea.invalid.validated {
  border-color: #d2373c;
  background-color: #fbebec;
}
.vl-textarea--error:focus, .vl-textarea.invalid.validated:focus {
  background-color: #fff;
}
.vl-textarea--success, .vl-textarea.valid.validated {
  border-color: #009e47;
  background-color: #e6f5ed;
  display: inline;
}
.vl-textarea--disabled, .vl-textarea[disabled] {
  background-color: #f3f5f6;
  border-color: #8695a8;
  color: var(--vl-theme-fg-color-70);
}
.vl-textarea--disabled:hover, .vl-textarea[disabled]:hover {
  border-width: 0.1rem;
  padding: 1rem;
}
`;
export default style;
