// @domg-wc/map/components/layer/wms-layer/vl-map-tiled-wms-layer/vl-map-tiled-wms-layer.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { webComponent } from '@domg-wc/common';
import OlTileLayer from 'ol/layer/Tile';
import { default as OlTileWMSSource } from 'ol/source/TileWMS';
import { VlMapWmsLayer } from '../vl-map-wms-layer';
class TileWMSSource extends OlTileWMSSource {
    constructor(layer) {
        super({ ...layer._sourceConfig });
        this.layer = layer;
    }
    tileUrlFunction(tileCoord, pixelRatio, projection) {
        const extent = this.layer.mapElement.map.getView().calculateExtent(this.layer.mapElement.map.getSize());
        this.updateParams({ ...this.layer._sourceConfig.params, tilesorigin: `${extent[0]},${extent[3]}` });
        return super.tileUrlFunction(tileCoord, pixelRatio, projection);
    }
}
let VlMapTiledWmsLayer = class VlMapTiledWmsLayer extends VlMapWmsLayer {
    constructor() {
        super(OlTileLayer, OlTileWMSSource);
    }
    get _sourceConfig() {
        return {
            ...super._sourceConfig,
            params: {
                ...super._sourceConfig.params,
                TILED: true,
            },
        };
    }
    __createSource() {
        return new TileWMSSource(this);
    }
};
VlMapTiledWmsLayer = __decorate([
    webComponent('vl-map-tiled-wms-layer'),
    __metadata("design:paramtypes", [])
], VlMapTiledWmsLayer);
export { VlMapTiledWmsLayer };
