// @domg-wc/components/atom/title/vl-title.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, webComponent } from '@domg-wc/common';
import { html } from 'lit-element';
import { choose } from 'lit/directives/choose.js';
import { classMap } from 'lit/directives/class-map.js';
import { vlTitleStyles } from './vl-title.css';
import { titleDefaults } from './vl-title.defaults';
let VlTitleComponent = class VlTitleComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.type = titleDefaults.type;
        this.underline = titleDefaults.underline;
        this.noSpaceBottom = titleDefaults.noSpaceBottom;
        this.alt = titleDefaults.alt;
    }
    static get styles() {
        return [vlTitleStyles];
    }
    static get properties() {
        return {
            type: { type: String },
            underline: { type: Boolean },
            noSpaceBottom: { type: Boolean, attribute: 'no-space-bottom' },
            alt: { type: Boolean },
        };
    }
    render() {
        const classes = {
            underline: this.underline,
            'no-space-bottom': this.noSpaceBottom,
            alt: this.alt,
        };
        return html `
            ${choose(this.type, [
            [
                'h1',
                () => html ` <h1 class=${classMap(classes)} part="h1">
                            <slot></slot>
                        </h1>`,
            ],
            [
                'h2',
                () => html ` <h2 class=${classMap(classes)} part="h2">
                            <slot></slot>
                        </h2>`,
            ],
            [
                'h3',
                () => html ` <h3 class=${classMap(classes)} part="h3">
                            <slot></slot>
                        </h3>`,
            ],
            [
                'h4',
                () => html ` <h4 class=${classMap(classes)} part="h4">
                            <slot></slot>
                        </h4>`,
            ],
            [
                'h5',
                () => html ` <h5 class=${classMap(classes)} part="h5">
                            <slot></slot>
                        </h5>`,
            ],
            [
                'h6',
                () => html ` <h6 class=${classMap(classes)} part="h6">
                            <slot></slot>
                        </h6>`,
            ],
        ], () => html ` <h1 class=${classMap(classes)} part="h1">
                    <slot></slot>
                </h1>`)}
        `;
    }
};
VlTitleComponent = __decorate([
    webComponent('vl-title')
], VlTitleComponent);
export { VlTitleComponent };
