// @domg/govflanders-style/component/upload/vl-upload.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-upload {
  position: relative;
  margin-bottom: 1rem;
  transition: background-color 0.3s ease-out;
}
.vl-upload--basic .vl-upload__element__input {
  width: 100%;
  height: auto;
  margin: 0;
  opacity: 1;
  pointer-events: auto;
  position: relative;
  overflow: auto;
}
.vl-upload--error .vl-upload__element, .vl-upload--error .vl-upload__element:hover {
  border-color: #d2373c;
  background-color: #fbebec;
}
.vl-upload--error .vl-upload__element .vl-upload__element__label:hover, .vl-upload--error .vl-upload__element:hover .vl-upload__element__label:hover {
  background-color: transparent;
}
.vl-upload--success .vl-upload__element, .vl-upload--success .vl-upload__element:hover {
  border-color: #009e47;
  background-color: #e6f5ed;
}
.vl-upload--success .vl-upload__element .vl-upload__element__label:hover, .vl-upload--success .vl-upload__element:hover .vl-upload__element__label:hover {
  background-color: transparent;
}
.vl-upload--disabled .vl-upload__element {
  background-color: #cbd2d9;
}
.vl-upload--disabled .vl-upload__element:hover, .vl-upload--disabled .vl-upload__element:focus {
  background-color: #cbd2d9;
  border: 0.1rem dashed #8695a8;
}
.vl-upload--disabled .vl-upload__element:hover .vl-upload__element__label, .vl-upload--disabled .vl-upload__element:focus .vl-upload__element__label {
  background-color: transparent;
  padding: 1rem;
}
.vl-upload--disabled .vl-upload__element .vl-upload__element__label {
  cursor: not-allowed;
}
.vl-upload--disabled .vl-upload__element .vl-upload__element__label .vl-upload__element__button {
  color: #687483;
  text-decoration: none;
  cursor: not-allowed;
}
.vl-upload__overlay {
  position: absolute;
  display: none;
  pointer-events: none;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: #fff;
  border: 0.1rem dashed #05c;
  z-index: 10;
  opacity: 0.9;
  font-family: "Flanders Art Sans", sans-serif;
}
.vl-upload__overlay__text {
  position: absolute;
  left: 50%;
  top: 50%;
  width: 80%;
  transform: translate(-50%, -50%);
  text-align: center;
  font-size: 1.4rem;
}
.vl-upload__overlay__text .vl-vi {
  font-size: 2rem;
  display: block;
  color: #05c;
}
.vl-upload__files {
  margin-bottom: 2rem;
  display: none;
  position: relative;
}
.vl-upload__files::before, .vl-upload__files::after {
  content: "";
  display: table;
}
.vl-upload__files::after {
  clear: both;
}
.vl-upload__files--has-files {
  display: block;
}
.vl-upload__files__close {
  margin: 1rem 0 0;
  float: right;
}
.vl-upload__files__close .vl-vi {
  margin-right: 0.5rem;
}
@media screen and (max-width: 767px) {
  .vl-upload__files__close {
    float: none;
  }
}
.vl-upload__file {
  position: relative;
  margin-bottom: 0.2rem;
  padding: 1rem 1.5rem;
  z-index: 5;
  font-family: "Flanders Art Sans", sans-serif;
  border: 0;
}
.vl-upload__file:last-child {
  border-bottom: 0;
}
.vl-upload__file:hover {
  background-color: rgba(179, 207, 245, 0.3);
}
.vl-upload__file__size {
  color: #687483;
}
.vl-upload__file__name {
  color: #333332;
  max-width: calc(100% - 2rem);
  word-break: break-all;
}
@supports (overflow-wrap: break-word) {
  .vl-upload__file__name {
    overflow-wrap: break-word;
    word-break: normal;
  }
}
.vl-upload__file__name__icon {
  display: inline-block;
  transform: translateY(-1px);
  line-height: 2rem;
  padding-right: 1rem;
  color: #687483;
}
.vl-upload__file__close {
  position: absolute;
  top: 50%;
  right: 0;
  transform: translateY(-50%);
  padding: 1rem;
  margin-right: 1rem;
  color: #05c;
  box-shadow: none;
  overflow: hidden;
}
.vl-upload__file__close .vl-vi {
  height: 1.1rem;
  width: 1.1rem;
  font-size: 1.6rem;
}
@media screen and (max-width: 1023px) {
  .vl-upload__file__close .vl-vi {
    font-size: 1.4rem;
  }
}
.vl-upload__file__close:hover {
  box-shadow: none;
}
.vl-upload__file__close:focus {
  box-shadow: 0 0 0 2px #fff, 0 0 0 5px rgba(0, 85, 204, 0.65);
}
.vl-upload__file.dz-error, .vl-upload__file--error {
  border: 0.1rem solid #edafb1;
  background-color: #fbebec;
  padding-left: 6.5rem;
}
@media screen and (max-width: 767px) {
  .vl-upload__file.dz-error, .vl-upload__file--error {
    padding-left: 5rem;
  }
}
.vl-upload__file.dz-error::before, .vl-upload__file--error::before {
  content: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'%3E%3Ctitle%3E_%3C/title%3E%3Cpath fill='%23db3434' d='M12,22A10,10,0,1,1,22,12,10,10,0,0,1,12,22ZM12,4a8,8,0,1,0,8,8A8,8,0,0,0,12,4ZM11,6h2v7H11V6Zm1,11.25A1.25,1.25,0,1,1,13.25,16,1.25,1.25,0,0,1,12,17.25Z'/%3E%3C/svg%3E");
  position: absolute;
  display: inline-flex;
  top: 50%;
  left: 2rem;
  transform: translateY(-50%);
  width: 24px;
  height: 24px;
}
@media screen and (max-width: 767px) {
  .vl-upload__file.dz-error::before, .vl-upload__file--error::before {
    left: 1.5rem;
    width: 18px;
    height: 18px;
  }
}
.vl-upload__file.dz-error .vl-upload__file__name, .vl-upload__file--error .vl-upload__file__name {
  font-weight: 500;
}
.vl-upload__file.dz-error .vl-upload__file__error,
.vl-upload__file.dz-error .dz-error-message, .vl-upload__file--error .vl-upload__file__error,
.vl-upload__file--error .dz-error-message {
  font-size: 0.9em;
}
.vl-upload__file.dz-error .vl-upload__file__name__icon, .vl-upload__file--error .vl-upload__file__name__icon {
  display: none;
}
.vl-upload__file.dz-error .vl-upload__file__close, .vl-upload__file--error .vl-upload__file__close {
  cursor: pointer;
}
.vl-upload__file.dz-error .vl-upload__file__close .vl-vi, .vl-upload__file--error .vl-upload__file__close .vl-vi {
  color: #333332;
}
.vl-upload__file.dz-error .vl-upload__file__close:focus, .vl-upload__file--error .vl-upload__file__close:focus {
  box-shadow: 0 0 0 2px #fbebec, 0 0 0 5px rgba(0, 85, 204, 0.65);
}
.vl-upload__file.dz-error.vl-upload__file--error-size .vl-upload__file__size, .vl-upload__file--error.vl-upload__file--error-size .vl-upload__file__size {
  color: #333332;
}
.vl-upload__file.dz-error:hover, .vl-upload__file--error:hover {
  background-color: #fbebec;
}
.vl-upload.upload--dragging .vl-upload__overlay {
  opacity: 1;
  display: block;
}
.vl-upload.upload--dragging .vl-upload__element {
  min-height: 8.75rem;
}
.vl-upload__element {
  font-family: "Flanders Art Sans", sans-serif;
  border: 0.1rem dashed #8695a8;
  border-radius: 0.3rem;
  text-align: center;
  position: relative;
  z-index: 4;
  min-height: 100%;
}
.vl-upload__element:hover {
  border-width: 0.2rem;
  border-color: rgba(0, 85, 204, 0.65);
}
.vl-upload__element:hover .vl-upload__element__label {
  background-color: rgba(179, 207, 245, 0.3);
  padding: 0.9rem;
}
.vl-upload__element__input {
  opacity: 0;
  pointer-events: none;
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 1rem;
  overflow: hidden;
}
.vl-upload__element__input:focus {
  outline: none;
}
.vl-upload__element__input:focus + label {
  background-color: rgba(179, 207, 245, 0.3);
}
.vl-upload__element__label {
  color: #05c;
  display: block;
  padding: 1rem;
  cursor: pointer;
  background-color: transparent;
  transition: background-color 0.3s ease-out;
  height: 100%;
}
.vl-upload__element__label .vl-vi {
  display: inline-block;
  font-size: 1.4rem;
  margin-right: 0.5rem;
  line-height: 2.4rem;
  top: 1px;
}
.vl-upload__element__label small {
  font-size: 1.4rem;
  color: #687483;
  display: none;
  text-decoration: none;
  text-align: center;
  line-height: 1.5;
  margin-top: 0.5rem;
}
.vl-upload .vl-upload__element__label small {
  display: block;
}
.vl-upload .vl-upload__element__label__error {
  color: #d2373c;
  font-size: 0.9em;
}

.vl-upload--dragging .vl-upload__overlay {
  opacity: 1;
  display: block;
}

.no-js .vl-upload__element__label {
  display: none;
}

.dz-image-preview {
  overflow: hidden;
}

.vl-upload__element__button {
  border-radius: 0;
  appearance: none;
  -webkit-appearance: none;
  border: 0;
  background-color: transparent;
  padding: 0;
  display: inline-flex;
  color: inherit;
  font-size: inherit;
  line-height: 2.4rem;
}

.vl-upload--dragging [data-vl-upload-full-body-drop] {
  position: fixed;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  min-height: 100vh;
  z-index: 10006;
  margin: 0;
}

.dropzone .dz-message .dz-button {
  border-radius: 0;
  appearance: none;
  -webkit-appearance: none;
  border: 0;
  background-color: transparent;
  padding: 0;
  width: 100%;
  font-size: inherit;
  position: relative;
}
.dropzone .dz-message .dz-button:focus::after {
  display: block;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 10;
  background-color: rgba(179, 207, 245, 0.3);
  content: "";
}
`;
export default style;
