// @domg-wc/map/components/legend-item/vl-map-legend-item.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement } from '@domg-wc/common';
import { html } from 'lit';
import { customElement } from 'lit/decorators.js';
import { vlMapLegendFluxStyles } from '../legend/vl-map-legend.flux-css';
import { vlMapLegendItemFluxStyles } from './vl-map-legend-item.flux-css';
let VlMapLegendItem = class VlMapLegendItem extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.icon = null;
        this.label = '';
        this.layer = '';
        this.iconText = '';
        this.iconTextColor = '';
    }
    static get styles() {
        return [vlMapLegendFluxStyles, vlMapLegendItemFluxStyles];
    }
    static get properties() {
        return {
            layer: { type: String, attribute: 'layer' },
            iconText: { type: String, attribute: 'icon-text' },
            iconTextColor: { type: String, attribute: 'icon-text-color', reflect: true },
        };
    }
    render() {
        return html `
            <div id="legend-item" class="flux-map-legend-item">
                <div class="flux-map-legend-icon-container">
                    ${this.icon}
                    <slot name="icon"></slot>
                </div>
                <span id="label" class="flux-map-legend-text">
                    ${this.label}
                    <slot name="label"></slot>
                </span>
            </div>
        `;
    }
};
VlMapLegendItem = __decorate([
    customElement('vl-map-legend-item')
], VlMapLegendItem);
export { VlMapLegendItem };
