// @domg-wc/components/block/description-data/vl-description-data.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement } from '@domg-wc/common';
import { vlGridStyles, vlResetStyles } from '@domg-wc/styles';
import { descriptionDataStyle } from '@domg/govflanders-style/component';
import { html } from 'lit';
import { customElement } from 'lit/decorators.js';
import { classMap } from 'lit/directives/class-map.js';
let VlDescriptionData = class VlDescriptionData extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.size = 0;
        this.mediumSize = 0;
        this.smallSize = 0;
        this.extraSmallSize = 0;
        this.bordered = false;
    }
    static get styles() {
        return [vlResetStyles, descriptionDataStyle, vlGridStyles];
    }
    static get properties() {
        return {
            size: { type: Number, attribute: 'items-size' },
            mediumSize: { type: Number, attribute: 'items-medium-size' },
            smallSize: { type: Number, attribute: 'items-small-size' },
            extraSmallSize: { type: Number, attribute: 'items-extra-small-size' },
            bordered: { type: Boolean, attribute: 'bordered' },
        };
    }
    firstUpdated() {
        const observer = new MutationObserver(() => {
            this.requestUpdate();
        });
        observer.observe(this, { subtree: true, childList: true });
    }
    render() {
        this.size = this.size || 12 / this.children.length;
        const classes = {
            'vl-description-data--bordered': this.bordered,
        };
        const columnClasses = {
            [`vl-column--${this.size}`]: this.size,
            [`vl-column--m-${this.mediumSize}`]: this.mediumSize,
            [`vl-column--s-${this.smallSize}`]: this.smallSize,
            [`vl-column--xs-${this.extraSmallSize}`]: this.extraSmallSize,
        };
        return html `
            <div class="vl-description-data ${classMap(classes)}">
                <div class="vl-grid">
                    ${[...Array.from(this.children)].map((child, index) => {
            const name = `item-${index}`;
            child.setAttribute('slot', name);
            return html `
                            <div class="vl-column ${classMap(columnClasses)}">
                                <slot name=${name}></slot>
                            </div>
                        `;
        })}
                </div>
            </div>
        `;
    }
};
VlDescriptionData = __decorate([
    customElement('vl-description-data')
], VlDescriptionData);
export { VlDescriptionData };
