// @domg-wc/components/block/breadcrumb/vl-breadcrumb-item.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement } from '@domg-wc/common';
import { resetStyle } from '@domg/govflanders-style/common';
import { breadcrumbStyle } from '@domg/govflanders-style/component';
import { html } from 'lit';
import { customElement } from 'lit/decorators.js';
import { vlBreadcrumbFluxStyles } from './vl-breadcrumb.flux-css';
let VlBreadcrumbItemComponent = class VlBreadcrumbItemComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.href = '';
        this.type = 'link';
    }
    static get properties() {
        return {
            href: {
                type: String,
                attribute: 'href',
                reflect: true,
            },
            type: {
                type: String,
                attribute: 'type',
                reflect: true,
            },
        };
    }
    static get styles() {
        return [resetStyle, breadcrumbStyle, vlBreadcrumbFluxStyles];
    }
    render() {
        if (this.type === 'link' && this.href) {
            return html `
                <a href=${this.href} class="vl-breadcrumb__list__item__cta">
                    <slot></slot>
                </a>
            `;
        }
        if (this.type === 'button') {
            return html `
                <button class="vl-breadcrumb__list__item__cta" type="button">
                    <slot></slot>
                </button>
            `;
        }
        return html `
            <span class="vl-breadcrumb__list__item__cta">
                <slot></slot>
            </span>
        `;
    }
};
VlBreadcrumbItemComponent = __decorate([
    customElement('vl-breadcrumb-item')
], VlBreadcrumbItemComponent);
export { VlBreadcrumbItemComponent };
