// @domg-wc/components/compliance/next/header/vl-header.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { awaitScript, BaseLitElement, legacyBreakpoint, legacyCore, registerWebComponents, webComponent, } from '@domg-wc/common';
import { headerContainerStyles, headerSkeletonStyles } from './vl-header.component.flux-css';
import { headerDefaults } from './vl-header.defaults';
registerWebComponents([legacyCore, legacyBreakpoint]);
let VlHeaderNext = class VlHeaderNext extends BaseLitElement {
    constructor() {
        super();
        // Properties
        this.applicationLinks = headerDefaults.applicationLinks;
        // Attributes
        this.authenticatedUserUrl = headerDefaults.authenticatedUserUrl;
        this.development = headerDefaults.development;
        this.identifier = headerDefaults.identifier;
        this.loginUrl = headerDefaults.loginUrl;
        this.logoutUrl = headerDefaults.logoutUrl;
        this.switchCapacityUrl = headerDefaults.switchCapacityUrl;
        this.simple = headerDefaults.simple;
        this.skeleton = headerDefaults.skeleton;
        this.allowCustomCSS = false;
    }
    static get properties() {
        return {
            authenticatedUserUrl: { type: String, attribute: 'authenticated-user-url' },
            development: { type: Boolean, attribute: 'development' },
            identifier: { type: String, attribute: 'identifier' },
            loginUrl: { type: String, attribute: 'login-url' },
            logoutUrl: { type: String, attribute: 'logout-url' },
            simple: { type: Boolean, attribute: 'simple' },
            skeleton: { type: Boolean, attribute: 'skeleton' },
            switchCapacityUrl: { type: String, attribute: 'switch-capacity-url' },
            applicationLinks: { type: Array },
        };
    }
    get headerContainer() {
        return document.querySelector('#header__container');
    }
    get headerContainerSkeleton() {
        return document.querySelector('#header__container__skeleton');
    }
    connectedCallback() {
        super.connectedCallback();
        this.injectHeaderContainer();
        if (this.skeleton) {
            this.injectHeaderContainerSkeleton();
        }
        this.loadWidget();
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.headerContainer?.remove();
        if (this.skeleton) {
            this.headerContainerSkeleton?.remove();
        }
        window.removeEventListener('widget.global_header.mounted', this.onReady);
    }
    willUpdate(changedProperties) {
        const sessionProperties = ['loginUrl', 'logoutUrl', 'switchCapacityUrl'];
        if (sessionProperties.some((property) => changedProperties.has(property))) {
            this.configureSession();
        }
    }
    createRenderRoot() {
        return this;
    }
    injectHeaderContainer() {
        if (this.headerContainer)
            return;
        (document.querySelector('body') || document.body).insertAdjacentHTML('afterbegin', '<header id="header__container"><div id="header"></div></header>');
        document.adoptedStyleSheets = [...document.adoptedStyleSheets, headerContainerStyles.styleSheet];
    }
    injectHeaderContainerSkeleton() {
        this.headerContainer?.insertAdjacentHTML('afterend', '<div id="header__skeleton"></div>');
        document.adoptedStyleSheets = [...document.adoptedStyleSheets, headerSkeletonStyles.styleSheet];
    }
    async isUserAuthenticated() {
        const response = await fetch(this.authenticatedUserUrl);
        return response.status === 200;
    }
    onReady() {
        this.dispatchEvent(new CustomEvent('ready'));
    }
    async loadWidget() {
        try {
            window.addEventListener('widget.global_header.mounted', this.onReady);
            await awaitScript('vl-header-widget', `https://widgets.${this.development ? 'tni-' : ''}vlaanderen.be/api/v2/widget/${this.identifier}/entry`);
            if (!window.globalHeaderClient) {
                console.error('De global header client werd niet geladen.');
                return;
            }
            const headerElement = this.headerContainer?.querySelector('#header') || undefined;
            await window.globalHeaderClient.mount(headerElement);
            if (this.applicationLinks.length > 0) {
                await window.globalHeaderClient.accessMenu.setApplicationMenuLinks(this.applicationLinks);
            }
            if (this.simple) {
                return;
            }
            this.configureSession();
            // Sinds v5 is het niet meer nodig om automatische logout requests te rejecten
            // widget.on('citizen_profile.session.logout.request', async (logoutRequest: any) => {...});
        }
        catch (error) {
            console.error('De global header werd niet geladen.', error);
        }
    }
    async configureSession() {
        const active = await this.isUserAuthenticated();
        window.globalHeaderClient?.accessMenu?.setProfile({
            active,
            loginUrl: this.loginUrl,
            logoutUrl: this.logoutUrl,
            switchCapacityUrl: this.switchCapacityUrl,
        });
    }
    get height() {
        return this.headerContainer?.getBoundingClientRect().height || 0;
    }
};
VlHeaderNext = __decorate([
    webComponent('vl-header-next'),
    __metadata("design:paramtypes", [])
], VlHeaderNext);
export { VlHeaderNext };
