// @domg-wc/components/block/rich-data-table/vl-rich-data-field.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseHTMLElement, registerWebComponents, webComponent } from '@domg-wc/common';
import { VlRichDataSorter } from './vl-rich-data-sorter.component';
let VlRichDataField = class VlRichDataField extends BaseHTMLElement {
    static get headerAttributes() {
        return ['name', 'label', 'sortable', 'sorting-direction', 'sorting-priority'];
    }
    static get bodyAttributes() {
        return ['selector', 'renderer'];
    }
    static get _observedAttributes() {
        return this.headerAttributes.concat(this.bodyAttributes);
    }
    static get is() {
        return 'vl-rich-data-field';
    }
    headerTemplate() {
        const th = document.createElement('th');
        const headerContent = this.__getHeaderContentElement();
        if (headerContent) {
            th.appendChild(headerContent);
        }
        if (this.sortable) {
            th.setAttribute('sortable', '');
        }
        return th;
    }
    valueTemplate(rowData) {
        const td = document.createElement('td');
        if (this.label) {
            td.setAttribute('data-title', this.label);
        }
        const element = this.__getValueContentElement(rowData);
        if (element) {
            td.appendChild(element);
        }
        else if (this._renderer) {
            this._renderer(td, rowData);
        }
        return td;
    }
    /**
     * Geeft de naam terug die gebruikt wordt om het veld te identificeren.
     * @return {string}
     */
    get name() {
        return this.getAttribute('name');
    }
    /**
     * Geeft de selector terug die gebruikt wordt om de juiste waarde uit de data te halen.
     * @return {string}
     */
    get selector() {
        return this.getAttribute('selector');
    }
    /**
     * Geeft de naam terug die getoond kan worden aan de gebruiker.
     * @return {string}
     */
    get label() {
        return this.getAttribute('label');
    }
    /**
     * Geeft terug of er op het veld gesorteerd kan worden.
     * @return {boolean}
     */
    get sortable() {
        return this.getAttribute('sortable') !== null;
    }
    /**
     * Geeft de sorteerrichting terug.
     * @return {asc | desc}
     */
    get sortingDirection() {
        return this.getAttribute('sorting-direction');
    }
    /**
     * Geeft de prioriteit van het sorteren terug.
     * @return {number}
     */
    get sortingPriority() {
        return this.getAttribute('sorting-priority');
    }
    get _labelSlotElement() {
        return this.querySelector('template[slot="label"]');
    }
    get _contentSlotElement() {
        return this.querySelector('template[slot="content"]');
    }
    set renderer(renderer) {
        this._renderer = renderer;
        this._changed(['renderer']);
    }
    _nameChangedCallback(oldValue, newValue) {
        if (oldValue !== newValue) {
            this._changed(['name']);
        }
    }
    _selectorChangedCallback(oldValue, newValue) {
        if (oldValue !== newValue) {
            this._changed(['selector']);
        }
    }
    _labelChangedCallback(oldValue, newValue) {
        if (oldValue !== newValue) {
            this._changed(['label']);
        }
    }
    _sortableChangedCallback(oldValue, newValue) {
        if (oldValue !== newValue) {
            this._changed(['sortable']);
        }
    }
    _sortingDirectionChangedCallback(oldValue, newValue) {
        if (oldValue !== newValue) {
            this._changed(['sorting-direction']);
        }
    }
    _sortingPriorityChangedCallback(oldValue, newValue) {
        if (oldValue !== newValue) {
            this._changed(['sorting-priority']);
        }
    }
    _changed(properties) {
        this.dispatchEvent(new CustomEvent('change', {
            detail: {
                properties: properties,
            },
        }));
    }
    get __headerContent() {
        return this.label || (this._labelSlotElement ? this._labelSlotElement.innerHTML : undefined);
    }
    __getHeaderContentElement() {
        const content = this.__headerContent;
        if (content) {
            if (this.sortable) {
                const direction = this.sortingDirection ? `direction="${this.sortingDirection}"` : '';
                const priority = this.sortingPriority ? `priority="${this.sortingPriority}"` : '';
                const sorter = `<vl-rich-data-sorter for="${this.name}" ${direction} ${priority}></vl-rich-data-sorter>`;
                return this._template(`<a>${content}${sorter}</a>`);
            }
            else {
                return this._template(`${content}`);
            }
        }
        else {
            return undefined;
        }
    }
    __getValueContentElement(data) {
        if (this.selector) {
            return this._template(`${this.selector.split('.').reduce((prev, curr) => (prev ? prev[curr] : null), data)}`);
        }
        else if (this._contentSlotElement) {
            const literal = `${this.querySelector('template[slot="content"]')?.innerHTML}`;
            const template = ((literal, item) => new Function('item', 'return `' + literal + '`')(item)).call(this, literal, data);
            return this._template(template);
        }
        else {
            return null;
        }
    }
};
(() => {
    registerWebComponents([VlRichDataSorter]);
})();
VlRichDataField = __decorate([
    webComponent('vl-rich-data-field')
], VlRichDataField);
export { VlRichDataField };
