// @domg-wc/components/block/rich-data-table/vl-rich-data-table.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var VlRichDataTable_1;
import { registerWebComponents, webComponentPromised } from '@domg-wc/common';
import { VlRichData } from '../rich-data/vl-rich-data.component';
import { VlTableComponent } from '../table/vl-table.component';
import { VlRichDataField } from './vl-rich-data-field.component';
import { VlRichDataSorter } from './vl-rich-data-sorter.component';
let VlRichDataTable = VlRichDataTable_1 = class VlRichDataTable extends VlRichData {
    constructor() {
        const content = `
            <vl-table slot="content">
                <table>
                    <thead>
                        <tr></tr>
                    </thead>
                    <tbody></tbody>
                </table>
            </vl-table>
        `;
        super(content);
        this.__observeSorters();
    }
    static get _observedAttributes() {
        return super._observedAttributes.concat([
            'data',
            'collapsed-m',
            'collapsed-s',
            'collapsed-xs',
            'zebra',
            'flux-zebra',
        ]);
    }
    static get _tableAttributes() {
        return ['collapsed-m', 'collapsed-s', 'collapsed-xs', 'zebra', 'flux-zebra'];
    }
    static get is() {
        return 'vl-rich-data-table';
    }
    /**
     * Geeft de data terug die in de tabel wordt getoond.
     * @return {Object[]}
     */
    get data() {
        return super.data;
    }
    /**
     * Stelt in welke data de tabel moet tonen.
     * @param {data: Object[]} object - Een Array van objecten die de data voorstellen.
     */
    set data(object) {
        const previousData = this.data ? this.data.data : undefined;
        super.data = object;
        const hasNewData = previousData !== this.data.data;
        if (hasNewData) {
            try {
                this._validate(this.data.data);
                this._renderBody();
            }
            catch (error) {
                if (this._data)
                    this._data.data = [];
                throw error;
            }
        }
    }
    get __activeSorters() {
        if (Boolean(this.__sorters) && this.__sorters !== undefined) {
            const sorters = Array.from(this.__sorters);
            return sorters
                .filter((sorter) => sorter.direction !== undefined)
                .sort(VlRichDataSorter.PRIORITY_COMPARATOR);
        }
        else {
            return undefined;
        }
    }
    get __contentColumn() {
        return this.shadowRoot?.querySelector('#content');
    }
    get __fields() {
        return this.querySelectorAll(VlRichDataField.is);
    }
    get __richDataFields() {
        return [...this.__fields].filter((field) => field.constructor === VlRichDataField);
    }
    get __sorters() {
        return this.__tableHeaderRow ? this.__tableHeaderRow.querySelectorAll(VlRichDataSorter.is) : undefined;
    }
    get __sortingState() {
        if (this.__activeSorters && this.__activeSorters.length > 0) {
            return this.__activeSorters.map((criteria) => {
                return {
                    name: criteria.for,
                    priority: criteria.priority,
                    direction: criteria.direction,
                };
            });
        }
        else {
            return null;
        }
    }
    get __table() {
        return this.shadowRoot?.querySelector('vl-table');
    }
    get __tableHeader() {
        return this.__table?.querySelector('thead');
    }
    get __tableHeaderRow() {
        const header = this.__tableHeader;
        if (header) {
            return header.querySelector('tr');
        }
        else {
            return undefined;
        }
    }
    get __tableBody() {
        return this.__table?.querySelector('tbody');
    }
    get _isMultisortingEnabled() {
        return this.getAttribute('multi-sort') !== null;
    }
    // @ts-expect-error: TS2611: _sorting is defined as a property in class VlRichData,
    // but is overridden here in VlRichDataTable as an accessor.
    set _sorting(sorting) {
        if (sorting) {
            if (this.__sorters)
                this.__sorters.forEach((sorter) => {
                    const matchedSorter = sorting.find((sort) => sort.name === sorter.for);
                    sorter.direction = matchedSorter ? matchedSorter.direction : undefined;
                    sorter.priority = matchedSorter ? matchedSorter.priority : undefined;
                });
        }
    }
    get _hasResults() {
        return Boolean(this._data);
    }
    connectedCallback() {
        super.connectedCallback();
        this._render();
        this.__observeFields();
    }
    attributeChangedCallback(attr, oldValue, newValue) {
        super.attributeChangedCallback(attr, oldValue, newValue);
        if (VlRichDataTable_1._tableAttributes.includes(attr)) {
            this.__table?.toggleAttribute(attr);
        }
    }
    __getState({ paging }) {
        const state = super.__getState({ paging });
        state.sorting = this.__sortingState;
        return state;
    }
    _validate(data) {
        if (data) {
            if (!Array.isArray(data)) {
                throw new Error('vl-rich-data-table verwacht een Array als data');
            }
        }
    }
    _render() {
        this._renderHeaders();
        this._renderBody();
    }
    _renderHeaders() {
        const { __tableHeaderRow, __richDataFields, __addHeaderColumn } = this;
        if (__tableHeaderRow)
            __tableHeaderRow.innerHTML = '';
        const headerColumns = __richDataFields.map((field) => field.headerTemplate());
        const atLeastOneHeaderColumnHasContent = headerColumns.some((header) => !!header.textContent);
        if (atLeastOneHeaderColumnHasContent) {
            headerColumns.forEach(__addHeaderColumn.bind(this));
            this.__showHeader();
        }
        else {
            this.__hideHeader();
        }
    }
    __addHeaderColumn(header) {
        this.__initializeSortingOnHeaderColumn(header);
        if (this.__tableHeaderRow)
            this.__tableHeaderRow.appendChild(header);
    }
    __hideHeader() {
        if (this.__tableHeader)
            this.__tableHeader.setAttribute('hidden', '');
    }
    __showHeader() {
        if (this.__tableHeader)
            this.__tableHeader.removeAttribute('hidden');
    }
    __initializeSortingOnHeaderColumn(header) {
        const sorterButton = header.querySelector('th[sortable] > a');
        if (sorterButton) {
            sorterButton.addEventListener('click', () => {
                sorterButton.querySelector('vl-rich-data-sorter').nextDirection();
            });
        }
    }
    _renderBody() {
        const { __tableBody, data } = this;
        if (data && data.data && __tableBody) {
            __tableBody.innerHTML = '';
            this.data.data.forEach((rowData) => {
                const rowTemplate = this._template(`<tr></tr>`).firstElementChild;
                this.__richDataFields.map((field) => {
                    rowTemplate?.appendChild(field.valueTemplate(rowData));
                });
                __tableBody.appendChild(rowTemplate);
            });
        }
    }
    _dataChangedCallback(oldValue, newValue) {
        this.data = JSON.parse(newValue);
    }
    __listenToFieldChanges(field) {
        field.addEventListener('change', this.__fieldChanged.bind(this));
    }
    __stopListeningToFieldChanges(field) {
        field.removeEventListener('change', this.__fieldChanged.bind(this));
    }
    __listenToSortChanges(sorter) {
        sorter.addEventListener('change', this.__sortingChanged.bind(this));
    }
    __stopListeningToSortChanges(sorter) {
        sorter.removeEventListener('change', this.__sortingChanged.bind(this));
    }
    __fieldChanged(event) {
        const propertiesChanged = event.detail.properties;
        if (propertiesChanged) {
            if (propertiesChanged.some((property) => VlRichDataField.headerAttributes.includes(property))) {
                this._renderHeaders();
            }
            if (propertiesChanged.some((property) => VlRichDataField.bodyAttributes.includes(property))) {
                this._renderBody();
            }
        }
    }
    __sortingChanged(event) {
        if (this._isMultisortingEnabled) {
            // eslint-disable-next-line @typescript-eslint/no-unused-expressions
            this.__activeSorters &&
                this.__activeSorters.forEach((sorter, index) => (sorter.priority = index + 1));
        }
        else {
            // eslint-disable-next-line @typescript-eslint/no-unused-expressions
            this.__activeSorters &&
                this.__activeSorters
                    .filter((sorter) => sorter !== event.target)
                    .forEach((sorter) => (sorter.direction = undefined));
        }
        this.__onStateChange(event);
    }
    __createObserver(doWhenNodeIsAdded, doWhenNodeIsRemoved, render) {
        return new MutationObserver((mutationsList) => {
            let shouldRender = false;
            mutationsList.forEach((mutation) => {
                if (mutation.addedNodes || mutation.removedNodes) {
                    shouldRender = true;
                    if (mutation.addedNodes) {
                        mutation.addedNodes.forEach(doWhenNodeIsAdded);
                    }
                    if (mutation.removedNodes) {
                        mutation.removedNodes.forEach(doWhenNodeIsRemoved);
                    }
                }
            });
            if (render && shouldRender) {
                this._render();
            }
        });
    }
    __observeFields() {
        this.__fields.forEach(this.__listenToFieldChanges.bind(this));
        const observer = this.__createObserver(this.__listenToFieldChanges.bind(this), this.__stopListeningToFieldChanges.bind(this), true);
        observer.observe(this, { childList: true });
    }
    __observeSorters() {
        const nodeToSorter = (doWithSorter) => {
            return (node) => {
                const sorter = node.querySelector(VlRichDataSorter.is);
                if (sorter) {
                    doWithSorter(sorter);
                }
            };
        };
        if (this.__tableHeaderRow) {
            this.__createObserver(nodeToSorter((sorter) => this.__listenToSortChanges(sorter)), nodeToSorter((sorter) => this.__stopListeningToSortChanges(sorter)), false).observe(this.__tableHeaderRow, { childList: true });
        }
    }
};
(() => {
    registerWebComponents([VlTableComponent]);
})();
VlRichDataTable = VlRichDataTable_1 = __decorate([
    webComponentPromised([customElements.whenDefined(VlRichDataField.is), customElements.whenDefined(VlRichDataSorter.is)], 'vl-rich-data-table'),
    __metadata("design:paramtypes", [])
], VlRichDataTable);
export { VlRichDataTable };
