// @domg-wc/components/block/rich-data-table/vl-rich-data-sorter.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var VlRichDataSorter_1;
import { BaseHTMLElement, registerWebComponents, webComponent } from '@domg-wc/common';
import { vlLegacyStyles } from '@domg-wc/styles';
import { VlIconComponent } from '../../atom/icon';
import { vlRichDataTableFluxStyles } from './vl-rich-data-table.flux-css';
let VlRichDataSorter = VlRichDataSorter_1 = class VlRichDataSorter extends BaseHTMLElement {
    constructor() {
        const html = `
            <div id="container" class="vl-u-visually-hidden">
                <vl-icon id="direction"></vl-icon>
                <label id="priority"></label>
            </div>
        `;
        const customStyleSheet = new CSSStyleSheet();
        customStyleSheet.replaceSync(`
            div {
              display: inline;
            }

            #direction {
              vertical-align: middle;
            }

            #priority {
              font-size: x-small;
              vertical-align: super;
            }
        `);
        const styleSheets = [
            ...vlLegacyStyles.map((style) => style.styleSheet),
            vlRichDataTableFluxStyles.styleSheet,
            customStyleSheet,
        ];
        super(html, styleSheets);
    }
    static get DIRECTIONS() {
        return {
            descending: 'desc',
            ascending: 'asc',
        };
    }
    static get _observedAttributes() {
        return ['direction', 'priority'];
    }
    static get is() {
        return 'vl-rich-data-sorter';
    }
    static get PRIORITY_COMPARATOR() {
        return ({ priority: firstPriority }, { priority: secondPriority }) => {
            if (secondPriority === undefined || (firstPriority ? firstPriority : 0) < secondPriority) {
                return -1;
            }
            else if (firstPriority === undefined || firstPriority > secondPriority) {
                return 1;
            }
            else {
                return 0;
            }
        };
    }
    get for() {
        return this.getAttribute('for');
    }
    get direction() {
        return this.__direction;
    }
    set direction(direction) {
        if (this.__direction !== direction) {
            this.__direction = direction;
            const directionIcon = this._directionIcon;
            if (directionIcon) {
                this.__directionElement?.setAttribute('icon', directionIcon);
                this.__containerElement?.classList.remove('vl-u-visually-hidden');
            }
            else {
                this.__containerElement?.classList.add('vl-u-visually-hidden');
            }
            if (direction === undefined) {
                this.priority = undefined;
            }
        }
    }
    get _directionIcon() {
        switch (this.direction) {
            case VlRichDataSorter_1.DIRECTIONS.ascending:
                return 'arrow-down';
            case VlRichDataSorter_1.DIRECTIONS.descending:
                return 'arrow-up';
            default:
                return null;
        }
    }
    get priority() {
        return this.__priority;
    }
    set priority(priority) {
        if (this.__priority !== priority) {
            this.__priority = Number(priority) || undefined;
            this.__priorityElement.textContent = String(this.priority);
        }
    }
    get __directionElement() {
        return this.shadowRoot?.querySelector('#direction');
    }
    get __containerElement() {
        return this.shadowRoot?.querySelector('#container');
    }
    get __priorityElement() {
        return this.shadowRoot?.querySelector('#priority');
    }
    _setDirectionAndPublishEvent(direction) {
        if (this.direction !== direction) {
            this.direction = direction;
            this._changed();
        }
    }
    nextDirection() {
        switch (this.direction) {
            case VlRichDataSorter_1.DIRECTIONS.descending:
                this._setDirectionAndPublishEvent(undefined);
                break;
            case VlRichDataSorter_1.DIRECTIONS.ascending:
                this._setDirectionAndPublishEvent(VlRichDataSorter_1.DIRECTIONS.descending);
                break;
            default:
                this._setDirectionAndPublishEvent(VlRichDataSorter_1.DIRECTIONS.ascending);
        }
    }
    _directionChangedCallback(oldValue, newValue) {
        this.direction = newValue;
    }
    _priorityChangedCallback(oldValue, newValue) {
        this.priority = newValue;
    }
    _changed() {
        this.dispatchEvent(new CustomEvent('change', {
            detail: {
                direction: this.direction,
                priority: this.priority,
            },
        }));
    }
};
(() => {
    registerWebComponents([VlIconComponent]);
})();
VlRichDataSorter = VlRichDataSorter_1 = __decorate([
    webComponent('vl-rich-data-sorter'),
    __metadata("design:paramtypes", [])
], VlRichDataSorter);
export { VlRichDataSorter };
