// @domg-wc/components/form/datepicker/vl-datepicker.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { isSafari, webComponent } from '@domg-wc/common';
import { vlGroupStyles } from '@domg-wc/styles';
import { baseStyle, resetStyle } from '@domg/govflanders-style/common';
import { datepickerStyle, iconStyle, tooltipStyle } from '@domg/govflanders-style/component';
import Cleave from 'cleave.js';
import flatpickr from 'flatpickr';
import Dutch from 'flatpickr/dist/l10n/nl.js';
import { html, nothing } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import { live } from 'lit/directives/live.js';
import { vlInputAddonStyles } from '../../atom/button';
import { FormControl } from '../form-control';
import { inputFieldStyles } from '../input-field';
import { createDateMask, createTimeMask } from './masks';
import { maskValidator, rangeOverflowValidator, rangeUnderflowValidator } from './validators';
import { datepickerDefaults } from './vl-datepicker.defaults';
import { vlDatepickerFluxStyles } from './vl-datepicker.flux-css';
const dateRangeSeparator = ' tot en met ';
const dateRangeSeparatorCharacter = '/';
let VlDatepickerComponent = class VlDatepickerComponent extends FormControl {
    constructor() {
        super(...arguments);
        // Properties
        this.regex = datepickerDefaults.regex; // Wordt enkel gebruikt in de pattern validator
        // Attributes
        this.type = datepickerDefaults.type;
        this.minDate = datepickerDefaults.minDate;
        this.maxDate = datepickerDefaults.maxDate;
        this.minTime = datepickerDefaults.minTime;
        this.maxTime = datepickerDefaults.maxTime;
        this.format = datepickerDefaults.format;
        this.block = datepickerDefaults.block;
        this.readonly = datepickerDefaults.readonly;
        this.value = datepickerDefaults.value;
        this.placeholder = datepickerDefaults.placeholder;
        this.autocomplete = datepickerDefaults.autocomplete;
        this.amPm = datepickerDefaults.amPm;
        this.disableMaskValidation = datepickerDefaults.disableMaskValidation; // Wordt enkel gebruikt in de mask validator
        this.pattern = datepickerDefaults.pattern; // Wordt enkel gebruikt in de mask validator
        this.disableMobileNativeInput = datepickerDefaults.disableMobileNativeInput;
        this.position = datepickerDefaults.position;
        this.isStatic = datepickerDefaults.isStatic;
        // Variables
        this.initialValue = '';
        this.inputHasFocus = false;
        this.flatpickrInstance = null;
        this.maskOptions = null; // Wordt enkel gebruikt in de mask validator
        this.cleaveInstance = null;
        this.inputValue = ''; // Houdt de waarde van het getoonde inputveld bij
        this.dispatchInput = false;
        this.handleOpenChange = (isOpen) => {
            this.isOpen = isOpen;
            if (isOpen) {
                this.addAccessibilityAttributes();
            }
        };
        this.toggleCalendar = () => {
            this.calculateCalendarPlaceholderPosition();
            this.flatpickrInstance?.toggle();
        };
        this.onInputFocus = () => {
            this.inputHasFocus = true;
        };
        this.onInputBlur = () => {
            this.inputHasFocus = false;
        };
        this.onInput = (event) => {
            this.handleInputValueChanged(event.target?.value ?? '');
        };
        this.parseRangeISODateString = (isoDateString) => isoDateString.split(dateRangeSeparatorCharacter).map((dateString) => flatpickr.parseDate(dateString, 'Y-m-d'));
        this.formatInputValue = (date) => flatpickr.formatDate(date, this.format);
        this.formatRangeInputValue = (dates) => dates.map((date) => this.formatInputValue(date)).join(Dutch?.nl?.rangeSeparator || dateRangeSeparator);
        this.handleDatePickerChange = (dates) => {
            if (dates.length === 1) {
                this.inputValue = this.formatInputValue(dates[0]);
                this.value = this.getISODateString(dates[0]);
            }
            else if (dates.length === 2) {
                this.value = this.getISODateString(dates[0], dates[1]);
                this.inputValue = this.formatRangeInputValue(dates);
            }
        };
        this.handleOpenCalendar = () => {
            this.handleOpenChange(true);
            this.addEventListener('click', this.handleCalendarClicked);
        };
        this.handleCloseCalendar = () => {
            this.handleOpenChange(false);
            this.removeEventListener('click', this.handleCalendarClicked);
        };
        this.updateFormControlValue = (inputValue) => {
            const detail = { value: this.value };
            const date = this.flatpickrInstance?.parseDate(inputValue, this.format);
            // indien de waarde getypt wordt, updaten we de flatpickr instance zodat deze de nieuwe waarde toont
            if (this.inputValue !== this.flatpickrInstance?.input.value && date) {
                this.flatpickrInstance?.setDate(date, false, this.format);
            }
            this.setValue(this.value ?? '');
            this.dispatchEvent(new CustomEvent('vl-change', { composed: true, bubbles: true, detail }));
            if (this.dispatchInput) {
                this.dispatchEvent(new CustomEvent('vl-input', { composed: true, bubbles: true, detail }));
                this.dispatchInput = false;
            }
            this.dispatchEventIfValid(detail);
        };
    }
    static get styles() {
        return [
            resetStyle,
            baseStyle,
            iconStyle,
            inputFieldStyles,
            tooltipStyle,
            datepickerStyle,
            vlDatepickerFluxStyles,
            vlGroupStyles,
            vlInputAddonStyles,
        ];
    }
    static get properties() {
        return {
            block: { type: Boolean },
            readonly: { type: Boolean },
            value: { type: String }, // Bevat de waarde van het component
            placeholder: { type: String },
            autocomplete: { type: String },
            type: { type: String },
            format: { type: String },
            amPm: { type: Boolean, attribute: 'am-pm' },
            minDate: { type: String, attribute: 'min-date' },
            maxDate: { type: String, attribute: 'max-date' },
            minTime: { type: String, attribute: 'min-time' },
            maxTime: { type: String, attribute: 'max-time' },
            pattern: { type: String },
            disableMaskValidation: { type: Boolean, attribute: 'disable-mask-validation' },
            disableMobileNativeInput: { type: Boolean, attribute: 'disable-mobile-native-input' },
            rawValue: { type: Boolean, attribute: 'raw-value' },
            inputValue: { type: String, state: true }, // Houdt de waarde van het getoonde inputveld bij
            isOpen: { type: Boolean, state: true },
            position: { type: String },
            isStatic: { type: Boolean, attribute: 'static' },
        };
    }
    get validationTarget() {
        return this.shadowRoot?.querySelector('input');
    }
    connectedCallback() {
        super.connectedCallback();
        if (Dutch?.nl) {
            Dutch.nl = {
                ...Dutch.nl,
                rangeSeparator: dateRangeSeparator,
                yearAriaLabel: 'Jaar',
                monthAriaLabel: 'Maand',
                hourAriaLabel: 'Uur',
                minuteAriaLabel: 'Minuut',
            };
            flatpickr.l10ns.default.rangeSeparator = dateRangeSeparator;
        }
        if (!this.initialValue && typeof this.value === 'string') {
            this.initialValue = this.value;
        }
    }
    firstUpdated(changedProperties) {
        super.firstUpdated(changedProperties);
        // passen formaat aan indien niet opgegeven
        if (changedProperties.has('type') && !this.format) {
            const timeFormat = 'H:i';
            const dateFormat = 'd.m.Y';
            switch (this.type) {
                case 'date':
                    this.format = dateFormat;
                    break;
                case 'time':
                    this.format = timeFormat;
                    break;
                case 'date-time':
                    this.format = `${dateFormat} ${timeFormat}`;
                    break;
                default:
                    this.format = dateFormat;
            }
        }
        this.maskOptions = this.composeMaskForFormat(this.format, this.type);
        if (this.maskOptions && !this.disableMaskValidation) {
            this.cleaveInstance = new Cleave(this.validationTarget, this.maskOptions);
        }
        this.initializeComponent();
        this.setInitialValue();
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        const options = this.getDynamicOptions();
        const dynamicAttributes = ['disabled', 'readonly', 'minDate', 'maxDate', 'minTime', 'maxTime', 'position'];
        const nativeInputAttributes = ['disabled', 'readonly', 'placeholder', 'required', 'block', 'error', 'success'];
        if (dynamicAttributes.some((prop) => changedProperties.has(prop))) {
            this.updateOptionsForInstance(options);
        }
        const changedNativeAttributes = nativeInputAttributes.filter((prop) => changedProperties.has(prop));
        if (changedNativeAttributes.length) {
            this.updateOptionsForNativeInput(changedNativeAttributes, nativeInputAttributes);
        }
        if (changedProperties.has('value')) {
            // alleen als de inputValue niet aangepast is, updaten we deze met de nieuwe waarde
            // dit gebeurt typisch wanneer de waarde van buitenaf aangepast wordt
            if (!changedProperties.has('inputValue') && typeof this.value === 'string') {
                switch (this.type) {
                    case 'date-time':
                    case 'date': {
                        const date = flatpickr.parseDate(this.value, 'Z');
                        if (date) {
                            this.inputValue = flatpickr.formatDate(date, this.format);
                        }
                        else if (!date && !this.value) {
                            this.flatpickrInstance?.clear();
                            this.inputValue = '';
                        }
                        break;
                    }
                    case 'time': {
                        const date = flatpickr.parseDate(this.value, this.format);
                        if (date) {
                            this.inputValue = flatpickr.formatDate(date, this.format);
                        }
                        else if (!date && !this.value) {
                            this.flatpickrInstance?.clear();
                            this.inputValue = '';
                        }
                        break;
                    }
                    default:
                        this.inputValue = this.value;
                }
            }
        }
        if (changedProperties.has('inputValue')) {
            this.updateFormControlValue(this.inputValue ?? '');
        }
        if (changedProperties.has('block')) {
            if (this.block) {
                this.getFlatpickrWrapper()?.classList.add('flatpickr-wrapper--block');
            }
            else {
                this.getFlatpickrWrapper()?.classList.remove('flatpickr-wrapper--block');
            }
        }
        if (this.flatpickrInstance?.isMobile && !this.disableMobileNativeInput) {
            this.getNativeDateInput()?.classList.add('js-vl-datepicker-input', 'vl-input-field', 'flatpickr-input', 'flatpickr-mobile');
            this.getNativeDateInput()?.classList.remove('vl-input-group');
        }
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.flatpickrInstance?.destroy();
    }
    render() {
        const inputClasses = {
            'vl-input-field': true,
            'vl-input-field--input-group-left': true,
            'js-vl-datepicker-toggle': true,
            'vl-input-field--error': this.error || this.isInvalid,
            'vl-input-field--success': this.success,
            'vl-input-field--block': this.block,
            'vl-input-field--disabled': this.disabled,
        };
        const buttonClasses = {
            'vl-input-addon': true,
            'js-vl-datepicker-toggle': true,
            'vl-input-addon--error': this.error || this.isInvalid,
            'vl-input-addon--success': this.success,
            'vl-input-addon--disabled': this.disabled,
        };
        return html `
            <div class="vl-group vl-group--input-group" id="datepicker-wrapper">
                ${!(this.flatpickrInstance?.isMobile && !this.disableMobileNativeInput)
            ? html `
                          <input
                              id=${this.id || nothing}
                              name=${this.name || nothing}
                              class=${classMap(inputClasses)}
                              type="text"
                              aria-label=${this.label || nothing}
                              aria-invalid=${this.isInvalid || nothing}
                              ?required=${this.required}
                              ?disabled=${this.disabled}
                              ?error=${this.error}
                              ?readonly=${this.readonly}
                              .value=${live(this.inputValue)}
                              placeholder=${this.placeholder || nothing}
                              autocomplete=${this.autocomplete || nothing}
                              pattern=${this.pattern || nothing}
                              inputmode=${this.cleaveInstance ? 'numeric' : nothing}
                              @focus="${this.onInputFocus}"
                              @blur="${this.onInputBlur}"
                              @input=${!this.cleaveInstance ? this.onInput : nothing}
                          />
                          <button
                              id="toggle-calendar"
                              type="button"
                              class=${classMap(buttonClasses)}
                              ?disabled=${this.disabled || this.readonly}
                              aria-label="datumkiezer${this.label ? ` ${this.label}` : ''}"
                              aria-expanded=${this.isOpen}
                              aria-controls=${this.id || nothing}
                              @click=${this.toggleCalendar}
                          >
                              <span
                                  class="vl-icon vl-icon--small vl-vi vl-vi-${this.type === 'time'
                ? 'clock'
                : 'calendar'}"
                                  aria-hidden="true"
                              ></span>
                          </button>
                      `
            : nothing}
            </div>
            <div id="datepicker-calendar-placeholder"></div>
        `;
    }
    resetFormControl() {
        super.resetFormControl();
        this.setInitialValue();
    }
    getRawValue() {
        return this.cleaveInstance?.getRawValue();
    }
    getDates() {
        return this.flatpickrInstance?.selectedDates;
    }
    setInitialValue() {
        this.value = this.initialValue;
        if (this.type === 'range' && this.initialValue) {
            const dates = this.parseRangeISODateString(this.initialValue);
            if (dates.length === 1) {
                this.inputValue = this.formatInputValue(dates[0]);
                this.flatpickrInstance?.setDate(dates[0], true);
            }
            else if (dates.length === 2) {
                this.inputValue = this.formatRangeInputValue(dates);
                this.flatpickrInstance?.setDate(dates, true);
            }
            return;
        }
        const initialDate = this.flatpickrInstance?.parseDate(this.initialValue, 'Z');
        if (initialDate instanceof Date && !isNaN(initialDate) && this.type !== 'range') {
            this.flatpickrInstance?.setDate(initialDate, true);
            this.inputValue = flatpickr.formatDate(initialDate, this.format);
            this.dispatchInput = false;
        }
        else if (this.type === 'time' && this.initialValue) {
            this.inputValue = this.initialValue;
        }
        else {
            this.flatpickrInstance?.clear();
            this.inputValue = '';
        }
    }
    parseTodayDate(dateString) {
        const formatDate = (date) => flatpickr.formatDate(date, this.format);
        if (dateString === 'today') {
            return formatDate(new Date());
        }
        else if (dateString) {
            const parsedDate = flatpickr.parseDate(dateString, 'Z');
            return parsedDate ? formatDate(parsedDate) : undefined;
        }
        else {
            return undefined;
        }
    }
    getDynamicOptions() {
        const minimumDateTime = flatpickr.parseDate(this.minTime, this.format);
        return {
            allowInput: this.inputHasFocus && !(this.disabled || this.readonly),
            maxDate: this.maxDate,
            minDate: this.minDate,
            minTime: this.minTime,
            maxTime: this.maxTime,
            defaultHour: minimumDateTime?.getHours() ?? 12,
            defaultMinute: minimumDateTime?.getMinutes() ?? 0,
            position: this.position || 'auto',
        };
    }
    addAccessibilityAttributes() {
        const calendar = this.shadowRoot?.querySelector('.flatpickr-calendar');
        calendar?.querySelectorAll('.flatpickr-day').forEach((day) => {
            // extend aria-label with the day of the week
            const dateString = day.getAttribute('aria-label');
            if (dateString) {
                const dateObj = new Date(dateString);
                const dayOfWeek = dateObj.toLocaleDateString('nl-NL', { weekday: 'long' });
                day.setAttribute('aria-label', `${dateString}, ${dayOfWeek}`);
                day.setAttribute('role', `button`);
            }
        });
        calendar?.querySelectorAll('.flatpickr-prev-month, .flatpickr-month')?.forEach((button) => {
            button.setAttribute('aria-label', button.classList.contains('flatpickr-prev-month') ? 'Vorige maand' : 'Volgende maand');
            button.setAttribute('role', 'button');
            button.querySelector('svg')?.setAttribute('aria-hidden', 'true');
        });
        calendar?.querySelector('.flatpickr-weekdays')?.setAttribute('aria-hidden', 'true');
    }
    getOptions() {
        const datepickerButton = this.shadowRoot?.querySelector('button');
        const defaultDate = this.type !== 'range' && this.parseTodayDate(this.initialValue);
        const staticOptions = {
            dateFormat: this.format,
            locale: Dutch.nl,
            clickOpens: false,
            onChange: this.handleDatePickerChange,
            onOpen: this.handleOpenCalendar,
            onClose: this.handleCloseCalendar,
            positionElement: datepickerButton,
            static: this.isStatic,
            appendTo: this.getCalendarPlaceholder(),
            defaultDate: defaultDate,
            enableTime: this.type === 'time' || this.type === 'date-time',
            noCalendar: this.type === 'time',
            time_24hr: !this.amPm,
            mode: this.type !== 'range' ? 'single' : 'range',
            disableMobile: this.disableMobileNativeInput,
        };
        const options = {
            ...staticOptions,
            ...this.getDynamicOptions(),
        };
        Object.keys(options).forEach((key) => {
            if (options[key] === undefined)
                delete options[key];
        });
        return options;
    }
    getDatePicker() {
        return this.shadowRoot?.querySelector('#datepicker-wrapper');
    }
    getFlatpickrWrapper() {
        return this.shadowRoot?.querySelector('.flatpickr-wrapper');
    }
    getNativeDateInput() {
        return this.renderRoot?.querySelector('input[type="date"]');
    }
    getCalendarPlaceholder() {
        return this.shadowRoot?.querySelector('#datepicker-calendar-placeholder');
    }
    updateOptionsForInstance(options) {
        Object.keys(options)
            .map((key) => key)
            .forEach((key) => {
            this.flatpickrInstance?.set(key, options[key]);
        });
    }
    updateOptionsForNativeInput(changedNativeAttributes, nativeInputAttributes) {
        nativeInputAttributes.forEach((attribute) => {
            if (changedNativeAttributes.includes(attribute) && this.getNativeDateInput()) {
                this.updateInputForAttribute(attribute, this.getNativeDateInput());
            }
        });
    }
    updateInputForAttribute(attribute, inputElement) {
        const attributeKey = attribute;
        if (this[attributeKey]) {
            inputElement.setAttribute(attribute, typeof this[attributeKey] === 'boolean' ? '' : this[attributeKey]);
            inputElement.classList.add(`vl-input-field--${attributeKey}`);
        }
        else {
            inputElement.removeAttribute(attribute);
            inputElement.classList.remove(`vl-input-field--${attributeKey}`);
        }
    }
    initializeComponent() {
        if (this.getDatePicker() && !this.flatpickrInstance) {
            this.flatpickrInstance = flatpickr(this.getDatePicker(), this.getOptions());
            this.getDatePicker()?.classList.add('static');
            this.getDatePicker()?.removeAttribute('readonly');
            if (!this.isStatic) {
                this.calculateCalendarPlaceholderPosition();
            }
        }
    }
    calculateCalendarPlaceholderPosition() {
        if (this.getDatePicker()) {
            const { top, left, height } = this.getDatePicker().getBoundingClientRect();
            const calendarPlaceholder = this.getCalendarPlaceholder();
            calendarPlaceholder.style.top = isSafari ? `calc(-${top}px - ${height}px)` : `-${top}px`;
            calendarPlaceholder.style.left = `-${left}px`;
        }
    }
    handleCalendarClicked() {
        this.dispatchInput = true;
    }
    getISODateString(date, date2) {
        if (!date)
            return '';
        switch (this.type) {
            case 'time':
                return flatpickr.formatDate(date, this.format?.includes('S') ? 'H:i:S' : 'H:i');
            case 'date-time':
                return flatpickr.formatDate(date, 'Y-m-dTH:i');
            case 'range':
                return `${flatpickr.formatDate(date, 'Y-m-d')}${date2 ? dateRangeSeparatorCharacter + flatpickr.formatDate(date2, 'Y-m-d') : ''}`;
            case 'date':
            default:
                return flatpickr.formatDate(date, 'Y-m-d');
        }
    }
    // functie die de start- en einddatum zal extraheren uit de geformatteerde invoerwaarde met scheidingsteken (tot en met)
    getDatesFromInputValue(inputValue) {
        const rangeSeparator = Dutch?.nl?.rangeSeparator;
        const dateStrings = (rangeSeparator && inputValue.split(rangeSeparator)) || [];
        const dates = dateStrings?.map((dateString) => flatpickr.parseDate(dateString, this.format));
        return dates.length && dates.every((date) => date instanceof Date) ? dates : undefined;
    }
    handleInputValueChanged(dateString, isValidDateString = true) {
        this.dispatchInput = true;
        let parsedDate, isValidFormat;
        // we verwachten een fout die opgeworpen wordt wanneer de waarde niet conform het formaat is
        // daarom gebruiken we een try-catch
        try {
            // indien een RegExp opgegeven is, controleren we of de waarde conform het formaat is
            const patternRegExp = this.pattern ? new RegExp(this.pattern) : undefined;
            const regex = this.regex || patternRegExp;
            isValidFormat = regex ? regex.test(dateString) : true;
            if (isValidFormat)
                parsedDate = flatpickr.parseDate(dateString, this.format);
            // eslint-disable-line @typescript-eslint/no-unused-vars
        }
        catch (error) {
            // we vangen de error op, maar behandelen we die niet en gaan verder met de default waarde
        }
        finally {
            if (isValidDateString && isValidFormat && parsedDate && this.type !== 'range') {
                this.value = this.getISODateString(parsedDate);
            }
            else if (this.type === 'range') {
                const dates = this.getDatesFromInputValue(dateString);
                if (dates?.length) {
                    this.value = this.getISODateString(dates[0], dates[1]);
                }
                else {
                    this.value = dateString;
                }
                this.inputValue = dateString;
            }
            else {
                // indien de waarde niet conform het verwachte formaat is of niet kon geconverteerd worden
                // stellen we de value gelijk aan de inputValue
                this.inputValue = dateString;
                this.value = dateString;
            }
            if (parsedDate instanceof Date && !isNaN(parsedDate)) {
                this.flatpickrInstance?.setDate(dateString, false, this.format);
            }
        }
    }
    composeMaskForFormat(format, type) {
        if (!format)
            return null;
        let maskOptions = null;
        switch (type) {
            case 'date':
                maskOptions = createDateMask(format, this.minDate, this.maxDate);
                break;
            case 'time':
                maskOptions = createTimeMask(format);
                break;
            default:
                break;
        }
        return maskOptions
            ? {
                ...maskOptions,
                // als we een cleave mask gebruiken, willen we de transformed value verkrijgen
                onValueChanged: ({ target: { value, rawValue }, }) => {
                    // flatpickr gaat soms een string parsen naar een Date, ook al is de string niet conform het verwachte formaat
                    // daarom voegen we hier nog een extra check toe om te voorkomen dat we een ongeldige datum in de flatpickr instance instellen
                    const isValidDateString = maskOptions?.regex?.test(rawValue);
                    this.handleInputValueChanged(value ?? '', isValidDateString);
                },
            }
            : null;
    }
};
VlDatepickerComponent.formControlValidators = [
    ...FormControl.formControlValidators,
    maskValidator,
    rangeOverflowValidator,
    rangeUnderflowValidator,
];
VlDatepickerComponent = __decorate([
    webComponent('vl-datepicker')
], VlDatepickerComponent);
export { VlDatepickerComponent };
