// @domg-wc/components/block/video-player/vl-video-player.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, webComponent } from '@domg-wc/common';
import { html } from 'lit';
import { PlyrLayout, VidstackPlayer } from 'vidstack/global/player';
import videoPlayerStyles from './vl-video-player.css';
import { plyrTranslations } from './vl-video-player.translations';
let VlVideoPlayerComponent = class VlVideoPlayerComponent extends BaseLitElement {
    static get styles() {
        return [...videoPlayerStyles];
    }
    static get properties() {
        return {
            title: { type: String },
            poster: { type: String },
            source: { type: String },
            subtitles: { type: String },
        };
    }
    async firstUpdated(changedProperties) {
        super.firstUpdated(changedProperties);
        const subtitleTrack = {
            src: this.subtitles,
            label: 'Nederlands',
            language: 'nl-BE',
            kind: 'subtitles',
            default: true,
        };
        const video = this.shadowRoot?.querySelector('video');
        if (!video) {
            return;
        }
        this.playerInstance = await VidstackPlayer.create({
            target: video,
            title: this.title,
            src: this.source,
            poster: this.poster,
            layout: new PlyrLayout({ translations: plyrTranslations, clickToPlay: true }),
        });
        this.playerInstance?.textTracks.add(subtitleTrack);
    }
    render() {
        return html `
            <div>
                <video controls crossorigin playsinline></video>
            </div>
        `;
    }
};
VlVideoPlayerComponent = __decorate([
    webComponent('vl-video-player')
], VlVideoPlayerComponent);
export { VlVideoPlayerComponent };
