// @domg-wc/components/block/cascader/vl-cascader-item.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { vlLayoutStyles } from '@domg-wc/styles';
import { html } from 'lit';
import { BaseLitElement, findNodesForSlot } from '@domg-wc/common';
import { customElement } from 'lit/decorators.js';
import { resetStyle } from '@domg/govflanders-style/common';
import { vlCascaderItemFluxStyles } from './vl-cascader-item.flux-css';
import { CASCADER_MESSAGES, CASCADER_SLOTS } from './vl-cascader.model';
import { defaultItemActionTemplate, getDefaultItemTemplate, getTemplateFunctionForType } from './vl-cascader.utils';
let VlCascaderItemComponent = class VlCascaderItemComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.item = { label: CASCADER_MESSAGES.LABEL_MISSING };
        this.label = CASCADER_MESSAGES.LABEL_MISSING;
    }
    static get properties() {
        return {
            label: { type: String },
            templateType: { type: String, attribute: 'template-type' },
            item: { type: Object },
            annotation: { type: String },
        };
    }
    static get styles() {
        return [resetStyle, vlLayoutStyles, vlCascaderItemFluxStyles];
    }
    connectedCallback() {
        super.connectedCallback();
        this.updateItemData();
    }
    updateItemData() {
        if (this.label != null) {
            this.item.label = this.label;
            this.item.templateType = this.templateType;
        }
        this.item.annotation = this.annotation;
    }
    render() {
        const cascaderRef = this.cascaderRef;
        const hasLabelSlot = Boolean(findNodesForSlot(this, CASCADER_SLOTS.LABEL)?.length);
        const { templateType } = this.item;
        let templateResultForNode;
        if (templateType && cascaderRef) {
            templateResultForNode = getTemplateFunctionForType(this.item, cascaderRef);
        }
        else if (cascaderRef) {
            templateResultForNode = getDefaultItemTemplate(this.item, cascaderRef, hasLabelSlot);
        }
        return cascaderRef && templateResultForNode
            ? html `${templateResultForNode} ${html `<slot name="content"></<slot>`}`
            : html `
                  ${!hasLabelSlot ? defaultItemActionTemplate(this.item) : html `<slot name="label"></slot>`}
                  <slot name="content"></slot>
              `;
    }
};
VlCascaderItemComponent = __decorate([
    customElement('vl-cascader-item')
], VlCascaderItemComponent);
export { VlCascaderItemComponent };
