// @domg-wc/components/block/cascader/vl-cascader.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, findNodesForSlot, registerWebComponents } from '@domg-wc/common';
import { resetStyle } from '@domg/govflanders-style/common';
import { breadcrumbStyle } from '@domg/govflanders-style/component';
import { html, nothing } from 'lit';
import { customElement } from 'lit/decorators.js';
import { classMap } from 'lit/directives/class-map.js';
import { VlIconComponent } from '../../atom/icon';
import { VlLinkComponent } from '../../atom/link';
import { vlTitleStyles } from '../../atom/title/vl-title.css';
import { VlCascaderItemComponent } from './vl-cascader-item.component';
import { cascaderDefaults } from './vl-cascader.defaults';
import { CASCADER_SLOTS } from './vl-cascader.model';
import { vlCascaderFluxStyles } from './vl-cascader.flux-css';
import { getDefaultItemTemplate, getTemplateFunctionForType } from './vl-cascader.utils';
import { vlLayoutStyles } from '@domg-wc/styles';
let VlCascaderComponent = class VlCascaderComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        // Attributes
        this.hideBreadcrumb = cascaderDefaults.hideBreadcrumb;
        this.level = cascaderDefaults.level;
        this.loadingMessage = cascaderDefaults.loadingMessage;
        this.headerText = cascaderDefaults.headerText;
        this.loading = cascaderDefaults.loading;
        // State
        this.nodeData = [];
        // Variables
        this.navigationLevelStack = []; // Stack to keep track of the navigation hierarchy history
        this.breadCrumbHistory = [];
        this.slidingIn = false;
        this.slidingOut = false;
        /**
         * Deze functie zal het volgende niveau laden op basis van het CascaderItem argument
         * @param item: CascaderItem
         */
        this.processNarrowDown = async (item) => {
            if (this.loading)
                return;
            const { label, children, narrowDown } = item;
            if (children?.length) {
                this.pushItemStack(label, this.nodeData);
                this.nodeData = children;
                this.slidingIn = true;
                this.requestUpdate();
            }
            else if (narrowDown && this.itemListFn) {
                this.loading = true;
                this.pushItemStack(label, this.nodeData);
                this.requestUpdate();
                this.nodeData = await this.itemListFn(item);
                this.loading = false;
                this.slidingIn = true;
                this.requestUpdate();
            }
        };
        this.handleBreadcrumbClick = (index, label) => {
            this.jumpToLevel(index);
            const detail = { levelClicked: index, label };
            this.dispatchEvent(new CustomEvent('vl-click-breadcrumb', { composed: true, bubbles: true, detail }));
        };
        this.renderBreadcrumbHome = () => {
            const hasBreadcrumbPlaceholderSlot = Boolean(findNodesForSlot(this, CASCADER_SLOTS.HOME)?.length);
            return html `
            <li class="vl-breadcrumb__list__item">
                <span class="vl-breadcrumb__list__item__separator" aria-hidden="true"></span>
                ${!hasBreadcrumbPlaceholderSlot
                ? html `
                              <vl-icon
                                  icon="places-home"
                                  class="vl-breadcrumb__list__item__cta"
                                  @click=${() => this.handleBreadcrumbClick(0)}
                              ></vl-icon>
                          `
                : html `
                              <span
                                  @click=${() => this.handleBreadcrumbClick(0)}
                                  class="vl-breadcrumb__list__item__cta vl-breadcrumb-home-slot"
                              >
                                  <slot name="home"></slot>
                              </span>
                          `}
                </span>
            </li>
        `;
        };
        this.renderBreadcrumbItem = ({ label, index }) => {
            const historyLength = this.breadCrumbHistory?.length;
            const isActiveLevel = historyLength === index;
            const breadCrumbItemClasses = {
                'vl-breadcrumb__list__item--deactivated': isActiveLevel,
            };
            return html `
            <li class="vl-breadcrumb__list__item ${classMap(breadCrumbItemClasses)}">
                <span class="vl-breadcrumb__list__item__separator" aria-hidden="true"></span>
                <span class="vl-breadcrumb__list__item__cta" @click=${() => this.handleBreadcrumbClick(index, label)}
                    >${label}</span
                >
            </li>
        `;
        };
        this.renderItem = (item) => {
            const customTemplateResult = getTemplateFunctionForType(item, this);
            if (item.component) {
                // declaratieve gedefinieerde templates van afnemers renderen (via nested <vl-cascader-item>'s)
                return item.component;
            }
            else if (customTemplateResult) {
                // programmatisch gedefinieerde templates van afnemers renderen (via templates:Map<string, TemplateResult>)
                return customTemplateResult;
            }
            else {
                return this.defaultItemTemplate(item);
            }
        };
        this.getChildCascaderNodes = (component) => component.querySelectorAll(':scope > vl-cascader-item');
        /**
         * Gaat de DOM-tree recursief aflopen binnen de VlCascaderComponent,
         * om zo de hiërarchische data structuur van de VlCascaderItemComponent als een array van CascaderItem's op te slaan
         * @param component: VlCascaderItemComponent | VlCascaderComponent
         * @returns CascaderItem[]
         */
        this.traverseTreeAndMapItems = (component) => {
            const nodes = this.getChildCascaderNodes(component);
            const nodesForThisLevel = [];
            nodes.forEach((node) => {
                const children = this.traverseTreeAndMapItems(node);
                if (node.item) {
                    node.item.children = children;
                }
                node.cascaderRef = this;
                nodesForThisLevel.push({
                    ...node.item,
                    label: node.label,
                    component: node,
                    children: children,
                });
            });
            return nodesForThisLevel;
        };
    }
    static get properties() {
        return {
            headerText: { type: String, attribute: 'header-text' },
            level: { type: Number, reflect: true },
            hideBreadcrumb: { type: Boolean, attribute: 'hide-breadcrumb' },
            loadingMessage: { type: String, attribute: 'loading-message' },
            loading: { type: Boolean, reflect: true },
            itemListFn: { type: Function },
            templates: { type: Map },
            nodeData: { type: Array, state: true },
        };
    }
    static get styles() {
        return [resetStyle, breadcrumbStyle, vlCascaderFluxStyles, vlTitleStyles, vlLayoutStyles];
    }
    get items() {
        return this.nodeData;
    }
    set items(cascaderItems) {
        this.nodeData = cascaderItems;
    }
    connectedCallback() {
        super.connectedCallback();
        this.setData();
    }
    isDeclarativeMode() {
        return !this.items || !this.items.length;
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        if (changedProperties.has('level')) {
            this.jumpToLevel(this.level);
        }
        else if (changedProperties.has('items')) {
            this.nodeData = this.items;
        }
    }
    render() {
        const navSectionClasses = {
            'slide-in': this.slidingIn && !this.slidingOut,
            'slide-out': this.slidingOut && !this.slidingIn,
        };
        return html `
            <div>
                ${this.renderBreadcrumb()} ${this.renderHeader()}
                <div class="content">
                    <section class=${classMap(navSectionClasses)} @animationend=${this.handleAnimationEnd}>
                        ${!this.loading
            ? this.nodeData?.map(this.renderItem)
            : html ` <vl-loader text=${this.loadingMessage}></vl-loader> `}
                    </section>
                </div>
            </div>
        `;
    }
    setData() {
        const nodeTree = this.traverseTreeAndMapItems(this);
        if (this.isDeclarativeMode() && nodeTree?.length) {
            this.nodeData = nodeTree;
        }
        else {
            this.nodeData = this.items;
        }
    }
    /**
     * Update de navigationLevelStack en breadCrumbHistory
     * Wanneer aangeroepen zonder argumenten, gaan we terug in de geschiedenis
     * @param label?: string
     * @param items?: CascaderItem[]
     * @private
     */
    pushItemStack(label, items) {
        if (label && items) {
            this.level = this.navigationLevelStack.push(items);
            this.breadCrumbHistory.push({ label, index: this.level });
        }
    }
    /**
     * Update de navigationLevelStack en breadCrumbHistory
     * Wanneer aangeroepen zonder argumenten, gaan we terug in de geschiedenis
     * @private
     */
    popItemStack() {
        this.nodeData = this.navigationLevelStack.pop() || [];
        this.level = (this.breadCrumbHistory.pop()?.index || 1) - 1;
    }
    goBack() {
        if (this.loading)
            return;
        if (this.navigationLevelStack.length > 0) {
            this.slidingOut = true;
            this.popItemStack();
        }
        this.requestUpdate();
    }
    jumpToLevel(index) {
        if (this.loading)
            return;
        const length = this.navigationLevelStack.length;
        const regressionNumber = length - index;
        if (regressionNumber > 0) {
            Array.from({ length: regressionNumber }).map(() => {
                this.goBack();
            });
        }
    }
    renderHeader() {
        return findNodesForSlot(this, 'header')?.length
            ? html `
                  <header>
                      <slot name="header"></slot>
                  </header>
              `
            : this.headerText
                ? html `
                  <header class="vl-header">
                      <h4 class="vl-header__title vl-label vl-label--h4">${this.headerText}</h4>
                  </header>
              `
                : nothing;
    }
    renderBreadcrumb() {
        const historyLength = this.breadCrumbHistory?.length;
        const hasBreadcrumbPlaceholderSlot = Boolean(findNodesForSlot(this, CASCADER_SLOTS.BREADCRUMB_PLACEHOLDER)?.length);
        return html `
            ${historyLength && !this.hideBreadcrumb
            ? html `
                      <nav aria-label="U bent hier: " class="vl-breadcrumb">
                          <ol class="vl-breadcrumb__list">
                              ${this.renderBreadcrumbHome()} ${this.breadCrumbHistory?.map(this.renderBreadcrumbItem)}
                          </ol>
                      </nav>
                  `
            : hasBreadcrumbPlaceholderSlot || !this.hideBreadcrumb
                ? html `
                      <nav class="vl-breadcrumb-placeholder">
                          <slot name="breadcrumb-placeholder"></slot>
                      </nav>
                  `
                : nothing}
        `;
    }
    defaultItemTemplate(item) {
        return getDefaultItemTemplate(item, this);
    }
    handleAnimationEnd() {
        this.slidingIn = false;
        this.slidingOut = false;
        this.requestUpdate();
    }
};
(() => {
    registerWebComponents([VlCascaderItemComponent, VlIconComponent, VlLinkComponent]);
})();
VlCascaderComponent = __decorate([
    customElement('vl-cascader')
], VlCascaderComponent);
export { VlCascaderComponent };
