// @domg/govflanders-style/component/loader/vl-loader.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-loader {
  display: inline-block;
  position: relative;
  width: 3rem;
  height: 1.8rem;
}
.vl-loader::before {
  animation: loader-wave infinite 1s linear;
  position: absolute;
  display: block;
  content: "";
  width: 0.4rem;
  height: 0.4rem;
  box-shadow: 1rem 0 #687483, 2rem 0 #687483, 3rem 0 #687483;
  border-radius: 50%;
  margin-top: -0.2rem;
  margin-left: -2.2rem;
  top: 50%;
  left: 50%;
}
@keyframes loader-wave {
  0% {
    box-shadow: 10px 0px #687483, 20px 0px #687483, 30px 0px #687483;
  }
  10% {
    box-shadow: 10px -3px #687483, 20px 0px #687483, 30px 0px #687483;
  }
  20% {
    box-shadow: 10px -6px #687483, 20px -3px #687483, 30px 0px #687483;
  }
  30% {
    box-shadow: 10px -3px #687483, 20px -6px #687483, 30px -3px #687483;
  }
  40% {
    box-shadow: 10px 0px #687483, 20px -3px #687483, 30px -6px #687483;
  }
  50% {
    box-shadow: 10px 0px #687483, 20px 0px #687483, 30px -3px #687483;
  }
  60% {
    box-shadow: 10px 0px #687483, 20px 0px #687483, 30px 0px #687483;
  }
  100% {
    box-shadow: 10px 0px #687483, 20px 0px #687483, 30px 0px #687483;
  }
}
.vl-loader--light::before {
  animation: loader-light-wave infinite 1s linear;
}
@keyframes loader-light-wave {
  0% {
    box-shadow: 10px 0px #fff, 20px 0px #fff, 30px 0px #fff;
  }
  10% {
    box-shadow: 10px -3px #fff, 20px 0px #fff, 30px 0px #fff;
  }
  20% {
    box-shadow: 10px -6px #fff, 20px -3px #fff, 30px 0px #fff;
  }
  30% {
    box-shadow: 10px -3px #fff, 20px -6px #fff, 30px -3px #fff;
  }
  40% {
    box-shadow: 10px 0px #fff, 20px -3px #fff, 30px -6px #fff;
  }
  50% {
    box-shadow: 10px 0px #fff, 20px 0px #fff, 30px -3px #fff;
  }
  60% {
    box-shadow: 10px 0px #fff, 20px 0px #fff, 30px 0px #fff;
  }
  100% {
    box-shadow: 10px 0px #fff, 20px 0px #fff, 30px 0px #fff;
  }
}
.vl-loader--small::before {
  width: 0.3rem;
  height: 0.3rem;
}
.vl-loader + p {
  font-style: 1.6rem;
}

.vl-loader-alert--inline {
  display: inline-flex;
  vertical-align: middle;
}
.vl-loader-alert .vl-loader::before {
  margin-top: 0;
}
`;
export default style;
