// @domg-wc/components/compliance/next/footer/vl-footer.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { awaitScript, BaseLitElement, legacyBreakpoint, legacyCore, registerWebComponents, webComponent, } from '@domg-wc/common';
registerWebComponents([legacyCore, legacyBreakpoint]);
let VlFooterNext = class VlFooterNext extends BaseLitElement {
    static get properties() {
        return {
            development: {
                type: Boolean,
                attribute: 'development',
                reflect: true,
            },
            identifier: {
                type: String,
                attribute: 'identifier',
                reflect: true,
            },
        };
    }
    constructor() {
        super();
        this.development = false;
        this.identifier = '';
        this.allowCustomCSS = false;
    }
    get footerContainer() {
        return document.querySelector('#footer__container');
    }
    injectFooterContainer() {
        const vlBody = document.querySelector('body');
        if (this.footerContainer)
            return;
        (vlBody || document.body).insertAdjacentHTML('beforeend', '<footer id="footer__container"><div id="footer"></div></footer>');
    }
    onReady() {
        this.dispatchEvent(new CustomEvent('ready'));
    }
    async loadWidget() {
        try {
            window.addEventListener('widget.global_footer.mounted', this.onReady);
            await awaitScript('vl-footer-widget', `https://widgets.${this.development ? 'tni-' : ''}vlaanderen.be/api/v2/widget/${this.identifier}/entry`);
            if (!window.globalFooterClient) {
                console.error('De global footer client werd niet geladen.');
                return;
            }
            const footerElement = this.footerContainer?.querySelector('#footer') || undefined;
            await window.globalFooterClient.mount(footerElement);
        }
        catch (error) {
            console.error('De global footer werd niet geladen.', error);
        }
    }
    render() {
        this.footerContainer?.remove();
        this.injectFooterContainer();
        this.loadWidget();
    }
    createRenderRoot() {
        return this;
    }
    disconnectedCallback() {
        window.removeEventListener('widget.global_footer.mounted', this.onReady);
    }
};
VlFooterNext = __decorate([
    webComponent('vl-footer-next'),
    __metadata("design:paramtypes", [])
], VlFooterNext);
export { VlFooterNext };
