// @domg-wc/map/components/action/layer-action/select-action/select-actions/vl-map-select-actions.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { webComponent } from '@domg-wc/common';
import { VlSelectActions } from '../../../../../actions/select/select-actions';
import { VlMapLayerStyle } from '../../../../layer-style/vl-map-layer-style';
import { VlMapSelectAction } from '../vl-map-select-action';
let VlMapSelectActions = class VlMapSelectActions extends VlMapSelectAction {
    set layers(layers) {
        this.layerNames = layers;
    }
    get style() {
        return this._style;
    }
    set style(style) {
        if (style instanceof VlMapLayerStyle) {
            this._style = style.style;
        }
        else {
            this._style = style;
        }
        this._processAction(this.olLayers);
    }
    connectedCallback() {
        super.connectedCallback();
        const olLayers = this.layerNames?.map((name) => {
            return this._mapElement.querySelector(`[is-layer][name="${name}"]`)?._layer;
        });
        this.olLayers = olLayers;
        this._processAction(this.olLayers);
    }
    _createAction(layers) {
        const options = {
            style: this.style,
            cluster: this._cluster !== undefined,
            filter: this.appliesTo.bind(this),
        };
        return new VlSelectActions(layers, this._callback, options);
    }
};
VlMapSelectActions = __decorate([
    webComponent('vl-map-select-actions')
], VlMapSelectActions);
export { VlMapSelectActions };
