// @domg-wc/map/components/search/vl-map-search.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var VlMapSearch_1;
import { BaseHTMLElement, registerWebComponents, webComponent } from '@domg-wc/common';
import { VlSearchComponent } from '@domg-wc/components/block';
import { SelectRichPosition } from '@domg-wc/components/form';
import { vlLegacyStyles } from '@domg-wc/styles';
import OlOverlay from 'ol/Overlay';
import { VlMap } from '../../vl-map';
import { VlSelectLocationComponent } from '../select-location/vl-select-location';
let VlMapSearch = VlMapSearch_1 = class VlMapSearch extends BaseHTMLElement {
    static get _observedAttributes() {
        return ['placeholder', 'search-placeholder', 'search-empty-text', 'search-no-results-text'];
    }
    static get _observedClassAttributes() {
        return ['with-offset'];
    }
    get _classPrefix() {
        return 'vl-map-search--';
    }
    constructor() {
        const html = `
            <vl-search id="search" inline>
                <vl-select-location slot="input" not-deletable position=${SelectRichPosition.BOTTOM}></vl-select-location>
            </vl-search>
        `;
        const customStyleSheet = new CSSStyleSheet();
        customStyleSheet.replaceSync(`
            :host {
              display: block;
            }

            ::part(vl-select-rich__combobox) {
                background-color: white;
            }

            ::part(vl-select-rich__combobox)::before {
                display: none;
            }

            vl-search {
                display: block;
                height: 3.5rem;
            }
        `);
        const styleSheets = [...vlLegacyStyles.map((style) => style.styleSheet), customStyleSheet];
        super(html, styleSheets);
        this.changeLocation = async () => {
            try {
                const location = await this._selectElement.location;
                if (location) {
                    if (this._onSelect) {
                        this._onSelect(location);
                    }
                    else {
                        this.zoomTo(location);
                    }
                }
            }
            catch (error) {
                console.error('Locatie kan niet opgehaald worden.', error);
            }
        };
        this.stopPropagation = (e) => {
            e.stopPropagation();
        };
        this.configure();
    }
    connectedCallback() {
        this.addEventListener('vl-input', this.changeLocation);
        this.addEventListener('keypress', this.stopPropagation);
        if (this.map?.isLambert2008) {
            this._selectElement.setAttribute('lambert2008', '');
        }
    }
    get _selectElement() {
        return this._shadow?.querySelector('vl-select-location');
    }
    bindMap(map) {
        this._map = map;
        if (map.isLambert2008) {
            this._selectElement.setAttribute('lambert2008', '');
        }
    }
    get map() {
        return this._map || (this.parentNode && this.parentNode instanceof VlMap ? this.parentNode : null);
    }
    /**
     * Bepaal callback die uitgevoerd wordt bij selectie van een locatie via de map search.
     *
     * @param {Function} callback
     */
    onSelect(callback) {
        this._onSelect = callback;
    }
    zoomTo(boundingBox) {
        this._map.zoomTo(boundingBox, 14);
    }
    configure() {
        customElements.whenDefined('vl-map').then(() => {
            // @ts-expect-error: The parentNode is expected to have a map property
            if (this.parentNode && this.parentNode.map) {
                // @ts-expect-error: The parentNode is expected to have a shadow property
                this._map = this.parentNode?._shadow?.host;
                this._map.map.addOverlay(new OlOverlay({
                    className: 'vl-map-search__overlaycontainer',
                    element: this,
                }));
            }
        });
    }
    _placeholderChangedCallback(oldValue, newValue) {
        this._dispatchSelectAttribute('placeholder', newValue);
    }
    _searchPlaceholderChangedCallback(oldValue, newValue) {
        this._dispatchSelectAttribute('search-placeholder', newValue);
    }
    _searchEmptyTextChangedCallback(oldValue, newValue) {
        this._dispatchSelectAttribute('search-empty-text', newValue);
    }
    _searchNoResultsTextChangedCallback(oldValue, newValue) {
        this._dispatchSelectAttribute('search-no-results-text', newValue);
    }
    _dispatchSelectAttribute(attribute, value) {
        if (value != undefined) {
            this._selectElement.setAttribute(`${VlMapSearch_1.attributePrefix}${attribute}`, value);
        }
        else {
            this._selectElement.removeAttribute(`${VlMapSearch_1.attributePrefix}${attribute}`);
        }
    }
    disconnectedCallback() {
        this.removeEventListener('keypress', this.stopPropagation);
        this._selectElement.removeEventListener('vl-input', this.changeLocation);
    }
};
(() => {
    registerWebComponents([VlSelectLocationComponent, VlSearchComponent]);
})();
VlMapSearch = VlMapSearch_1 = __decorate([
    webComponent('vl-map-search'),
    __metadata("design:paramtypes", [])
], VlMapSearch);
export { VlMapSearch };
