// @domg/govflanders-style/component/description-data/vl-description-data.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-description-data__label, .vl-description-data__subdata {
  display: block;
  font-size: 1.6rem;
  font-weight: 400;
  color: #687483;
  margin-bottom: 0.75rem;
}
@media screen and (max-width: 767px) {
  .vl-description-data__label, .vl-description-data__subdata {
    font-size: 1.4rem;
  }
}
.vl-description-data__label {
  display: inline-flex;
  gap: 0.5rem;
}
.vl-description-data__label__icon, .vl-description-data__label__icon.vl-icon {
  position: absolute;
  font-size: 2.2rem;
  color: #687483;
  top: -0.4rem;
  left: 0;
}
@media screen and (max-width: 767px) {
  .vl-description-data__label__icon, .vl-description-data__label__icon.vl-icon {
    font-size: 1.8rem;
    top: -0.2rem;
  }
}
.vl-description-data__label--icon {
  position: relative;
  padding: 0 0 0 4rem;
}
.vl-description-data__label--icon + .vl-description-data__value {
  padding: 0 0 0 4rem;
}
@media screen and (max-width: 767px) {
  .vl-description-data__label--icon {
    padding: 0 0 0 3rem;
  }
  .vl-description-data__label--icon + .vl-description-data__value {
    padding: 0 0 0 3rem;
  }
}
.vl-description-data__label--icon--top {
  padding: 5rem 0 0;
}
.vl-description-data__label--icon--top .vl-description-data__label__icon {
  font-size: 3.2rem;
}
.vl-description-data__label--icon--top + .vl-description-data__value {
  padding: 0;
}
.vl-description-data__label + .vl-description-data__value > .vl-description-data__subdata {
  font-size: inherit;
  color: inherit;
  margin-top: 0;
}
.vl-description-data__subdata {
  margin-top: 0.5rem;
  margin-bottom: 0;
}
.vl-description-data__value, .vl-description-data__data {
  display: block;
  font-size: 1.8rem;
  font-weight: 500;
  color: #333332;
}
@media screen and (max-width: 767px) {
  .vl-description-data__value, .vl-description-data__data {
    font-size: 1.6rem;
  }
}
.vl-description-data--bordered {
  min-height: 6px;
  border: 0;
  background: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 6.04 5.99'%3E%3Cdefs%3E%3Cstyle%3E.cls-1%7Bfill:%23bec5cf;%7D%3C/style%3E%3C/defs%3E%3Ctitle%3EAsset 1%3C/title%3E%3Cg id='Layer_2' data-name='Layer 2'%3E%3Cg id='Layer_1-2' data-name='Layer 1'%3E%3Crect class='cls-1' x='1.01' y='3.99' width='1.01' height='1'/%3E%3Crect class='cls-1' y='4.99' width='1.01' height='1'/%3E%3Crect class='cls-1' x='3.02' y='2' width='1.01' height='1'/%3E%3Crect class='cls-1' x='2.01' y='2.99' width='1.01' height='1'/%3E%3Crect class='cls-1' x='5.04' width='1.01' height='1'/%3E%3Crect class='cls-1' x='4.03' y='1' width='1.01' height='1'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E") repeat-x;
  background-size: 6px 6px;
  padding: 3rem 0;
}
.vl-description-data--bordered::after {
  content: "";
  width: 100%;
  display: block;
  min-height: 6px;
  border: 0;
  background: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 6.04 5.99'%3E%3Cdefs%3E%3Cstyle%3E.cls-1%7Bfill:%23bec5cf;%7D%3C/style%3E%3C/defs%3E%3Ctitle%3EAsset 1%3C/title%3E%3Cg id='Layer_2' data-name='Layer 2'%3E%3Cg id='Layer_1-2' data-name='Layer 1'%3E%3Crect class='cls-1' x='1.01' y='3.99' width='1.01' height='1'/%3E%3Crect class='cls-1' y='4.99' width='1.01' height='1'/%3E%3Crect class='cls-1' x='3.02' y='2' width='1.01' height='1'/%3E%3Crect class='cls-1' x='2.01' y='2.99' width='1.01' height='1'/%3E%3Crect class='cls-1' x='5.04' width='1.01' height='1'/%3E%3Crect class='cls-1' x='4.03' y='1' width='1.01' height='1'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E") repeat-x;
  background-size: 6px 6px;
  height: 3rem;
  background-position: bottom center;
}
`;
export default style;
