// @domg-wc/map/components/layer-switcher/vl-map-layer-switcher.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { awaitUntil, BaseLitElement, registerWebComponents } from '@domg-wc/common';
import { formLabelStyles, VlCheckboxComponent } from '@domg-wc/components/form';
import { vlLegacyStyles } from '@domg-wc/styles';
import { html } from 'lit';
import { customElement } from 'lit/decorators.js';
import { vlMapLayerSwitcherFluxStyles } from './vl-map-layer-switcher.flux-css';
let VlMapLayerSwitcher = class VlMapLayerSwitcher extends BaseLitElement {
    constructor() {
        super(...arguments);
        // Attributen
        this.componentTitle = 'Kaartlagen';
        this.layers = null;
        // Private properties
        this.vlMapLayers = [];
        this.mapElement = null;
        this.layerObserver = null;
        this.computeCheckboxAttributes = () => {
            const resolution = this.mapElement?.resolution;
            this.vlMapLayers.forEach((layer) => {
                const checkbox = this.shadowRoot?.querySelector(`vl-checkbox[layer="${layer.title}"]`);
                if (!layer.isVisibleAtResolution(resolution)) {
                    checkbox?.setAttribute('disabled', '');
                }
                else {
                    checkbox?.removeAttribute('disabled');
                }
                if (layer.visible) {
                    checkbox?.setAttribute('checked', '');
                }
                else {
                    checkbox?.removeAttribute('checked');
                }
            });
        };
    }
    static get styles() {
        return [vlLegacyStyles, vlMapLayerSwitcherFluxStyles, formLabelStyles];
    }
    static get properties() {
        return {
            // Gebruik 'componentTitle' omdat 'title' een native property is van HtmlElement.
            componentTitle: {
                type: String,
                attribute: 'title',
                reflect: true,
            },
            layers: {
                type: Array,
                attribute: 'layers',
            },
            vlMapLayers: {
                // Voeg 'vlMapLayers' toe om de Lit lifecycle te triggeren als deze verandert.
                attribute: false,
            },
        };
    }
    async connectedCallback() {
        super.connectedCallback();
        this.mapElement = this.closest('vl-map');
        await this.layersReady();
        this.vlMapLayers = this.getVlMapLayers();
        this.mapElement?.on('moveend', this.computeCheckboxAttributes);
        if (!this.layers) {
            this.observeMapLayers();
        }
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.layerObserver?.disconnect();
        this.mapElement?.un('moveend', this.computeCheckboxAttributes);
    }
    async willUpdate(changedProperties) {
        if (changedProperties.has('layers')) {
            await this.layersReady();
            this.vlMapLayers = this.getVlMapLayers();
        }
        if (changedProperties.has('vlMapLayers') && !!this.vlMapLayers) {
            this.vlMapLayers?.forEach(({ layer }) => {
                layer.on('change:visible', () => {
                    this.computeCheckboxAttributes();
                });
            });
        }
    }
    updated() {
        this.computeCheckboxAttributes();
    }
    render() {
        return html `
            <div>
                <label class="vl-form__label">${this.componentTitle}</label>
                ${this.vlMapLayers.map((layer) => html `
                        <vl-checkbox
                            label=${layer.title}
                            layer=${layer.title}
                            checked=${layer.visible}
                            @vl-input=${() => (layer.visible = !layer.visible)}
                            >${layer.title}
                        </vl-checkbox>
                    `)}
            </div>
        `;
    }
    async layersReady() {
        if (!this.mapElement) {
            return new Promise((_resolve, reject) => reject());
        }
        return Promise.all(this.mapElement?.nonBaseLayers.map((layer) => awaitUntil(() => layer.ready)));
    }
    getVlMapLayers() {
        if (!this.layers || !Array.isArray(this.layers)) {
            return this.mapElement?.nonBaseLayers || [];
        }
        return this.mapElement?.nonBaseLayers.filter((layer) => this.layers?.includes(layer.name)) || [];
    }
    observeMapLayers() {
        this.layerObserver = new MutationObserver((mutations) => {
            mutations.forEach((mutation) => {
                mutation.addedNodes.forEach((node) => {
                    if (node instanceof HTMLElement && node.hasAttribute('is-layer')) {
                        this.vlMapLayers = [...this.vlMapLayers, node];
                    }
                });
                mutation.removedNodes.forEach((node) => {
                    if (node instanceof HTMLElement && node.hasAttribute('is-layer')) {
                        this.vlMapLayers = this.vlMapLayers.filter((layer) => node !== layer);
                    }
                });
            });
        });
        this.layerObserver.observe(this.mapElement, { subtree: true, childList: true });
    }
};
(() => {
    registerWebComponents([VlCheckboxComponent]);
})();
VlMapLayerSwitcher = __decorate([
    customElement('vl-map-layer-switcher')
], VlMapLayerSwitcher);
export { VlMapLayerSwitcher };
