// @domg-wc/components/form/checkbox/vl-checkbox.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { webComponent } from '@domg-wc/common';
import { vlLegacyStyles } from '@domg-wc/styles';
import { baseStyle, resetStyle } from '@domg/govflanders-style/common';
import { checkboxStyle } from '@domg/govflanders-style/component';
import { html, nothing } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import { FormControl } from '../form-control';
import { vlCheckboxComponentFluxStyles } from './vl-checkbox.component.flux-css';
import { checkboxDefaults } from './vl-checkbox.defaults';
let VlCheckboxComponent = class VlCheckboxComponent extends FormControl {
    constructor() {
        super(...arguments);
        // Attributes
        this.block = checkboxDefaults.block;
        this.value = checkboxDefaults.value;
        this.checked = checkboxDefaults.checked;
        this.isSwitch = checkboxDefaults.isSwitch;
        this.indeterminate = checkboxDefaults.indeterminate;
        // Variables
        this.initialValue = null;
        this.initialCheckedValue = false;
        this.dispatchInput = false;
    }
    static get styles() {
        return [resetStyle, baseStyle, vlLegacyStyles, checkboxStyle, vlCheckboxComponentFluxStyles];
    }
    static get properties() {
        return {
            block: { type: Boolean },
            value: { type: String },
            checked: { type: Boolean, reflect: true },
            indeterminate: { type: Boolean, reflect: true },
            isSwitch: { type: Boolean, attribute: 'switch' },
        };
    }
    get validationTarget() {
        return this.shadowRoot?.querySelector('input');
    }
    connectedCallback() {
        super.connectedCallback();
        if (!this.initialValue) {
            this.initialValue = this.value;
            this.initialCheckedValue = this.checked;
        }
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        if (changedProperties.has('checked') || changedProperties.has('value')) {
            const value = this.checked ? this.value || 'on' : null;
            const detail = {
                checked: this.checked,
                currentTarget: this,
            };
            if (this.checked) {
                detail.value = value;
            }
            this.setValue(value);
            this.dispatchEvent(new CustomEvent('vl-change', { composed: true, bubbles: true, detail }));
            if (this.dispatchInput) {
                this.dispatchEvent(new CustomEvent('vl-input', { bubbles: true, composed: true, detail }));
                this.dispatchInput = false;
            }
            this.dispatchEventIfValid(detail);
        }
    }
    render() {
        return html ` ${!this.isSwitch ? this.renderCheckboxDefault() : this.renderCheckboxSwitch()} `;
    }
    resetFormControl() {
        super.resetFormControl();
        this.checked = this.initialCheckedValue;
        this.value = this.initialValue;
    }
    renderCheckboxDefault() {
        const classes = {
            'vl-checkbox': true,
            'vl-checkbox--disabled': this.disabled,
            'vl-checkbox--error': this.isInvalid || this.error,
            'vl-checkbox--success': this.success,
            'vl-checkbox--block': this.block,
        };
        return html `
            <label class=${classMap(classes)}>
                <input
                    id=${this.id || nothing}
                    name=${this.name || nothing}
                    class="vl-checkbox__toggle"
                    type="checkbox"
                    aria-label=${this.label || nothing}
                    aria-invalid=${this.isInvalid || nothing}
                    ?required=${this.required}
                    ?disabled=${this.disabled}
                    ?error=${this.error}
                    .value=${this.value}
                    .checked=${this.checked}
                    .indeterminate=${this.indeterminate && !this.checked}
                    @click=${this.toggle}
                />
                <div class="vl-checkbox__label">
                    <i class="vl-checkbox__box" aria-hidden="true"></i>
                    <span>
                        <slot></slot>
                    </span>
                </div>
            </label>
        `;
    }
    renderCheckboxSwitch() {
        const classes = {
            'vl-checkbox--switch__wrapper': true,
            'vl-checkbox--disabled': this.disabled,
            'vl-checkbox--error': this.isInvalid || this.error,
            'vl-checkbox--success': this.success,
            'vl-checkbox--block': this.block,
        };
        return html `
            <div class=${classMap(classes)}>
                <input
                    id=${this.id || nothing}
                    name=${this.name || nothing}
                    type="checkbox"
                    class="vl-checkbox--switch"
                    role="switch"
                    ?required=${this.required}
                    ?disabled=${this.disabled}
                    ?error=${this.error}
                    .value=${this.value}
                    .checked=${this.checked}
                    @click=${this.toggle}
                />
                <label for=${this.id} class="vl-checkbox__label">
                    <span class="vl-checkbox--switch__label">
                        <span aria-hidden="true"></span>
                    </span>
                    <span>
                        <slot></slot>
                    </span>
                </label>
            </div>
        `;
    }
    toggle() {
        this.checked = !this.checked;
        this.dispatchInput = true;
    }
};
VlCheckboxComponent = __decorate([
    webComponent('vl-checkbox')
], VlCheckboxComponent);
export { VlCheckboxComponent };
