// @domg-wc/components/block/doormat/vl-doormat.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, webComponent } from '@domg-wc/common';
import { doormatDefaults } from './vl-doormat.defaults';
import { html, nothing } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import doormatStyle from './vl-doormat.css';
import { vlLinkStyles } from '../../atom/link-style/vl-link-style.css';
let VlDoormatComponent = class VlDoormatComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.href = doormatDefaults.href;
        this.external = doormatDefaults.external;
        this.alt = doormatDefaults.alt;
        this.imageSrc = doormatDefaults.imageSrc;
        this.imageAlt = doormatDefaults.imageAlt;
        this.imageWidth = doormatDefaults.imageWidth;
        this.imageHeight = doormatDefaults.imageHeight;
        this.graphic = doormatDefaults.graphic;
    }
    static get styles() {
        return [vlLinkStyles(), doormatStyle];
    }
    static get properties() {
        return {
            href: { type: String },
            external: { type: Boolean },
            alt: { type: Boolean },
            imageSrc: { type: String, attribute: 'image-src' },
            imageAlt: { type: String, attribute: 'image-alt' },
            imageWidth: { type: Number, attribute: 'image-width' },
            imageHeight: { type: Number, attribute: 'image-height' },
            graphic: { type: Boolean },
        };
    }
    render() {
        const classes = {
            'vl-doormat': true,
            'vl-doormat--alt': this.alt,
            'vl-doormat--graphic': this.graphic,
        };
        const target = this.external ? '_blank' : nothing;
        return html `
            <a class=${classMap(classes)} href=${this.href} target=${target}>
                ${this.imageSrc ? this.renderImage() : nothing}
                <div class="vl-doormat__content">
                    <h2 class="vl-doormat__title">
                        <slot name="title"></slot>
                    </h2>
                    <div class="vl-doormat__text">
                        <slot name="text"></slot>
                    </div>
                </div>
            </a>
        `;
    }
    renderImage() {
        if (this.graphic) {
            return html `
                <div class="vl-doormat__graphic-wrapper">
                    <img
                        class="vl-doormat__graphic"
                        src=${this.imageSrc}
                        alt=${this.imageAlt || nothing}
                        width=${this.imageWidth || nothing}
                        height=${this.imageHeight || nothing}
                    />
                </div>
            `;
        }
        return html `
            <img
                class="vl-doormat__image"
                src=${this.imageSrc}
                alt=${this.imageAlt || nothing}
                width=${this.imageWidth || nothing}
                height=${this.imageHeight || nothing}
            />
        `;
    }
};
VlDoormatComponent = __decorate([
    webComponent('vl-doormat')
], VlDoormatComponent);
export { VlDoormatComponent };
