// @domg/govflanders-style/component/infotext/vl-infotext.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-infotext {
  display: block;
  font-family: "Flanders Art Sans", sans-serif;
  font-weight: 500;
  color: #687483;
  text-align: center;
  text-decoration: none;
}
@media screen and (max-width: 767px) {
  .vl-infotext:not(.vl-infotext--badge) .vl-infotext__value {
    font-size: 2.4rem;
  }
}
.vl-infotext--badge, .vl-infotext--accent {
  position: relative;
  display: block;
  background-color: #f7f9fc;
  border: 1px solid #8695a8;
  border-radius: 50%;
  padding-top: 100%;
}
.vl-infotext--badge .vl-infotext__value,
.vl-infotext--badge .vl-infotext__text, .vl-infotext--accent .vl-infotext__value,
.vl-infotext--accent .vl-infotext__text {
  position: absolute;
  text-align: center;
  line-height: 1;
  right: 0;
  left: 0;
}
.vl-infotext--badge .vl-infotext__value, .vl-infotext--accent .vl-infotext__value {
  padding: 0 1.1rem;
  bottom: 45%;
}
.vl-infotext--badge .vl-infotext__text, .vl-infotext--accent .vl-infotext__text {
  font-size: 1.3rem;
  font-weight: 400;
  padding: 1rem 1rem 0;
  top: 52%;
}
@media screen and (max-width: 1023px) {
  .vl-infotext--badge .vl-infotext__text, .vl-infotext--accent .vl-infotext__text {
    font-size: 1.2rem;
  }
}
.vl-infotext--accent {
  background-color: var(--vl-theme-primary-color);
  border: 1px solid var(--vl-theme-primary-color);
}
.vl-infotext-wrapper {
  max-width: 14rem;
}
.vl-infotext__value {
  font-size: 4.8rem;
  line-height: 1.5;
  white-space: nowrap;
}
.vl-infotext__text {
  font-size: 1.8rem;
}
@media screen and (max-width: 767px) {
  .vl-infotext__text {
    font-size: 1.4rem;
  }
}

a.vl-infotext {
  color: #05c;
}
a.vl-infotext:hover, a.vl-infotext:focus {
  color: #003bb0;
}
a.vl-infotext:hover .vl-infotext__text, a.vl-infotext:focus .vl-infotext__text {
  text-decoration: underline;
}
`;
export default style;
