// @domg-wc/map/components/select-location/vl-select-location.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { debounce, webComponent } from '@domg-wc/common';
import { VlSelectRichComponent } from '@domg-wc/components/form';
import LambertCoordinaat from '../../utils/lambert-coordinaat';
import { placeholder, searchPlaceholder } from './vl-select-location.defaults';
import { isCoord, isLocation } from './vl-select-location.model';
let VlSelectLocationComponent = class VlSelectLocationComponent extends VlSelectRichComponent {
    constructor() {
        super();
        this.url = 'https://geo.api.vlaanderen.be/geolocation';
        this.searchUrl = `${this.url}/Suggestion?q=`;
        this.locationUrl = `${this.url}/Location?q=`;
        this.locationXyUrl = `${this.url}/Location?c=5&xy=`;
        this.onSearch = (event) => {
            if (event[0]?.detail?.value) {
                const { value } = event[0].detail;
                const lambertCoordinaat = LambertCoordinaat.of(value);
                if (LambertCoordinaat.isLambertCoordinaat(lambertCoordinaat)) {
                    this.searchChoicesByLambertCoordinaat(lambertCoordinaat);
                }
                else {
                    this.searchChoicesByValue(value);
                }
            }
        };
        this.debouncedOnSearch = debounce(this.onSearch, 300);
        this.setChoices = (values) => this.choices?.setChoices(values, 'value', 'label', true);
        this.searchChoicesByLambertCoordinaat = (lambertCoordinaat) => fetch(`${this.locationXyUrl + lambertCoordinaat.x},${lambertCoordinaat.y}`)
            .then((response) => response.json())
            .then((data) => this.setChoices([this.mapLambertCoordinaatToChoice(lambertCoordinaat)].concat(this.mapLocationResultToChoices(data, lambertCoordinaat))));
        this.mapLambertCoordinaatToChoice = (lambertCoordinaat) => ({
            value: lambertCoordinaat,
            label: `Lambert-coördinaat: ${lambertCoordinaat.toString()}`,
        });
        this.mapLocationResultToChoices = (data, lambertCoordinaat) => data?.LocationResult?.map((locationResult) => ({
            value: lambertCoordinaat,
            label: locationResult.FormattedAddress,
        })) || [];
        this.searchChoicesByValue = (searchValue) => fetch(this.searchUrl + encodeURIComponent(searchValue))
            .then((response) => response.json())
            .then((data) => this.setChoices(this.mapSuggestionResultToChoices(data)));
        this.mapSuggestionResultToChoices = (data) => data?.SuggestionResult?.map((result) => ({
            value: result,
            label: result,
        })) || [];
        this.placeholder = placeholder;
        this.search = true;
        this.searchPlaceholder = searchPlaceholder;
    }
    /**
     * Geeft de bounding box op basis van de geselecteerde locatie.
     *
     * @return {Promise}
     */
    get location() {
        if (!this.choices)
            return undefined;
        const value = this.choices.getValue(true);
        if (!value)
            return undefined;
        if (isLocation(value)) {
            return Promise.resolve(this.isLambert2008
                ? [
                    value.BoundingBox.LowerLeft.X_Lambert2008,
                    value.BoundingBox.LowerLeft.Y_Lambert2008,
                    value.BoundingBox.UpperRight.X_Lambert2008,
                    value.BoundingBox.UpperRight.Y_Lambert2008,
                ]
                : [
                    value.BoundingBox.LowerLeft.X_Lambert72,
                    value.BoundingBox.LowerLeft.Y_Lambert72,
                    value.BoundingBox.UpperRight.X_Lambert72,
                    value.BoundingBox.UpperRight.Y_Lambert72,
                ]);
        }
        else if (isCoord(value) && LambertCoordinaat.isLambertCoordinaat(value)) {
            return Promise.resolve([value.x - 1, value.y - 1, value.x + 1, value.y + 1]);
        }
        else if (typeof value === 'string') {
            return fetch(this.locationUrl + encodeURIComponent(value))
                .then((response) => response.json())
                .then(({ LocationResult }) => this.isLambert2008
                ? [
                    LocationResult[0].BoundingBox.LowerLeft.X_Lambert2008,
                    LocationResult[0].BoundingBox.LowerLeft.Y_Lambert2008,
                    LocationResult[0].BoundingBox.UpperRight.X_Lambert2008,
                    LocationResult[0].BoundingBox.UpperRight.Y_Lambert2008,
                ]
                : [
                    LocationResult[0].BoundingBox.LowerLeft.X_Lambert72,
                    LocationResult[0].BoundingBox.LowerLeft.Y_Lambert72,
                    LocationResult[0].BoundingBox.UpperRight.X_Lambert72,
                    LocationResult[0].BoundingBox.UpperRight.Y_Lambert72,
                ]);
        }
        return undefined;
    }
    connectedCallback() {
        super.connectedCallback();
        this.addEventListener('vl-select-search', this.debouncedOnSearch);
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.removeEventListener('vl-select-search', this.debouncedOnSearch);
    }
    /**
     * Voorlopig is dit een opt-in attribuut. In de toekomst wordt Lambert 2008 de default.
     */
    get isLambert2008() {
        return this.hasAttribute('lambert2008');
    }
};
VlSelectLocationComponent = __decorate([
    webComponent('vl-select-location'),
    __metadata("design:paramtypes", [])
], VlSelectLocationComponent);
export { VlSelectLocationComponent };
