// @domg-wc/components/block/tooltip/vl-tooltip.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { html } from 'lit-html';
import { classMap } from 'lit-html/directives/class-map.js';
import { customElement } from 'lit/decorators.js';
import { VlPopoverComponent } from '../popover/vl-popover.component';
import { POPOVER_TYPE } from '../popover/vl-popover.model';
import { vlTooltipFluxStyles } from './vl-tooltip.flux-css';
import { TOOLTIP_TYPE } from './vl-tooltip.model';
let VlTooltipComponent = class VlTooltipComponent extends VlPopoverComponent {
    static get properties() {
        return {
            for: { type: String, attribute: 'for' },
            open: { type: Boolean, attribute: 'open', reflect: true },
            placement: { type: String, attribute: 'placement', reflect: true },
            distance: { type: Number, attribute: 'distance' },
            hideArrow: { type: Boolean, attribute: 'hide-arrow' },
            strategy: { type: String, attribute: 'strategy' },
            type: { type: String, attribute: 'type' },
        };
    }
    static get styles() {
        return [...super.styles, vlTooltipFluxStyles];
    }
    constructor() {
        super();
        this.type = TOOLTIP_TYPE.DESCRIPTION;
        this.trigger = 'hover focus';
    }
    render() {
        const classes = {
            'popover-content': true,
        };
        return html `
            <div class=${classMap(classes)} role="tooltip" aria-hidden="${!this.open}">
                <slot></slot>
                ${!this.hideArrow ? html `<i id="popover-arrow" part="arrow" role="presentation"></i>` : null}
            </div>
        `;
    }
    get popupOptions() {
        const options = super.popupOptions;
        return {
            ...options,
            trigger: 'hover focus',
            type: POPOVER_TYPE.TOOLTIP,
            hideOnClick: false,
            ariaType: this.type,
        };
    }
};
VlTooltipComponent = __decorate([
    customElement('vl-tooltip'),
    __metadata("design:paramtypes", [])
], VlTooltipComponent);
export { VlTooltipComponent };
