// @domg/govflanders-style/component/content-header/vl-content-header.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-content-header {
  background-color: var(--vl-theme-primary-color);
  min-width: 768px;
  overflow: hidden;
  max-height: 40rem;
  min-height: 10rem;
  position: relative;
  color: #333332;
}
@media screen and (max-width: 767px) {
  .vl-content-header > .vl-layout {
    position: static;
  }
}
.vl-content-header__logo-wrapper {
  display: flex;
  align-items: center;
}
.vl-content-header__logo-wrapper + * {
  margin-top: 0.4rem;
}
.vl-content-header__wrapper {
  overflow: hidden;
}
.vl-content-header--has-logo {
  background-color: #fff;
}
.vl-content-header--no-image {
  border-bottom: 1px solid #cbd2da;
}
@media screen and (max-width: 767px) {
  .vl-content-header {
    min-width: 0;
    max-height: none;
    display: none;
  }
  .vl-content-header.vl-content-header--show-mobile {
    display: block;
  }
}
@media screen and (max-width: 767px) {
  .vl-content-header:not(.vl-content-header--show-mobile) .vl-content-header__context {
    position: absolute;
    float: none;
    left: 0;
    max-width: 100%;
    bottom: 100%;
    margin-bottom: 1rem;
    padding: 0.3rem 1rem 0.4rem;
  }
}

.vl-content-header--alt .vl-content-header__context__content, .vl-content-header--alt .vl-content-header__context__link,
.vl-content-header--alt .vl-content-header__title__content,
.vl-content-header--alt .vl-content-header__title__link {
  color: #333332;
}
.vl-content-header--alt .vl-content-header__context__content:hover, .vl-content-header--alt .vl-content-header__context__content:focus, .vl-content-header--alt .vl-content-header__context__link:hover, .vl-content-header--alt .vl-content-header__context__link:focus,
.vl-content-header--alt .vl-content-header__title__content:hover,
.vl-content-header--alt .vl-content-header__title__content:focus,
.vl-content-header--alt .vl-content-header__title__link:hover,
.vl-content-header--alt .vl-content-header__title__link:focus {
  color: #333332;
  background: linear-gradient(-110deg, transparent 20px, var(--vl-theme-primary-color-70) 21px);
}
.vl-content-header--alt .vl-content-header__context__content,
.vl-content-header--alt .vl-content-header__title__content {
  background-color: var(--vl-theme-primary-color);
}
.vl-content-header--alt .vl-content-header__context__content:hover, .vl-content-header--alt .vl-content-header__context__content:focus,
.vl-content-header--alt .vl-content-header__title__content:hover,
.vl-content-header--alt .vl-content-header__title__content:focus {
  background: var(--vl-theme-primary-color-70);
}
.vl-content-header--alt.vl-content-header--has-context .vl-content-header__title__link {
  background: linear-gradient(70deg, transparent 20px, var(--vl-theme-primary-color) 21px);
}
.vl-content-header--alt.vl-content-header--has-context .vl-content-header__title__link:hover, .vl-content-header--alt.vl-content-header--has-context .vl-content-header__title__link:focus {
  background: linear-gradient(70deg, transparent 20px, var(--vl-theme-primary-color-70) 21px);
}

.vl-content-header__bg {
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  height: 100%;
  max-height: 40rem;
}
.vl-content-header__bg img {
  position: absolute;
  top: -9999px;
  bottom: -9999px;
  left: -9999px;
  right: -9999px;
  margin: auto;
  width: auto;
  height: auto;
  min-width: 100%;
  min-height: 100%;
  display: block;
  max-width: none;
}
@supports (object-fit: cover) {
  .vl-content-header__bg img {
    object-fit: cover;
    object-position: center;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
  }
}
.vl-content-header--half-image .vl-content-header__bg {
  position: absolute;
  left: 50%;
  right: -20%;
  transform: skew(20deg);
  overflow: hidden;
  transform-origin: 100% 50%;
}
.vl-content-header--half-image .vl-content-header__bg img {
  transform-origin: 100% 0%;
  transform: skew(-20deg);
}

.vl-content-header--large, .vl-content-header--small, .vl-content-header--xsmall {
  max-height: none;
}
.vl-content-header--large .vl-content-header__wrapper, .vl-content-header--small .vl-content-header__wrapper, .vl-content-header--xsmall .vl-content-header__wrapper {
  display: flex;
  flex-direction: row;
}
.vl-content-header--large .vl-content-header__wrapper > .vl-layout, .vl-content-header--small .vl-content-header__wrapper > .vl-layout, .vl-content-header--xsmall .vl-content-header__wrapper > .vl-layout {
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  flex: 1 1 auto;
}
.vl-content-header--large .vl-content-header__content, .vl-content-header--small .vl-content-header__content, .vl-content-header--xsmall .vl-content-header__content {
  position: relative;
  bottom: auto;
}
.vl-content-header--large .vl-content-header__content > *, .vl-content-header--small .vl-content-header__content > *, .vl-content-header--xsmall .vl-content-header__content > * {
  vertical-align: bottom;
}
.vl-content-header--small .vl-content-header__wrapper > .vl-layout {
  min-height: 15rem;
  padding-top: 4rem;
  padding-bottom: 4rem;
}
@media screen and (max-width: 1023px) {
  .vl-content-header--small .vl-content-header__wrapper > .vl-layout {
    min-height: 11rem;
    padding-bottom: 2rem;
    padding-top: 2rem;
  }
}
@media screen and (max-width: 767px) {
  .vl-content-header--small .vl-content-header__wrapper > .vl-layout {
    min-height: 8rem;
    padding-bottom: 1rem;
    padding-top: 1rem;
  }
}
.vl-content-header--large .vl-content-header__wrapper > .vl-layout {
  min-height: 31.5rem;
  padding-bottom: 6rem;
  padding-top: 6rem;
}
@media screen and (max-width: 1023px) {
  .vl-content-header--large .vl-content-header__wrapper > .vl-layout {
    min-height: 25.5rem;
    padding-bottom: 3rem;
    padding-top: 3rem;
  }
}
@media screen and (max-width: 767px) {
  .vl-content-header--large .vl-content-header__wrapper > .vl-layout {
    min-height: 11rem;
    padding-bottom: 2rem;
    padding-top: 2rem;
  }
}
.vl-content-header--xsmall {
  min-height: 9.5rem;
}
@media screen and (max-width: 767px) {
  .vl-content-header--xsmall {
    min-height: 5.5rem;
  }
}
.vl-content-header--xsmall .vl-content-header {
  min-height: 9.5rem;
}
.vl-content-header--xsmall .vl-content-header__wrapper > .vl-layout {
  min-height: 9.5rem;
  padding-bottom: 2.5rem;
  padding-top: 2.5rem;
}
@media screen and (max-width: 767px) {
  .vl-content-header--xsmall .vl-content-header__wrapper > .vl-layout {
    min-height: 5.5rem;
    padding-bottom: 1.5rem;
    padding-top: 1.5rem;
  }
}

.vl-content-header__logo {
  background-color: #fff;
  white-space: nowrap;
  position: relative;
  color: #333332;
  display: inline-block;
  vertical-align: middle;
}
.vl-content-header__logo__img {
  background-color: var(--vl-theme-primary-color);
  width: 8rem;
  height: 9rem;
  display: inline-block;
  vertical-align: middle;
  position: relative;
}
@media screen and (max-width: 767px) {
  .vl-content-header__logo__img {
    width: 4.5rem;
    height: 5rem;
  }
}
.vl-content-header__logo__img svg {
  width: 3.9rem;
  height: 6rem;
  margin-top: 1.5rem;
  margin-left: 1.5rem;
  fill: #333332;
}
@media screen and (max-width: 767px) {
  .vl-content-header__logo__img svg {
    width: 2.9rem;
    height: 3rem;
    margin-top: 1rem;
    margin-left: 0.3rem;
  }
}
.vl-content-header__logo__img::after {
  display: block;
  content: "";
  position: absolute;
  right: 0;
  bottom: 0;
  border-top: 9rem solid #fff;
  border-left: 3.0781812899rem solid transparent;
}
@media screen and (max-width: 767px) {
  .vl-content-header__logo__img::after {
    border-top: 5rem solid #fff;
    border-left: 1.7101007166rem solid transparent;
  }
}
.vl-content-header--no-image .vl-content-header__logo__img, .vl-content-header--half-image .vl-content-header__logo__img {
  background-color: transparent;
}
.vl-content-header--no-image .vl-content-header__logo__img::after, .vl-content-header--half-image .vl-content-header__logo__img::after {
  height: 100%;
  border: 0;
  width: 1px;
  background-color: #333332;
  transform: rotate(-20deg);
  transform-origin: 100% 80%;
}
.vl-content-header--no-image .vl-content-header__logo, .vl-content-header--half-image .vl-content-header__logo {
  background-color: transparent;
  color: #333332;
}
.vl-content-header__logo__text {
  display: inline-block;
  vertical-align: middle;
  position: relative;
  padding-right: 2rem;
  margin-left: -0.2rem;
  line-height: 1;
}
@media screen and (max-width: 767px) {
  .vl-content-header__logo__text {
    padding-right: 1rem;
  }
}
.vl-content-header__logo__title {
  font-weight: 500;
  font-size: 3rem;
  margin-bottom: 0.3rem;
}
@media screen and (max-width: 1023px) {
  .vl-content-header__logo__title {
    font-size: 2.4rem;
  }
}
@media screen and (max-width: 767px) {
  .vl-content-header__logo__title {
    font-size: 1.6rem;
  }
}
.vl-content-header__logo__tagline {
  display: block;
  font-size: 1.7rem;
  margin-left: 5px;
  font-weight: normal;
  margin-top: 2px;
}
@media screen and (max-width: 1023px) {
  .vl-content-header__logo__tagline {
    font-size: 1.35rem;
  }
}
@media screen and (max-width: 767px) {
  .vl-content-header__logo__tagline {
    margin-left: 3px;
    font-size: 1.1rem;
  }
}

.vl-content-header__entity-logo + .vl-content-header__logo {
  vertical-align: top;
  margin-left: -1.5rem;
}
@media screen and (max-width: 1023px) {
  .vl-content-header__entity-logo + .vl-content-header__logo {
    margin-left: -1.5rem;
  }
}
@media screen and (max-width: 767px) {
  .vl-content-header__entity-logo + .vl-content-header__logo {
    display: none;
  }
}

.vl-content-header__entity-logo {
  display: inline-block;
  vertical-align: top;
  padding: 1rem 2rem;
  font-size: 2.6rem;
  line-height: 0.95;
  text-transform: uppercase;
  background-color: var(--vl-theme-primary-color);
  color: #333332;
  text-decoration: none;
}
.vl-content-header__entity-logo--right {
  text-align: right;
}
.vl-content-header__entity-logo--lowercase {
  text-transform: none;
}
.vl-content-header__entity-logo--serif {
  font-family: "Flanders Art Serif", serif;
}
.vl-content-header__entity-logo--neutral {
  color: #333332;
}
@media screen and (max-width: 1023px) {
  .vl-content-header__entity-logo {
    font-size: 2rem;
  }
}
@media screen and (max-width: 767px) {
  .vl-content-header__entity-logo {
    font-size: 1.6rem;
    padding: 0.5rem 1rem;
    text-align: left;
  }
}
.vl-content-header__entity-logo__prefix, .vl-content-header__entity-logo__title {
  display: block;
}
.vl-content-header__entity-logo__title {
  font-weight: 500;
}
.vl-content-header__entity-logo--small {
  font-size: 1.6rem;
}
.vl-content-header--alt .vl-content-header__entity-logo {
  color: #333332;
  background-color: #fff;
}
.vl-content-header--has-logo .vl-content-header__entity-logo {
  height: 9rem;
  display: inline-flex;
  flex-direction: column;
  justify-content: center;
}
@media screen and (max-width: 767px) {
  .vl-content-header--has-logo .vl-content-header__entity-logo {
    height: 5rem;
  }
}
.vl-content-header--has-logo .vl-content-header__entity-logo + .vl-content-header__logo {
  vertical-align: top;
  margin-left: -1.5rem;
}

a.vl-content-header__entity-logo:hover, a.vl-content-header__entity-logo:focus {
  background-color: var(--vl-theme-primary-color);
  color: #333332;
}
.vl-content-header--alt a.vl-content-header__entity-logo:hover, .vl-content-header--alt a.vl-content-header__entity-logo:focus {
  background-color: white;
  color: #003bb0;
}

.vl-content-header__img-logo {
  display: inline-block;
}
.vl-content-header__img-logo__img {
  max-width: 30rem;
  max-height: 7.5rem;
  width: auto;
  height: auto;
}
@media screen and (max-width: 767px) {
  .vl-content-header__img-logo__img {
    max-width: 15rem;
    max-height: 3.5rem;
  }
}

.vl-content-header--has-actions::before {
  content: "";
  display: block;
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  height: 0.3rem;
  background-color: #f7f9fc;
  z-index: 2;
}
@media screen and (max-width: 767px) {
  .vl-content-header--has-actions::before {
    display: none;
  }
}
.vl-content-header__actions {
  background-color: #f7f9fc;
}
.vl-content-header .vl-content-header__action {
  color: #333332;
}
.vl-content-header .vl-content-header__action + .vl-content-header__action {
  border-left: 1px solid #333332;
}
.vl-content-header .vl-content-header__action a,
.vl-content-header .vl-content-header__action button {
  color: #333332;
}
.vl-content-header .vl-content-header__action a:hover, .vl-content-header .vl-content-header__action a:focus,
.vl-content-header .vl-content-header__action button:hover,
.vl-content-header .vl-content-header__action button:focus {
  color: #05c;
}

.vl-content-header__actions {
  padding: 1.5rem 0.5rem 1rem;
  float: right;
  display: flex;
  background-color: #f7f9fc;
  position: absolute;
  top: 0;
  right: 3rem;
}
.vl-content-header__actions::before, .vl-content-header__actions::after {
  content: "";
  display: table;
}
.vl-content-header__actions::after {
  clear: both;
}
@media screen and (max-width: 767px) {
  .vl-content-header__actions {
    display: none;
  }
}
.vl-content-header .vl-content-header__action {
  padding: 0 0.5rem;
  font-size: 1.4rem;
  line-height: 1;
  color: #333332;
}
.vl-content-header .vl-content-header__action + .vl-content-header__action {
  border-left: 1px solid #333332;
}
.vl-content-header .vl-content-header__action a,
.vl-content-header .vl-content-header__action button {
  line-height: inherit;
  color: #333332;
  margin: 0;
}

.vl-content-header__title {
  display: block;
  color: #333332;
}
.vl-content-header__context--has-link + .vl-content-header__title .vl-content-header__title__content {
  padding: 0.5rem 2rem 1.1rem;
  font-weight: 400;
  font-size: 4.8rem;
  line-height: 1.1;
  margin-top: 1px;
  background: white;
}
.vl-content-header__title--has-link {
  display: inline-block;
}
@media screen and (max-width: 767px) {
  .vl-content-header__title--has-link {
    display: none;
  }
}
.vl-content-header__title__content, .vl-content-header__title__link {
  padding: 0.4rem 2rem 0.7rem;
  font-weight: 500;
  font-size: 3.4rem;
  line-height: 1.2;
  display: inline-block;
  background-color: white;
  color: #333332;
  text-decoration: none;
}
@media screen and (max-width: 767px) {
  .vl-content-header__title__content, .vl-content-header__title__link {
    font-size: 1.8rem;
    padding: 0.3rem 1rem 0.4rem;
    font-weight: 500;
  }
}
.vl-content-header--has-context .vl-content-header__title__content, .vl-content-header--has-context .vl-content-header__title__link {
  font-size: 2.3rem;
  line-height: 1.1;
  font-weight: 300;
  padding: 0.9rem 2rem 0.9rem 3.5rem;
  font-size: 2.4rem;
  line-height: 1.1;
  background: linear-gradient(70deg, transparent 20px, white 21px);
}

.vl-content-header__subtitle {
  background-color: white;
}

.vl-content-header__context {
  font-size: 2.3rem;
  line-height: 1.1;
  font-weight: 500;
  background: linear-gradient(-110deg, transparent 20px, white 21px);
  display: inline-block;
}
.vl-content-header--alt .vl-content-header__context {
  background: linear-gradient(-110deg, transparent 20px, var(--vl-theme-primary-color) 21px);
}
.vl-content-header__context__link {
  font-size: 2.3rem;
  line-height: 1.1;
  font-weight: 500;
  font-size: 2.4rem;
  line-height: 1.1;
  display: block;
  padding: 0.9rem 3.5rem 0.9rem 2rem;
  background: linear-gradient(-110deg, transparent 20px, white 21px);
  color: #333332;
  text-decoration: none;
}
@media screen and (max-width: 767px) {
  .vl-content-header__context__link {
    background: white;
    font-size: 1.8rem;
    font-weight: 500;
    padding: 0.3rem 1rem 0.4rem;
  }
}
.vl-content-header--alt .vl-content-header__context__link {
  background: linear-gradient(-110deg, transparent 20px, var(--vl-theme-primary-color) 21px);
  color: #333332;
}
.vl-content-header__context--has-link {
  margin-right: -2.6rem;
}
@media screen and (max-width: 767px) {
  .vl-content-header__context--has-link {
    margin-right: 0;
  }
}

.vl-content-header__line {
  position: absolute;
  bottom: 0;
  transform-origin: 100%;
  transition: transform 0.1s;
}
.vl-content-header__line::before {
  display: block;
  content: "";
  height: 2px;
  background-color: var(--vl-theme-primary-color);
  width: 1000vw;
  transform-origin: 50%;
}
.vl-content-header__line-1 {
  left: 8rem;
}
@media screen and (max-width: 767px) {
  .vl-content-header__line-1 {
    left: 1rem;
  }
}
.vl-content-header__line-1::before {
  transform: translateX(-50%) rotate(71deg);
  background-color: #c6cdd3;
}
.vl-content-header__line-2 {
  left: -977vw;
  bottom: 3.5rem;
}
@media screen and (max-width: 767px) {
  .vl-content-header__line-2 {
    bottom: 0;
  }
}
.vl-content-header__line-2::before {
  transform: translateX(50%) rotate(-25deg);
}
.vl-content-header__line-3 {
  left: -970vw;
  bottom: 0;
}
@media screen and (max-width: 767px) {
  .vl-content-header__line-3 {
    right: 31rem;
  }
}
.vl-content-header__line-3::before {
  transform: translateX(50%) rotate(-40deg);
  background-color: rgba(198, 205, 211, 0.6);
}
`;
export default style;
