// @domg-wc/components/block/spotlight/vl-spotlight.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement } from '@domg-wc/common';
import { vlLegacyStyles, vlResetStyles } from '@domg-wc/styles';
import { documentStyle, iconListStyle, spotlightStyle } from '@domg/govflanders-style/component';
import { html, nothing } from 'lit';
import { customElement } from 'lit/decorators.js';
import { classMap } from 'lit/directives/class-map.js';
import { vlSpotlightFluxStyles } from './vl-spotlight.flux-css';
import { SIZE } from './vl-spotlight.model';
let VlSpotlight = class VlSpotlight extends BaseLitElement {
    constructor() {
        super(...arguments);
        // Attributes
        this.link = '';
        this.external = false;
        this.alt = false;
        this.noBorder = false;
        this.size = '';
        this.imgSrc = '';
        this.imgAlt = '';
    }
    static get styles() {
        return [vlResetStyles, ...vlLegacyStyles, spotlightStyle, documentStyle, iconListStyle, vlSpotlightFluxStyles];
    }
    static get properties() {
        return {
            link: {
                type: String,
                attribute: 'link',
            },
            external: {
                type: Boolean,
                attribute: 'external',
            },
            alt: {
                type: Boolean,
                attribute: 'alt',
                reflect: true,
            },
            noBorder: {
                type: Boolean,
                attribute: 'no-border',
                reflect: true,
            },
            size: {
                type: String,
                attribute: 'size',
                reflect: true,
            },
            imgSrc: {
                type: String,
                attribute: 'img-src',
            },
            imgAlt: {
                type: String,
                attribute: 'img-alt',
            },
        };
    }
    __getSlot(slotName) {
        return html ` <slot name="${slotName}"></slot>`;
    }
    __processSlotTitle() {
        return this._getTitleTemplateWithValue(this.__getSlot('title'));
    }
    __processSlotSubTitle() {
        return this._getSubTitleTemplateWithValue(this.__getSlot('subtitle'));
    }
    __processSlotTitleInHeader() {
        return this._getTitleTemplateWithValue(this.__getSlot('title'));
    }
    __processSlotContent() {
        return this._getContentTemplateWithValue(this.__getSlot('content'));
    }
    __processSlotText() {
        return this._getTextTemplateWithValue(this.__getSlot('text'));
    }
    _getTitleTemplateWithValue(value) {
        return html `<h3 class="vl-spotlight__title">${value}</h3>`;
    }
    _getSubTitleTemplateWithValue(value) {
        return html `<p class="vl-spotlight__subtitle">${value}</p>`;
    }
    _getContentTemplateWithValue(value) {
        return html ` <div class="vl-spotlight__content">${value}</div>`;
    }
    _getTextTemplateWithValue(value) {
        return html `<p class="vl-spotlight__text">${value}</p>`;
    }
    __fallbackIfEmpty(value, templateResult) {
        if (value && value.length > 0) {
            return templateResult;
        }
        return ``;
    }
    render() {
        const classes = {
            'vl-spotlight': true,
            'vl-spotlight--alt': this.alt,
            'vl-spotlight--xs': this.size === SIZE.XS,
            'vl-spotlight--s': this.size === SIZE.S,
            'vl-spotlight--l': this.size === SIZE.L,
            'vl-spotlight--no-border': this.noBorder,
        };
        if (this.link) {
            return html `<a
                href="${this.link}"
                class="${classMap(classes)}"
                target=${this.external ? '_blank' : nothing}
            >
                <article role="none">
                    ${this.__processHeader()} ${this.__processSlotTitle()} ${this.__processSlotSubTitle()}
                    ${this.__processSlotContent()} ${this.__processSlotText()}
                </article>
            </a>`;
        }
        return html `
            <article class="${classMap(classes)}" role="none">
                ${this.__processHeader()} ${this.__processSlotTitle()} ${this.__processSlotSubTitle()}
                ${this.__processSlotContent()} ${this.__processSlotText()}
            </article>
        `;
    }
    __processHeader() {
        if (!this.imgSrc)
            return html ``;
        return html ` <header role="none" class="vl-spotlight__header">
            <div class="vl-spotlight__image vl-spotlight__image--focus-center-center">
                <img class="vl-spotlight__image__img" src="${this.imgSrc}" alt="${this.imgAlt}" />
            </div>
            <div class="vl-spotlight__title-wrapper">${this.__processSlotTitleInHeader()}</div>
        </header>`;
    }
};
VlSpotlight = __decorate([
    customElement('vl-spotlight')
], VlSpotlight);
export { VlSpotlight };
