// @domg-wc/components/block/side-navigation/vl-side-navigation-item.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { BaseLitElement, findDeepestElementThroughShadowRoot, webComponent } from '@domg-wc/common';
let VlSideNavigationItemComponent = class VlSideNavigationItemComponent extends BaseLitElement {
    constructor() {
        super();
        this.parent = false;
        this.classList.add('vl-side-navigation__item');
    }
    static get properties() {
        return {
            parent: { type: Boolean },
        };
    }
    createRenderRoot() {
        return this;
    }
    get scrollOffset() {
        const vlHeader = findDeepestElementThroughShadowRoot(document.documentElement, 'vl-header') ||
            findDeepestElementThroughShadowRoot(document.documentElement, 'vl-header-next');
        const vlFunctionalHeaderSticky = findDeepestElementThroughShadowRoot(document.documentElement, 'vl-functional-header[sticky]');
        return (vlHeader?.height || 0) + (vlFunctionalHeaderSticky?.height || 0);
    }
    get hashSync() {
        const parent = this.closest('vl-side-navigation');
        return parent ? parent.hasAttribute('hash-sync') : false;
    }
    firstUpdated() {
        this.setAttribute('role', 'listitem');
        const a = this.querySelector('a');
        if (!a) {
            console.warn('vl-side-navigation-item vereist een anchor element (<a>) als child.', this);
            return;
        }
        a.addEventListener('click', (e) => {
            e.preventDefault();
            const href = a.getAttribute('href');
            if (!href) {
                console.warn('vl-side-navigation-item vereist een geldige href op het anchor element.');
                return;
            }
            const element = findDeepestElementThroughShadowRoot(this.getRootNode(), href);
            if (!element) {
                console.warn(`Element met id "${href}" niet gevonden in de DOM.`);
                return;
            }
            const rect = element.getBoundingClientRect();
            const scrollTop = window.scrollY + rect.top - this.scrollOffset;
            window.scrollTo({ top: scrollTop, behavior: 'smooth' });
            if (this.hashSync) {
                history.pushState(null, '', href);
            }
        });
    }
};
VlSideNavigationItemComponent = __decorate([
    webComponent('vl-side-navigation-item'),
    __metadata("design:paramtypes", [])
], VlSideNavigationItemComponent);
export { VlSideNavigationItemComponent };
