// @domg-wc/components/block/side-navigation/vl-side-navigation.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var VlSideNavigationComponent_1;
import { BaseLitElement, findDeepestElementThroughShadowRoot, legacyBreakpoint, legacyCore, registerWebComponents, unwrap, webComponent, } from '@domg-wc/common';
import { vlContentBlockStyles, vlGridStyles, vlLegacyStyles, vlSectionStyles } from '@domg-wc/styles';
import { vlIconStyles } from '../../atom/icon-style/vl-icon-style.css';
import { vlSideNavigationStyles } from './vl-side-navigation.css';
import { sideNavigationDefaults } from './vl-side-navigation.defaults';
import './vl-side-navigation.lib.js';
registerWebComponents([legacyCore, legacyBreakpoint]);
let VlSideNavigationComponent = VlSideNavigationComponent_1 = class VlSideNavigationComponent extends BaseLitElement {
    constructor() {
        super();
        this.hashSync = sideNavigationDefaults.hashSync;
        this.processClasses();
        this.rerender();
    }
    static get properties() {
        return {
            hashSync: { type: Boolean, attribute: 'hash-sync' },
        };
    }
    createRenderRoot() {
        return this;
    }
    get scrollOffset() {
        const vlHeader = findDeepestElementThroughShadowRoot(document.documentElement, 'vl-header') ||
            findDeepestElementThroughShadowRoot(document.documentElement, 'vl-header-next');
        const vlFunctionalHeaderSticky = findDeepestElementThroughShadowRoot(document.documentElement, 'vl-functional-header[sticky]');
        return (vlHeader?.height || 0) + (vlFunctionalHeaderSticky?.height || 0);
    }
    connectedCallback() {
        super.connectedCallback();
        // Gezien dit component geen shadow dom heeft, moeten de styles van de children toegevoegd worden
        // aan de adoptedStyleSheets van de omliggende shadow dom, anders zullen de styles niet toegepast worden
        const shadowRoot = this.shadowRoot || this.getRootNode();
        if (shadowRoot instanceof ShadowRoot) {
            shadowRoot.adoptedStyleSheets = [
                ...shadowRoot.adoptedStyleSheets,
                vlSideNavigationStyles.styleSheet,
                vlGridStyles.styleSheet,
                vlSectionStyles.styleSheet,
                vlContentBlockStyles.styleSheet,
                vlIconStyles.styleSheet,
                ...vlLegacyStyles.map((style) => style.styleSheet),
            ];
        }
        else {
            document.adoptedStyleSheets = [
                ...document.adoptedStyleSheets,
                vlSideNavigationStyles.styleSheet,
                vlIconStyles.styleSheet,
                ...vlLegacyStyles.map((style) => style.styleSheet),
            ];
        }
        this.processAttributes();
        this.initialScroll();
        this.closest('vl-header, vl-header-next')?.addEventListener('ready', () => {
            this.processAttributes();
        });
    }
    initialScroll() {
        if (!this.hashSync) {
            return;
        }
        const id = location.hash.slice(1);
        if (id) {
            const element = findDeepestElementThroughShadowRoot(this.getRootNode(), `#${id}`);
            if (element) {
                const rect = element.getBoundingClientRect();
                const scrollTop = window.scrollY + rect.top - this.scrollOffset;
                window.scrollTo({ top: scrollTop, behavior: 'smooth' });
            }
        }
    }
    rerender() {
        const sideNavigationId = this.getAttribute('side-navigation-id');
        /*
            Hack voor UIG-2897
            Proza messages werden niet afgebeeld tot er een resize van de window gebeurde.
            Door de undress() - dress() niet in een timeout te wrappen is dit probleem opgelost.
            Dit zorgt echter voor andere problemen als de side navigation overgaat van desktop naar mobile view.
            Omdat dit component bij elke resize volledig afgebroken en opnieuw opgebouwd wordt, is het onmogelijk om state bij te houden als er geresized wordt.
            Als oplossing moet de afnemer een uniek side-navigation-id attribuut meegeven, dit id stoppen we in een static variabele.
            Als de static variabele nog niet is ingevuld of is ingevuld met het id van een andere side-navigation,
            kunnen we er van uitgaan dat dit de eerste keer is dat deze side-navigation rendert en de correcte code uitvoeren.
        */
        if (sideNavigationId && VlSideNavigationComponent_1.initializedSideNavigationId !== sideNavigationId) {
            VlSideNavigationComponent_1.initializedSideNavigationId = sideNavigationId;
            this.undress();
            this.dress();
        }
        else {
            setTimeout(() => {
                this.undress();
                this.dress();
            }, 200);
        }
    }
    dress() {
        vl.sideNavigation.dress(this);
        this.style.position = '';
    }
    /*
     * DV's component doesn't foresee user changing breakpoint at runtime
     * without "undress" styles for mobile will be applied to desktop template & vice versa
     */
    undress() {
        // remove .vl-u-no-overflow
        vl.util.removeClass(document.body, vl.ns + 'u-no-overflow');
        // delete .js-vl-scrollspy__toggle
        const scrollSpyToggleElements = document.querySelectorAll('.js-vl-scrollspy__toggle');
        if (scrollSpyToggleElements.length) {
            vl.util.each(Array.from(scrollSpyToggleElements), (element) => element.remove());
        }
        // delete .js-vl-scrollspy__close
        const scrollSpyCloseElements = document.querySelectorAll('.js-vl-scrollspy__close');
        if (scrollSpyCloseElements.length) {
            vl.util.each(Array.from(scrollSpyCloseElements), (element) => element.remove());
        }
        // unwrap .js-vl-scrollspy-placeholder - remove parent specific for mobile template, keeping child elements
        const scrollSpyPlaceholderElement = document.querySelector('.js-vl-scrollspy-placeholder');
        if (scrollSpyPlaceholderElement)
            unwrap(scrollSpyPlaceholderElement);
    }
    processAttributes() {
        this.setAttribute('side-navigation', '');
        this.setAttribute('side-navigation-scrollable', '');
        this.setAttribute('scrollspy', '');
        this.setAttribute('scrollspy-mobile', 'Navigatie');
        this.setAttribute('sticky', '');
        this.setAttribute('sticky-offset-top', `${this.scrollOffset}`);
    }
    processClasses() {
        this.classList.add('vl-side-navigation');
        this.classList.add('js-vl-side-navigation');
        this.classList.add('js-vl-sticky');
        this.classList.add('js-vl-scrollspy');
    }
};
VlSideNavigationComponent.initializedSideNavigationId = '';
VlSideNavigationComponent = VlSideNavigationComponent_1 = __decorate([
    webComponent('vl-side-navigation'),
    __metadata("design:paramtypes", [])
], VlSideNavigationComponent);
export { VlSideNavigationComponent };
