// @domg-wc/components/block/side-navigation/vl-side-navigation-title.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, webComponent } from '@domg-wc/common';
class VlSideNavigationTitleComponent extends BaseLitElement {
    connectedCallback() {
        super.connectedCallback();
        this.classList.add('vl-side-navigation__title');
    }
    createRenderRoot() {
        return this;
    }
}
let VlSideNavigationH1 = class VlSideNavigationH1 extends VlSideNavigationTitleComponent {
};
VlSideNavigationH1 = __decorate([
    webComponent('vl-side-navigation-h1')
], VlSideNavigationH1);
export { VlSideNavigationH1 };
let VlSideNavigationH2 = class VlSideNavigationH2 extends VlSideNavigationTitleComponent {
};
VlSideNavigationH2 = __decorate([
    webComponent('vl-side-navigation-h2')
], VlSideNavigationH2);
export { VlSideNavigationH2 };
let VlSideNavigationH3 = class VlSideNavigationH3 extends VlSideNavigationTitleComponent {
};
VlSideNavigationH3 = __decorate([
    webComponent('vl-side-navigation-h3')
], VlSideNavigationH3);
export { VlSideNavigationH3 };
let VlSideNavigationH4 = class VlSideNavigationH4 extends VlSideNavigationTitleComponent {
};
VlSideNavigationH4 = __decorate([
    webComponent('vl-side-navigation-h4')
], VlSideNavigationH4);
export { VlSideNavigationH4 };
let VlSideNavigationH5 = class VlSideNavigationH5 extends VlSideNavigationTitleComponent {
};
VlSideNavigationH5 = __decorate([
    webComponent('vl-side-navigation-h5')
], VlSideNavigationH5);
export { VlSideNavigationH5 };
let VlSideNavigationH6 = class VlSideNavigationH6 extends VlSideNavigationTitleComponent {
};
VlSideNavigationH6 = __decorate([
    webComponent('vl-side-navigation-h6')
], VlSideNavigationH6);
export { VlSideNavigationH6 };
