// @domg-wc/components/block/side-navigation/vl-side-navigation-toggle.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { BaseLitElement, findDeepestElementThroughShadowRoot, registerWebComponents, webComponent } from '@domg-wc/common';
import { VlIconComponent } from '@domg-wc/components/atom';
registerWebComponents([VlIconComponent]);
let VlSideNavigationToggleComponent = class VlSideNavigationToggleComponent extends BaseLitElement {
    constructor() {
        super();
        this.href = '';
        this.child = '';
        this.classList.add('vl-side-navigation__toggle');
    }
    static get properties() {
        return {
            href: { type: String },
            child: { type: String },
        };
    }
    createRenderRoot() {
        return this;
    }
    hasStickyHeader() {
        return !!this.getRootNode().querySelector('vl-header');
    }
    get scrollOffset() {
        const vlHeader = findDeepestElementThroughShadowRoot(document.documentElement, 'vl-header') ||
            findDeepestElementThroughShadowRoot(document.documentElement, 'vl-header-next');
        const vlFunctionalHeaderSticky = findDeepestElementThroughShadowRoot(document.documentElement, 'vl-functional-header[sticky]');
        return (vlHeader?.height || 0) + (vlFunctionalHeaderSticky?.height || 0);
    }
    get hashSync() {
        const parent = this.closest('vl-side-navigation');
        return parent ? parent.hasAttribute('hash-sync') : false;
    }
    firstUpdated() {
        if (this.querySelector('a')) {
            return;
        }
        if (!this.href) {
            console.warn('vl-side-navigation-toggle vereist een geldige href.');
            return;
        }
        const childNodes = this.childNodes;
        const aNode = document.createElement('a');
        aNode.setAttribute('href', this.href);
        const iconNode = document.createElement('vl-icon');
        iconNode.setAttribute('icon', 'arrow-right-fat');
        aNode.append(...childNodes, iconNode);
        this.appendChild(aNode);
        aNode.addEventListener('click', (e) => {
            e.preventDefault();
            const element = findDeepestElementThroughShadowRoot(this.getRootNode(), this.href);
            if (!element) {
                console.warn(`Element met id "${this.href}" niet gevonden in de DOM.`);
                return;
            }
            const rect = element.getBoundingClientRect();
            const scrollTop = window.scrollY + rect.top - this.scrollOffset;
            window.scrollTo({ top: scrollTop, behavior: 'smooth' });
            if (this.hashSync) {
                history.pushState(null, '', this.href);
            }
        });
    }
};
VlSideNavigationToggleComponent = __decorate([
    webComponent('vl-side-navigation-toggle'),
    __metadata("design:paramtypes", [])
], VlSideNavigationToggleComponent);
export { VlSideNavigationToggleComponent };
