// @domg-wc/components/form/textarea/vl-textarea.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { webComponent } from '@domg-wc/common';
import { baseStyle, resetStyle } from '@domg/govflanders-style/common';
import { textareaStyle } from '@domg/govflanders-style/component';
import { maxLengthValidator, minLengthValidator } from '@open-wc/form-control';
import { html, nothing } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import { live } from 'lit/directives/live.js';
import { FormControl } from '../form-control/form-control';
import { textareaDefaults } from './vl-textarea.defaults';
let VlTextareaComponent = class VlTextareaComponent extends FormControl {
    constructor() {
        super(...arguments);
        // Attributes
        this.block = textareaDefaults.block;
        this.readonly = textareaDefaults.readonly;
        this.value = textareaDefaults.value;
        this.placeholder = textareaDefaults.placeholder;
        this.autocomplete = textareaDefaults.autocomplete;
        this.minLength = textareaDefaults.minLength;
        this.maxLength = textareaDefaults.maxLength;
        this.rows = textareaDefaults.rows;
        this.cols = textareaDefaults.cols;
        // Variables
        this.initialValue = '';
        this.dispatchInput = false;
    }
    static get styles() {
        return [resetStyle, baseStyle, textareaStyle];
    }
    static get properties() {
        return {
            block: { type: Boolean },
            readonly: { type: Boolean },
            value: { type: String, reflect: true },
            placeholder: { type: String },
            autocomplete: { type: String },
            minLength: { type: Number, attribute: 'min-length' },
            maxLength: { type: Number, attribute: 'max-length' },
            rows: { type: Number },
            cols: { type: Number },
        };
    }
    connectedCallback() {
        super.connectedCallback();
        if (!this.initialValue) {
            this.initialValue = this.value;
        }
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        if (changedProperties.has('value')) {
            const detail = { value: this.value };
            this.setValue(this.value);
            this.dispatchEvent(new CustomEvent('vl-change', { composed: true, bubbles: true, detail }));
            if (this.dispatchInput) {
                this.dispatchEvent(new CustomEvent('vl-input', { composed: true, bubbles: true, detail }));
                this.dispatchInput = false;
            }
            this.dispatchEventIfValid(detail);
        }
    }
    render() {
        const classes = {
            'vl-textarea': true,
            'vl-textarea--disabled': this.disabled,
            'vl-textarea--error': this.isInvalid || this.error,
            'vl-textarea--success': this.success,
            'vl-textarea--block': this.block,
        };
        return html `
            <textarea
                id=${this.id || nothing}
                name=${this.name || nothing}
                class=${classMap(classes)}
                aria-label=${this.label || nothing}
                aria-invalid=${this.isInvalid || nothing}
                ?required=${this.required}
                ?disabled=${this.disabled}
                ?error=${this.error}
                ?readonly=${this.readonly}
                .value=${live(this.value)}
                placeholder=${this.placeholder || nothing}
                autocomplete=${this.autocomplete || nothing}
                minlength=${this.minLength ?? nothing}
                maxlength=${this.maxLength ?? nothing}
                rows=${this.rows ?? nothing}
                cols=${this.cols ?? nothing}
                @input=${this.onInput}
            />
        `;
    }
    get validationTarget() {
        return this.shadowRoot?.querySelector('textarea');
    }
    resetFormControl() {
        super.resetFormControl();
        this.value = this.initialValue;
    }
    onKeydown(event) {
        if (event.key === 'Enter') {
            // @ts-expect-error: event.bubbles is readonly
            event['bubbles'] = false;
        }
        super.onKeydown(event);
    }
    onInput(event) {
        this.dispatchInput = true;
        this.value = event?.target?.value;
    }
};
VlTextareaComponent.formControlValidators = [...FormControl.formControlValidators, minLengthValidator, maxLengthValidator];
VlTextareaComponent = __decorate([
    webComponent('vl-textarea')
], VlTextareaComponent);
export { VlTextareaComponent };
