// @domg-wc/map/components/controls/measure-control/vl-map-measure-control.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { BaseLitElement, registerWebComponents, webComponent } from '@domg-wc/common';
import { VlButtonComponent } from '@domg-wc/components/atom';
import { unByKey } from 'ol/Observable';
import { CONTROL_TYPE, IDENTIFIER } from '../../../vl-map.model';
import { VlMapControl } from '../vl-map-control.mixin';
let VlMapMeasureControl = class VlMapMeasureControl extends VlMapControl(BaseLitElement) {
    constructor() {
        super();
        this.controlElement = null;
        this.controlElement = document.createElement('vl-button');
        this.controlElement.setAttribute('tertiary', '');
        this.controlElement.setAttribute('toggle', ''); // maak er een toggle knop van
        this.controlElement.innerText = 'Meten';
        this.identifier = IDENTIFIER.MEASURE;
        this.type = CONTROL_TYPE.ACTION;
    }
    connectedCallback() {
        super.connectedCallback();
        this.clickListener = this.controlElement.addEventListener('click', () => this.handleMeasureControlClick());
    }
    getAction() {
        return this.map.getActionWithIdentifier(this.identifier);
    }
    handleMeasureControlClick() {
        // de default click handler van de button zorgt (omdat hij in toggle mode staat) al voor de 'on' wissel
        const measureAction = this.getAction();
        if (measureAction) {
            if (this.controlElement.on) {
                measureAction.element.activate();
                this.controlElement.removeAttribute('tertiary');
            }
            else {
                measureAction.element.deactivate();
                this.controlElement.setAttribute('tertiary', '');
            }
        }
    }
    setActive(active) {
        if (active) {
            this.controlElement.on = true;
            this.controlElement.removeAttribute('tertiary');
        }
        else {
            this.controlElement.on = false;
            this.controlElement.setAttribute('tertiary', '');
        }
    }
    setDisabled(set) {
        this.controlElement.disabled = set;
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        unByKey(this.clickListener);
    }
};
(() => {
    registerWebComponents([VlButtonComponent]);
})();
VlMapMeasureControl = __decorate([
    webComponent('vl-map-measure-control'),
    __metadata("design:paramtypes", [])
], VlMapMeasureControl);
export { VlMapMeasureControl };
