// @domg-wc/components/form/form-message/vl-form-message.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, webComponent } from '@domg-wc/common';
import { resetStyle } from '@domg/govflanders-style/common';
import { formMessageStyle } from '@domg/govflanders-style/component';
import { html } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import { formMessageDefaults } from './vl-form-message.defaults';
import { vFormMessageFluxStyles } from './vl-form-message.flux-css';
export const FORM_MESSAGE_CUSTOM_TAG = 'vl-form-message';
let VlFormMessageComponent = class VlFormMessageComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        // Attributes
        this.show = formMessageDefaults.show;
        this.preLine = formMessageDefaults.preLine;
        this.for = formMessageDefaults.for; // Wordt enkel gebruikt in de form-control basis klasse
        this.state = formMessageDefaults.state; // Wordt enkel gebruikt in de form-control basis klasse
        this.validationMessage = '';
    }
    static get styles() {
        return [resetStyle, formMessageStyle, vFormMessageFluxStyles];
    }
    static get properties() {
        return {
            show: { type: Boolean, reflect: true },
            preLine: { type: Boolean, attribute: 'pre-line' },
            for: { type: String },
            state: { type: String },
            validationMessage: { type: String, attribute: 'validation-message' },
        };
    }
    render() {
        const classes = {
            'vl-form__error': true,
            'vl-pre-line': this.preLine,
        };
        return html `
            <p class=${classMap(classes)} part="text" ?hidden=${!this.show}>
                <slot>${this.validationMessage}</slot>
            </p>
        `;
    }
};
VlFormMessageComponent = __decorate([
    webComponent(FORM_MESSAGE_CUSTOM_TAG)
], VlFormMessageComponent);
export { VlFormMessageComponent };
