// @domg-wc/map/components/current-location/vl-map-current-location.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { BaseLitElement } from '@domg-wc/common';
import { vlIconStyles } from '@domg-wc/components/atom';
import { css, html, unsafeCSS } from 'lit';
import { customElement } from 'lit/decorators.js';
import proj4 from 'proj4';
import { DEFAULT_TOOLTIP, DEFAULT_ZOOM } from './vl-map-current-location.defaults';
import { vlMapCurrentLocationStyles } from './vl-map-current-location.flux-css';
let VlMapCurrentLocation = class VlMapCurrentLocation extends BaseLitElement {
    constructor() {
        super();
        this.zoom = DEFAULT_ZOOM;
        this.tooltip = DEFAULT_TOOLTIP;
    }
    static get styles() {
        return [
            css `
                ${unsafeCSS(vlMapCurrentLocationStyles)}
            `,
            vlIconStyles,
        ];
    }
    static get properties() {
        return {
            zoom: {
                type: Number,
                attribute: 'zoom',
                reflect: true,
            },
            tooltip: {
                type: String,
                attribute: 'tooltip',
                reflect: true,
            },
        };
    }
    connectedCallback() {
        super.connectedCallback();
        this._mapElement = this.closest('vl-map');
    }
    _currentLocation() {
        navigator.geolocation.getCurrentPosition((position) => {
            const source = proj4.Proj('EPSG:4326');
            const dest = proj4.Proj(this._mapElement.map.projection.getCode());
            const point = proj4.Point(position.coords.longitude, position.coords.latitude);
            const transformedPoint = proj4.transform(source, dest, point);
            this._mapElement.map.getView().setCenter([transformedPoint.x, transformedPoint.y]);
            this._mapElement.map.getView().setZoom(this.zoom);
        });
    }
    render() {
        return html ` <div class="flux-map-current-location">
            <button @click=${() => this._currentLocation()} type="button" title="${this.tooltip}">
                <span class="vl-icon vl-icon--location-gps"></span>
            </button>
        </div>`;
    }
};
VlMapCurrentLocation = __decorate([
    customElement('vl-map-current-location'),
    __metadata("design:paramtypes", [])
], VlMapCurrentLocation);
export { VlMapCurrentLocation };
