// @domg-wc/components/block/tabs/vl-tabs-pane.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, webComponent } from '@domg-wc/common';
import { html } from 'lit';
let VlTabsPaneComponent = class VlTabsPaneComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.titleSlotObserver = null;
    }
    static get is() {
        return 'vl-tabs-pane';
    }
    static get properties() {
        return {
            id: { type: String },
            title: { type: String },
        };
    }
    render() {
        return html ` <slot></slot>`;
    }
    connectedCallback() {
        super.connectedCallback();
        this.processTitleSlot();
    }
    disconnectedCallback() {
        if (this.titleSlotObserver) {
            this.titleSlotObserver.disconnect();
        }
    }
    get paneId() {
        return this.getAttribute('id') || '';
    }
    get paneTitle() {
        return this.getAttribute('title') || '';
    }
    get titleSlot() {
        return this.querySelector('[slot="title"]');
    }
    processTitleSlot() {
        if (this.titleSlot) {
            this.moveTabPaneTitleSlot();
            this.titleSlotObserver = this.observeTitleSlot(() => this.moveTabPaneTitleSlot());
        }
    }
    moveTabPaneTitleSlot() {
        if (this.titleSlot) {
            const titleClone = this.titleSlot.cloneNode(true);
            titleClone.setAttribute('slot', `${this.paneId}-title-slot`);
            const slot = this.parentElement?.querySelector(`[slot="${this.paneId}-title-slot"]`);
            if (slot) {
                this.parentElement?.replaceChild(titleClone, slot);
            }
            else {
                this.parentElement?.appendChild(titleClone);
            }
        }
    }
    observeTitleSlot(callback) {
        const observer = new MutationObserver((mutations) => callback(mutations));
        if (this.titleSlot) {
            observer.observe(this.titleSlot, { childList: true, subtree: true, characterData: true });
        }
        return observer;
    }
    titleChangedCallback(oldValue, newValue) {
        if (this.hasAttribute('observe-title')) {
            const tab = this.parentElement?.shadowRoot?.querySelector(`a.vl-tab__link#${this.paneId}`);
            if (tab) {
                tab.innerHTML = `<slot name="${this.paneId}-title-slot">${newValue}</slot>`;
            }
        }
    }
};
VlTabsPaneComponent = __decorate([
    webComponent('vl-tabs-pane')
], VlTabsPaneComponent);
export { VlTabsPaneComponent };
