// @domg-wc/components/block/tabs/vl-tab-section.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, webComponent } from '@domg-wc/common';
let VlTabSectionComponent = class VlTabSectionComponent extends BaseLitElement {
    static get is() {
        return 'vl-tab-section';
    }
    connectedCallback() {
        super.connectedCallback();
        this.processClasses();
        this.processAttributes();
    }
    createRenderRoot() {
        return this;
    }
    processClasses() {
        this.classList.add('vl-tab__pane');
    }
    processAttributes() {
        this.setAttribute('tab-pane', '');
        this.setAttribute('tabindex', '0');
        this.setAttribute('role', 'tabpanel');
        this.setAttribute('hidden', 'hidden');
        this.setAttribute('aria-labelledby', `${this.id}-tab`);
    }
};
VlTabSectionComponent = __decorate([
    webComponent('vl-tab-section')
], VlTabSectionComponent);
export { VlTabSectionComponent };
