// @domg-wc/components/block/tabs/vl-tab.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, webComponent } from '@domg-wc/common';
let VlTabComponent = class VlTabComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.href = '';
        this.disableLink = false;
        this.withinFunctionalHeader = false;
    }
    static get is() {
        return 'vl-tab';
    }
    static get properties() {
        return {
            href: { type: String },
            disableLink: { type: Boolean, attribute: 'disable-link' },
            withinFunctionalHeader: { type: Boolean, attribute: 'within-functional-header' },
        };
    }
    createRenderRoot() {
        return this;
    }
    connectedCallback() {
        super.connectedCallback();
        this.processClasses();
        this.processLinkElement();
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        if (changedProperties.has('href')) {
            if (!this.hasAttribute('disable-link')) {
                this.linkElement?.setAttribute('href', this.href);
                this.linkElement?.setAttribute('aria-controls', `${this.href}-pane`);
            }
        }
        if (changedProperties.has('id')) {
            if (this.linkElement) {
                this.linkElement.id = this.id;
            }
        }
        if (changedProperties.has('withinFunctionalHeader')) {
            if (this.withinFunctionalHeader) {
                this.linkElement?.classList.add('vl-link');
            }
            else {
                this.linkElement?.classList.remove('vl-link');
            }
        }
    }
    activate() {
        this.linkElement?.click();
    }
    get isActive() {
        return this.classList.contains('vl-tab--active');
    }
    get linkElement() {
        return this.querySelector('.vl-tab__link');
    }
    get linkElementTemplate() {
        const a = document.createElement('a');
        a.classList.add('vl-tab__link');
        a.setAttribute('tab', '');
        return a;
    }
    processClasses() {
        this.classList.add('vl-tab');
    }
    processLinkElement() {
        const a = this.linkElementTemplate;
        const slot = this.querySelector('slot');
        if (slot) {
            a.appendChild(slot);
        }
        a.addEventListener('click', (event) => {
            this.dispatchActiveTabChangedEvent(event);
        });
        this.appendChild(a);
    }
    dispatchActiveTabChangedEvent(event) {
        if (!this.isActive) {
            this.dispatchEvent(new CustomEvent('change', { detail: { activeTab: this.id }, composed: true }));
        }
        // enkel als de gebruiker expliciet klikte een `vl-click` event sturen
        if (event.screenX && event.screenY) {
            this.dispatchEvent(new CustomEvent('vl-click', { detail: { activeTab: this.id }, composed: true }));
        }
    }
};
VlTabComponent = __decorate([
    webComponent('vl-tab')
], VlTabComponent);
export { VlTabComponent };
