// @domg-wc/components/block/tabs/vl-tabs.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var VlTabsComponent_1;
import { awaitUntil, BaseLitElement, legacyBreakpoint, legacyCore, registerWebComponents, webComponent, } from '@domg-wc/common';
import './vl-tabs.lib.js';
import { vlLegacyStyles } from '@domg-wc/styles';
import { baseStyle, resetStyle } from '@domg/govflanders-style/common';
import { linkStyle } from '@domg/govflanders-style/component';
import { html } from 'lit';
import { VlTabSectionComponent } from './vl-tab-section.component';
import { VlTabComponent } from './vl-tab.component';
import { VlTabsPaneComponent } from './vl-tabs-pane.component';
import { tabsStyle } from './vl-tabs.css';
import { vlTabsFluxStyles } from './vl-tabs.flux-css';
registerWebComponents([legacyCore, legacyBreakpoint]);
let VlTabsComponent = VlTabsComponent_1 = class VlTabsComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.abortController = new AbortController();
        this.alt = false;
        this.responsiveLabel = '';
        this.activeTab = '';
        this.href = '';
        this.disableLinks = false;
        this.withinFunctionalHeader = false;
        this.displayStyle = 'default';
    }
    static get is() {
        return 'vl-tabs';
    }
    static get properties() {
        return {
            alt: { type: Boolean },
            responsiveLabel: { type: String, attribute: 'responsive-label' },
            activeTab: { type: String, attribute: 'active-tab' },
            href: { type: String },
            disableLinks: { type: Boolean, attribute: 'disable-links' },
            withinFunctionalHeader: { type: Boolean, attribute: 'within-functional-header' },
            displayStyle: { type: String, attribute: 'display-style' }, // 'default' | 'tabs' | 'collapsed'
        };
    }
    static get dressedAttributeName() {
        return 'tabs-dressed';
    }
    get dressed() {
        return this.hasAttribute(VlTabsComponent_1.dressedAttributeName);
    }
    get tabsElement() {
        return this.shadowRoot?.getElementById('tabs');
    }
    get tabListElement() {
        return this.shadowRoot?.getElementById('tab-list');
    }
    get tabsToggleElement() {
        return this.shadowRoot?.querySelector('.vl-tabs__toggle');
    }
    get responsiveLabelElement() {
        return this.shadowRoot?.getElementById('tabs-responsive-label');
    }
    get tabPanes() {
        return [...this.querySelectorAll(VlTabsPaneComponent.is)];
    }
    get hrefWithFallback() {
        return this.getAttribute('href') || window.location.pathname + window.location.search;
    }
    connectedCallback() {
        super.connectedCallback();
    }
    disconnectedCallback() {
        this.observer?.disconnect();
        this.abortController.abort('disconnectedCallback');
    }
    static get styles() {
        return [resetStyle, tabsStyle, vlTabsFluxStyles, linkStyle, baseStyle, ...vlLegacyStyles];
    }
    render() {
        return html `
            <div id="tabs" tabs tabs-responsive-label="Navigatie">
                <div id="tabs-wrapper" class="vl-tabs__wrapper">
                    <ul id="tab-list" class="vl-tabs" tabs-list role="tablist" aria-label="tabs"></ul>
                    <button type="button" tabs-toggle class="vl-tabs__toggle" close="false">
                        <span id="tabs-responsive-label">Navigatie</span>
                    </button>
                </div>
            </div>
        `;
    }
    firstUpdated(_changedProperties) {
        super.firstUpdated(_changedProperties);
        this.renderTabs();
        if (!this.hasAttribute('within-functional-header')) {
            this.renderSections();
        }
        this.dress();
        this.observer = this.observeTabPanes((mutations) => this.processTabPane(mutations));
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        if (changedProperties.has('alt')) {
            if (this.alt) {
                this.tabListElement.classList.add('vl-tabs--alt');
            }
            else {
                this.tabListElement.classList.remove('vl-tabs--alt');
            }
        }
        if (changedProperties.has('responsiveLabel')) {
            const value = this.responsiveLabel || 'Navigatie';
            this.tabsElement.setAttribute('tabs-responsive-label', value);
            this.responsiveLabelElement.innerHTML = value;
        }
        if (changedProperties.has('activeTab')) {
            this.ready().then(() => {
                const tab = [...this.tabListElement.children].find((tab) => tab.id == this.activeTab);
                if (tab && !tab.isActive) {
                    tab.activate();
                    if (this.tabsToggleElement && this.tabListElement.getAttribute('show') === 'true') {
                        // Als de tabsToggle aanwezig is en de tabList open is, klik op de tabsToggle om de tabList te sluiten.
                        this.tabsToggleElement.click();
                    }
                }
            });
        }
        if (changedProperties.has('withinFunctionalHeader')) {
            if (this.withinFunctionalHeader) {
                this.classList.add('vl-tabs--within-functional-header');
            }
            else {
                this.classList.remove('vl-tabs--within-functional-header');
            }
        }
        if (changedProperties.has('href')) {
            [...this.tabListElement.children].forEach((tab) => tab.setAttribute('href', `${this.href}#${tab.id}`));
        }
    }
    async dress(forced) {
        if (!this.dressed || forced) {
            await customElements.whenDefined('vl-tab');
            await customElements.whenDefined('vl-tab-section');
            vl.tabs.dress(this);
            this.setAttribute(VlTabsComponent_1.dressedAttributeName, '');
        }
    }
    async ready() {
        return awaitUntil(() => this.dressed);
    }
    addTab({ tabPane, index }) {
        const element = this.getTabTemplate({
            id: tabPane.paneId,
            title: tabPane.paneTitle,
        });
        if (index && index >= 0) {
            this.tabListElement?.insertBefore(element, this.tabListElement.children[index]);
        }
        else {
            this.tabListElement?.appendChild(element);
        }
    }
    getTabTemplate({ id, title }) {
        const tab = document.createElement('vl-tab');
        tab.id = `${id}`;
        tab.role = 'tab';
        tab.setAttribute('href', `${this.hrefWithFallback}#${id}`);
        if (this.hasAttribute('within-functional-header')) {
            tab.setAttribute('within-functional-header', '');
        }
        if (this.hasAttribute('disable-links')) {
            tab.setAttribute('disable-link', '');
        }
        const slot = document.createElement('slot');
        slot.name = `${id}-title-slot`;
        slot.innerHTML = `${title}`;
        tab.appendChild(slot);
        return tab;
    }
    removeTab(id) {
        const element = this.tabListElement?.querySelector(`[id="${id}"]`);
        if (element) {
            this.tabListElement?.removeChild(element);
        }
    }
    addTabSection({ id, index }) {
        this.tabPanes[index].setAttribute('slot', `${id}-slot`);
        const element = this.getTabSectionTemplate({ id });
        if (index && index >= 0) {
            this.tabsElement?.insertBefore(element, this.tabsElement.children[++index]);
        }
        else {
            this.tabsElement?.appendChild(element);
        }
    }
    getTabSectionTemplate({ id }) {
        const tabSection = document.createElement('vl-tab-section');
        tabSection.id = `${id}-pane`;
        const slot = document.createElement('slot');
        slot.name = `${id}-slot`;
        tabSection.appendChild(slot);
        return tabSection;
    }
    removeTabSection(id) {
        const element = this.tabsElement?.querySelector(`#${id}-pane`);
        if (element) {
            this.tabsElement?.removeChild(element);
        }
    }
    renderTabs() {
        if (this.tabListElement)
            this.tabListElement.innerHTML = '';
        this.tabPanes.forEach((tabPane) => {
            this.addTab({ tabPane: tabPane });
        });
    }
    renderSections() {
        this.tabPanes.forEach((tabPane, index) => this.addTabSection({ id: tabPane.paneId, index }));
    }
    observeTabPanes(callback) {
        const node = this;
        const observer = new MutationObserver(callback);
        observer.observe(node, { childList: true });
        return observer;
    }
    processTabPane(mutations) {
        const tabPanesToAdd = mutations
            .flatMap((mutation) => [...mutation.addedNodes])
            .filter((node) => node instanceof VlTabsPaneComponent);
        tabPanesToAdd.forEach((tabPane) => this.addTabAndSection(tabPane));
        const tabPanesToDelete = mutations
            .flatMap((mutation) => [...mutation.removedNodes])
            .filter((node) => node instanceof VlTabsPaneComponent);
        tabPanesToDelete.forEach((tabPane) => this.removeTabAndSection(tabPane));
        this.dress(true);
    }
    addTabAndSection(tabPane) {
        const index = this.tabPanes.indexOf(tabPane);
        this.addTab({ tabPane, index });
        if (!this.hasAttribute('within-functional-header')) {
            this.addTabSection({ id: tabPane.getAttribute('id') || '', index });
        }
    }
    removeTabAndSection(tabPane) {
        this.removeTab(tabPane.paneId);
        if (!this.hasAttribute('within-functional-header')) {
            this.removeTabSection(tabPane.paneId);
        }
    }
};
(() => {
    registerWebComponents([VlTabComponent, VlTabSectionComponent, VlTabsPaneComponent]);
})();
VlTabsComponent = VlTabsComponent_1 = __decorate([
    webComponent('vl-tabs')
], VlTabsComponent);
export { VlTabsComponent };
