// @govflanders/vl-widget-shared/dist/i18n.js
import i18next from 'i18next';
import I18NextVue from 'i18next-vue';
import { useI18nStore } from './store';
import { fallbackLanguage as fallback } from './settings';
export const getHtmlLang = (allowedLanguages, fallbackLanguage) => {
    const lang = document.documentElement.getAttribute('lang')?.split('-')[0].toLocaleLowerCase();
    return allowedLanguages.includes(lang) ? lang : fallbackLanguage;
};
export const useI18n = ({ app, languageResources: { nl, fr, en, de }, i18nConfig }) => {
    const allowedLanguages = Object.keys(i18nConfig?.translations || {}).map((lang) => lang);
    const fallbackLanguage = i18nConfig?.default || fallback;
    const initialLanguage = getHtmlLang(allowedLanguages, fallbackLanguage);
    i18next
        .init({
        lng: initialLanguage,
        fallbackLng: fallbackLanguage,
        debug: false,
        resources: {
            en: { translation: en },
            nl: { translation: nl },
            de: { translation: de },
            fr: { translation: fr },
        },
    })
        .then(() => {
        const i18nStore = useI18nStore();
        i18nStore.currentLanguage = initialLanguage;
        const observer = new MutationObserver((mutationsList) => {
            mutationsList.forEach((mutation) => {
                if (mutation.type === 'attributes' && mutation.attributeName === 'lang') {
                    const newLang = getHtmlLang(allowedLanguages, fallbackLanguage);
                    i18nStore.currentLanguage = newLang;
                    i18next.changeLanguage(newLang, (err) => {
                        if (err)
                            return console.warn('something went wrong loading', err);
                    });
                }
            });
        });
        // Start observing changes in the lang attribute
        observer.observe(document.documentElement, {
            attributes: true,
            attributeFilter: ['lang'],
        });
    });
    app.use(I18NextVue, { i18next });
};
