// @govflanders/vl-widget-shared/dist/store.js
import { defineStore } from 'pinia';
import { get, merge, unset, set } from 'lodash-es';
import { languages } from './types';
import { fallbackLanguage } from './settings';
const emptyConfig = {
    i18n: undefined,
    extensions: [],
};
export const useI18nStore = defineStore('i18nStore', {
    state: () => ({
        config: emptyConfig,
        currentLanguage: 'nl',
    }),
    getters: {
        defaultLanguage() {
            return this.config.i18n?.default || 'nl';
        },
        translateObject() {
            const merged = merge(this.config, this.config.i18n?.translations[fallbackLanguage], this.config.i18n?.translations[this.defaultLanguage], this.config.i18n?.translations[this.currentLanguage]);
            const extensions = [];
            for (const extension of merged.extensions || []) {
                extensions.push(merge(extension, extension.i18n?.translations[fallbackLanguage], extension.i18n?.translations[this.defaultLanguage], extension.i18n?.translations[this.currentLanguage]));
            }
            merged.extensions = extensions;
            return merged;
        },
    },
    actions: {
        init(config) {
            this.config = config;
        },
        set(value, path, extensionKey) {
            const extension = this.config.extensions?.find((extension) => extension.pluginTypeId === extensionKey);
            const base = extension || this.config;
            if (isTranslatable(value)) {
                for (const language of languages) {
                    set(base, ['i18n', 'translations', language, ...path], value[language]);
                }
            }
            else {
                for (const language of languages) {
                    unset(base, ['i18n', 'translations', language, ...path]);
                }
                set(base, path, value);
            }
        },
        setMultiple(values, path, extensionKey) {
            this.remove(path, extensionKey);
            for (const value of values) {
                this.add(value, path, extensionKey);
            }
        },
        add(value, path, extensionKey) {
            const extension = this.config.extensions?.find((extension) => extension.pluginTypeId === extensionKey);
            const base = extension || this.config;
            if (isTranslatable(value)) {
                for (const [language, v] of Object.entries(value)) {
                    const array = (get(base, ['i18n', 'translations', language, ...path]) || []);
                    array.push(v);
                    set(base, ['i18n', 'translations', language, ...path], array);
                }
            }
            else {
                const array = (get(base, path) || []);
                array.push(value);
                set(base, path, array);
            }
        },
        addMultiple(values, path, extensionKey) {
            for (const value of values) {
                this.add(value, path, extensionKey);
            }
        },
        remove(path, extensionKey) {
            const extension = this.config.extensions?.find((extension) => extension.pluginTypeId === extensionKey);
            const base = extension || this.config;
            unset(base, path);
            for (const language of languages) {
                unset(base, ['i18n', 'translations', language, ...path]);
            }
        },
        translate(item) {
            if (typeof item === 'object' && item !== null && this.currentLanguage in item) {
                return item[this.currentLanguage];
            }
            return item;
        },
    },
});
const isLanguageKey = (key) => {
    return Object.values(languages).includes(key);
};
// helper function to check if a value is translatable
export const isTranslatable = (value) => {
    return value !== null && typeof value === 'object' && !Array.isArray(value) && Object.keys(value).every((key) => isLanguageKey(key));
};
