// @domg-wc/components/form/input-field/validators.js
export const minValueValidator = {
    attribute: 'min',
    key: 'rangeUnderflow',
    message(instance) {
        if (instance.minExclusive) {
            return `Value must be greater than ${instance.min}.`;
        }
        return `Value must be greater than or equal to ${instance.min}.`;
    },
    isValid(instance, value) {
        if (!value || (!instance.min && instance.min !== 0)) {
            return true;
        }
        if (parseFloat(value) < instance.min) {
            return false;
        }
        if (instance.minExclusive && parseFloat(value) === instance.min) {
            return false;
        }
        return true;
    },
};
export const maxValueValidator = {
    attribute: 'max',
    key: 'rangeOverflow',
    message(instance) {
        if (instance.maxExclusive) {
            return `Value must be lesser than ${instance.max}.`;
        }
        return `Value must be lesser than or equal to ${instance.max}.`;
    },
    isValid(instance, value) {
        if (!value || (!instance.max && instance.max !== 0)) {
            return true;
        }
        if (parseFloat(value) > instance.max) {
            return false;
        }
        if (instance.maxExclusive && parseFloat(value) === instance.max) {
            return false;
        }
        return true;
    },
};
export const patternValidator = {
    attribute: 'pattern',
    key: 'patternMismatch',
    message: 'Please match the requested format',
    isValid(instance, value) {
        if (!value || (!instance.pattern && !instance.regex)) {
            return true;
        }
        const regex = instance.pattern || instance.regex;
        const regExp = new RegExp(regex);
        return regExp.test(value);
    },
};
