// @domg-wc/components/form/input-field/vl-input-field.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { webComponent } from '@domg-wc/common';
import { maxLengthValidator, minLengthValidator } from '@open-wc/form-control';
import { html, nothing } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import { live } from 'lit/directives/live.js';
import { FormControl } from '../form-control';
import { maxValueValidator, minValueValidator, patternValidator } from './validators';
import { inputFieldStyles } from './vl-input-field.css';
import { inputFieldDefaults } from './vl-input-field.defaults';
let VlInputFieldComponent = class VlInputFieldComponent extends FormControl {
    constructor() {
        super(...arguments);
        // Properties
        this.regex = inputFieldDefaults.regex; // Wordt enkel gebruikt in de pattern validator
        // Attributes
        this.block = inputFieldDefaults.block;
        this.readonly = inputFieldDefaults.readonly;
        this.type = inputFieldDefaults.type;
        this.value = inputFieldDefaults.value;
        this.placeholder = inputFieldDefaults.placeholder;
        this.autocomplete = inputFieldDefaults.autocomplete;
        this.minLength = inputFieldDefaults.minLength;
        this.maxLength = inputFieldDefaults.maxLength;
        this.min = inputFieldDefaults.min;
        this.max = inputFieldDefaults.max;
        this.minExclusive = inputFieldDefaults.minExclusive; // Wordt enkel gebruikt in de min validator
        this.maxExclusive = inputFieldDefaults.maxExclusive; // Wordt enkel gebruikt in de max validator
        this.pattern = inputFieldDefaults.pattern;
        this.inputGroup = inputFieldDefaults.inputGroup;
        // Variables
        this.initialValue = '';
        this.dispatchInput = false;
        this.isInputGroupPosition = (position) => {
            switch (position) {
                case 'first':
                    return this.parentElement?.firstElementChild === this;
                case 'last':
                    return this.parentElement?.lastElementChild === this;
                default:
                    return false;
            }
        };
    }
    static get styles() {
        return [inputFieldStyles];
    }
    static get properties() {
        return {
            block: { type: Boolean },
            readonly: { type: Boolean },
            type: { type: String },
            value: { type: String, reflect: true },
            placeholder: { type: String },
            autocomplete: { type: String },
            minLength: { type: Number, attribute: 'min-length' },
            maxLength: { type: Number, attribute: 'max-length' },
            min: { type: Number },
            max: { type: Number },
            minExclusive: { type: Boolean, attribute: 'min-exclusive' },
            maxExclusive: { type: Boolean, attribute: 'max-exclusive' },
            inputMode: { type: String, state: true },
            pattern: { type: String },
            inputGroup: { type: Boolean, attribute: 'input-group' },
            regex: { type: Object },
        };
    }
    connectedCallback() {
        super.connectedCallback();
        if (!this.initialValue) {
            this.initialValue = this.value;
        }
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        this.onUpdated(changedProperties);
    }
    render() {
        const classes = {
            'vl-input-field': true,
            'vl-input-field--disabled': this.disabled,
            'vl-input-field--error': this.isInvalid || this.error,
            'vl-input-field--success': this.success,
            'vl-input-field--block': this.block,
            'vl-input-field--input-group-left': this.inputGroup && this.isInputGroupPosition('first'),
            'vl-input-field--input-group-right': this.inputGroup && this.isInputGroupPosition('last'),
        };
        return html `
            <input
                id=${this.id || nothing}
                part="input"
                name=${this.name || nothing}
                class=${classMap(classes)}
                type=${this.type}
                aria-label=${this.label || nothing}
                aria-invalid=${this.isInvalid || nothing}
                ?required=${this.required}
                ?disabled=${this.disabled}
                ?error=${this.error}
                ?readonly=${this.readonly}
                .value=${live(this.value)}
                placeholder=${this.placeholder || nothing}
                autocomplete=${this.autocomplete || nothing}
                minlength=${this.minLength ?? nothing}
                maxlength=${this.maxLength ?? nothing}
                min=${this.min ?? nothing}
                max=${this.max ?? nothing}
                pattern=${this.pattern}
                inputmode=${this.inputMode}
                @input=${this.onInput}
            />
        `;
    }
    get validationTarget() {
        return this.shadowRoot?.querySelector('input');
    }
    resetFormControl() {
        super.resetFormControl();
        this.value = this.initialValue;
    }
    onInput(event) {
        this.dispatchInput = true;
        this.value = event?.target?.value;
    }
    onUpdated(changedProperties) {
        if (changedProperties.has('value')) {
            const detail = { value: this.value };
            this.setValue(this.value);
            this.dispatchEvent(new CustomEvent('vl-change', { composed: true, bubbles: true, detail }));
            if (this.dispatchInput) {
                this.dispatchEvent(new CustomEvent('vl-input', { composed: true, bubbles: true, detail }));
                this.dispatchInput = false;
            }
            this.dispatchEventIfValid(detail);
        }
    }
};
VlInputFieldComponent.formControlValidators = [
    ...FormControl.formControlValidators,
    minLengthValidator,
    maxLengthValidator,
    minValueValidator,
    maxValueValidator,
    patternValidator,
];
VlInputFieldComponent = __decorate([
    webComponent('vl-input-field')
], VlInputFieldComponent);
export { VlInputFieldComponent };
