// @domg-wc/components/atom/text/vl-text.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, webComponent } from '@domg-wc/common';
import { html } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import { vlTextStyles } from './vl-text.css';
import { textDefaults } from './vl-text.defaults';
let VlTextComponent = class VlTextComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.bold = textDefaults.bold;
        this.italic = textDefaults.italic;
        this.underline = textDefaults.underline;
        this.success = textDefaults.success;
        this.warning = textDefaults.warning;
        this.error = textDefaults.error;
        this.annotation = textDefaults.annotation;
        this.small = textDefaults.small;
    }
    static get styles() {
        return [vlTextStyles];
    }
    static get properties() {
        return {
            bold: { type: Boolean },
            italic: { type: Boolean },
            underline: { type: Boolean },
            success: { type: Boolean },
            warning: { type: Boolean },
            error: { type: Boolean },
            annotation: { type: Boolean },
            small: { type: Boolean },
        };
    }
    render() {
        const classes = {
            'vl-text--bold': this.bold,
            'vl-text--italic': this.italic,
            'vl-text--underline': this.underline,
            'vl-text--success': this.success,
            'vl-text--warning': this.warning,
            'vl-text--error': this.error,
            'vl-text--annotation': this.annotation,
            'vl-text--small': this.small,
        };
        return html `
            <span class="vl-text ${classMap(classes)}">
                <slot></slot>
            </span>
        `;
    }
};
VlTextComponent = __decorate([
    webComponent('vl-text')
], VlTextComponent);
export { VlTextComponent };
