// @domg-wc/components/compliance/accessibility/vl-accessibility.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { BaseLitElement, registerWebComponents } from '@domg-wc/common';
import { vlGroupStyles, vlLegacyStyles, vlMarginStyles, vlResetStyles } from '@domg-wc/styles';
import { html } from 'lit';
import { customElement } from 'lit/decorators.js';
import { VlTitleComponent } from '../../atom';
import { VlLinkComponent } from '../../atom/link';
import { VlFunctionalHeaderComponent } from '../../block/functional-header';
import { content, contentElements } from './child/content.component';
import { header, headerElements } from './child/header.component';
import { title, titleElements } from './child/title.component';
let VlAccessibility = class VlAccessibility extends BaseLitElement {
    constructor() {
        super();
        this.allowCustomCSS = false;
        this.application = 'deze applicatie';
        this.compliance = 'PARTIALLY_COMPLIANT';
        this.date = '20 juli 2021';
        this.dateModified = '20 juli 2021';
        this.disableBackLink = false;
        this.hideBackLink = false;
        this.evaluation = 'NOT_EVALUATED';
        this.version = '1.0.0';
    }
    static get styles() {
        return [vlResetStyles, ...vlLegacyStyles, vlGroupStyles, vlMarginStyles];
    }
    static get properties() {
        return {
            application: {
                type: String,
                attribute: 'application',
                reflect: true,
            },
            compliance: {
                type: String,
                attribute: 'compliance',
                reflect: true,
            },
            date: {
                type: String,
                attribute: 'date',
                reflect: true,
            },
            dateModified: {
                type: String,
                attribute: 'date-modified',
                reflect: true,
            },
            disableBackLink: {
                type: Boolean,
                attribute: 'disable-back-link',
                reflect: true,
            },
            hideBackLink: {
                type: Boolean,
                attribute: 'hide-back-link',
                reflect: true,
            },
            evaluation: {
                type: String,
                attribute: 'evaluation',
                reflect: true,
            },
            version: {
                type: String,
                attribute: 'version',
                reflect: true,
            },
            limitations: {
                type: Object,
            },
        };
    }
    render() {
        const props = {
            application: this.application,
            compliance: this.compliance,
            date: this.date,
            dateModified: this.dateModified,
            disableBackLink: this.disableBackLink,
            hideBackLink: this.hideBackLink,
            evaluation: this.evaluation,
            version: this.version,
            limitations: this.limitations,
        };
        return html ` <slot name="header">${header(props)}</slot> ${title(props)} ${content(props)} `;
    }
};
(() => {
    registerWebComponents([
        ...new Set([VlFunctionalHeaderComponent, ...contentElements(), ...headerElements(), ...titleElements()]),
        VlTitleComponent,
        VlLinkComponent,
    ]);
})();
VlAccessibility = __decorate([
    customElement('vl-accessibility'),
    __metadata("design:paramtypes", [])
], VlAccessibility);
export { VlAccessibility };
