// @domg/govflanders-style/component/video-player/vl-video-player.css.js
import { css } from 'lit';
const style = css `
@charset "UTF-8";
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-video-player {
  /* stylelint-disable-next-line */
  /* stylelint-disable-next-line */
  /* stylelint-disable-next-line */
}
@keyframes plyr-progress {
  to {
    background-position: 25px 0;
  }
}
@keyframes plyr-popup {
  0% {
    opacity: 0.5;
    transform: translateY(10px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
}
@keyframes plyr-fade-in {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}
.vl-video-player .plyr {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  direction: ltr;
  font-family: inherit;
  font-variant-numeric: tabular-nums;
  font-weight: 500;
  line-height: 1.7;
  max-width: 100%;
  min-width: 200px;
  position: relative;
  text-shadow: none;
  transition: box-shadow 0.3s ease;
}
.vl-video-player .plyr video,
.vl-video-player .plyr audio {
  border-radius: inherit;
  height: auto;
  vertical-align: middle;
  width: 100%;
}
.vl-video-player .plyr button {
  font: inherit;
  line-height: inherit;
  width: auto;
}
.vl-video-player .plyr:focus {
  outline: 0;
}
.vl-video-player .plyr--full-ui {
  box-sizing: border-box;
}
.vl-video-player .plyr--full-ui *,
.vl-video-player .plyr--full-ui *::after,
.vl-video-player .plyr--full-ui *::before {
  box-sizing: inherit;
}
.vl-video-player .plyr--full-ui a,
.vl-video-player .plyr--full-ui button,
.vl-video-player .plyr--full-ui input,
.vl-video-player .plyr--full-ui label {
  touch-action: manipulation;
}
.vl-video-player .plyr__badge {
  background: #4f5b5f;
  border-radius: 2px;
  color: #fff;
  font-size: 9px;
  line-height: 1;
  padding: 3px 4px;
}
.vl-video-player .plyr--full-ui ::-webkit-media-text-track-container {
  display: none;
}
.vl-video-player .plyr__captions {
  animation: plyr-fade-in 0.3s ease;
  bottom: 0;
  color: #fff;
  display: none;
  font-size: 14px;
  left: 0;
  padding: 15px;
  position: absolute;
  text-align: center;
  transform: translateY(-60px);
  transition: transform 0.4s ease-in-out;
  width: 100%;
}
.vl-video-player .plyr__captions span {
  background: rgba(0, 0, 0, 0.8);
  border-radius: 2px;
  box-decoration-break: clone;
  line-height: 185%;
  padding: 0.2em 0.5em;
  white-space: pre-wrap;
}
.vl-video-player .plyr__captions span div {
  display: inline;
}
.vl-video-player .plyr__captions span:empty {
  display: none;
}
@media (min-width: 480px) {
  .vl-video-player .plyr__captions {
    font-size: 16px;
    padding: 30px;
  }
}
@media (min-width: 768px) {
  .vl-video-player .plyr__captions {
    font-size: 18px;
  }
}
.vl-video-player .plyr--captions-active .plyr__captions {
  display: block;
}
.vl-video-player .plyr--hide-controls .plyr__captions {
  transform: translateY(-22.5px);
}
.vl-video-player .plyr__control {
  background: transparent;
  border: 0;
  border-radius: 3px;
  color: inherit;
  cursor: pointer;
  flex-shrink: 0;
  overflow: visible;
  padding: 10.5px;
  position: relative;
  transition: all 0.3s ease;
}
.vl-video-player .plyr__control svg {
  display: block;
  fill: currentColor;
  height: 18px;
  pointer-events: none;
  width: 18px;
}
.vl-video-player .plyr__control:focus {
  outline: 0;
}
.vl-video-player .plyr__control.plyr__tab-focus {
  box-shadow: 0 0 0 3px rgba(0, 85, 204, 0.35);
  outline: 0;
}
.vl-video-player .plyr__control[aria-pressed=false] .icon--pressed,
.vl-video-player .plyr__control[aria-pressed=true] .icon--not-pressed,
.vl-video-player .plyr__control[aria-pressed=false] .label--pressed,
.vl-video-player .plyr__control[aria-pressed=true] .label--not-pressed {
  display: none;
}
.vl-video-player .plyr--audio .plyr__control.plyr__tab-focus, .vl-video-player .plyr--audio .plyr__control:hover, .vl-video-player .plyr--audio .plyr__control[aria-expanded=true] {
  background: #05c;
  color: #fff;
}
.vl-video-player .plyr__control--overlaid {
  background: rgba(0, 85, 204, 0.8);
  border: 0;
  border-radius: 100%;
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.15);
  color: #fff;
  display: none;
  left: 50%;
  padding: 23px;
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
  z-index: 2;
}
.vl-video-player .plyr__control--overlaid svg {
  height: 20px;
  left: 2px;
  position: relative;
  width: 20px;
}
.vl-video-player .plyr__control--overlaid:hover, .vl-video-player .plyr__control--overlaid:focus {
  background: #05c;
}
.vl-video-player .plyr--playing .plyr__control--overlaid {
  opacity: 0;
  visibility: hidden;
}
.vl-video-player .plyr--full-ui.plyr--video .plyr__control--overlaid {
  display: block;
}
.vl-video-player .plyr--full-ui ::-webkit-media-controls {
  display: none;
}
.vl-video-player .plyr__controls {
  align-items: center;
  display: flex;
  text-align: center;
}
.vl-video-player .plyr__controls > .plyr__control,
.vl-video-player .plyr__controls .plyr__progress,
.vl-video-player .plyr__controls .plyr__time,
.vl-video-player .plyr__controls .plyr__menu {
  margin-left: 7.5px;
}
.vl-video-player .plyr__controls > .plyr__control:first-child, .vl-video-player .plyr__controls > .plyr__control:first-child + [data-plyr=pause],
.vl-video-player .plyr__controls .plyr__progress:first-child,
.vl-video-player .plyr__controls .plyr__progress:first-child + [data-plyr=pause],
.vl-video-player .plyr__controls .plyr__time:first-child,
.vl-video-player .plyr__controls .plyr__time:first-child + [data-plyr=pause],
.vl-video-player .plyr__controls .plyr__menu:first-child,
.vl-video-player .plyr__controls .plyr__menu:first-child + [data-plyr=pause] {
  margin-left: 0;
}
.vl-video-player .plyr__controls .plyr__volume {
  margin-left: 7.5px;
}
@media (min-width: 480px) {
  .vl-video-player .plyr__controls > .plyr__control,
  .vl-video-player .plyr__controls .plyr__progress,
  .vl-video-player .plyr__controls .plyr__time,
  .vl-video-player .plyr__controls .plyr__menu {
    margin-left: 15px;
  }
  .vl-video-player .plyr__controls > .plyr__control + .plyr__control,
  .vl-video-player .plyr__controls .plyr__menu + .plyr__control,
  .vl-video-player .plyr__controls > .plyr__control + .plyr__menu {
    margin-left: 7.5px;
  }
}
.vl-video-player .plyr--video .plyr__controls {
  background: linear-gradient(rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.7));
  border-bottom-left-radius: inherit;
  border-bottom-right-radius: inherit;
  bottom: 0;
  color: #fff;
  left: 0;
  padding: 52.5px 15px 15px;
  padding-top: 0;
  position: absolute;
  right: 0;
  transition: opacity 0.4s ease-in-out, transform 0.4s ease-in-out;
  z-index: 2;
}
@media (min-width: 480px) {
  .vl-video-player .plyr--video .plyr__controls {
    padding-top: 52.5px;
  }
}
.vl-video-player .plyr--video .plyr__controls .plyr__control svg {
  filter: drop-shadow(0 1px 1px rgba(0, 0, 0, 0.15));
}
.vl-video-player .plyr--video .plyr__controls .plyr__control.plyr__tab-focus, .vl-video-player .plyr--video .plyr__controls .plyr__control:hover, .vl-video-player .plyr--video .plyr__controls .plyr__control[aria-expanded=true] {
  background: #05c;
  color: #fff;
}
.vl-video-player .plyr--audio .plyr__controls {
  background: #fff;
  border-radius: inherit;
  color: #4f5b5f;
  padding: 15px;
}
.vl-video-player .plyr--video.plyr--hide-controls .plyr__controls {
  opacity: 0;
  pointer-events: none;
  transform: translateY(100%);
}
.vl-video-player .plyr [data-plyr=captions],
.vl-video-player .plyr [data-plyr=pip],
.vl-video-player .plyr [data-plyr=airplay],
.vl-video-player .plyr [data-plyr=fullscreen] {
  display: none;
}
.vl-video-player .plyr--captions-enabled [data-plyr=captions],
.vl-video-player .plyr--pip-supported [data-plyr=pip],
.vl-video-player .plyr--airplay-supported [data-plyr=airplay],
.vl-video-player .plyr--fullscreen-enabled [data-plyr=fullscreen] {
  display: inline-block;
}
.vl-video-player .plyr__video-embed {
  height: 0;
  padding-bottom: 5625%;
  position: relative;
}
.vl-video-player .plyr__video-embed iframe {
  border: 0;
  height: 100%;
  left: 0;
  position: absolute;
  top: 0;
  user-select: none;
  width: 100%;
}
.vl-video-player .plyr--full-ui .plyr__video-embed iframe {
  pointer-events: none;
}
.vl-video-player .plyr--full-ui .plyr__video-embed > .plyr__video-embed__container {
  padding-bottom: 24000%;
  position: relative;
  transform: translateY(-3828.125%);
}
.vl-video-player .plyr__menu {
  display: flex;
  position: relative;
}
.vl-video-player .plyr__menu .plyr__control svg {
  transition: transform 0.3s ease;
}
.vl-video-player .plyr__menu .plyr__control[aria-expanded=true] svg {
  transform: rotate(90deg);
}
.vl-video-player .plyr__menu .plyr__control[aria-expanded=true] .plyr__tooltip {
  display: none;
}
.vl-video-player .plyr__menu__container {
  animation: plyr-popup 0.2s ease;
  background: rgba(255, 255, 255, 0.9);
  border-radius: 4px;
  bottom: 100%;
  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.15);
  color: #4f5b5f;
  font-size: 16px;
  margin-bottom: 10px;
  position: absolute;
  right: -3px;
  text-align: left;
  white-space: nowrap;
  z-index: 3;
}
.vl-video-player .plyr__menu__container > div {
  overflow: hidden;
  transition: height 0.35s cubic-bezier(0.4, 0, 0.2, 1), width 0.35s cubic-bezier(0.4, 0, 0.2, 1);
}
.vl-video-player .plyr__menu__container::after {
  border: 4px solid transparent;
  border-top-color: rgba(255, 255, 255, 0.9);
  content: "";
  height: 0;
  position: absolute;
  right: 15px;
  top: 100%;
  width: 0;
}
.vl-video-player .plyr__menu__container ul {
  list-style: none;
  margin: 0;
  overflow: hidden;
  padding: 10.5px;
}
.vl-video-player .plyr__menu__container ul li {
  margin-top: 2px;
}
.vl-video-player .plyr__menu__container ul li:first-child {
  margin-top: 0;
}
.vl-video-player .plyr__menu__container .plyr__control {
  align-items: center;
  color: #4f5b5f;
  display: flex;
  font-size: 14px;
  padding: 6px 21px;
  user-select: none;
  width: 100%;
}
.vl-video-player .plyr__menu__container .plyr__control::after {
  border: 4px solid transparent;
  content: "";
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
}
.vl-video-player .plyr__menu__container .plyr__control--forward {
  padding-right: 42px;
}
.vl-video-player .plyr__menu__container .plyr__control--forward::after {
  border-left-color: rgba(79, 91, 95, 0.8);
  right: 5px;
}
.vl-video-player .plyr__menu__container .plyr__control--forward.plyr__tab-focus::after, .vl-video-player .plyr__menu__container .plyr__control--forward:hover::after {
  border-left-color: currentColor;
}
.vl-video-player .plyr__menu__container .plyr__control--back {
  font-weight: 500;
  margin: 10.5px;
  margin-bottom: 5px;
  padding-left: 42px;
  position: relative;
  width: calc(100% - 21px);
}
.vl-video-player .plyr__menu__container .plyr__control--back::after {
  border-right-color: rgba(79, 91, 95, 0.8);
  left: 10.5px;
}
.vl-video-player .plyr__menu__container .plyr__control--back::before {
  background: #b7c5cd;
  box-shadow: 0 1px 0 #fff;
  content: "";
  height: 1px;
  left: 0;
  margin-top: 6px;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 100%;
}
.vl-video-player .plyr__menu__container .plyr__control--back.plyr__tab-focus::after, .vl-video-player .plyr__menu__container .plyr__control--back:hover::after {
  border-right-color: currentColor;
}
.vl-video-player .plyr__menu__container label.plyr__control {
  padding-left: 10.5px;
}
.vl-video-player .plyr__menu__container label.plyr__control input[type=radio] + span {
  background: rgba(0, 0, 0, 0.1);
  border-radius: 100%;
  display: block;
  flex-shrink: 0;
  height: 16px;
  margin-right: 15px;
  position: relative;
  transition: all 0.3s ease;
  width: 16px;
}
.vl-video-player .plyr__menu__container label.plyr__control input[type=radio] + span::after {
  background: #fff;
  border-radius: 100%;
  content: "";
  height: 6px;
  left: 5px;
  opacity: 0;
  position: absolute;
  top: 5px;
  transform: scale(0);
  transition: transform 0.3s ease, opacity 0.3s ease;
  width: 6px;
}
.vl-video-player .plyr__menu__container label.plyr__control input[type=radio]:checked + span {
  background: #05c;
}
.vl-video-player .plyr__menu__container label.plyr__control input[type=radio]:checked + span::after {
  opacity: 1;
  transform: scale(1);
}
.vl-video-player .plyr__menu__container label.plyr__control input[type=radio]:focus + span {
  box-shadow: 0 0 0 3px rgba(0, 85, 204, 0.35);
  outline: 0;
}
.vl-video-player .plyr__menu__container label.plyr__control.plyr__tab-focus input[type=radio] + span, .vl-video-player .plyr__menu__container label.plyr__control:hover input[type=radio] + span {
  background: rgba(0, 0, 0, 0.1);
}
.vl-video-player .plyr__menu__container .plyr__menu__value {
  align-items: center;
  display: flex;
  margin-left: auto;
  margin-right: -10.5px;
  overflow: hidden;
  padding-left: 37px;
  pointer-events: none;
}
.vl-video-player .plyr__progress {
  display: flex;
  flex: 1;
  position: relative;
  margin-right: 14px;
  left: 7px;
}
.vl-video-player .plyr__progress input[type=range] {
  position: relative;
  z-index: 2;
  width: calc(100% + 14px) !important;
  margin: 0 -7px !important;
}
.vl-video-player .plyr__progress .plyr__tooltip {
  font-size: 14px;
  left: 0;
}
.vl-video-player .plyr__progress--buffer {
  -webkit-appearance: none; /* stylelint-disable-line */
  background: transparent;
  border: 0;
  border-radius: 100px;
  height: 6px;
  left: 0;
  margin: -3px 0 0;
  padding: 0;
  position: absolute;
  top: 50%;
  width: 100%;
}
.vl-video-player .plyr__progress--buffer::-webkit-progress-bar {
  background: transparent;
  transition: width 0.2s ease;
}
.vl-video-player .plyr__progress--buffer::-webkit-progress-value {
  background: currentColor;
  border-radius: 100px;
  min-width: 6px;
}
.vl-video-player .plyr__progress--buffer::-moz-progress-bar {
  background: currentColor;
  border-radius: 100px;
  min-width: 6px;
  transition: width 0.2s ease;
}
.vl-video-player .plyr__progress--buffer::-ms-fill {
  border-radius: 100px;
  transition: width 0.2s ease;
}
.vl-video-player .plyr--video .plyr__progress--buffer {
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.15);
  color: rgba(255, 255, 255, 0.25);
}
.vl-video-player .plyr--audio .plyr__progress--buffer {
  color: rgba(183, 197, 205, 0.66);
}
.vl-video-player .plyr--loading .plyr__progress--buffer {
  animation: plyr-progress 1s linear infinite;
  background-image: linear-gradient(-45deg, rgba(47, 52, 61, 0.6) 25%, transparent 25%, transparent 50%, rgba(47, 52, 61, 0.6) 50%, rgba(47, 52, 61, 0.6) 75%, transparent 75%, transparent);
  background-repeat: repeat-x;
  background-size: 25px 25px;
  color: transparent;
}
.vl-video-player .plyr--video.plyr--loading .plyr__progress--buffer {
  background-color: rgba(255, 255, 255, 0.25);
}
.vl-video-player .plyr--audio.plyr--loading .plyr__progress--buffer {
  background-color: rgba(183, 197, 205, 0.66);
}
.vl-video-player .plyr__poster {
  background-color: #000;
  background-position: 50% 50%;
  background-repeat: no-repeat;
  background-size: contain;
  height: 100%;
  left: 0;
  opacity: 0;
  position: absolute;
  top: 0;
  transition: opacity 0.3s ease;
  width: 100%;
  z-index: 1;
  pointer-events: none;
}
.vl-video-player .plyr--stopped.plyr__poster-enabled .plyr__poster {
  opacity: 1;
}
.vl-video-player .plyr--full-ui input[type=range] {
  -webkit-appearance: none; /* stylelint-disable-line */
  background: transparent;
  border: 0;
  border-radius: 28px;
  color: var(--vl-theme-primary-color);
  display: block;
  height: 20px;
  margin: 0;
  padding: 0;
  transition: box-shadow 0.3s ease;
  width: 100%;
}
.vl-video-player .plyr--full-ui input[type=range]::-webkit-slider-runnable-track {
  background: transparent;
  border: 0;
  border-radius: 3px;
  height: 6px;
  transition: all 0.3s ease;
  user-select: none;
  background-image: linear-gradient(to right, currentColor var(--value, 0%), transparent var(--value, 0%));
}
.vl-video-player .plyr--full-ui input[type=range]::-webkit-slider-thumb {
  background: #fff;
  border: 0;
  border-radius: 100%;
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.15), 0 0 0 1px rgba(47, 52, 61, 0.2);
  box-sizing: border-box;
  height: 14px;
  position: relative;
  transition: all 0.2s ease;
  width: 14px;
  -webkit-appearance: none; /* stylelint-disable-line */
  margin-top: -4px;
}
.vl-video-player .plyr--full-ui input[type=range]::-moz-range-track {
  background: transparent;
  border: 0;
  border-radius: 3px;
  height: 6px;
  transition: all 0.3s ease;
  user-select: none;
}
.vl-video-player .plyr--full-ui input[type=range]::-moz-range-thumb {
  background: #fff;
  border: 0;
  border-radius: 100%;
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.15), 0 0 0 1px rgba(47, 52, 61, 0.2);
  box-sizing: border-box;
  height: 14px;
  position: relative;
  transition: all 0.2s ease;
  width: 14px;
}
.vl-video-player .plyr--full-ui input[type=range]::-moz-range-progress {
  background: currentColor;
  border-radius: 3px;
  height: 6px;
}
.vl-video-player .plyr--full-ui input[type=range]::-ms-track {
  background: transparent;
  border: 0;
  border-radius: 3px;
  height: 6px;
  transition: all 0.3s ease;
  user-select: none;
  color: transparent;
}
.vl-video-player .plyr--full-ui input[type=range]::-ms-fill-upper {
  background: transparent;
  border: 0;
  border-radius: 3px;
  height: 6px;
  transition: all 0.3s ease;
  user-select: none;
}
.vl-video-player .plyr--full-ui input[type=range]::-ms-fill-lower {
  background: transparent;
  border: 0;
  border-radius: 3px;
  height: 6px;
  transition: all 0.3s ease;
  user-select: none;
  background: currentColor;
}
.vl-video-player .plyr--full-ui input[type=range]::-ms-thumb {
  background: #fff;
  border: 0;
  border-radius: 100%;
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.15), 0 0 0 1px rgba(47, 52, 61, 0.2);
  box-sizing: border-box;
  height: 14px;
  position: relative;
  transition: all 0.2s ease;
  width: 14px;
  margin-top: 0;
}
.vl-video-player .plyr--full-ui input[type=range]::-ms-tooltip {
  display: none;
}
.vl-video-player .plyr--full-ui input[type=range]:focus {
  outline: 0;
}
.vl-video-player .plyr--full-ui input[type=range]::-moz-focus-outer {
  border: 0;
}
.vl-video-player .plyr--full-ui input[type=range].plyr__tab-focus::-webkit-slider-runnable-track {
  box-shadow: 0 0 0 3px rgba(0, 85, 204, 0.35);
  outline: 0;
}
.vl-video-player .plyr--full-ui input[type=range].plyr__tab-focus::-moz-range-track {
  box-shadow: 0 0 0 3px rgba(0, 85, 204, 0.35);
  outline: 0;
}
.vl-video-player .plyr--full-ui input[type=range].plyr__tab-focus::-ms-track {
  box-shadow: 0 0 0 3px rgba(0, 85, 204, 0.35);
  outline: 0;
}
.vl-video-player .plyr--full-ui.plyr--video input[type=range]::-webkit-slider-runnable-track {
  background-color: rgba(255, 255, 255, 0.25);
}
.vl-video-player .plyr--full-ui.plyr--video input[type=range]::-moz-range-track {
  background-color: rgba(255, 255, 255, 0.25);
}
.vl-video-player .plyr--full-ui.plyr--video input[type=range]::-ms-track {
  background-color: rgba(255, 255, 255, 0.25);
}
.vl-video-player .plyr--full-ui.plyr--video input[type=range]:active::-webkit-slider-thumb {
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.15), 0 0 0 1px rgba(47, 52, 61, 0.2), 0 0 0 3px rgba(255, 255, 255, 0.5);
}
.vl-video-player .plyr--full-ui.plyr--video input[type=range]:active::-moz-range-thumb {
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.15), 0 0 0 1px rgba(47, 52, 61, 0.2), 0 0 0 3px rgba(255, 255, 255, 0.5);
}
.vl-video-player .plyr--full-ui.plyr--video input[type=range]:active::-ms-thumb {
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.15), 0 0 0 1px rgba(47, 52, 61, 0.2), 0 0 0 3px rgba(255, 255, 255, 0.5);
}
.vl-video-player .plyr--full-ui.plyr--audio input[type=range]::-webkit-slider-runnable-track {
  background-color: rgba(183, 197, 205, 0.66);
}
.vl-video-player .plyr--full-ui.plyr--audio input[type=range]::-moz-range-track {
  background-color: rgba(183, 197, 205, 0.66);
}
.vl-video-player .plyr--full-ui.plyr--audio input[type=range]::-ms-track {
  background-color: rgba(183, 197, 205, 0.66);
}
.vl-video-player .plyr--full-ui.plyr--audio input[type=range]:active::-webkit-slider-thumb {
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.15), 0 0 0 1px rgba(47, 52, 61, 0.2), 0 0 0 3px rgba(0, 0, 0, 0.1);
}
.vl-video-player .plyr--full-ui.plyr--audio input[type=range]:active::-moz-range-thumb {
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.15), 0 0 0 1px rgba(47, 52, 61, 0.2), 0 0 0 3px rgba(0, 0, 0, 0.1);
}
.vl-video-player .plyr--full-ui.plyr--audio input[type=range]:active::-ms-thumb {
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.15), 0 0 0 1px rgba(47, 52, 61, 0.2), 0 0 0 3px rgba(0, 0, 0, 0.1);
}
.vl-video-player .plyr__time {
  font-size: 14px;
}
.vl-video-player .plyr__time + .plyr__time::before {
  content: "⁄";
  margin-right: 15px;
}
@media (max-width: 767px) {
  .vl-video-player .plyr__time + .plyr__time {
    display: none;
  }
}
.vl-video-player .plyr--video .plyr__time {
  text-shadow: 0 1px 1px rgba(0, 0, 0, 0.15);
}
.vl-video-player .plyr__tooltip {
  background: #fff;
  border-radius: 0;
  bottom: 100%;
  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.15);
  color: #333332;
  font-size: 14px;
  font-weight: 500;
  line-height: 1.3;
  margin-bottom: 6px;
  opacity: 0;
  padding: 3px 4.5px;
  pointer-events: none;
  position: absolute;
  transform: translate(-50%, 10px) scale(0.8);
  transform-origin: 50% 100%;
  transition: transform 0.2s 0.1s ease, opacity 0.2s 0.1s ease;
  white-space: nowrap;
  z-index: 2;
}
.vl-video-player .plyr__tooltip::before {
  border-left: 6px solid transparent;
  border-right: 6px solid transparent;
  border-top: 6px solid #fff;
  bottom: -6px;
  content: "";
  height: 0;
  left: 50%;
  position: absolute;
  transform: translateX(-50%);
  width: 0;
  z-index: 2;
}
.vl-video-player .plyr .plyr__control:hover .plyr__tooltip,
.vl-video-player .plyr .plyr__control.plyr__tab-focus .plyr__tooltip,
.vl-video-player .plyr__tooltip--visible {
  opacity: 1;
  transform: translate(-50%, 0) scale(1);
}
.vl-video-player .plyr .plyr__control:hover .plyr__tooltip {
  z-index: 3;
}
.vl-video-player .plyr__controls > .plyr__control:first-child .plyr__tooltip,
.vl-video-player .plyr__controls > .plyr__control:first-child + .plyr__control .plyr__tooltip {
  left: 0;
  transform: translate(0, 10px) scale(0.8);
  transform-origin: 0 100%;
}
.vl-video-player .plyr__controls > .plyr__control:first-child .plyr__tooltip::before,
.vl-video-player .plyr__controls > .plyr__control:first-child + .plyr__control .plyr__tooltip::before {
  left: 19.5px;
}
.vl-video-player .plyr__controls > .plyr__control:last-child .plyr__tooltip {
  right: 0;
  transform: translate(0, 10px) scale(0.8);
  transform-origin: 100% 100%;
}
.vl-video-player .plyr__controls > .plyr__control:last-child .plyr__tooltip::before {
  left: auto;
  right: 19.5px;
  transform: translateX(50%);
}
.vl-video-player .plyr__controls > .plyr__control:first-child:hover .plyr__tooltip, .vl-video-player .plyr__controls > .plyr__control:first-child.plyr__tab-focus .plyr__tooltip,
.vl-video-player .plyr__controls > .plyr__control:first-child .plyr__tooltip--visible,
.vl-video-player .plyr__controls > .plyr__control:first-child + .plyr__control:hover .plyr__tooltip,
.vl-video-player .plyr__controls > .plyr__control:first-child + .plyr__control.plyr__tab-focus .plyr__tooltip,
.vl-video-player .plyr__controls > .plyr__control:first-child + .plyr__control .plyr__tooltip--visible,
.vl-video-player .plyr__controls > .plyr__control:last-child:hover .plyr__tooltip,
.vl-video-player .plyr__controls > .plyr__control:last-child.plyr__tab-focus .plyr__tooltip,
.vl-video-player .plyr__controls > .plyr__control:last-child .plyr__tooltip--visible {
  transform: translate(0, 0) scale(1);
}
.vl-video-player .plyr--video {
  overflow: hidden;
}
.vl-video-player .plyr--video.plyr--menu-open {
  overflow: visible;
}
.vl-video-player .plyr__video-wrapper {
  background: #000;
  border-radius: inherit;
  overflow: hidden;
  position: relative;
  z-index: 0;
}
.vl-video-player .plyr__volume {
  flex: 1;
  position: relative;
}
.vl-video-player .plyr__volume input[type=range] {
  position: relative;
  z-index: 2;
}
@media (min-width: 480px) {
  .vl-video-player .plyr__volume {
    max-width: 50px;
  }
}
@media (min-width: 768px) {
  .vl-video-player .plyr__volume {
    max-width: 80px;
  }
}
.vl-video-player .plyr--is-ios .plyr__volume {
  display: none !important;
}
.vl-video-player .plyr--is-ios.plyr--vimeo [data-plyr=mute] {
  display: none !important;
}
.vl-video-player .plyr:fullscreen {
  background: #000;
  border-radius: 0 !important;
  height: 100%;
  margin: 0;
  width: 100%;
}
.vl-video-player .plyr:fullscreen video {
  height: 100%;
}
.vl-video-player .plyr:fullscreen .plyr__video-wrapper {
  height: 100%;
  width: 100%;
}
.vl-video-player .plyr:fullscreen .plyr__video-embed {
  overflow: visible;
}
.vl-video-player .plyr:fullscreen.plyr--vimeo .plyr__video-wrapper {
  height: 0;
  top: 50%;
  transform: translateY(-50%);
}
.vl-video-player .plyr:fullscreen .plyr__control .icon--exit-fullscreen {
  display: block;
}
.vl-video-player .plyr:fullscreen .plyr__control .icon--exit-fullscreen + svg {
  display: none;
}
.vl-video-player .plyr:fullscreen.plyr--hide-controls {
  cursor: none;
}
@media (min-width: 1024px) {
  .vl-video-player .plyr:fullscreen .plyr__captions {
    font-size: 21px;
  }
}
.vl-video-player .plyr:-webkit-full-screen {
  background: #000;
  border-radius: 0 !important;
  height: 100%;
  margin: 0;
  width: 100%;
}
.vl-video-player .plyr:-webkit-full-screen video {
  height: 100%;
}
.vl-video-player .plyr:-webkit-full-screen .plyr__video-wrapper {
  height: 100%;
  width: 100%;
}
.vl-video-player .plyr:-webkit-full-screen .plyr__video-embed {
  overflow: visible;
}
.vl-video-player .plyr:-webkit-full-screen.plyr--vimeo .plyr__video-wrapper {
  height: 0;
  top: 50%;
  transform: translateY(-50%);
}
.vl-video-player .plyr:-webkit-full-screen .plyr__control .icon--exit-fullscreen {
  display: block;
}
.vl-video-player .plyr:-webkit-full-screen .plyr__control .icon--exit-fullscreen + svg {
  display: none;
}
.vl-video-player .plyr:-webkit-full-screen.plyr--hide-controls {
  cursor: none;
}
@media (min-width: 1024px) {
  .vl-video-player .plyr:-webkit-full-screen .plyr__captions {
    font-size: 21px;
  }
}
.vl-video-player .plyr:-moz-full-screen {
  background: #000;
  border-radius: 0 !important;
  height: 100%;
  margin: 0;
  width: 100%;
}
.vl-video-player .plyr:-moz-full-screen video {
  height: 100%;
}
.vl-video-player .plyr:-moz-full-screen .plyr__video-wrapper {
  height: 100%;
  width: 100%;
}
.vl-video-player .plyr:-moz-full-screen .plyr__video-embed {
  overflow: visible;
}
.vl-video-player .plyr:-moz-full-screen.plyr--vimeo .plyr__video-wrapper {
  height: 0;
  top: 50%;
  transform: translateY(-50%);
}
.vl-video-player .plyr:-moz-full-screen .plyr__control .icon--exit-fullscreen {
  display: block;
}
.vl-video-player .plyr:-moz-full-screen .plyr__control .icon--exit-fullscreen + svg {
  display: none;
}
.vl-video-player .plyr:-moz-full-screen.plyr--hide-controls {
  cursor: none;
}
@media (min-width: 1024px) {
  .vl-video-player .plyr:-moz-full-screen .plyr__captions {
    font-size: 21px;
  }
}
.vl-video-player .plyr:-ms-fullscreen {
  background: #000;
  border-radius: 0 !important;
  height: 100%;
  margin: 0;
  width: 100%;
}
.vl-video-player .plyr:-ms-fullscreen video {
  height: 100%;
}
.vl-video-player .plyr:-ms-fullscreen .plyr__video-wrapper {
  height: 100%;
  width: 100%;
}
.vl-video-player .plyr:-ms-fullscreen .plyr__video-embed {
  overflow: visible;
}
.vl-video-player .plyr:-ms-fullscreen.plyr--vimeo .plyr__video-wrapper {
  height: 0;
  top: 50%;
  transform: translateY(-50%);
}
.vl-video-player .plyr:-ms-fullscreen .plyr__control .icon--exit-fullscreen {
  display: block;
}
.vl-video-player .plyr:-ms-fullscreen .plyr__control .icon--exit-fullscreen + svg {
  display: none;
}
.vl-video-player .plyr:-ms-fullscreen.plyr--hide-controls {
  cursor: none;
}
@media (min-width: 1024px) {
  .vl-video-player .plyr:-ms-fullscreen .plyr__captions {
    font-size: 21px;
  }
}
.vl-video-player .plyr--fullscreen-fallback {
  background: #000;
  border-radius: 0 !important;
  height: 100%;
  margin: 0;
  width: 100%;
  bottom: 0;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 10000000;
}
.vl-video-player .plyr--fullscreen-fallback video {
  height: 100%;
}
.vl-video-player .plyr--fullscreen-fallback .plyr__video-wrapper {
  height: 100%;
  width: 100%;
}
.vl-video-player .plyr--fullscreen-fallback .plyr__video-embed {
  overflow: visible;
}
.vl-video-player .plyr--fullscreen-fallback.plyr--vimeo .plyr__video-wrapper {
  height: 0;
  top: 50%;
  transform: translateY(-50%);
}
.vl-video-player .plyr--fullscreen-fallback .plyr__control .icon--exit-fullscreen {
  display: block;
}
.vl-video-player .plyr--fullscreen-fallback .plyr__control .icon--exit-fullscreen + svg {
  display: none;
}
.vl-video-player .plyr--fullscreen-fallback.plyr--hide-controls {
  cursor: none;
}
@media (min-width: 1024px) {
  .vl-video-player .plyr--fullscreen-fallback .plyr__captions {
    font-size: 21px;
  }
}
.vl-video-player .plyr__ads {
  border-radius: inherit;
  bottom: 0;
  cursor: pointer;
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  z-index: -1;
}
.vl-video-player .plyr__ads > div,
.vl-video-player .plyr__ads > div iframe {
  height: 100%;
  position: absolute;
  width: 100%;
}
.vl-video-player .plyr__ads::after {
  background: rgba(47, 52, 61, 0.8);
  border-radius: 2px;
  bottom: 15px;
  color: #fff;
  content: attr(data-badge-text);
  font-size: 11px;
  padding: 2px 6px;
  pointer-events: none;
  position: absolute;
  right: 15px;
  z-index: 3;
}
.vl-video-player .plyr__ads::after:empty {
  display: none;
}
.vl-video-player .plyr__cues {
  background: currentColor;
  display: block;
  height: 6px;
  left: 0;
  margin: -3px 0 0;
  opacity: 0.8;
  position: absolute;
  top: 50%;
  width: 3px;
  z-index: 3;
}
.vl-video-player .plyr--no-transition {
  transition: none !important;
}
.vl-video-player .plyr__sr-only {
  clip: rect(1px, 1px, 1px, 1px);
  overflow: hidden;
  border: 0 !important;
  height: 1px !important;
  padding: 0 !important;
  position: absolute !important;
  width: 1px !important;
}
.vl-video-player .plyr__control {
  border-radius: 0.3rem;
}
.vl-video-player .plyr > .plyr__control {
  padding: 15px 2rem;
}
`;
export default style;
