// @domg-wc/map/components/controls/action-control/vl-map-action-control.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, registerWebComponents, webComponent } from '@domg-wc/common';
import { VlButtonComponent } from '@domg-wc/components/atom';
import { VlMapControl } from '../vl-map-control.mixin';
let VlMapActionControl = class VlMapActionControl extends VlMapControl(BaseLitElement) {
    constructor() {
        super(...arguments);
        this.actionId = '';
        this.icon = '';
        this.label = '';
        this.defaultActive = false;
        this.controlElement = null;
    }
    static get properties() {
        return {
            actionId: {
                type: String,
                attribute: 'action-id',
            },
            icon: {
                type: String,
                attribute: 'icon',
            },
            label: {
                type: String,
                attribute: 'label',
            },
            defaultActive: {
                type: Boolean,
                attribute: 'default-active',
                state: false,
            },
        };
    }
    connectedCallback() {
        this.controlElement = document.createElement('vl-button');
        this.controlElement.on = false;
        this.controlElement.setAttribute('tertiary', '');
        this.controlElement.setAttribute('toggle', ''); // maak er een toggle knop van
        this.controlElement.addEventListener('click', () => this.handleClickToggle());
        if (this.defaultActive) {
            this.activate();
        }
        else {
            this.deactivate();
        }
        if (this.icon) {
            this.controlElement.setAttribute('icon', this.icon);
        }
        if (this.label) {
            this.controlElement.innerText = this.label;
        }
        else {
            this.controlElement.setAttribute('text-hidden', '');
        }
        super.connectedCallback();
    }
    get active() {
        return this.controlElement.on;
    }
    activate() {
        this.controlElement.on = true;
        this.controlElement.removeAttribute('tertiary');
        if (this.action) {
            this.action.active = true;
        }
    }
    deactivate() {
        this.controlElement.on = false;
        this.controlElement.setAttribute('tertiary', '');
        if (this.action) {
            this.action.active = false;
        }
    }
    get action() {
        return this.closest('vl-map')?.querySelector(`#${this.actionId}`);
    }
    handleClickToggle() {
        if (this.controlElement.on) {
            this.activate();
        }
        else {
            this.deactivate();
        }
        this.dispatchEvent(new CustomEvent('change-control', {
            detail: {
                isActive: this.controlElement.on,
            },
        }));
    }
};
(() => {
    registerWebComponents([VlButtonComponent]);
})();
VlMapActionControl = __decorate([
    webComponent('vl-map-action-control')
], VlMapActionControl);
export { VlMapActionControl };
