// @domg-wc/components/form/select-rich/vl-select-rich.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var VlSelectRichComponent_1;
import { webComponent } from '@domg-wc/common';
import { baseStyle, resetStyle } from '@domg/govflanders-style/common';
import { iconStyle } from '@domg/govflanders-style/component';
import Choices from 'choices.js';
import { html, nothing } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import { FormControl } from '../form-control';
import { inputFieldStyles } from '../input-field';
import multiselectStyle from './styles/vl-multiselect.dv-css';
import { vlSelectRichFluxStyles } from './styles/vl-select-rich.flux-css';
import selectStyle from './styles/vl-select.dv-css';
import { selectRichDefaults } from './vl-select-rich.defaults';
let VlSelectRichComponent = VlSelectRichComponent_1 = class VlSelectRichComponent extends FormControl {
    constructor() {
        super();
        // Properties
        this.options = selectRichDefaults.options;
        this.initialOptions = selectRichDefaults.initialOptions;
        this.placeholder = selectRichDefaults.placeholder;
        this.search = selectRichDefaults.search;
        this.searchPlaceholder = selectRichDefaults.searchPlaceholder;
        // Variables
        this.choices = null;
        // Attributes
        this.notDeletable = selectRichDefaults.notDeletable;
        this.multiple = selectRichDefaults.multiple;
        this.position = selectRichDefaults.position;
        this.resultLimit = selectRichDefaults.resultLimit;
        this.noResultsText = selectRichDefaults.noResultsText;
        this.noChoicesText = selectRichDefaults.noChoicesText;
        // State
        this.value = null;
        this.dropdownInitialised = false;
        this.isDropdownOpen = false;
        this.dispatchInput = false;
        this.initialised = false;
        this.callbackOnInit = async () => {
            this.getChoicesElement()?.addEventListener('click', this.onClickChoices);
            this.internals.labels[0]?.addEventListener('click', this.onClickChoices);
            this.getChoicesElement()?.addEventListener('showDropdown', () => {
                const dropdownElement = this.getChoicesElement()?.querySelector('.vl-select__list--dropdown');
                if (dropdownElement && !this.dropdownInitialised) {
                    dropdownElement.setAttribute('role', 'group');
                    dropdownElement.setAttribute('id', 'vl-select__list');
                    this.dropdownInitialised = true;
                }
                this.isDropdownOpen = true;
            });
            this.getChoicesElement()?.addEventListener('hideDropdown', () => {
                this.isDropdownOpen = false;
            });
            this.setChoicesInputAttributes();
            await Promise.resolve();
            // Fix voor required validator
            if (!this.value) {
                this.setValue(null);
            }
            // Fix voor Choices.js search event dat niet afgevuurd wordt als de search value verwijderd wordt.
            this.choices?.input?.element?.addEventListener('input', this.onSearchInput);
            this.initialised = true;
        };
        this.onClickChoices = () => {
            if (!this.disabled) {
                this.choices?.showDropdown();
            }
        };
        this.onSearchInput = (event) => {
            const value = event?.target?.value;
            this.dispatchEvent(new CustomEvent('vl-select-search', { bubbles: true, composed: true, detail: { value } }));
        };
        this.submitFormOnEnter = false;
    }
    static get styles() {
        return [
            resetStyle,
            baseStyle,
            inputFieldStyles,
            selectStyle,
            multiselectStyle,
            iconStyle,
            vlSelectRichFluxStyles,
        ];
    }
    static get properties() {
        return {
            initialOptions: { type: Array, attribute: 'initial-options' },
            options: {
                type: Array,
                hasChanged: (value, oldValue) => {
                    if (value instanceof Array && oldValue instanceof Array) {
                        // We vergelijken de arrays
                        return JSON.stringify([...value]) !== JSON.stringify([...oldValue]);
                    }
                    else {
                        return value !== oldValue;
                    }
                },
            },
            placeholder: { type: String },
            notDeletable: { type: Boolean, attribute: 'not-deletable' },
            multiple: { type: Boolean },
            search: { type: Boolean },
            position: { type: String },
            resultLimit: { type: Number, attribute: 'result-limit' },
            noResultsText: { type: String, attribute: 'no-results-text' },
            noChoicesText: { type: String, attribute: 'no-choices-text' },
            searchPlaceholder: { type: String, attribute: 'search-placeholder' },
            value: {
                type: FormData,
                state: true,
                hasChanged: (value, oldValue) => this.compareValue(oldValue, value),
            },
        };
    }
    connectedCallback() {
        super.connectedCallback();
        if (this.initialised) {
            this.choices = new Choices(this.validationTarget, this.getChoicesConfig());
            this.initialOptions = structuredClone(this.options);
        }
    }
    async firstUpdated(changedProperties) {
        super.firstUpdated(changedProperties);
        this.choices = new Choices(this.validationTarget, this.getChoicesConfig());
        this.initialOptions = structuredClone(this.options);
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        if (!this.choices) {
            return;
        }
        if (changedProperties.has('options')) {
            if (this.choices.initialised) {
                this.choices.clearStore();
                this.choices.setChoices(this.options, 'value', 'label', true);
                this.updateSelectedOptions(this.options);
            }
            if (VlSelectRichComponent_1.compareValue(this.value, changedProperties.has('value'))) {
                this.value = this.collectFormData();
            }
        }
        if (changedProperties.has('value')) {
            const detail = { value: this.getSelected() };
            this.setValue(this.value);
            this.dispatchEvent(new CustomEvent('vl-change', { bubbles: true, composed: true, detail }));
            if (this.validity.valid) {
                this.dispatchEventIfValid(detail);
            }
        }
        if (changedProperties.has('disabled')) {
            if (this.disabled) {
                this.choices.disable();
            }
            else {
                this.choices.enable();
            }
        }
        if (changedProperties.has('error')) {
            this.internals.setValidity({ customError: this.error }, 'custom-error');
        }
        if (changedProperties.has('resultLimit')) {
            this.choices.config.searchResultLimit = this.resultLimit;
        }
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.getChoicesElement()?.removeEventListener('click', this.onClickChoices);
        this.internals.labels[0]?.removeEventListener('click', this.onClickChoices);
        this.choices?.input?.element?.removeEventListener('input', this.onSearchInput);
        this.choices?.destroy();
    }
    render() {
        const classes = {
            'vl-select': !this.multiple,
            'vl-multiselect': this.multiple,
            'vl-select--disabled': this.disabled,
            'vl-select--error': this.isInvalid || this.error,
            'vl-select--success': this.success,
        };
        return html `
            <select
                id=${this.id || nothing}
                name=${this.name || nothing}
                class=${classMap(classes)}
                aria-label=${this.label || nothing}
                aria-invalid=${this.isInvalid || nothing}
                ?required=${this.required}
                ?disabled=${this.disabled}
                ?error=${this.error}
                ?multiple=${this.multiple}
                @change=${this.onInput}
                @addItem=${this.onChange}
                @removeItem=${this.onChange}
            ></select>
        `;
    }
    get validationTarget() {
        return this.shadowRoot?.querySelector('select');
    }
    /**
     * Reset de form control naar de initiële staat.
     */
    resetFormControl() {
        super.resetFormControl();
        this.choices?.clearStore();
        this.choices?.setChoices(this.options, 'value', 'label', true);
        this.updateSelectedOptions(this.initialOptions);
        this.value = this.collectFormData();
    }
    /**
     * Vervangt de huidige opties van de select.
     * @param options
     */
    setOptions(options) {
        if (!options || !options.length) {
            return;
        }
        this.options = structuredClone(options);
    }
    /**
     * Update de opties van de select met de opgegeven opties.
     * @param options
     */
    updateSelectedOptions(options) {
        const selectedValues = options.filter((option) => option.selected).map((option) => option.value);
        const unselectedValues = options.filter((option) => !option.selected).map((option) => option.value);
        this.removeSelectionByValue(unselectedValues);
        this.selectByValue(selectedValues);
    }
    /**
     * Stelt de geselecteerde optie(s) in op basis van de opgegeven waarde(s).
     * @param value
     */
    setSelectedValues(value) {
        this.removeAllSelections();
        this.selectByValue(value);
    }
    getSelected() {
        return this.multiple ? this.getSelectedValues() : this.getSelectedValues()[0] || null;
    }
    /**
     * Selecteert 1 of meerdere keuzes op basis van de opgegeven waarde(s).
     * @param value
     */
    selectByValue(value) {
        if (!this.choices) {
            return;
        }
        this.choices.setChoiceByValue(value);
        this.setValue(this.collectFormData());
    }
    /**
     * Deselecteert 1 of meerdere keuzes op basis van de opgegeven waarde(s).
     * @param value
     */
    removeSelectionByValue(value) {
        if (!this.choices) {
            return;
        }
        if (Array.isArray(value)) {
            value.forEach((val) => this.choices.removeActiveItemsByValue(val));
        }
        else {
            this.choices.removeActiveItemsByValue(value);
        }
        this.setValue(this.collectFormData());
    }
    /**
     * Verwijdert alle selecties.
     */
    removeAllSelections() {
        if (!this.choices) {
            return;
        }
        this.choices.removeActiveItems();
        this.setValue(this.collectFormData());
    }
    onKeydown(event) {
        // de keyboard-events mogen niet buiten deze component bubbelen
        // om te vermijden dat - bij integratie in bvb. de tabs - navigeren met de pijltjes een tab wissel veroorzaakt
        event.stopPropagation();
        super.onKeydown(event);
    }
    setChoicesInputAttributes() {
        if (this.choices?.input?.element) {
            const inputElement = this.choices.input.element;
            inputElement.setAttribute('type', 'text');
            inputElement.classList.add('vl-input-field', 'vl-input-field-cloned');
            inputElement.setAttribute('autocomplete', 'off');
            inputElement.setAttribute('autocapitalize', 'off');
            inputElement.setAttribute('spellcheck', 'false');
            inputElement.setAttribute('role', 'textbox');
            inputElement.setAttribute('aria-autocomplete', 'list');
            inputElement.setAttribute('aria-label', 'zoek item');
        }
    }
    getSelectedValues() {
        const selectedOptions = this.validationTarget.selectedOptions;
        return (Array.from(selectedOptions)
            // Filter placeholders en niet-geselecteerde opties eruit
            .filter((option) => option.value && option.hasAttribute('selected'))
            .map((option) => option.value));
    }
    collectFormData() {
        const name = this.name || this.id;
        const selectedValues = this.getSelectedValues();
        return selectedValues?.length
            ? selectedValues.reduce((formData, string, currentIndex) => {
                // eslint-disable-next-line @typescript-eslint/no-unused-expressions
                currentIndex ? formData.append(name, string) : formData.set(name, string);
                return formData;
            }, new FormData())
            : null;
    }
    getChoicesElement() {
        return this.shadowRoot?.querySelector('.js-vl-select');
    }
    getChoicesConfig() {
        return {
            callbackOnInit: this.callbackOnInit,
            shouldSort: false,
            removeItemButton: !this.notDeletable,
            removeItems: !this.notDeletable,
            searchEnabled: this.search,
            placeholder: !!this.placeholder,
            placeholderValue: this.placeholder,
            position: this.position,
            noResultsText: this.noResultsText,
            searchResultLimit: this.resultLimit,
            noChoicesText: this.noChoicesText,
            searchPlaceholderValue: this.searchPlaceholder,
            shadowRoot: this.shadowRoot,
            classNames: {
                ...Choices.defaults.allOptions.classNames,
                containerOuter: 'js-vl-select',
                containerInner: 'vl-select__inner',
                input: 'vl-input-field',
                inputCloned: 'vl-input-field-cloned',
                list: 'vl-select__list',
                listItems: 'vl-select__list--multiple',
                listSingle: 'vl-select__list--single',
                listDropdown: ['vl-select__list', 'vl-select__list--dropdown'],
                item: 'vl-select__item',
                itemSelectable: 'vl-select__item--selectable',
                itemDisabled: 'vl-select__item--disabled',
                itemChoice: 'vl-select__item--choice',
                placeholder: 'vl-select__placeholder',
                group: 'vl-select__group',
                groupHeading: 'vl-select__heading',
                button: 'vl-select__button',
            },
            callbackOnCreateTemplates: (template) => {
                return {
                    containerOuter: () => {
                        return template(`
                            <div
                                class="js-vl-select vl-vi vl-vi-nav-down"
                                data-type="${this.multiple ? 'select-multiple' : 'select-one'}"
                                ${this.search ? 'aria-autocomplete="list"' : ''}
                                part="vl-select-rich__combobox"
                                role="combobox"
                                aria-haspopup="true"
                                aria-expanded=${this.isDropdownOpen ? 'true' : 'false'}
                                tabindex="0"
                                aria-controls="vl-select__list"
                                aria-label="${this.multiple ? 'selecteer één of meerdere opties' : 'selecteer één optie'}"
                            ></div>
                            `);
                    },
                    item: (_config, data) => {
                        const isPlaceholder = data.placeholder === true;
                        if (this.notDeletable) {
                            return template(`
                            <div class="vl-select__item
                                ${data.highlighted ? 'is-highlighted' : 'vl-select__item--selectable'}
                                ${this.multiple ? 'vl-pill' : ''}
                                ${data.placeholder ? 'vl-select__placeholder' : ''}"
                                role="option"
                                data-item
                                data-id="${data.id}"
                                data-value="${data.value}"
                                ${data.disabled ? 'aria-disabled="true"' : ''}
                            >
                                ${data.label}
                            </div>
                        `);
                        }
                        return template(`<div class="
                                    vl-select__item
                                    ${data.highlighted ? 'is-highlighted' : ''}
                                    ${!data.disabled ? 'vl-select__item--selectable' : ''}
                                    ${this.multiple ? 'vl-pill' : ''}
                                    ${data.placeholder ? 'vl-select__placeholder' : ''}"
                                    data-item
                                    data-id="${data.id}"
                                    data-value="${data.value}"
                                            ${isPlaceholder ? 'role="option"' : ''}
                                    ${data.disabled ? 'aria-disabled="true"' : 'data-deletable'}
                                >
                                    <span>${data.label}</span>
                                    <button type="button"
                                    ${isPlaceholder ? '' : 'role="option"'}
                                     class="vl-pill__close ${!this.multiple ? 'vl-vi vl-vi-close' : ''}" data-button aria-label="verwijder ${data.label}">
                                        ${this.multiple
                            ? `<span class="vl-pill__close__icon vl-vi vl-vi-close" aria-hidden="true"></span>`
                            : ''}
                                    </button>
                                </div>`);
                    },
                    itemList: () => template(`<div class="vl-input-field ${!this.multiple ? '' : 'vl-select__list--multiple'}" role="listbox"></div>`),
                    choiceList: () => template(`<div class="vl-select__list" role="listbox" aria-label="item lijst" tabindex="0"></div>`),
                };
            },
        };
    }
    /**
     * Event handler voor de change event van de select. Deze wordt aangeroepen wanneer de waarde van de
     * select verandert, ongeacht de bron (bijv. door de gebruiker of door code).
     * @private
     */
    onChange() {
        this.value = this.collectFormData();
    }
    /**
     * Event handler voor de input event van de select. Deze wordt aangeroepen wanneer de gebruiker
     * de waarde van de select wijzigt.
     * @private
     */
    onInput() {
        this.dispatchEvent(new CustomEvent('vl-input', { bubbles: true, composed: true, detail: { value: this.getSelected() } }));
    }
};
VlSelectRichComponent.compareValue = (value, oldValue) => {
    if (value instanceof FormData && oldValue instanceof FormData) {
        // We vergelijken de letterlijke inhoud van de entries van dit FormData object, omdat default FormData vergelijking niet voldoet
        return JSON.stringify([...value.entries()]) !== JSON.stringify([...oldValue.entries()]);
    }
    else {
        return value !== oldValue;
    }
};
VlSelectRichComponent = VlSelectRichComponent_1 = __decorate([
    webComponent('vl-select-rich'),
    __metadata("design:paramtypes", [])
], VlSelectRichComponent);
export { VlSelectRichComponent };
