// @domg-wc/components/atom/link/vl-link.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, ICON_PLACEMENT, webComponent } from '@domg-wc/common';
import { html, nothing } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import { vlIconStyles } from '../icon-style/vl-icon-style.css';
import { vlLinkIconStyles } from '../link-style/vl-link-icon-style.css';
import { vlLinkStyles } from '../link-style/vl-link-style.css';
import { buttonAsLinkStyles } from './vl-button-as-link.css';
import { linkDefaults } from './vl-link.defaults';
import { vlLinkFluxStyles } from './vl-link.flux-css';
let VlLinkComponent = class VlLinkComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.href = linkDefaults.href;
        this.label = linkDefaults.label;
        this.bold = linkDefaults.bold;
        this.small = linkDefaults.small;
        this.large = linkDefaults.large;
        this.error = linkDefaults.error;
        this.external = linkDefaults.external;
        this.icon = linkDefaults.icon;
        this.iconPlacement = linkDefaults.iconPlacement;
        this.buttonAsLink = linkDefaults.buttonAsLink;
    }
    static get styles() {
        return [vlLinkFluxStyles, vlLinkStyles(), vlLinkIconStyles, buttonAsLinkStyles, vlIconStyles];
    }
    static get properties() {
        return {
            href: { type: String },
            label: { type: String },
            block: { type: Boolean },
            bold: { type: Boolean },
            error: { type: Boolean },
            small: { type: Boolean },
            large: { type: Boolean },
            external: { type: Boolean },
            icon: { type: String },
            iconPlacement: { type: String, attribute: 'icon-placement' },
            buttonAsLink: { type: Boolean, attribute: 'button-as-link' },
        };
    }
    render() {
        const classes = {
            bold: this.bold,
            error: this.error,
            small: this.small,
            large: this.large,
        };
        const target = this.external ? '_blank' : nothing;
        const rel = this.external ? 'noopener noreferrer nofollow' : nothing;
        const positionIconBefore = this.iconPlacement !== ICON_PLACEMENT.AFTER;
        return !this.buttonAsLink
            ? html `
                  <a
                      class=${classMap(classes)}
                      href=${this.href}
                      target=${target}
                      rel=${rel}
                      part="link"
                      aria-label=${this.label || nothing}
                  >
                      ${positionIconBefore ? this.renderIcon() : nothing}
                      <slot></slot>
                      ${!positionIconBefore ? this.renderIcon() : nothing}
                      ${this.external ? this.renderExternalIcon() : ''}
                  </a>
              `
            : html `
                  <button
                      class="vl-button-as-link ${classMap(classes)}"
                      part="button"
                      aria-label=${this.label || nothing}
                  >
                      ${positionIconBefore ? this.renderIcon() : nothing}
                      <slot></slot>
                      ${!positionIconBefore ? this.renderIcon() : nothing}
                      ${this.external ? this.renderExternalIcon() : ''}
                  </button>
              `;
    }
    renderIcon() {
        const beforeClass = !this.buttonAsLink ? 'vl-icon--right-margin' : 'vl-link__icon--before';
        const afterClass = !this.buttonAsLink ? 'vl-icon--left-margin' : 'vl-link__icon--after';
        const classes = {
            'vl-icon': true,
            [`vl-icon--${this.icon}`]: true,
            'vl-link__icon': this.buttonAsLink,
            [beforeClass]: this.iconPlacement === 'before',
            [afterClass]: this.iconPlacement === 'after',
        };
        return this.icon ? html `<span class=${classMap(classes)} part="icon" aria-hidden="true"></span>` : nothing;
    }
    renderExternalIcon() {
        return html `<span class="vl-icon vl-icon--external vl-icon--after" part="icon" aria-hidden="true"></span>`;
    }
};
VlLinkComponent = __decorate([
    webComponent('vl-link')
], VlLinkComponent);
export { VlLinkComponent };
