// @domg/govflanders-style/component/form-structure/vl-form-structure.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-form {
  font-size: 1.6rem;
}
@media screen and (max-width: 767px) {
  .vl-form {
    font-size: 1.6rem;
  }
}
.vl-form__group {
  background-color: #f7f9fc;
  padding: 2rem 1.5rem;
  margin: 2rem 0 0;
}
@media screen and (max-width: 767px) {
  .vl-form__group {
    padding: 1.5rem 1rem;
  }
}
@media screen and (max-width: 767px) {
  .vl-form__group--mobile-full-width {
    padding-left: 3%;
    padding-right: 3%;
    position: relative;
    width: 100vw;
    left: 50%;
    right: 50%;
    margin-left: -50vw;
    margin-right: -50vw;
  }
}
.vl-form__row + .vl-form__row {
  margin-top: 1.5rem;
}
@media screen and (max-width: 767px) {
  .vl-form__row .vl-form__row__group + .vl-form__row__group {
    margin-top: 1.5rem;
  }
}
.vl-form__row--addon {
  display: flex;
  flex-direction: row;
}
.vl-form__row--addon .vl-form__addon {
  flex-grow: 0;
}
.vl-form__row--inline {
  display: flex;
  margin-left: -3rem;
}
.vl-form__row--inline::before, .vl-form__row--inline::after {
  content: "";
  display: table;
}
.vl-form__row--inline::after {
  clear: both;
}
@media screen and (max-width: 767px) {
  .vl-form__row--inline {
    display: block;
    margin-left: -1.5rem;
  }
}
.vl-form__row--inline .vl-form__label {
  float: left;
  width: 25%;
  min-width: 25%;
  max-width: 25%;
  flex-basis: 25%;
  overflow: hidden;
  text-overflow: ellipsis;
  padding-top: 0.4rem;
  padding-left: 3rem;
}
@media screen and (max-width: 767px) {
  .vl-form__row--inline .vl-form__label {
    display: block;
    width: auto;
    max-width: none;
    padding-left: 1.5rem;
  }
}
.vl-form__row--inline .vl-form__input {
  float: right;
  width: 75%;
  margin-left: 3rem;
}
@media screen and (max-width: 767px) {
  .vl-form__row--inline .vl-form__input {
    width: auto;
    margin-left: 1.5rem;
  }
}
.vl-form__row--inline .vl-form__input .vl-radio,
.vl-form__row--inline .vl-form__input .vl-checkbox,
.vl-form__row--inline .vl-form__input input[type=file] {
  margin-top: 0.6rem;
}
@media screen and (max-width: 767px) {
  .vl-form__row--inline .vl-form__input,
  .vl-form__row--inline .vl-form__label {
    float: none;
    width: auto;
  }
}
`;
export default style;
