// @domg/govflanders-style/component/data-table/vl-data-table.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-data-table {
  width: 100%;
  max-width: 100%;
}
.vl-data-table caption {
  text-align: left;
  margin: 15px 0 5px 0;
  font-size: 1.8rem;
  font-weight: 500;
}
.vl-data-table thead tr {
  border-top: 0.1rem #cbd2da solid;
  border-bottom: 0.2rem #cbd2da solid;
}
.vl-data-table tfoot tr {
  border-top: 0.2rem #cbd2da solid;
}
.vl-data-table tfoot td {
  font-weight: 500;
  white-space: nowrap;
}
.vl-data-table tfoot td:first-child {
  padding-left: 0;
}
.vl-data-table tfoot td:last-child {
  padding-right: 0;
}
.vl-data-table tbody tr.vl-data-table__element--error,
.vl-data-table tbody td.vl-data-table__element--error {
  background: #fbebec;
  border-bottom: 0.1rem solid #d2373c;
}
.vl-data-table tbody tr.vl-data-table__element--warning,
.vl-data-table tbody td.vl-data-table__element--warning {
  background: #fff6e7;
  border-bottom: 0.1rem solid #ffA10a;
}
.vl-data-table tbody tr.vl-data-table__element--success,
.vl-data-table tbody td.vl-data-table__element--success {
  background: #e6f5ed;
  border-bottom: 0.1rem solid #009e47;
}
.vl-data-table tbody tr {
  border-bottom: 0.1rem #cbd2da solid;
}
.vl-data-table tbody tr[data-vl-table-selectable] {
  cursor: pointer;
  transition: background 0.2s ease-in-out;
}
.vl-data-table tbody tr[data-vl-table-selectable]:hover {
  background: #f3f5f6;
}
.vl-data-table tbody tr.vl-data-table__grouped-row:not(.vl-data-table__grouped-row--last) {
  border-bottom: 0;
}
.vl-data-table tbody tr th {
  border-left: 0.2rem #cbd2da solid;
  border-right: 0.2rem #cbd2da solid;
}
.vl-data-table tbody tr th:first-child {
  border-left: none;
}
.vl-data-table tbody tr th:last-child {
  border-right: none;
}
.vl-data-table td,
.vl-data-table th {
  font-size: 1.6rem;
  line-height: 1.3;
  text-align: left;
  vertical-align: top;
  padding: 1.2rem 1rem;
}
@media screen and (max-width: 767px) {
  .vl-data-table td,
  .vl-data-table th {
    font-size: 1.4rem;
    padding: 1rem;
  }
}
.vl-data-table td.vl-data-table__icon-container,
.vl-data-table th.vl-data-table__icon-container {
  background-color: #f3f5f6;
  color: #333332;
}
.vl-data-table td.vl-data-table__icon-container .vl-vi,
.vl-data-table th.vl-data-table__icon-container .vl-vi {
  color: #687483;
  font-size: 3rem;
}
.vl-vi .vl-data-table td.vl-data-table__icon-container,
.vl-vi .vl-data-table th.vl-data-table__icon-container {
  text-align: center;
}
.vl-data-table th {
  font-weight: 500;
  white-space: nowrap;
}
.vl-data-table th > * {
  white-space: normal;
}
.vl-data-table .vl-data-table__grouped-row td {
  padding: 0.3rem 1rem 0.3rem 0;
}
@media screen and (max-width: 767px) {
  .vl-data-table .vl-data-table__grouped-row td {
    padding: 0.3rem 1rem 0.3rem 0;
  }
}
.vl-data-table .vl-data-table__grouped-row--first td {
  padding-top: 1.2rem;
}
@media screen and (max-width: 767px) {
  .vl-data-table .vl-data-table__grouped-row--first td {
    padding-top: 1rem;
  }
}
.vl-data-table .vl-data-table__grouped-row--last td {
  padding-bottom: 1.2rem;
}
@media screen and (max-width: 767px) {
  .vl-data-table .vl-data-table__grouped-row--last td {
    padding-bottom: 1rem;
  }
}
.vl-data-table__header-title--sortable {
  text-decoration: none;
}
.vl-data-table__header-title--sortable .vl-data-table__header-title__sort-icon {
  opacity: 0;
}
.vl-data-table__header-title--sortable:hover, .vl-data-table__header-title--sortable:focus {
  text-decoration: underline;
}
.vl-data-table__header-title--sortable:hover .vl-data-table__header-title__sort-icon, .vl-data-table__header-title--sortable:focus .vl-data-table__header-title__sort-icon {
  opacity: 0.5;
}
.vl-data-table__header-title--sortable-active .vl-data-table__header-title__sort-icon {
  opacity: 1;
}
.vl-data-table__body-title {
  max-width: 30rem;
}
.vl-data-table--alt tr th,
.vl-data-table--alt tr td {
  border-left: 0.2rem #cbd2da solid;
  border-right: 0.2rem #cbd2da solid;
}
.vl-data-table--alt tr th:first-child,
.vl-data-table--alt tr td:first-child {
  border-left: none;
}
.vl-data-table--alt tr th:last-child,
.vl-data-table--alt tr td:last-child {
  border-right: none;
}
.vl-data-table--alt tr th:not(:first-child) {
  padding: 0 1.2rem 1.2rem;
}
.vl-data-table--alt tr td:not(:first-child) {
  padding: 1.2rem;
}
.vl-data-table--double-spacing tr td,
.vl-data-table--double-spacing tr th {
  padding: 1.2rem 3rem;
}
.vl-data-table--no-header tbody tr:first-child {
  border-top: 3px #cbd2da solid;
}
.vl-data-table .vl-pill {
  vertical-align: middle;
}
@media screen and (max-width: 767px) {
  .vl-data-table .vl-pill {
    font-size: 1.4rem;
    height: 2rem;
    line-height: 2rem;
    padding: 0 0.5rem;
  }
}

.vl-u-table-overflow {
  width: 100%;
  overflow-x: auto;
}
.vl-u-table-overflow .vl-data-table {
  overflow: auto;
}

.no-js [data-vl-table-check-all] + span {
  display: none !important;
}

.vl-data-table--hover tbody tr {
  transition: background 0.2s ease-in-out;
}
.vl-data-table--hover tbody tr:hover {
  background: #f3f5f6;
}

.vl-data-table--matrix tr th {
  border-left: 0.2rem #cbd2da solid;
  border-right: 0.2rem #cbd2da solid;
}
.vl-data-table--matrix tr th:first-child {
  border-left: none;
}
.vl-data-table--matrix tr th:last-child {
  border-right: none;
}

.vl-data-table--grid td,
.vl-data-table--grid th {
  padding: 1.2rem;
  border-left: 0.1rem #cbd2da solid;
  border-right: 0.1rem #cbd2da solid;
}
.vl-data-table--grid td:first-child,
.vl-data-table--grid th:first-child {
  border-left: none;
}
.vl-data-table--grid td:last-child,
.vl-data-table--grid th:last-child {
  border-right: none;
}
.vl-data-table--grid th[scope=row] {
  border-width: 0.2rem;
}

.vl-data-table--zebra tbody tr:not(.vl-data-table__element--warning):not(.vl-data-table__element--error):not(.vl-data-table__element--success):nth-child(odd) {
  background-color: #f3f5f6;
}
.vl-data-table--zebra tbody tr:not(.vl-data-table__element--warning):not(.vl-data-table__element--error):not(.vl-data-table__element--success):nth-child(odd):hover {
  background-color: #edf0f2;
}

.vl-data-table__actions--top {
  margin: 0 0 2rem;
}
.vl-data-table__actions--bottom {
  margin: 2rem 0 0;
}
@media screen and (max-width: 500px) {
  .vl-data-table__actions--bottom .vl-data-table__actions__list {
    margin: 0 0 1rem;
  }
}
.vl-data-table__actions__list .vl-data-table__action:not(:last-child) {
  margin-right: 0.6rem;
}
.vl-data-table__action {
  display: inline-block;
  vertical-align: middle;
  margin-bottom: 0.6rem;
}
.vl-data-table__action__toggle {
  display: flex;
  align-items: center;
  background: none;
  font-family: "Flanders Art Sans", sans-serif;
  font-size: 1.6rem;
  font-weight: 400;
  line-height: 1;
  color: #333332;
  border: 0.1rem #cbd2da solid;
  text-decoration: none;
  text-align: left;
  cursor: pointer;
  padding: 0.9rem 1.5rem 0.8rem;
}
@media screen and (max-width: 767px) {
  .vl-data-table__action__toggle {
    font-size: 1.5rem;
  }
}
[data-vl-disable=true] .vl-data-table__action__toggle {
  color: #687483;
  cursor: default;
}
.vl-data-table__action__toggle__icon {
  font-size: 1.2rem;
}
.vl-data-table__action__toggle__icon:first-child {
  margin-right: 0.5rem;
}
.vl-data-table__action__toggle__icon:last-child {
  margin-left: 0.5rem;
}
@media screen and (max-width: 500px) {
  .vl-data-table__action__toggle--contract-xs span {
    display: none;
  }
  .vl-data-table__action__toggle--contract-xs .vl-vi::before {
    margin: 0;
  }
}

@media screen and (max-width: 500px) {
  .vl-data-table--collapsed-xs {
    position: relative;
  }
  .vl-data-table--collapsed-xs table,
  .vl-data-table--collapsed-xs thead,
  .vl-data-table--collapsed-xs tbody,
  .vl-data-table--collapsed-xs th,
  .vl-data-table--collapsed-xs td,
  .vl-data-table--collapsed-xs tr {
    display: block;
  }
  .vl-data-table--collapsed-xs thead tr {
    position: absolute;
    top: -9999px;
    left: -9999px;
  }
  .vl-data-table--collapsed-xs td {
    border-bottom: 1px solid #cbd2da;
    display: flex;
    gap: 10px;
    flex-grow: 2;
    text-align: left;
  }
  .vl-data-table--collapsed-xs td::before,
  .vl-data-table--collapsed-xs th::before {
    content: attr(data-title);
    position: relative;
    flex-shrink: 0;
    width: 35%;
    overflow: hidden;
    text-align: left;
    font-weight: 500;
  }
  .vl-data-table--collapsed-xs.vl-data-table--collapsed-xs td, .vl-data-table--collapsed-xs.vl-data-table--collapsed-xs th, .vl-data-table--collapsed-xs.vl-data-table--collapsed-s td, .vl-data-table--collapsed-xs.vl-data-table--collapsed-s th {
    padding: 1rem;
  }
  .vl-data-table--collapsed-xs.vl-data-table--collapsed-m td, .vl-data-table--collapsed-xs.vl-data-table--collapsed-m th {
    padding: 1.2rem;
  }
  .vl-data-table--collapsed-xs.vl-data-table--collapsed-truncate td::before,
  .vl-data-table--collapsed-xs.vl-data-table--collapsed-truncate th::before {
    white-space: nowrap;
    text-overflow: ellipsis;
  }
}
@media screen and (max-width: 767px) {
  .vl-data-table--collapsed-s {
    position: relative;
  }
  .vl-data-table--collapsed-s table,
  .vl-data-table--collapsed-s thead,
  .vl-data-table--collapsed-s tbody,
  .vl-data-table--collapsed-s th,
  .vl-data-table--collapsed-s td,
  .vl-data-table--collapsed-s tr {
    display: block;
  }
  .vl-data-table--collapsed-s thead tr {
    position: absolute;
    top: -9999px;
    left: -9999px;
  }
  .vl-data-table--collapsed-s td {
    border-bottom: 1px solid #cbd2da;
    display: flex;
    gap: 10px;
    flex-grow: 2;
    text-align: left;
  }
  .vl-data-table--collapsed-s td::before,
  .vl-data-table--collapsed-s th::before {
    content: attr(data-title);
    position: relative;
    flex-shrink: 0;
    width: 35%;
    overflow: hidden;
    text-align: left;
    font-weight: 500;
  }
  .vl-data-table--collapsed-s.vl-data-table--collapsed-xs td, .vl-data-table--collapsed-s.vl-data-table--collapsed-xs th, .vl-data-table--collapsed-s.vl-data-table--collapsed-s td, .vl-data-table--collapsed-s.vl-data-table--collapsed-s th {
    padding: 1rem;
  }
  .vl-data-table--collapsed-s.vl-data-table--collapsed-m td, .vl-data-table--collapsed-s.vl-data-table--collapsed-m th {
    padding: 1.2rem;
  }
  .vl-data-table--collapsed-s.vl-data-table--collapsed-truncate td::before,
  .vl-data-table--collapsed-s.vl-data-table--collapsed-truncate th::before {
    white-space: nowrap;
    text-overflow: ellipsis;
  }
}
@media screen and (max-width: 1023px) {
  .vl-data-table--collapsed-m {
    position: relative;
  }
  .vl-data-table--collapsed-m table,
  .vl-data-table--collapsed-m thead,
  .vl-data-table--collapsed-m tbody,
  .vl-data-table--collapsed-m th,
  .vl-data-table--collapsed-m td,
  .vl-data-table--collapsed-m tr {
    display: block;
  }
  .vl-data-table--collapsed-m thead tr {
    position: absolute;
    top: -9999px;
    left: -9999px;
  }
  .vl-data-table--collapsed-m td {
    border-bottom: 1px solid #cbd2da;
    display: flex;
    gap: 10px;
    flex-grow: 2;
    text-align: left;
  }
  .vl-data-table--collapsed-m td::before,
  .vl-data-table--collapsed-m th::before {
    content: attr(data-title);
    position: relative;
    flex-shrink: 0;
    width: 35%;
    overflow: hidden;
    text-align: left;
    font-weight: 500;
  }
  .vl-data-table--collapsed-m.vl-data-table--collapsed-xs td, .vl-data-table--collapsed-m.vl-data-table--collapsed-xs th, .vl-data-table--collapsed-m.vl-data-table--collapsed-s td, .vl-data-table--collapsed-m.vl-data-table--collapsed-s th {
    padding: 1rem;
  }
  .vl-data-table--collapsed-m.vl-data-table--collapsed-m td, .vl-data-table--collapsed-m.vl-data-table--collapsed-m th {
    padding: 1.2rem;
  }
  .vl-data-table--collapsed-m.vl-data-table--collapsed-truncate td::before,
  .vl-data-table--collapsed-m.vl-data-table--collapsed-truncate th::before {
    white-space: nowrap;
    text-overflow: ellipsis;
  }
}

.vl-data-table--sticky {
  table-layout: auto;
}
.vl-data-table--sticky .vl-data-table__cell--sticky {
  position: sticky;
  left: 0;
  background-color: inherit;
  border: 0;
  z-index: 1;
}
.vl-data-table--sticky thead .vl-data-table__cell--sticky {
  top: 0;
  z-index: 2;
}
.vl-data-table--sticky thead .vl-data-table__cell--sticky::before {
  display: block;
  position: absolute;
  bottom: -3px;
  left: 0;
  right: 0;
  height: 3px;
  background-color: #cbd2da;
  content: "";
}
.vl-data-table--sticky tbody .vl-data-table__cell--sticky::before {
  display: block;
  position: absolute;
  bottom: 0;
  top: 0;
  right: -1px;
  width: 1px;
  background-color: #cbd2da;
  content: "";
}
.vl-data-table--sticky th,
.vl-data-table--sticky td {
  min-width: 200px;
}
.vl-data-table--sticky tr {
  background-color: #fff;
}

.vl-data-table__sticky-wrapper {
  width: 100%;
  max-height: 75vh;
  overflow-x: auto;
  overflow-y: auto;
}

.vl-u-table-overflow {
  background: linear-gradient(to right, white, white, rgba(255, 255, 255, 0) 30px), radial-gradient(farthest-side at 0 50%, rgba(0, 0, 0, 0.3), rgba(255, 255, 255, 0)), linear-gradient(to left, white, white, rgba(255, 255, 255, 0) 30px), radial-gradient(farthest-side at 100% 50%, rgba(0, 0, 0, 0.3), rgba(255, 255, 255, 0)) 100%;
  background-color: white;
  background-repeat: no-repeat;
  background-attachment: local, scroll, local, scroll;
  background-size: 100% 100%, 15px 100%, 100% 100%, 15px 100%;
}

.vl-data-table__navigation {
  position: relative;
  z-index: 1;
  display: flex;
  justify-content: flex-end;
}
`;
export default style;
