// @domg/govflanders-style/component/input-addon/vl-input-addon.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-input-addon {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  height: 3.5rem;
  line-height: 3.5rem;
  min-width: 3.5rem;
  border: 0.1rem solid #8695a8;
  text-decoration: none;
  padding: 0 1rem;
  font-weight: 500;
  font-size: 1.6rem;
  color: #333332;
  background: #fff;
  margin: 0;
}
.vl-input-addon[type=button] .vl-vi, .vl-input-addon[href] .vl-vi {
  color: #05c;
}
.vl-input-addon[type=button]:hover, .vl-input-addon[href]:hover {
  box-shadow: inset 0 0 0 0.1rem rgba(0, 85, 204, 0.65);
  border-color: rgba(0, 85, 204, 0.65);
  background: rgba(179, 207, 245, 0.3);
}
.vl-input-addon[type=button]:hover .vl-vi, .vl-input-addon[type=button]:focus .vl-vi, .vl-input-addon[href]:hover .vl-vi, .vl-input-addon[href]:focus .vl-vi {
  color: #003bb0;
}
.vl-input-addon[type=button]:focus, .vl-input-addon[href]:focus {
  box-shadow: 0 0 0 2px #fff, 0 0 0 5px rgba(0, 85, 204, 0.65);
  outline: transparent solid 0.2rem;
}
@supports (outline-offset: 2px) {
  .vl-input-addon[type=button]:focus, .vl-input-addon[href]:focus {
    box-shadow: none;
    outline: 3px solid rgba(0, 85, 204, 0.65);
    outline-offset: 2px;
  }
}
@supports (outline-offset: 2px) {
  .vl-input-addon[type=button]:focus, .vl-input-addon[href]:focus {
    box-shadow: none;
    outline: 3px solid rgba(0, 85, 204, 0.65);
    outline-offset: 2px;
  }
}
.vl-input-addon[type=button]:hover:focus, .vl-input-addon[href]:hover:focus {
  box-shadow: inset 0 0 0 0.1rem rgba(0, 85, 204, 0.65), 0 0 0 2px #fff, 0 0 0 5px rgba(0, 85, 204, 0.65);
}
.vl-input-addon--disabled[type=button], .vl-input-addon--disabled[href] {
  background-color: #cbd2d9;
  cursor: not-allowed;
}
.vl-input-addon--disabled[type=button] .vl-vi, .vl-input-addon--disabled[href] .vl-vi {
  color: #687483;
}
.vl-input-addon--disabled[type=button]:hover, .vl-input-addon--disabled[type=button]:active, .vl-input-addon--disabled[href]:hover, .vl-input-addon--disabled[href]:active {
  background-color: #cbd2d9;
  border-color: #cbd2d9;
}
.vl-input-addon--disabled[type=button]:hover .vl-vi, .vl-input-addon--disabled[type=button]:active .vl-vi, .vl-input-addon--disabled[href]:hover .vl-vi, .vl-input-addon--disabled[href]:active .vl-vi {
  color: #687483;
}
.vl-input-addon--disabled[type=button]:focus .vl-vi, .vl-input-addon--disabled[href]:focus .vl-vi {
  color: #687483;
}
.vl-input-addon .vl-vi {
  display: inline-flex;
  color: inherit;
}
.vl-input-addon .vl-vi::before, .vl-input-addon .vl-vi::after {
  font-size: 1.8rem;
}
`;
export default style;
