// @domg-wc/components/block/popover/vl-floating-ui.controller.js
import { arrow, computePosition, flip, offset, platform, shift, } from '@floating-ui/dom';
import { offsetParent } from 'composed-offset-position';
import { POPOVER_ARIA_TYPE, POPOVER_TYPE } from './vl-popover.model';
export default class FloatingController {
    constructor(host, options) {
        this.hoverTimeout = 0;
        this.hostHover = false;
        this.handleClick = () => {
            this.host.toggle();
        };
        this.handleMouseOver = () => {
            clearTimeout(this.hoverTimeout);
            this.hoverTimeout = window.setTimeout(() => this.host.show(), this.options.showDelay);
        };
        this.handleMouseOut = () => {
            clearTimeout(this.hoverTimeout);
            this.hoverTimeout = window.setTimeout(() => {
                if (!this.hostHover)
                    this.host.hide();
                else
                    this.handleMouseOut();
            }, 
            // A minimum delay of 100ms is added to prevent the popover from hiding too quickly when moving the mouse from the reference to the popover.
            this.options.hideDelay > 100 ? this.options.hideDelay : 100);
        };
        this.handleHostMouseOver = () => {
            this.hostHover = true;
        };
        this.handleHostMouseOut = () => {
            this.hostHover = false;
        };
        this.handleFocusIn = () => {
            this.host.show();
        };
        this.handleFocusOut = () => {
            setTimeout(() => {
                if (!this.host.matches(':focus') && !this.host.matches(':focus-within')) {
                    this.host.hide();
                }
            }, 100);
        };
        this.handleHostFocusOut = () => {
            if (!this.host.matches(':focus-within')) {
                this.host.hide();
            }
        };
        this.handleClickOutside = (event) => {
            const target = event.target;
            if (!this.host.contains(target) && !this.getReferenceElement()?.contains(target) && event.button === 0) {
                this.host.hide();
            }
        };
        this.handleKeyDown = (event) => {
            if (this.host.open && event.key === 'Escape') {
                event.stopPropagation();
                this.host.hide();
            }
        };
        this.handleResize = () => {
            if (this.host.open) {
                this.host.hide();
            }
        };
        this.setExpanded = (expanded) => {
            if (!this.isTooltip) {
                this.getReferenceElement()?.setAttribute('aria-expanded', expanded ? 'true' : 'false');
            }
        };
        this.setOptions(options);
        this.host = host;
        this.host.addController(this);
    }
    get isTooltip() {
        return this.options.type === POPOVER_TYPE.TOOLTIP;
    }
    hostConnected() {
        this.addEventListeners();
        if (!this.host.id) {
            this.host.id = `popover-${Math.random().toString().substring(2, 10)}`;
        }
        if (this.isTooltip) {
            this.host.shadowRoot?.querySelector('.popover-content')?.setAttribute('role', 'tooltip');
            if (this.options.ariaType === POPOVER_ARIA_TYPE.DESCRIPTION) {
                this.addAttributeToReferenceElement('aria-describedby', this.host.id);
            }
            else if (this.options.ariaType === POPOVER_ARIA_TYPE.LABEL &&
                !this.getReferenceElement()?.hasAttribute('aria-label')) {
                this.addAttributeToReferenceElement('aria-labelledby', this.host.id);
            }
        }
        else {
            this.addAttributeToReferenceElement('aria-controls', this.host.id);
            this.addAttributeToReferenceElement('aria-haspopup', 'true');
        }
    }
    hostDisconnected() {
        this.removeEventListeners();
    }
    getReferenceElement() {
        const referenceId = `#${this.options.reference}`;
        const hostRootNode = this.host.getRootNode();
        const referenceElement = document.querySelector(referenceId) || hostRootNode.querySelector(referenceId);
        return referenceElement ? referenceElement : null;
    }
    getArrowElement() {
        return this.host.shadowRoot.querySelector('i#popover-arrow');
    }
    buildMiddlewares() {
        return [
            // https://floating-ui.com/docs/offset
            // offset() should generally be placed at the beginning of your middleware array.
            offset(this.options.distance),
            // https://floating-ui.com/docs/flip
            flip(),
            // https://floating-ui.com/docs/shift
            shift(),
            // https://floating-ui.com/docs/arrow
            // arrow() should generally be placed toward the end of your middleware array, after shift().
            arrow({ element: this.getArrowElement(), padding: 8 }),
        ];
    }
    async updatePosition() {
        if (!this.getReferenceElement) {
            console.warn(this.host.tagName, ' could not find reference element with id: #', this.options?.reference);
            return;
        }
        const { x, y, placement, middlewareData } = await computePosition(this.getReferenceElement(), this.host, {
            placement: this.options.placement,
            middleware: this.buildMiddlewares(),
            platform: {
                ...platform,
                getOffsetParent: (element) => platform.getOffsetParent(element, offsetParent),
            },
        });
        Object.assign(this.host.style, {
            position: this.options.strategy,
            left: `${x}px`,
            top: `${y}px`,
        });
        const staticSide = { top: 'bottom', right: 'left', bottom: 'top', left: 'right' }[placement.split('-')[0]];
        if (middlewareData.arrow && this.getArrowElement()) {
            const { x, y } = middlewareData.arrow;
            Object.assign(this.getArrowElement().style, {
                left: x != null ? `${x}px` : '',
                top: y != null ? `${y}px` : '',
                [staticSide]: `${-this.getArrowElement().offsetWidth / 2}px`,
            });
        }
    }
    setOptions(options) {
        this.options = options;
    }
    addEventListeners() {
        const referenceElement = this.getReferenceElement();
        referenceElement?.addEventListener('keydown', this.handleKeyDown);
        if (this.hasTrigger('click')) {
            referenceElement?.addEventListener('click', this.handleClick);
            document.addEventListener('click', this.handleClickOutside, true);
            if (this.options.hideOnClick)
                this.host.addEventListener('click', this.host.hide);
        }
        if (this.hasTrigger('hover')) {
            referenceElement?.addEventListener('mouseover', this.handleMouseOver);
            referenceElement?.addEventListener('mouseout', this.handleMouseOut);
        }
        if (this.hasTrigger('focus')) {
            referenceElement?.addEventListener('focusin', this.handleFocusIn, true);
            referenceElement?.addEventListener('focusout', this.handleFocusOut, true);
        }
        this.host.addEventListener('mouseover', this.handleHostMouseOver);
        this.host.addEventListener('mouseout', this.handleHostMouseOut);
        this.host.addEventListener('focusout', this.handleHostFocusOut);
        window.addEventListener('resize', this.handleResize);
    }
    removeEventListeners() {
        const referenceElement = this.getReferenceElement();
        if (!referenceElement) {
            console.warn(this.host.tagName, ' could not find reference element with id: #', this.options?.reference);
            return;
        }
        referenceElement.removeEventListener('keydown', this.handleKeyDown);
        if (this.hasTrigger('click')) {
            referenceElement.removeEventListener('click', this.handleClick);
            document.removeEventListener('click', this.handleClickOutside, true);
            this.host.removeEventListener('click', this.host.hide);
        }
        if (this.hasTrigger('hover')) {
            referenceElement.removeEventListener('mouseover', this.handleMouseOver);
            referenceElement.removeEventListener('mouseout', this.handleMouseOut);
        }
        if (this.hasTrigger('focus')) {
            referenceElement.removeEventListener('focusin', this.handleFocusIn, true);
            referenceElement.removeEventListener('focusout', this.handleFocusOut, true);
        }
        this.host.removeEventListener('mouseover', this.handleHostMouseOver);
        this.host.removeEventListener('mouseout', this.handleHostMouseOut);
        this.host.removeEventListener('focusout', this.handleHostFocusOut);
        window.removeEventListener('resize', this.handleResize);
    }
    hasTrigger(trigger) {
        return this.options.trigger.split(' ').includes(trigger);
    }
    addAttributeToReferenceElement(attribute, value) {
        if (!this.getReferenceElement()?.hasAttribute(attribute)) {
            this.getReferenceElement()?.setAttribute(attribute, value);
        }
    }
}
