// @domg-wc/components/block/popover/vl-floating-ui.controller.js
import { isSafari } from '@domg-wc/common';
import { arrow, computePosition, flip, offset, platform, shift, } from '@floating-ui/dom';
import { offsetParent } from 'composed-offset-position';
import { POPOVER_ARIA_TYPE, POPOVER_TYPE } from './vl-popover.model';
export default class FloatingController {
    constructor(host, options) {
        this.hoverTimeout = 0;
        this.hostHover = false;
        this.ignoreNextClick = false;
        this.handleClick = () => {
            if (this.ignoreNextClick) {
                this.ignoreNextClick = false;
                return;
            }
            this.host.toggle();
        };
        this.handleHideOnClick = () => {
            this.hide();
        };
        this.handleMouseOver = () => {
            clearTimeout(this.hoverTimeout);
            this.hoverTimeout = window.setTimeout(() => this.show(), this.options.showDelay);
        };
        this.handleMouseOut = () => {
            clearTimeout(this.hoverTimeout);
            this.hoverTimeout = window.setTimeout(() => {
                if (!this.hostHover)
                    this.hide();
                else
                    this.handleMouseOut();
            }, 
            // A minimum delay of 100ms is added to prevent the popover from hiding too quickly when moving the mouse from the reference to the popover.
            this.options.hideDelay > 100 ? this.options.hideDelay : 100);
        };
        this.handleHostMouseOver = () => {
            this.hostHover = true;
        };
        this.handleHostMouseOut = () => {
            this.hostHover = false;
        };
        this.handleFocusIn = () => {
            this.show();
            // Ignore the next click coming from the same event
            this.ignoreNextClick = true;
        };
        this.handleFocusOut = () => {
            setTimeout(() => {
                if (!this.host.matches(':focus') && !this.host.matches(':focus-within')) {
                    this.hide();
                }
            }, 100);
        };
        this.handleHostFocusOut = ({ relatedTarget }) => {
            if (isSafari && relatedTarget instanceof HTMLElement && this.host.contains(relatedTarget)) {
                // In Safari, tabbing from one popover-action to another triggers the focusout event.
                // As a result the popover closes when you tab from one action to another, which would make any action but
                // the first one unreachable using the keyboard. If the focus stays within the popover, we don't hide it.
                return;
            }
            if (!this.host.matches(':focus-within')) {
                this.hide();
            }
        };
        this.handleClickOutside = (e) => {
            const path = e.composedPath();
            if (path.includes(this.referenceElement))
                return;
            if (path.includes(this.triggerButton))
                return;
            if (path.includes(this.host))
                return;
            this.host.blur();
            this.hide();
        };
        this.handleKeyDown = (event) => {
            if (this.host.open && event.key === 'Escape') {
                event.stopPropagation();
                this.hide();
            }
        };
        this.handleResize = () => {
            if (this.host.open) {
                this.hide();
            }
        };
        this.setExpanded = (expanded) => {
            if (!this.isTooltip) {
                this.triggerButton.setAttribute('aria-expanded', expanded ? 'true' : 'false');
            }
        };
        this.setOptions(options);
        this.host = host;
        this.host.addController(this);
    }
    get isTooltip() {
        return this.options.type === POPOVER_TYPE.TOOLTIP;
    }
    hostConnected() {
        this.addEventListeners();
        if (!this.host.id) {
            this.host.id = `popover-${Math.random().toString().substring(2, 10)}`;
        }
        if (this.isTooltip) {
            this.host.shadowRoot?.querySelector('.popover-content')?.setAttribute('role', 'tooltip');
            if (this.options.ariaType === POPOVER_ARIA_TYPE.DESCRIPTION) {
                this.addAttributeToTrigger('aria-describedby', this.host.id);
            }
            else if (this.options.ariaType === POPOVER_ARIA_TYPE.LABEL &&
                !this.getReferenceElement()?.hasAttribute('aria-label')) {
                this.addAttributeToTrigger('aria-labelledby', this.host.id);
            }
        }
        else {
            this.addAttributeToTrigger('aria-controls', this.host.id);
            this.addAttributeToTrigger('aria-haspopup', 'true');
        }
    }
    hostDisconnected() {
        this.removeEventListeners();
    }
    // For popovers that are not tooltips, we need to distinguish between the reference element itself and the trigger button
    // The trigger button:
    // - captures the events
    // - is the button on which we need to set the aria attributes and the event listeners
    // The reference element:
    // - is the element that is used to position the popover
    // - captures the keydown event to close the popover on 'Escape'
    // They are the same element if the reference element is a button
    get triggerButton() {
        if (this.isTooltip) {
            // For tooltips, if the reference element is not a button, we can use the reference element itself as the trigger
            return this.referenceElement;
        }
        if (this.referenceElement instanceof HTMLButtonElement) {
            return this.referenceElement;
        }
        if (this.referenceElement instanceof HTMLElement && this.referenceElement.getAttribute('role') === 'button') {
            return this.referenceElement;
        }
        // If the reference element is not a button, then find the first button within the reference element.
        // This is necessary to place the aria attributes. These must be on the button itself.
        const firstButton = this.referenceElement.querySelector('button') ||
            this.referenceElement.shadowRoot?.querySelector('button') ||
            this.referenceElement.querySelector('[role="button"]') ||
            this.referenceElement.shadowRoot?.querySelector('[role="button"]') ||
            null;
        if (!firstButton) {
            const message = `Het referentie element (#${this.options.reference}) bevat geen button. Gebruik een button of pas \`role="button"\` correct toe (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/button_role).`;
            console.warn(this.host.tagName, message);
            return this.referenceElement;
        }
        return firstButton;
    }
    get referenceElement() {
        return this.getReferenceElement();
    }
    getReferenceElement() {
        const referenceId = `#${this.options.reference}`;
        const hostRootNode = this.host.getRootNode();
        const referenceElement = document.querySelector(referenceId) || hostRootNode.querySelector(referenceId);
        if (!referenceElement) {
            const message = `Het referentie element (#${this.options.reference}) kon niet gevonden worden.`;
            console.warn(this.host.tagName, message);
            // Fallback to the parent element of the host if the reference element is not found
            return this.host.parentElement;
        }
        return referenceElement;
    }
    getArrowElement() {
        return this.host.shadowRoot.querySelector('i#popover-arrow');
    }
    buildMiddlewares() {
        return [
            // https://floating-ui.com/docs/offset
            // offset() should generally be placed at the beginning of your middleware array.
            offset(this.options.distance),
            // https://floating-ui.com/docs/flip
            flip(),
            // https://floating-ui.com/docs/shift
            shift(),
            // https://floating-ui.com/docs/arrow
            // arrow() should generally be placed toward the end of your middleware array, after shift().
            arrow({ element: this.getArrowElement(), padding: 8 }),
        ];
    }
    async updatePosition() {
        const { x, y, placement, middlewareData } = await computePosition(this.referenceElement, this.host, {
            placement: this.options.placement,
            middleware: this.buildMiddlewares(),
            platform: {
                ...platform,
                getOffsetParent: (element) => platform.getOffsetParent(element, offsetParent),
            },
        });
        Object.assign(this.host.style, {
            position: this.options.strategy,
            left: `${x}px`,
            top: `${y}px`,
        });
        const staticSide = { top: 'bottom', right: 'left', bottom: 'top', left: 'right' }[placement.split('-')[0]];
        if (middlewareData.arrow && this.getArrowElement()) {
            const { x, y } = middlewareData.arrow;
            Object.assign(this.getArrowElement().style, {
                left: x != null ? `${x}px` : '',
                top: y != null ? `${y}px` : '',
                [staticSide]: `${-this.getArrowElement().offsetWidth / 2}px`,
            });
        }
    }
    setOptions(options) {
        this.options = options;
    }
    show() {
        this.host.show();
        document.addEventListener('click', this.handleClickOutside, true);
    }
    hide() {
        this.host.hide();
        document.removeEventListener('click', this.handleClickOutside, true);
    }
    toggle() {
        if (this.host.open) {
            this.hide();
        }
        else {
            this.show();
        }
    }
    addEventListeners() {
        this.referenceElement.addEventListener('keydown', this.handleKeyDown);
        if (this.hasTrigger('click')) {
            this.triggerButton.addEventListener('click', this.handleClick);
            document.addEventListener('click', this.handleClickOutside, true);
        }
        if (this.options.hideOnClick) {
            this.host.addEventListener('click', this.handleHideOnClick);
        }
        if (this.hasTrigger('hover')) {
            this.triggerButton.addEventListener('mouseover', this.handleMouseOver);
            this.triggerButton.addEventListener('mouseout', this.handleMouseOut);
        }
        this.triggerButton.addEventListener('focusin', this.handleFocusIn, true);
        this.triggerButton.addEventListener('focusout', this.handleFocusOut, true);
        this.host.addEventListener('mouseover', this.handleHostMouseOver);
        this.host.addEventListener('mouseout', this.handleHostMouseOut);
        this.host.addEventListener('focusout', this.handleHostFocusOut);
        this.host.addEventListener('keydown', this.handleKeyDown);
        window.addEventListener('resize', this.handleResize);
    }
    removeEventListeners() {
        this.referenceElement.removeEventListener('keydown', this.handleKeyDown);
        if (this.hasTrigger('click')) {
            this.triggerButton.removeEventListener('click', this.handleClick);
            document.removeEventListener('click', this.handleClickOutside, true);
        }
        this.host.removeEventListener('click', this.handleHideOnClick);
        if (this.hasTrigger('hover')) {
            this.triggerButton.removeEventListener('mouseover', this.handleMouseOver);
            this.triggerButton.removeEventListener('mouseout', this.handleMouseOut);
        }
        this.triggerButton.removeEventListener('focusin', this.handleFocusIn, true);
        this.triggerButton.removeEventListener('focusout', this.handleFocusOut, true);
        this.host.removeEventListener('mouseover', this.handleHostMouseOver);
        this.host.removeEventListener('mouseout', this.handleHostMouseOut);
        this.host.removeEventListener('focusout', this.handleHostFocusOut);
        this.host.removeEventListener('keydown', this.handleKeyDown);
        window.removeEventListener('resize', this.handleResize);
    }
    hasTrigger(trigger) {
        return this.options.trigger.includes(trigger);
    }
    addAttributeToTrigger(attribute, value) {
        if (!this.triggerButton.hasAttribute(attribute)) {
            this.triggerButton.setAttribute(attribute, value);
        }
    }
}
