// @domg-wc/components/block/popover/vl-popover-action.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement } from '@domg-wc/common';
import { vlIconStyles } from '@domg-wc/components/atom';
import { resetStyle } from '@domg/govflanders-style/common';
import { html, nothing } from 'lit';
import { customElement } from 'lit/decorators.js';
import { vlPopoverActionFluxStyles } from './vl-popover-action.flux-css';
let VlPopoverActionComponent = class VlPopoverActionComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.selected = false;
        this.icon = '';
    }
    static get styles() {
        return [resetStyle, vlIconStyles, vlPopoverActionFluxStyles];
    }
    static get properties() {
        return {
            icon: { type: String, reflect: true },
            action: { type: String },
            selected: { type: Boolean },
            href: { type: String },
            target: { type: String },
            rel: { type: String },
        };
    }
    updated(changedProperties) {
        if (changedProperties.has('selected')) {
            if (this.selected) {
                this.setAttribute('aria-selected', 'true');
            }
            else {
                this.removeAttribute('aria-selected');
            }
        }
    }
    renderIcon() {
        return this.icon && this.icon !== '' ? html `<span class="vl-icon vl-icon--${this.icon}"></span>` : nothing;
    }
    render() {
        return this.href
            ? html ` <a
                  href="${this.href}"
                  target="${this.target || nothing}"
                  rel="${this.rel || nothing}"
                  role="menuitem"
              >
                  ${this.renderIcon()} <span><slot></slot></span>
              </a>`
            : html ` <button type="button" role="menuitem">
                  ${this.renderIcon()} <span><slot></slot></span>
              </button>`;
    }
};
VlPopoverActionComponent = __decorate([
    customElement('vl-popover-action')
], VlPopoverActionComponent);
export { VlPopoverActionComponent };
