// @domg-wc/components/atom/paragraph/vl-paragraph.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, webComponent } from '@domg-wc/common';
import { html } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import { vlParagraphStyles } from '../paragraph-style/vl-paragraph-style.css';
import { paragraphDefaults } from './vl-paragraph.defaults';
let VlParagraphComponent = class VlParagraphComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.bold = paragraphDefaults.bold;
        this.introduction = paragraphDefaults.introduction;
    }
    static get styles() {
        return [vlParagraphStyles];
    }
    static get properties() {
        return {
            bold: { type: Boolean },
            introduction: { type: Boolean },
        };
    }
    render() {
        const classes = {
            bold: this.bold,
            introduction: this.introduction,
        };
        return html `
            <p class=${classMap(classes)}>
                <slot></slot>
            </p>
        `;
    }
};
VlParagraphComponent = __decorate([
    webComponent('vl-paragraph')
], VlParagraphComponent);
export { VlParagraphComponent };
