// @domg/govflanders-style/component/progress-bar/vl-progress-bar.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-u-text--ellipse {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.vl-u-text--uppercase {
  text-transform: uppercase;
}

.vl-u-text--lowercase {
  text-transform: lowercase;
}

.vl-u-text--capitalize::first-letter {
  text-transform: uppercase;
}

.vl-u-text--success {
  color: #007a37;
}

.vl-u-text--warning {
  color: #9f5804;
}

.vl-u-text--error {
  color: #d2373c;
}

.vl-u-text--bold, .vl-progress-bar__step--active .vl-progress-bar__bullet__text {
  font-weight: 500;
}

.vl-u-text--italic {
  font-style: italic;
}

.vl-u-text--small {
  font-size: 1.4rem !important;
}

.vl-u-text--medium {
  font-size: 1.6rem !important;
}

.vl-u-text--xsmall {
  font-size: 1.2rem !important;
}

.vl-u-text--xxsmall {
  font-size: 1rem !important;
}

mark,
.vl-u-text--marked {
  background-color: transparent;
  box-shadow: inset 0 -10px 0 0 rgba(255, 197, 21, 0.3);
}

.vl-progress-bar {
  display: table;
  table-layout: fixed;
  width: 100%;
  height: 2.2rem;
  margin-bottom: 2.7rem;
}
@media screen and (max-width: 767px) {
  .vl-progress-bar {
    height: 1.8rem;
    margin-bottom: 0px;
  }
}
.vl-progress-bar__step {
  position: relative;
  display: table-cell;
}
.vl-progress-bar__step::before {
  display: block;
  position: absolute;
  background-color: #8695a8;
  content: "";
  height: 0.3rem;
  top: 50%;
  right: 2.3rem;
  left: 0.2rem;
  transform: translate(0%, -50%);
}
@media screen and (max-width: 767px) {
  .vl-progress-bar__step::before {
    right: 1.9rem;
  }
}
.vl-progress-bar__step--active::before {
  background: #8695a8;
  right: 2.7rem;
}
@media screen and (max-width: 767px) {
  .vl-progress-bar__step--active::before {
    right: 2.3rem;
  }
}
.vl-progress-bar__step--active .vl-progress-bar__bullet {
  background-color: #8695a8;
  border-color: #8695a8;
  width: 2.2rem;
  height: 2.2rem;
}
@media screen and (max-width: 767px) {
  .vl-progress-bar__step--active .vl-progress-bar__bullet {
    width: 1.8rem;
    height: 1.8rem;
  }
}
.vl-progress-bar__step--active ~ .vl-progress-bar__step .vl-progress-bar__bullet, .vl-progress-bar__step--active ~ .vl-progress-bar__step::before {
  background-color: #cfd5dd;
  border-color: #cfd5dd;
}
.vl-region--alt .vl-progress-bar__step--active ~ .vl-progress-bar__step .vl-progress-bar__bullet, .vl-region--alt .vl-progress-bar__step--active ~ .vl-progress-bar__step::before {
  background-color: #cfd5dd;
}
.vl-progress-bar__step--active ~ .vl-progress-bar__step .vl-progress-bar__bullet::after {
  color: #687483;
}
.vl-progress-bar__step--active ~ .vl-progress-bar__step .vl-progress-bar__bullet__text {
  color: #687483;
}
.vl-progress-bar__step:last-child .vl-progress-bar__bullet__text {
  max-width: 8rem;
  overflow: hidden;
  text-overflow: ellipsis;
}
.vl-progress-bar__bullet {
  position: absolute;
  background-color: #8695a8;
  width: 1.8rem;
  height: 1.8rem;
  border-radius: 50%;
  top: 50%;
  right: 0.2rem;
  appearance: none;
  border-style: solid;
  border-color: #8695a8;
  padding: 0;
  transform: translate(0%, -50%);
}
@media screen and (max-width: 767px) {
  .vl-progress-bar__bullet {
    width: 1.4rem;
    height: 1.4rem;
  }
}
.vl-progress-bar__bullet::after {
  color: #333332;
}
.vl-progress-bar__bullet__text {
  position: absolute;
  transform: translate(-50%, 1.6rem);
  font-size: 1.6rem;
  white-space: nowrap;
}
.vl-progress-bar a .vl-progress-bar__bullet:hover::after {
  color: #003bb0;
}
.vl-progress-bar--numeric {
  counter-reset: progressbar-counter;
  height: 4.2rem;
  margin-bottom: 2.6rem;
}
@media screen and (max-width: 767px) {
  .vl-progress-bar--numeric {
    height: 3.5rem;
    margin-bottom: 0px;
  }
}
.vl-progress-bar--numeric .vl-progress-bar__step::before {
  right: 3.7rem;
}
@media screen and (max-width: 767px) {
  .vl-progress-bar--numeric .vl-progress-bar__step::before {
    right: 3rem;
  }
}
.vl-progress-bar--numeric .vl-progress-bar__step .vl-progress-bar__bullet {
  width: 3.2rem;
  height: 3.2rem;
  text-decoration: none;
  background-color: var(--vl-theme-primary-color);
  border-color: var(--vl-theme-primary-color);
}
@media screen and (max-width: 767px) {
  .vl-progress-bar--numeric .vl-progress-bar__step .vl-progress-bar__bullet {
    width: 2.5rem;
    height: 2.5rem;
  }
}
.vl-progress-bar--numeric .vl-progress-bar__step .vl-progress-bar__bullet::after {
  position: absolute;
  content: counter(progressbar-counter);
  counter-increment: progressbar-counter;
  font-size: 1.8rem;
  font-weight: 500;
  width: 100%;
  height: 100%;
  text-align: center;
  top: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
}
@media screen and (max-width: 767px) {
  .vl-progress-bar--numeric .vl-progress-bar__step .vl-progress-bar__bullet::after {
    font-size: 1.4rem;
  }
}
.vl-progress-bar--numeric .vl-progress-bar__step .vl-progress-bar__bullet__text {
  transform: translate(-50%, 2.6rem);
}
.vl-progress-bar--numeric .vl-progress-bar__step--active::before {
  right: 4.7rem;
}
@media screen and (max-width: 767px) {
  .vl-progress-bar--numeric .vl-progress-bar__step--active::before {
    right: 4rem;
  }
}
.vl-progress-bar--numeric .vl-progress-bar__step--active .vl-progress-bar__bullet {
  width: 4.2rem;
  height: 4.2rem;
}
@media screen and (max-width: 767px) {
  .vl-progress-bar--numeric .vl-progress-bar__step--active .vl-progress-bar__bullet {
    width: 3.5rem;
    height: 3.5rem;
  }
}
.vl-progress-bar--numeric .vl-progress-bar__step--active .vl-progress-bar__bullet::after {
  font-size: 2.2rem;
}
@media screen and (max-width: 767px) {
  .vl-progress-bar--numeric .vl-progress-bar__step--active .vl-progress-bar__bullet::after {
    font-size: 1.8rem;
  }
}
.vl-progress-bar--numeric .vl-progress-bar__step--active ~ .vl-progress-bar__step .vl-progress-bar__bullet, .vl-progress-bar--numeric .vl-progress-bar__step--active ~ .vl-progress-bar__step::before {
  background-color: #cfd5dd;
  border-color: #cfd5dd;
}
`;
export default style;
