// @domg/govflanders-style/component/search/vl-search.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-search {
  position: relative;
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 2rem 1.5rem;
  background-color: #f7f9fc;
}
.vl-search .vl-search__reset {
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  right: 0px;
  z-index: 1;
  background: none;
  border: none;
  background-image: url("data:image/svg+xml,%3Csvg width='10' height='11' viewBox='0 0 10 11' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath fill-rule='evenodd' clip-rule='evenodd' d='M8.70383 0.5L5 4.20383L1.29617 0.5L0 1.79617L3.70386 5.5L0 9.20383L1.29617 10.5L5 6.79614L8.70383 10.5L10 9.20383L6.29617 5.5L10 1.79617L8.70383 0.5Z' fill='%238695A8'/%3E%3C/svg%3E");
  background-repeat: no-repeat;
  background-position: center center;
  background-size: 10px 11px;
  min-height: 2.5rem;
  width: 2.5rem;
}
@media screen and (max-width: 767px) {
  .vl-search {
    display: block;
    padding: 1.5rem 1rem;
  }
}
.vl-search .vl-search__label {
  color: #687483;
  font-weight: 400;
  font-size: 1.6rem;
  line-height: 1;
  white-space: nowrap;
}
@media screen and (max-width: 767px) {
  .vl-search .vl-search__label {
    font-size: 1.6rem;
  }
}
.vl-search .vl-search__input-wrapper {
  display: flex;
  width: 100%;
  align-items: center;
  position: relative;
}
@media screen and (max-width: 767px) {
  .vl-search .vl-search__input-wrapper {
    flex-direction: column;
    align-items: flex-start;
  }
}
.vl-search .vl-search__input {
  flex: 6;
  margin: 0 2rem;
}
@media screen and (max-width: 1023px) {
  .vl-search .vl-search__input {
    margin: 0 1rem;
    flex: 4;
  }
}
@media screen and (max-width: 767px) {
  .vl-search .vl-search__input {
    margin: 0;
    width: 100%;
    margin-bottom: 1.5rem;
  }
}
.vl-search .vl-search__reset {
  right: 25px;
}
@media screen and (max-width: 1023px) {
  .vl-search .vl-search__reset {
    right: 15px;
  }
}
@media screen and (max-width: 767px) {
  .vl-search .vl-search__reset {
    right: 5px;
    top: calc(50% - 0.75rem);
  }
}
.vl-search.vl-search--alt {
  background-color: transparent;
  padding: 0;
  background: none;
}
@media screen and (max-width: 767px) {
  .vl-search.vl-search--alt {
    padding: 0;
  }
}
.vl-search--inline {
  background-color: transparent;
}
@media screen and (max-width: 767px) {
  .vl-search--inline {
    display: block;
    width: auto;
  }
}
.vl-search--inline .vl-search__label {
  display: block;
  padding: 0;
  position: absolute;
  top: 0;
  right: 0;
  height: 100%;
  width: 4.3rem;
  color: #fff;
  font-size: 0 !important;
}
.vl-search--inline .vl-search__submit {
  padding: 0;
  position: absolute;
  right: 5px;
  font-size: 1.9rem;
  z-index: 1;
  top: 50%;
  min-height: 2.5rem;
  width: 2.5rem;
  transform: translateY(-50%);
}
.vl-search--inline .vl-search__label {
  color: #333332;
}
.vl-search--inline .vl-search__reset {
  right: 35px;
  top: 50%;
}
.vl-search--inline .vl-search__input {
  padding-right: 30px;
  flex: unset;
  margin: 0;
  text-overflow: ellipsis;
  width: 100%;
}
.vl-search--inline .vl-search__input:focus, .vl-search--inline .vl-search__input:hover {
  padding-right: 30px;
}
.vl-search--inline.vl-search--has-input-value .vl-search__input {
  padding-right: 60px;
}
.vl-search--inline.vl-search--has-input-value .vl-search__input:focus, .vl-search--inline.vl-search--has-input-value .vl-search__input:hover {
  padding-right: 60px;
}
.vl-search--inline .vl-search__submit:focus {
  transition: none;
}
.vl-search--inline .vl-search__input-wrapper {
  display: block;
  width: 100%;
  text-align: left;
  position: relative;
}
`;
export default style;
