// @domg/govflanders-style/common/layout.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-page {
  position: relative;
  width: 100%;
  background-color: #fff;
}
.vl-page .vl-main-content > .vl-region:last-child {
  padding-bottom: 10rem;
}
@media screen and (max-width: 767px) {
  .vl-page .vl-main-content > .vl-region:last-child {
    padding-bottom: 7rem;
  }
}

.vl-main-content {
  outline: none;
  position: relative;
}

.vl-region {
  margin: 0 auto;
  padding: 3rem 0 6rem;
}
@media screen and (max-width: 767px) {
  .vl-region {
    padding: 3rem 0;
  }
}

.vl-region:not(.vl-region--alt) + .vl-region:not(.vl-region--alt) {
  padding-top: 0;
}

.vl-region--alt {
  background-color: #f7f9fc;
}

.vl-region--overlap {
  background: linear-gradient(to bottom, transparent calc(50% - 30px), #f7f9fc calc(50% - 30px), #f7f9fc 100%);
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr="#00000000", endColorstr="#000000",GradientType=0 );
}
.vl-region--overlap .vl-layout {
  border: 1px #cbd2da solid;
  padding-top: 50px;
  padding-bottom: 50px;
  background: #fff;
}
@media only screen and (max-width: 1295px) {
  .vl-region--overlap .vl-layout {
    margin: 15px;
  }
}
@media screen and (max-width: 1023px) {
  .vl-region--overlap .vl-layout {
    padding-top: 20px;
    padding-bottom: 20px;
  }
}

.vl-region--overlap + .vl-region--alt {
  padding-top: 0 !important;
}

.vl-region:not(.vl-region--alt) + .vl-region--alt,
.vl-region--alt + .vl-region:not(.vl-region--alt) {
  padding-top: 6rem;
}
@media screen and (max-width: 767px) {
  .vl-region:not(.vl-region--alt) + .vl-region--alt,
  .vl-region--alt + .vl-region:not(.vl-region--alt) {
    padding-top: 3rem;
  }
}

.vl-region:first-child {
  padding-top: 6rem;
}
@media screen and (max-width: 767px) {
  .vl-region:first-child {
    padding-top: 2rem;
  }
}

.vl-region--small {
  padding: 1.5rem 0;
}
@media screen and (max-width: 767px) {
  .vl-region--small {
    padding: 2rem 0;
  }
}

.vl-region--medium {
  padding: 3rem 0;
}
@media screen and (max-width: 767px) {
  .vl-region--medium {
    padding: 2rem 0;
  }
}

.vl-region--bordered + .vl-region--bordered {
  border-top: 1px solid #f7f9fc;
}
.vl-region--bordered + .vl-region--bordered.vl-region--alt {
  border-top-color: #fff;
}

.vl-region.vl-region--no-space {
  padding: 0;
}

.vl-region.vl-region--no-space-bottom {
  padding-bottom: 0;
}

.vl-region.vl-region--no-space-top {
  padding-top: 0;
}

.vl-layout {
  position: relative;
  margin: 0 auto;
  min-width: 1024px;
  max-width: 1280px;
  padding: 0 3rem;
}
@media screen and (max-width: 1023px) {
  .vl-layout {
    width: auto;
    min-width: 768px;
    max-width: 1280px;
  }
}
@media screen and (max-width: 767px) {
  .vl-layout {
    width: auto;
    min-width: 0;
    padding: 0 1.5rem;
  }
}
`;
export default style;
