// @domg/govflanders-style/component/popover/vl-popover.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

@keyframes slideIn {
  0% {
    display: none;
  }
  1% {
    display: block;
    opacity: 0;
  }
  100% {
    opacity: 1;
    transform: translateY(0%);
  }
}
@keyframes slideOut {
  0% {
    transform: translateY(0%);
    opacity: 1;
    display: block;
  }
  99% {
    opacity: 0;
    transform: translateY(120%);
  }
  100% {
    display: none;
  }
}
.vl-popover {
  display: inline-block;
  position: relative;
}
.vl-popover .vl-popover__arrow {
  display: none;
  position: absolute;
  right: 0.5rem;
  bottom: -1.1rem;
  border: 0.9rem solid transparent;
  border-bottom-color: #cfd5dd;
  border-top-width: 0;
  z-index: 99999;
}
.vl-popover .vl-popover__arrow::before {
  content: "";
  position: absolute;
  border: 0.9rem solid transparent;
  border-bottom-color: #fff;
  border-top-width: 0;
  transform: translate(-50%, 1px);
}
@media screen and (max-width: 767px) {
  .vl-popover .vl-popover__arrow {
    display: none;
  }
}
.vl-popover--static {
  position: static;
}
.vl-popover--static .vl-popover__arrow {
  position: relative;
}
.vl-popover--align-left .vl-popover__arrow {
  left: 0.4rem;
  right: auto;
}
.vl-popover--align-center .vl-popover__arrow {
  left: 50%;
  transform: translateX(-50%);
  right: auto;
  margin: auto;
}
.vl-popover--align-top .vl-popover__arrow {
  top: -1rem;
  bottom: auto;
  transform: scale(-1);
}
.vl-popover .vl-popover__toggle {
  border-radius: 0;
  appearance: none;
  -webkit-appearance: none;
  border: 0;
  background-color: transparent;
  padding: 0;
  text-decoration: none;
  display: inline-block;
  font-size: 1.6rem;
  font-family: "Flanders Art Sans", sans-serif;
}
.vl-popover .vl-popover__toggle .vl-vi-arrow {
  transition: cubic-bezier(1, 0.1, 0, 0.9) 0.2s transform;
}
.vl-popover .vl-popover__toggle .vl-popover__toggle-icon {
  margin-left: 0.8rem;
}
.vl-popover .vl-popover__toggle .vl-badge {
  margin-top: 0.5rem;
}
.vl-popover.js-vl-popover--open .vl-popover__arrow {
  display: block;
}
@media screen and (max-width: 767px) {
  .vl-popover.js-vl-popover--open .vl-popover__arrow {
    display: none;
  }
}
@media screen and (max-width: 767px) {
  .vl-popover.js-vl-popover--open .vl-popover__content {
    display: block;
    opacity: 0;
    animation: slideIn 0.5s cubic-bezier(1, 0.1, 0, 0.9);
    animation-fill-mode: forwards;
  }
}
@media screen and (min-width: 767px) {
  .vl-popover.js-vl-popover--open .vl-popover__content {
    visibility: visible;
    opacity: 1;
  }
}
.vl-popover.js-vl-popover--open .vl-vi-arrow {
  transform: rotate(-180deg);
}
@media screen and (max-width: 767px) {
  .vl-popover.js-vl-popover--closing .vl-popover__content {
    will-change: transform;
    display: block;
    animation: slideOut 0.5s cubic-bezier(1, 0.1, 0, 0.9);
    animation-fill-mode: forwards;
  }
}
.vl-popover--closable .vl-popover__close-btn {
  display: inline-flex;
}
.vl-popover--closable .vl-popover__close-btn:hover {
  color: #05c;
}
.vl-popover--closable .vl-popover__close-btn:focus {
  color: #004099;
}
@media screen and (min-width: 767px) {
  .vl-popover--closable .vl-popover__close-btn {
    padding: 1.5rem;
    right: 0;
    width: auto;
  }
  .vl-popover--closable .vl-popover__close-btn .vl-button__icon {
    margin: 0;
  }
}

.vl-popover__close-btn {
  display: none;
  position: absolute;
  right: 0.5rem;
  top: 0;
  font-size: 1.4rem;
}
@media screen and (max-width: 767px) {
  .vl-popover__close-btn {
    display: inline-flex;
    top: -4.5rem;
  }
}

.vl-backdrop {
  position: fixed;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  z-index: 10008;
  background-color: rgba(232, 235, 238, 0.95);
  opacity: 0.95;
  opacity: 0;
  z-index: -1;
  transition: cubic-bezier(1, 0.1, 0, 0.9) 0.4s opacity, cubic-bezier(1, 0.1, 0, 0.9) 0.4s z-index;
  will-change: opacity;
}
@media screen and (max-width: 767px) {
  .vl-backdrop--active {
    opacity: 1;
    z-index: 10008;
  }
}

.vl-popover__content-overflow {
  max-height: 30rem;
  overflow-y: auto;
  padding: 2rem;
}
@media screen and (max-width: 767px) {
  .vl-popover__content-overflow {
    max-height: 65vh;
  }
}
@media screen and (min-width: 767px) {
  .vl-popover--closable .vl-popover__content-overflow {
    padding-right: 5rem;
  }
}

.vl-popover__content {
  background-color: #fff;
  box-shadow: 0 0 2.1rem 0 rgba(0, 0, 0, 0.1);
  width: 23rem;
  min-width: 23rem;
  max-width: 90vw;
  z-index: 10003;
  border: 0.1rem solid #cfd5dd;
}
@media screen and (max-width: 767px) {
  .vl-popover__content {
    width: 100%;
    max-width: 100%;
    margin: 0;
    position: fixed;
    bottom: 0;
    left: 0;
    top: auto;
    display: none;
    z-index: 10009;
    transform: translateY(120%);
  }
}
@media screen and (min-width: 767px) {
  .vl-popover__content {
    visibility: hidden;
    opacity: 0;
    margin: 0;
    margin-top: 0.3rem;
  }
}
@media screen and (min-width: 767px) {
  .vl-popover--large .vl-popover__content {
    width: 45rem;
  }
}

.vl-popover .vl-link-list,
.vl-popover__link-list {
  padding: 0;
}
.vl-popover .vl-link-list li,
.vl-popover .vl-link-list .vl-link-list__item,
.vl-popover__link-list li,
.vl-popover__link-list .vl-link-list__item {
  padding: 0.6rem 0;
}
.vl-popover .vl-link-list li:first-child,
.vl-popover .vl-link-list .vl-link-list__item:first-child,
.vl-popover__link-list li:first-child,
.vl-popover__link-list .vl-link-list__item:first-child {
  padding-top: 0;
}
.vl-popover--closable .vl-popover .vl-link-list li:first-child,
.vl-popover--closable .vl-popover .vl-link-list .vl-link-list__item:first-child,
.vl-popover--closable .vl-popover__link-list li:first-child,
.vl-popover--closable .vl-popover__link-list .vl-link-list__item:first-child {
  max-width: calc(100% - 4.5rem);
}
.vl-popover .vl-link-list li:last-of-type,
.vl-popover .vl-link-list .vl-link-list__item:last-of-type,
.vl-popover__link-list li:last-of-type,
.vl-popover__link-list .vl-link-list__item:last-of-type {
  padding-bottom: 0;
  margin-bottom: 0;
}
.vl-popover .vl-link-list li.vl-popover__link-list__separator,
.vl-popover .vl-link-list .vl-link-list__item.vl-popover__link-list__separator,
.vl-popover__link-list li.vl-popover__link-list__separator,
.vl-popover__link-list .vl-link-list__item.vl-popover__link-list__separator {
  height: 0.1rem;
  background: #cfd5dd;
  padding: 0;
  margin: 0.5rem 0;
}
.vl-popover .vl-link-list--bordered .vl-popover .vl-link-list__item:not(:last-of-type) .vl-link,
.vl-popover .vl-link-list--bordered .vl-popover__link-list__item:not(:last-of-type) .vl-link,
.vl-popover__link-list--bordered .vl-popover .vl-link-list__item:not(:last-of-type) .vl-link,
.vl-popover__link-list--bordered .vl-popover__link-list__item:not(:last-of-type) .vl-link {
  position: relative;
}
`;
export default style;
