// @domg/govflanders-style/component/button/vl-button.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

a.vl-button {
  cursor: pointer;
}

.vl-button {
  border-radius: 0;
  appearance: none;
  -webkit-appearance: none;
  border: 0;
  background-color: transparent;
  padding: 0;
  display: inline-flex;
  align-items: center;
  min-height: 3.5rem;
  font-size: 1.6rem;
  font-family: "Flanders Art Sans", sans-serif;
  font-weight: 500;
  padding: 0.5rem 2rem;
  background-color: var(--vl-theme-action-color, #05c);
  text-decoration: none;
  border-radius: 0.3rem;
  color: #fff;
  text-align: center;
  outline: 0;
  max-width: 100%;
  cursor: default;
}
@media screen and (max-width: 767px) {
  .vl-button {
    font-size: 1.6rem;
    padding: 1rem;
  }
}
.vl-button:hover, .vl-button:focus {
  color: #fff;
}
.vl-button:hover {
  background-color: var(--vl-theme-action-hover-color, #003bb0);
  text-decoration: none;
}
.vl-button:focus {
  box-shadow: 0 0 0 2px #fff, 0 0 0 5px rgba(0, 85, 204, 0.65);
  outline: transparent solid 0.2rem;
}
@supports (outline-offset: 2px) {
  .vl-button:focus {
    box-shadow: none;
    outline: 3px solid rgba(0, 85, 204, 0.65);
    outline-offset: 2px;
  }
}
.vl-button:active {
  background-color: var(--vl-theme-action-active-color, #004099);
}
.vl-button:visited {
  color: #fff;
}
.vl-button--ellipse {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
.vl-button--ellipse .vl-button__label {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
.vl-button--secondary {
  background-color: transparent;
  color: var(--vl-theme-action-color, #05c);
  border: 0.2rem solid currentColor;
  transition: color 0.2s, border-color 0.2s, box-shadow 0.2s;
}
.vl-button--secondary:visited {
  color: #05c;
}
.vl-button--secondary:focus {
  color: var(--vl-theme-action-active-color, #05c);
}
.vl-button--secondary:hover, .vl-button--secondary:active {
  color: var(--vl-theme-action-hover-color, #003bb0);
  background-color: transparent;
  border-color: var(--vl-theme-action-hover-color, #003bb0);
}
.vl-button--tertiary {
  background-color: transparent;
  color: var(--vl-theme-action-color, #05c);
  border: 0.1rem #c6cdd3 solid;
  transition: background-color 0.2s, box-shadow 0.2s;
}
.vl-button--tertiary:visited {
  color: var(--vl-theme-action-color, #05c);
}
.vl-button--tertiary:focus {
  color: var(--vl-theme-action-hover-color, #05c);
  border-color: var(--vl-theme-action-hover-color, #c6cdd3);
}
.vl-button--tertiary:hover, .vl-button--tertiary:active {
  color: var(--vl-theme-action-hover-color, #05c);
  background-color: transparent;
  border-color: var(--vl-theme-action-hover-color, #5990de);
  border-width: 0.2rem;
  padding: 0 calc(2rem - 0.1rem);
}
@media screen and (max-width: 767px) {
  .vl-button--tertiary:hover, .vl-button--tertiary:active {
    padding: calc(1rem - 0.1rem);
  }
}
.vl-button--tertiary.vl-button--large:hover, .vl-button--tertiary.vl-button--large:active {
  padding: calc(2rem - 0.1rem);
  padding-top: calc(1.5rem - 0.1rem);
  padding-bottom: calc(1.5rem - 0.1rem);
}
@media screen and (max-width: 767px) {
  .vl-button--tertiary.vl-button--large:hover, .vl-button--tertiary.vl-button--large:active {
    padding-left: calc(1rem - 0.1rem);
    padding-right: calc(1rem - 0.1rem);
  }
}
.vl-button--tertiary.vl-button--loading:hover, .vl-button--tertiary.vl-button--loading:active {
  padding: 0 calc(8rem - 0.1rem) 0 calc(4rem - 0.1rem);
}
@media screen and (max-width: 767px) {
  .vl-button--tertiary.vl-button--loading:hover, .vl-button--tertiary.vl-button--loading:active {
    padding: 1rem calc(8rem - 0.1rem) 1rem calc(4rem - 0.1rem);
  }
}
.vl-button--tertiary.vl-button--loading:focus {
  color: #fff;
}
.vl-button--block {
  display: flex;
  justify-content: center;
  width: 100%;
}
.vl-button--block.vl-button--icon-after, .vl-button--block.vl-button--icon-before {
  width: 100%;
}
.vl-button--icon {
  width: 3.5rem;
}
.vl-button--icon.vl-button {
  padding: 0;
}
.vl-button--icon.vl-button .vl-button__icon {
  margin: auto;
}
.vl-button--error {
  background-color: #d2373c;
}
.vl-button--error.vl-button--secondary {
  background-color: #fff;
  color: #d2373c;
}
.vl-button--error.vl-button--secondary:visited {
  color: #d2373c;
}
.vl-button--error.vl-button--secondary:hover, .vl-button--error.vl-button--secondary:active {
  color: #aa2729;
  border-color: #aa2729;
  background-color: transparent;
}
.vl-button--error.vl-button--tertiary {
  background-color: #fff;
  color: #d2373c;
  border-color: #d2373c;
}
.vl-button--error.vl-button--tertiary:visited {
  color: #d2373c;
}
.vl-button--error.vl-button--tertiary:hover, .vl-button--error.vl-button--tertiary:active {
  color: #aa2729;
  border-color: #aa2729;
  background-color: transparent;
}
.vl-button--error:hover, .vl-button--error:active {
  color: #fff;
  background-color: #aa2729;
}
.vl-button--large {
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
  font-size: 1.8rem;
}
.vl-button--wide {
  padding-left: 6rem;
  padding-right: 6rem;
}
.vl-button--narrow {
  padding-left: 1rem;
  padding-right: 1rem;
}
.vl-button--narrow.vl-button--tertiary:hover, .vl-button--narrow.vl-button--tertiary:active {
  padding: 0 calc(1rem - 0.1rem);
}
.vl-button--disabled {
  background-color: #cbd2d9;
  border-color: #cbd2d9;
  color: #687483;
  cursor: not-allowed;
}
.vl-button--disabled:hover, .vl-button--disabled:active {
  background-color: #cbd2d9;
  border-color: #cbd2d9;
  color: #687483;
}
.vl-button--loading {
  background-color: #cbd2d9;
  position: relative;
  color: #fff;
  padding: 1rem 8rem 1rem 4rem;
}
.vl-button--loading:hover, .vl-button--loading:active {
  background-color: #cbd2d9;
  color: #fff;
}
.vl-button--loading:focus {
  color: #fff;
}
.vl-button--loading::after {
  animation: waving-light infinite 1s linear;
  content: "";
  display: block;
  position: absolute;
  top: 50%;
  right: 4rem;
  margin-top: -0.2rem;
  margin-right: 3.2rem;
  width: 0.4rem;
  height: 0.4rem;
  background-color: #cbd2d9;
  border-radius: 50%;
  box-shadow: 1rem 0 #fff, 2rem 0 #fff, 3rem 0 #fff;
}
@keyframes waving-light {
  0% {
    box-shadow: 10px 0px #687483, 20px 0px #687483, 30px 0px #687483;
  }
  10% {
    box-shadow: 10px -3px #687483, 20px 0px #687483, 30px 0px #687483;
  }
  20% {
    box-shadow: 10px -6px #687483, 20px -3px #687483, 30px 0px #687483;
  }
  30% {
    box-shadow: 10px -3px #687483, 20px -6px #687483, 30px -3px #687483;
  }
  40% {
    box-shadow: 10px 0px #687483, 20px -3px #687483, 30px -6px #687483;
  }
  50% {
    box-shadow: 10px 0px #687483, 20px 0px #687483, 30px -3px #687483;
  }
  60% {
    box-shadow: 10px 0px #687483, 20px 0px #687483, 30px 0px #687483;
  }
  100% {
    box-shadow: 10px 0px #687483, 20px 0px #687483, 30px 0px #687483;
  }
}
.vl-button--naked, .vl-button--naked-action {
  background-color: transparent;
  border: 0;
  color: #333332;
}
.vl-button--naked:hover, .vl-button--naked:focus, .vl-button--naked:active, .vl-button--naked:visited, .vl-button--naked-action:hover, .vl-button--naked-action:focus, .vl-button--naked-action:active, .vl-button--naked-action:visited {
  border: 0;
  color: #333332;
  background-color: transparent;
}
.vl-button--naked.vl-button--disabled, .vl-button--naked-action.vl-button--disabled {
  color: #687483;
}
.vl-button--naked.vl-button--disabled:hover, .vl-button--naked.vl-button--disabled:focus, .vl-button--naked.vl-button--disabled:active, .vl-button--naked.vl-button--disabled:visited, .vl-button--naked-action.vl-button--disabled:hover, .vl-button--naked-action.vl-button--disabled:focus, .vl-button--naked-action.vl-button--disabled:active, .vl-button--naked-action.vl-button--disabled:visited {
  color: #687483;
}
.vl-button--naked-action {
  color: var(--vl-theme-action-color, #05c);
}
.vl-button--naked-action:hover, .vl-button--naked-action:focus, .vl-button--naked-action:active, .vl-button--naked-action:visited {
  color: var(--vl-theme-action-color, #05c);
}
.vl-button__icon {
  display: inline-flex;
}
.vl-button__icon.vl-icon {
  display: inline-flex;
}
.vl-button__icon--before {
  margin-right: 0.8rem;
}
.vl-button__icon--after {
  margin-left: 0.8rem;
}

.vl-button-group {
  display: flex;
  flex-wrap: wrap;
}
.vl-button-group .vl-button {
  margin-right: 1rem;
  margin-bottom: 1rem;
}
`;
export default style;
