// @domg/govflanders-style/component/typography/vl-typography.css.js
import { css } from 'lit';
const style = css `
@charset "UTF-8";
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-typography ul,
.ul {
  list-style-type: disc;
}

.vl-typography ol,
.ol {
  list-style-type: decimal;
}
.vl-typography ol ol,
.vl-typography ol .ol,
.ol ol,
.ol .ol {
  list-style-type: lower-latin;
}

.vl-typography ol,
.ol,
.vl-typography ul,
.ul {
  display: block;
  margin: 1.8rem 0 2rem 1rem;
  padding-left: 2rem;
}
.vl-typography ol:first-child,
.ol:first-child,
.vl-typography ul:first-child,
.ul:first-child {
  margin-top: 0;
}
.vl-typography ol:last-child,
.ol:last-child,
.vl-typography ul:last-child,
.ul:last-child {
  margin-bottom: 0;
}
.vl-typography ol li,
.ol li,
.vl-typography ul li,
.ul li {
  display: list-item;
  margin-bottom: 0.5em;
}
.vl-typography ol li:last-child,
.ol li:last-child,
.vl-typography ul li:last-child,
.ul li:last-child {
  margin-bottom: 0;
}
.vl-typography ol ul,
.ol ul,
.vl-typography ul ul,
.ul ul {
  list-style-type: circle;
}
.vl-typography ol ul,
.vl-typography ol ol,
.ol ul,
.ol ol,
.vl-typography ul ul,
.vl-typography ul ol,
.ul ul,
.ul ol {
  margin: 0.5em 0 0.5rem 1.5em;
}
.vl-typography ol ul:first-child,
.vl-typography ol ol:first-child,
.ol ul:first-child,
.ol ol:first-child,
.vl-typography ul ul:first-child,
.vl-typography ul ol:first-child,
.ul ul:first-child,
.ul ol:first-child {
  margin-top: 0;
}
.vl-typography ol ul:last-child,
.vl-typography ol ol:last-child,
.ol ul:last-child,
.ol ol:last-child,
.vl-typography ul ul:last-child,
.vl-typography ul ol:last-child,
.ul ul:last-child,
.ul ol:last-child {
  margin-bottom: 0;
}

.vl-typography p,
.p {
  margin-bottom: 1.8rem;
}
.vl-typography p:last-child,
.p:last-child {
  margin-bottom: 0;
}

.vl-typography ul li p:last-child,
.vl-typography ul li .p:last-child {
  margin-bottom: 1.8rem;
}

.vl-typography strong,
.vl-typography b {
  font-weight: 500;
}
.vl-typography em,
.vl-typography i {
  font-style: italic;
}
.vl-typography strike,
.vl-typography s {
  text-decoration: line-through;
}
.vl-typography hr {
  display: block;
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  margin-left: auto;
  margin-right: auto;
  border: 0;
  border-top: 1px solid #cbd2da;
}
.vl-typography table {
  width: 100%;
  max-width: 100%;
  margin-bottom: 2rem;
}
.vl-typography table thead tr {
  border-bottom: 3px #cbd2da solid;
}
.vl-typography table tbody tr {
  border-bottom: 1px #cbd2da solid;
}
.vl-typography table tbody tr[data-vl-table-selectable] {
  transition: background 0.2s ease-in-out;
  cursor: pointer;
}
.vl-typography table tbody tr[data-vl-table-selectable]:hover {
  background: #f3f5f6;
}
.vl-typography table tbody tr.vl-data-table__grouped-row:not(.vl-data-table__grouped-row--last) {
  border-bottom: 0;
}
.vl-typography table td,
.vl-typography table th {
  text-align: left;
  font-size: 1.6rem;
  line-height: 1.3;
  vertical-align: top;
  padding: 1.2rem 1rem;
}
@media screen and (max-width: 767px) {
  .vl-typography table td,
  .vl-typography table th {
    padding: 1rem;
  }
}
@media screen and (max-width: 767px) {
  .vl-typography table td,
  .vl-typography table th {
    font-size: 1.4rem;
  }
}
.vl-typography table td:first-child,
.vl-typography table th:first-child {
  border-left: 0;
}
.vl-typography table th {
  font-weight: 500;
}
.vl-typography table .vl-data-table__grouped-row td {
  padding: 0.3rem 1rem 0.3rem 0;
}
@media screen and (max-width: 767px) {
  .vl-typography table .vl-data-table__grouped-row td {
    padding: 0.3rem 1rem 0.3rem 0;
  }
}
.vl-typography table .vl-data-table__grouped-row--first td {
  padding-top: 1.2rem;
}
@media screen and (max-width: 767px) {
  .vl-typography table .vl-data-table__grouped-row--first td {
    padding-top: 1rem;
  }
}
.vl-typography table .vl-data-table__grouped-row--last td {
  padding-bottom: 1.2rem;
}
@media screen and (max-width: 767px) {
  .vl-typography table .vl-data-table__grouped-row--last td {
    padding-bottom: 1rem;
  }
}
.vl-typography table__header-title--sortable {
  text-decoration: none;
}
.vl-typography table__header-title--sortable .vl-data-table__header-title__sort-icon {
  opacity: 0;
}
.vl-typography table__header-title--sortable:hover, .vl-typography table__header-title--sortable:focus {
  text-decoration: underline;
}
.vl-typography table__header-title--sortable:hover .vl-data-table__header-title__sort-icon, .vl-typography table__header-title--sortable:focus .vl-data-table__header-title__sort-icon {
  opacity: 0.5;
}
.vl-typography table__header-title--sortable-active .vl-data-table__header-title__sort-icon {
  opacity: 1;
}
.vl-typography table__body-title {
  max-width: 30rem;
}
.vl-typography table--alt tr td:not(:first-child) {
  padding: 1.2rem;
}
.vl-typography table--alt tr th:not(:first-child) {
  padding: 0 1.2rem 1.2rem;
}
.vl-typography table--alt tr th:first-child,
.vl-typography table--alt tr td:first-child {
  border-right: 1px #cbd2da solid;
}
.vl-typography table.vl-data-table--no-header tbody tr:first-child {
  border-top: 3px #cbd2da solid;
}
.vl-typography table .vl-pill {
  vertical-align: middle;
}
@media screen and (max-width: 767px) {
  .vl-typography table .vl-pill {
    height: 2rem;
    line-height: 2rem;
    padding: 0 0.5rem;
    font-size: 1.4rem;
  }
}
.vl-typography blockquote {
  position: relative;
  padding-left: 9rem;
  font-size: 3.4rem;
  font-family: "Flanders Art Sans", sans-serif;
  font-weight: 400;
  margin-bottom: 2rem;
  line-height: 1.5;
}
@media screen and (max-width: 767px) {
  .vl-typography blockquote {
    padding-left: 5rem;
  }
}
.vl-typography blockquote::before {
  position: absolute;
  content: "“";
  font-size: 9rem;
  left: 0;
  top: 0;
  padding-top: 1rem;
  padding-left: 1rem;
  width: 7rem;
  height: 6rem;
  line-height: 1;
  color: var(--vl-theme-fg-color);
  background: linear-gradient(-110deg, transparent 20px, var(--vl-theme-primary-color) 20px);
}
@media screen and (max-width: 767px) {
  .vl-typography blockquote::before {
    width: 4.6rem;
    font-size: 6rem;
    height: 3.6rem;
    padding-right: 1.5rem;
    padding-top: 0.5rem;
    padding-left: 0.5rem;
    margin-right: 0rem;
  }
}
.vl-typography pre:not([class*=language]) {
  padding: 1.5rem;
  background-color: #333332;
  color: #fff;
  display: block;
  white-space: pre;
  overflow-x: scroll;
  tab-size: 2;
  word-break: normal;
  hyphens: none;
}
.vl-typography pre:not([class*=language]) code,
.vl-typography pre:not([class*=language]) kdb,
.vl-typography pre:not([class*=language]) samp {
  background-color: transparent;
  padding: 0;
}
.vl-typography code,
.vl-typography kdb,
.vl-typography samp {
  font-family: monospace;
  background: #e8ebee;
  padding: 0.2rem;
}
.vl-typography a,
.vl-typography .vl-link {
  display: inline;
  text-decoration: underline;
}
.vl-typography a:hover,
.vl-typography .vl-link:hover {
  text-decoration: none;
}
.vl-typography a:focus, .vl-typography a:active,
.vl-typography .vl-link:focus,
.vl-typography .vl-link:active {
  text-decoration: none;
}
.vl-typography--prewrapped {
  white-space: pre-wrap;
}
`;
export default style;
