// @domg-wc/components/block/content-header/vl-content-header.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { BaseHTMLElement, webComponent } from '@domg-wc/common';
import { contentHeaderStyle } from '@domg/govflanders-style/component';
import { baseStyle, layoutStyle, resetStyle } from '@domg/govflanders-style/common';
let VlContentHeaderComponent = class VlContentHeaderComponent extends BaseHTMLElement {
    constructor() {
        const html = `
          <header class="vl-content-header vl-content-header--large vl-content-header--show-mobile vl-content-header--has-context">
            <div class="vl-content-header__wrapper">
              <picture id="picture" class="vl-content-header__bg"></picture>
              <div class="vl-layout">
                <div class="vl-content-header__content">
                <div id="context" class="vl-content-header__context vl-content-header__context--has-link"></div>
                  <h2 id="title" class="vl-content-header__title vl-content-header__title--has-link"></h2>
                </div>
              </div>
            </div>
          </header>
        `;
        const styleSheets = [
            resetStyle.styleSheet,
            baseStyle.styleSheet,
            layoutStyle.styleSheet,
            contentHeaderStyle.styleSheet,
        ];
        super(html, styleSheets);
    }
    connectedCallback() {
        super.connectedCallback();
        this.__processSlotElements();
        this._observer = this.__observeSlotElements(() => this.__processSlotElements());
    }
    disconnectedCallback() {
        this._observer?.disconnect();
    }
    get pictureElement() {
        return this._shadow?.querySelector('#picture');
    }
    get pictureSlotElement() {
        return this.querySelector('img[slot="image"]');
    }
    get contextElement() {
        return this._shadow?.querySelector('#context');
    }
    get contextSlotElement() {
        return this.querySelector('a[slot="context-link"]');
    }
    get titleElement() {
        return this._shadow?.querySelector('#title');
    }
    get titleSlotElement() {
        return this.querySelector('a[slot="title-link"]');
    }
    __processSlotElements() {
        this.__processSlot(this.pictureElement, this.pictureSlotElement);
        const contextSlot = this.__processSlot(this.contextElement, this.contextSlotElement);
        contextSlot.classList.add('vl-content-header__context__link');
        const titleSlot = this.__processSlot(this.titleElement, this.titleSlotElement);
        titleSlot.classList.add('vl-content-header__title__link');
    }
    __clearChildren(element) {
        while (element.hasChildNodes()) {
            element.firstChild?.remove();
        }
    }
    __processSlot(container, slotElement) {
        this.__clearChildren(container);
        const slotClone = slotElement.cloneNode(true);
        container.appendChild(slotClone);
        return slotClone;
    }
    __observeSlotElements(callback) {
        const node = this;
        const observer = new MutationObserver(callback);
        observer.observe(node, {
            attributes: true,
            childList: true,
            characterData: true,
            subtree: true,
        });
        return observer;
    }
};
VlContentHeaderComponent = __decorate([
    webComponent('vl-content-header'),
    __metadata("design:paramtypes", [])
], VlContentHeaderComponent);
export { VlContentHeaderComponent };
