// @domg/govflanders-style/component/action-group/vl-action-group.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-action-group {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
}
.vl-action-group a,
.vl-action-group button,
.vl-action-group .vl-button {
  margin-right: 1.4rem;
}
.vl-action-group a:last-child,
.vl-action-group button:last-child,
.vl-action-group .vl-button:last-child {
  margin-right: 0;
}
.vl-action-group--align-right {
  justify-content: flex-end;
}
.vl-action-group--align-center {
  justify-content: center;
}
.vl-action-group--space-between {
  justify-content: space-between;
}
.vl-action-group .vl-link {
  white-space: nowrap;
}
.vl-action-group--bordered > * {
  position: relative;
  margin: 0 1.6rem;
}
.vl-action-group--bordered > *:not(:last-child) {
  margin-right: 1.6rem;
}
.vl-action-group--bordered > *:last-child {
  border: 0;
}
.vl-action-group--bordered > *:first-child {
  margin-left: 0;
}
.vl-action-group--bordered > *:not(:last-child)::after {
  content: "";
  background-color: #cbd2da;
  height: 1.5rem;
  width: 0.1rem;
  right: -1.6rem;
  top: 50%;
  transform: translateY(-50%);
  position: absolute;
}
@media screen and (min-width: 1023px) {
  .vl-action-group--collapse--l {
    flex-direction: column;
    align-items: flex-start;
  }
  .vl-action-group--collapse--l > *:not(:last-child) {
    margin-bottom: 1rem;
  }
  .vl-action-group--collapse--l.vl-action-group--align-center, .vl-action-group--collapse--l.vl-action-group--align-right {
    align-items: center;
  }
  .vl-action-group--collapse--l.vl-action-group--align-center .vl-button, .vl-action-group--collapse--l.vl-action-group--align-right .vl-button {
    margin-right: 0;
  }
  .vl-action-group--collapse--l.vl-action-group--align-right {
    align-items: flex-end;
  }
  .vl-action-group--collapse--l .vl-link {
    border-right: 0;
    display: block;
    padding-right: 0;
    text-align: left;
    flex: 1 0 100%;
    margin: 0;
  }
  .vl-action-group--collapse--l .vl-link:not(:last-child) {
    margin-bottom: 0.5rem;
  }
  .vl-action-group--collapse--l .vl-link:not(:last-child)::after {
    display: none;
  }
}
@media screen and (max-width: 1023px) {
  .vl-action-group--collapse--m {
    flex-direction: column;
    align-items: flex-start;
  }
  .vl-action-group--collapse--m > *:not(:last-child) {
    margin-bottom: 1rem;
  }
  .vl-action-group--collapse--m.vl-action-group--align-center, .vl-action-group--collapse--m.vl-action-group--align-right {
    align-items: center;
  }
  .vl-action-group--collapse--m.vl-action-group--align-center .vl-button, .vl-action-group--collapse--m.vl-action-group--align-right .vl-button {
    margin-right: 0;
  }
  .vl-action-group--collapse--m.vl-action-group--align-right {
    align-items: flex-end;
  }
  .vl-action-group--collapse--m .vl-link {
    border-right: 0;
    display: block;
    padding-right: 0;
    text-align: left;
    flex: 1 0 100%;
    margin: 0;
  }
  .vl-action-group--collapse--m .vl-link:not(:last-child) {
    margin-bottom: 0.5rem;
  }
  .vl-action-group--collapse--m .vl-link:not(:last-child)::after {
    display: none;
  }
}
@media screen and (max-width: 767px) {
  .vl-action-group--collapse--s {
    flex-direction: column;
    align-items: flex-start;
  }
  .vl-action-group--collapse--s > *:not(:last-child) {
    margin-bottom: 1rem;
  }
  .vl-action-group--collapse--s.vl-action-group--align-center, .vl-action-group--collapse--s.vl-action-group--align-right {
    align-items: center;
  }
  .vl-action-group--collapse--s.vl-action-group--align-center .vl-button, .vl-action-group--collapse--s.vl-action-group--align-right .vl-button {
    margin-right: 0;
  }
  .vl-action-group--collapse--s.vl-action-group--align-right {
    align-items: flex-end;
  }
  .vl-action-group--collapse--s .vl-link {
    border-right: 0;
    display: block;
    padding-right: 0;
    text-align: left;
    flex: 1 0 100%;
    margin: 0;
  }
  .vl-action-group--collapse--s .vl-link:not(:last-child) {
    margin-bottom: 0.5rem;
  }
  .vl-action-group--collapse--s .vl-link:not(:last-child)::after {
    display: none;
  }
}
@media screen and (max-width: 500px) {
  .vl-action-group--collapse--xs {
    flex-direction: column;
    align-items: flex-start;
  }
  .vl-action-group--collapse--xs > *:not(:last-child) {
    margin-bottom: 1rem;
  }
  .vl-action-group--collapse--xs.vl-action-group--align-center, .vl-action-group--collapse--xs.vl-action-group--align-right {
    align-items: center;
  }
  .vl-action-group--collapse--xs.vl-action-group--align-center .vl-button, .vl-action-group--collapse--xs.vl-action-group--align-right .vl-button {
    margin-right: 0;
  }
  .vl-action-group--collapse--xs.vl-action-group--align-right {
    align-items: flex-end;
  }
  .vl-action-group--collapse--xs .vl-link {
    border-right: 0;
    display: block;
    padding-right: 0;
    text-align: left;
    flex: 1 0 100%;
    margin: 0;
  }
  .vl-action-group--collapse--xs .vl-link:not(:last-child) {
    margin-bottom: 0.5rem;
  }
  .vl-action-group--collapse--xs .vl-link:not(:last-child)::after {
    display: none;
  }
}
`;
export default style;
