// @domg-wc/components/compliance/cookie-statement/cookie/vl-authentication-cookie.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { webComponent } from '@domg-wc/common';
import { VlCookie } from './vl-cookie.component';
let VlAuthenticationCookie = class VlAuthenticationCookie extends VlCookie {
    constructor() {
        super({
            title: 'Departement Omgeving toegangsbeheer cookies',
            name: ['KEYCLOAK_SESSION', 'KEYCLOAK_SESSION_LEGACY'],
            purpose: 'Sessiegebaseerde cookies die het mogelijk maken om gebruikers te herkennen op een webpagina van Departement Omgeving.',
            domain: window.location.hostname,
            processor: 'Departement Omgeving',
            validity: '10 uur',
        });
    }
};
VlAuthenticationCookie = __decorate([
    webComponent('vl-authentication-cookie'),
    __metadata("design:paramtypes", [])
], VlAuthenticationCookie);
export { VlAuthenticationCookie };
