// @domg-wc/components/compliance/cookie-statement/cookie/vl-header-cookie.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { webComponent } from '@domg-wc/common';
import { VlCookie } from './vl-cookie.component';
let VlHeaderCookie = class VlHeaderCookie extends VlCookie {
    constructor() {
        super({
            title: 'Vlaanderen header cookie',
            name: 'VOGANONUSER',
            purpose: 'De Reverse Proxy van de Vlaamse overheid plaats dit cookie in kader van de Vlaanderen header op Vlaanderen.be om de goede uitvoering van de verzending van communicatie over een elektronisch communicatienetwerk van de Vlaamse overheid te verzekeren.',
            domain: 'vlaanderen.be',
            processor: 'Vlaamse overheid',
            validity: 'Permanente cookies met een geldigheid van maximaal 24 uur',
        });
    }
};
VlHeaderCookie = __decorate([
    webComponent('vl-header-cookie'),
    __metadata("design:paramtypes", [])
], VlHeaderCookie);
export { VlHeaderCookie };
