// @domg-wc/components/compliance/cookie-statement/cookie/vl-header-authentication-cookie.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { webComponent } from '@domg-wc/common';
import { VlCookie } from './vl-cookie.component';
let VlHeaderAuthenticationCookie = class VlHeaderAuthenticationCookie extends VlCookie {
    constructor() {
        super({
            title: 'Vlaams toegangsbeheer cookies',
            name: [
                'AMWEBJCT!%2Fsps!JSESSIONID',
                'https%3A%2F%2Fauthenticatie.vlaanderen.be%2Fsps%2Fvidp%2Fsaml20FIMSAML20',
                'PD_STATEFUL_5bb64e42-0d53-11e2-a712-52540052f0ed',
                'PD-H-SESSION-ID',
                'tbsession',
            ],
            purpose: 'Sessiegebaseerde cookies die het mogelijk maken om gebruikers te herkennen op een webpagina van het Vlaams toegangsbeheer.',
            domain: 'authenticatie.vlaanderen.be',
            processor: 'Vlaamse overheid',
            validity: 'Sessie',
        });
    }
};
VlHeaderAuthenticationCookie = __decorate([
    webComponent('vl-header-authentication-cookie'),
    __metadata("design:paramtypes", [])
], VlHeaderAuthenticationCookie);
export { VlHeaderAuthenticationCookie };
