// @domg-wc/components/compliance/cookie-statement/cookie/vl-jsessionid-cookie.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { webComponent } from '@domg-wc/common';
import { VlCookie } from './vl-cookie.component';
let VlJSessionIdCookie = class VlJSessionIdCookie extends VlCookie {
    constructor() {
        super({
            title: 'Sessie cookie voor betere gebruikerservaring',
            name: ['JSESSIONID', 'KEYCLOAK_IDENTITY', 'KEYCLOAK_IDENTITY_LEGACY'],
            purpose: 'De cookie wordt gebruikt om een sessie tussen de applicatieserver en een gebruiker in stand te houden. Dankzij deze cookie kan een gebruiker door de server op een uniek manier geïdentificeerd worden.',
            domain: window.location.hostname,
            processor: 'Departement Omgeving',
            validity: 'Beperkt tot de duur van de sessie',
        });
    }
};
VlJSessionIdCookie = __decorate([
    webComponent('vl-jsessionid-cookie'),
    __metadata("design:paramtypes", [])
], VlJSessionIdCookie);
export { VlJSessionIdCookie };
