// @domg-wc/components/compliance/cookie-statement/cookie/vl-sticky-session-cookie.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { webComponent } from '@domg-wc/common';
import { VlCookie } from './vl-cookie.component';
let VlStickySessionCookie = class VlStickySessionCookie extends VlCookie {
    constructor() {
        super({
            title: 'Persistentie sessie cookie voor betere gebruikerservaring',
            name: [
                'BIGipServerPool-AUTO-* (vb. "BIGipServerPOOL-AUTO-app=2016879114.37407.0000")',
                'BIGipServerPool-sso-pr-* (vb. "BIGipServerPOOL-sso-pr-app=2016879114.37407.0000")',
            ],
            purpose: 'Dankzij de cookie kan er verzekerd worden dat verzoeken van een gebruiker voor de duur van een sessie worden gedistribueerd naar de server waarop ze zijn gestart. De cookies worden daarom vaak beschreven als "sticky sessions". Hierdoor kan een betere gebruikerservaring gerealiseerd worden. Bovendien kan de infrastructuur optimaal gebruikt worden.',
            domain: window.location.hostname,
            processor: 'Departement Omgeving',
            validity: 'Beperkt tot de duur van de sessie',
        });
    }
};
VlStickySessionCookie = __decorate([
    webComponent('vl-sticky-session-cookie'),
    __metadata("design:paramtypes", [])
], VlStickySessionCookie);
export { VlStickySessionCookie };
