// @domg-wc/components/form/radio-group/vl-radio.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, webComponent } from '@domg-wc/common';
import { vlLegacyStyles } from '@domg-wc/styles';
import { baseStyle, resetStyle } from '@domg/govflanders-style/common';
import { radioStyle } from '@domg/govflanders-style/component';
import { html, LitElement, nothing } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import { live } from 'lit/directives/live.js';
import { vlRadioComponentFluxStyles } from './vl-radio.component.flux-css';
import { radioDefaults } from './vl-radio.defaults';
let VlRadioComponent = class VlRadioComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        // Attributes
        this.id = radioDefaults.id;
        this.name = radioDefaults.name;
        this.value = radioDefaults.value;
        this.label = radioDefaults.label;
        this.block = radioDefaults.block;
        this.readonly = radioDefaults.readonly;
        this.disabled = radioDefaults.disabled;
        this.error = radioDefaults.error;
        this.success = radioDefaults.success;
        this.checked = radioDefaults.checked;
        // Variables
        this.dispatchInput = false;
    }
    static get styles() {
        return [resetStyle, baseStyle, vlLegacyStyles, radioStyle, vlRadioComponentFluxStyles];
    }
    static get properties() {
        return {
            id: { type: String },
            name: { type: String },
            value: { type: String },
            label: { type: String },
            disabled: { type: Boolean },
            readonly: { type: Boolean },
            error: { type: Boolean },
            success: { type: Boolean },
            block: { type: Boolean },
            checked: { type: Boolean, reflect: true },
        };
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        if (changedProperties.has('checked') || changedProperties.has('value')) {
            if (this.checked) {
                const detail = { checked: true, value: this.value };
                this.dispatchEvent(new CustomEvent('vl-change', { composed: true, bubbles: true, detail }));
                if (this.dispatchInput) {
                    this.dispatchEvent(new CustomEvent('vl-input', { bubbles: true, composed: true, detail }));
                    this.dispatchInput = false;
                }
                this.dispatchEvent(new CustomEvent('vl-valid', { composed: true, bubbles: true, detail }));
            }
        }
    }
    render() {
        const classes = {
            'vl-radio': true,
            'vl-radio--block': this.block,
            'vl-radio--disabled': this.disabled,
            'vl-radio--error': this.error,
            'vl-radio--success': this.success,
        };
        return html `
            <label id="radio-label" class=${classMap(classes)} for=${this.id}>
                <input
                    id=${this.id || nothing}
                    name=${this.name || nothing}
                    class="vl-radio__toggle"
                    type="radio"
                    aria-label=${this.label || nothing}
                    .value=${live(this.value)}
                    .checked=${this.checked}
                    ?disabled=${this.disabled}
                    ?readonly=${this.readonly}
                    @change=${this.onChange}
                    @input=${this.onInput}
                />
                <div class="vl-radio__label">
                    <span id="label-text">
                        <slot></slot>
                    </span>
                </div>
            </label>
        `;
    }
    get validationTarget() {
        return this.shadowRoot?.querySelector('input');
    }
    onChange() {
        this.checked = !this.checked;
    }
    onInput() {
        this.dispatchInput = true;
    }
};
VlRadioComponent.shadowRootOptions = { ...LitElement.shadowRootOptions, delegatesFocus: true };
VlRadioComponent = __decorate([
    webComponent('vl-radio')
], VlRadioComponent);
export { VlRadioComponent };
