// @domg-wc/map/components/action/vl-map-action.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { webComponent } from '@domg-wc/common';
import { BaseHTMLElement } from '@domg-wc/common';
let VlMapAction = class VlMapAction extends BaseHTMLElement {
    constructor() {
        super();
        this._active = false;
        this._controlledActive = false;
    }
    connectedCallback() {
        super.connectedCallback();
        this.__defineLayer();
    }
    static isVlMapAction() {
        return true;
    }
    get action() {
        return this._action;
    }
    get _mapElement() {
        return this.closest('vl-map');
    }
    get _defaultActive() {
        return this.hasAttribute('default-active');
    }
    get _callback() {
        return (...args) => (this.__callback ? this.__callback(...args) : null);
    }
    get _layerElement() {
        console.warn('_layerElement implementation is missing');
        return null;
    }
    get active() {
        return this._controlledActive;
    }
    set active(value) {
        this._controlledActive = value;
        if (value) {
            this.activate();
        }
        else {
            this.deactivate();
        }
    }
    get layer() {
        return this._layer;
    }
    set layer(layer) {
        this._layer = layer;
    }
    activate() {
        // Do not activate action if its layer is invisible
        if (this.action && this.action.layer.get('visible')) {
            this._mapElement.changeActiveAction(this.action);
        }
    }
    deactivate() {
        // Only deactivate if this action is currently active
        if (this.action && this.action === this._mapElement.activeAction) {
            const actionIsNotDefault = this.action !== this._mapElement.defaultAction;
            const layerIsVisible = this._mapElement.defaultAction && this._mapElement.defaultAction.layer.get('visible');
            this._mapElement.changeActiveAction(actionIsNotDefault && layerIsVisible && this._mapElement.defaultAction);
        }
    }
    _createAction(layer) {
        console.warn('_createAction implementation is missing');
        return null;
    }
    _processAction() {
        if (this.action) {
            this.action.element = this;
            this._mapElement.addAction(this.action);
            // Activate the action when
            // - the action is the default active action and no other action has yet been activated
            // - the controlled active state of the action is true
            if ((this._defaultActive && !this._mapElement.activeAction) || this.active) {
                this.activate();
            }
        }
    }
    __defineLayer() {
        if (this._layerElement) {
            this.layer = this._layerElement.layer;
        }
    }
    reset() {
        this._active = false;
    }
};
VlMapAction = __decorate([
    webComponent('vl-map-action'),
    __metadata("design:paramtypes", [])
], VlMapAction);
export { VlMapAction };
