// @domg-wc/components/block/share-buttons/vl-share-button.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement } from '@domg-wc/common';
import { baseStyle, resetStyle } from '@domg/govflanders-style/common';
import { iconStyle } from '@domg/govflanders-style/component';
import { html } from 'lit';
import { customElement } from 'lit/decorators.js';
import { vlShareButtonFluxStyles } from './vl-share-button.flux-css';
import { MEDIA_NAMES } from './vl-share-button.model';
let VlShareButton = class VlShareButton extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.medium = '';
        this.href = '';
    }
    static get styles() {
        return [resetStyle, baseStyle, vlShareButtonFluxStyles, iconStyle];
    }
    static get properties() {
        return {
            medium: { type: String, attribute: 'medium', reflect: true },
            href: { type: String, attribute: 'href', reflect: true },
        };
    }
    render() {
        const name = MEDIA_NAMES[this.medium];
        return html `<a
            href=${this.href}
            class="vl-share-button vl-share-button--${this.medium}"
            title="Deel op ${name}"
        >
            <i class="vl-vi vl-vi-${this.medium}" aria-hidden="true"></i>
            ${name}
        </a>`;
    }
};
VlShareButton = __decorate([
    customElement('vl-share-button')
], VlShareButton);
export { VlShareButton };
