// @domg-wc/components/compliance/privacy/vl-privacy.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { BaseLitElement, registerWebComponents } from '@domg-wc/common';
import { vlContentBlockStyles, vlGridStyles, vlLegacyStyles, vlResetStyles, vlSectionStyles, vlStackedStyles, } from '@domg-wc/styles';
import { html } from 'lit';
import { customElement } from 'lit/decorators.js';
import { VlTitleComponent } from '../../atom';
import { VlParagraphComponent } from '../../atom/paragraph';
import { VlContactCardComponent } from '../../block/contact-card';
import { VlDocumentComponent } from '../../block/document';
import { VlInfoblockComponent } from '../../block/infoblock';
import { VlPropertiesComponent } from '../../block/properties';
import { VlSideNavigationComponent } from '../../block/side-navigation';
import { VlTypography } from '../../block/typography';
import { privacyBottomSection } from './child/bottom.component';
import { privacyContentSection } from './child/content.component';
import { header, privacyHeaderElements } from './child/header.component';
import { privacyVersionSection } from './child/version.component';
import { privacyDefaults } from './vl-privacy.defaults';
let VlPrivacy = class VlPrivacy extends BaseLitElement {
    constructor() {
        super();
        this.date = privacyDefaults.date;
        this.version = privacyDefaults.version;
        this.disableBackLink = privacyDefaults.disableBackLink;
        this.hideBackLink = privacyDefaults.hideBackLink;
        this.allowCustomCSS = false;
    }
    static get styles() {
        return [vlResetStyles, vlLegacyStyles, vlGridStyles, vlContentBlockStyles, vlSectionStyles, vlStackedStyles];
    }
    static get properties() {
        return {
            date: { type: String, attribute: 'date' },
            disableBackLink: { type: Boolean, attribute: 'disable-back-link' },
            hideBackLink: { type: Boolean, attribute: 'hide-back-link' },
            version: { type: String, attribute: 'version' },
        };
    }
    render() {
        return html `
            <slot name="header"
                >${header({ disableBackLink: this.disableBackLink, hideBackLink: this.hideBackLink })}
            </slot>
            <section class="vl-section">
                <slot name="version"> ${privacyVersionSection(this.version, this.date)}</slot>
            </section>
            <section id="content" class="vl-section">
                <slot name="content" @slotchange=${this.handleContentSlotChanged}> ${privacyContentSection()}</slot>
            </section>
            <section class="vl-section vl-section--overlap">
                <slot name="bottom"> ${privacyBottomSection()}</slot>
            </section>
        `;
    }
    handleContentSlotChanged(event) {
        // wanneer de inhoud van het content slot verandert
        // verplaatsen we de inhoud ervan naar de shadow DOM van deze component
        // we verwachten dat er in dit slot een side-navigation element zit en die vormt een uitdaging als het in de light DOM zit
        const contentSlot = this.shadowRoot?.querySelector('#content');
        const content = this.querySelector('[slot="content"]');
        if (contentSlot && content) {
            contentSlot.textContent = '';
            contentSlot.appendChild(content);
        }
    }
};
(() => {
    registerWebComponents([
        // components
        VlContactCardComponent,
        VlDocumentComponent,
        VlInfoblockComponent,
        VlTypography,
        VlTitleComponent,
        VlSideNavigationComponent,
        VlPropertiesComponent,
        VlParagraphComponent,
        // child components
        ...privacyHeaderElements(),
    ]);
})();
VlPrivacy = __decorate([
    customElement('vl-privacy'),
    __metadata("design:paramtypes", [])
], VlPrivacy);
export { VlPrivacy };
