// @domg-wc/map/utils/capabilities.js
import WMTSCapabilities from 'ol/format/WMTSCapabilities';
import { geoApiVlaanderenCapabilitiesXML } from './capabilities-response';
const parser = new WMTSCapabilities();
const geoApiVlaanderenCapabilities = parser.read(geoApiVlaanderenCapabilitiesXML);
export const Lambert2008Identifier = 'BPL2008VL';
const getLambert2008TileSet = () => geoApiVlaanderenCapabilities.Contents.TileMatrixSet.find(({ Identifier }) => Identifier === Lambert2008Identifier);
export const getLambert2008Extent = () => getLambert2008TileSet()?.BoundingBox;
export const getLambert2008Code = () => getLambert2008TileSet()?.SupportedCRS || 'EPSG:3812';
export const Lambert72Identifier = 'BPL72VL';
const getLambert72TileSet = () => geoApiVlaanderenCapabilities.Contents.TileMatrixSet.find(({ Identifier }) => Identifier === Lambert72Identifier);
export const getLambert72Extent = () => getLambert72TileSet()?.BoundingBox;
export const getLambert72Code = () => getLambert72TileSet()?.SupportedCRS || 'EPSG:31370';
const getGRBLayer = () => geoApiVlaanderenCapabilities.Contents.Layer.find(({ Identifier }) => Identifier === 'grb_bsk');
export const getGRBLayerExtent = () => getGRBLayer().WGS84BoundingBox;
export const getWGS84Code = () => 'EPSG:4326';
