// @domg-wc/map/components/action/draw-action/draw-action-style/vl-map-draw-action-style.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseHTMLElement, webComponent } from '@domg-wc/common';
import OlStyleFill from 'ol/style/Fill';
import OlStyleStroke from 'ol/style/Stroke';
import OlStyleCircle from 'ol/style/Circle';
import OlStyle from 'ol/style/Style';
let VlMapDrawActionStyle = class VlMapDrawActionStyle extends BaseHTMLElement {
    connectedCallback() {
        super.connectedCallback();
        this._setStyleOnParent();
    }
    get fillColor() {
        return this.getAttribute('color') || 'rgba(2, 85, 204, 0.8)';
    }
    get strokeColor() {
        return this.getAttribute('border-color') || 'rgba(2, 85, 204, 1)';
    }
    get strokeSize() {
        return this.getAttribute('border-size') || 1;
    }
    get circleRadius() {
        return this.getAttribute('circle-size') || 4;
    }
    get circleColor() {
        return this.getAttribute('circle-color') || 'rgba(2, 85, 204, 0.8)';
    }
    get circleStrokeColor() {
        return this.getAttribute('circle-border-color') || 'rgba(2, 85, 204, 1)';
    }
    get circleStrokeSize() {
        return this.getAttribute('circle-border-size') || 1;
    }
    // @ts-expect-error: style exists in HTMLElement
    get style() {
        return this._styleFunction();
    }
    get _styleFunction() {
        return () => {
            const styleConfig = {
                fill: new OlStyleFill({
                    color: this.fillColor,
                }),
                stroke: new OlStyleStroke({
                    color: this.strokeColor,
                    width: this.strokeSize,
                }),
                image: new OlStyleCircle({
                    radius: this.circleRadius,
                    fill: new OlStyleFill({
                        color: this.circleColor,
                    }),
                    stroke: new OlStyleStroke({
                        color: this.circleStrokeColor,
                        width: this.circleStrokeSize,
                    }),
                }),
                text: undefined,
            };
            return new OlStyle(styleConfig);
        };
    }
    _setStyleOnParent() {
        if (this.parentElement) {
            customElements.whenDefined(this.parentElement.tagName.toLowerCase()).then(() => {
                // @ts-expect-error: style property is readonly
                this.parentElement.style = this;
            });
        }
    }
};
VlMapDrawActionStyle = __decorate([
    webComponent('vl-map-draw-action-style')
], VlMapDrawActionStyle);
export { VlMapDrawActionStyle };
