// @domg-wc/components/block/wizard/vl-wizard-pane.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement } from '@domg-wc/common';
import { html, LitElement, nothing } from 'lit';
import { customElement } from 'lit/decorators.js';
let VlWizardPane = class VlWizardPane extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.isActive = false;
        this.name = '';
    }
    static get properties() {
        return {
            isActive: { type: Boolean },
            name: { type: String, attribute: 'name', reflect: true },
        };
    }
    updated(changed) {
        [...changed].forEach(([prop]) => prop === 'name' && this.parentElement?.onslotchange?.(prop));
        if (changed.has('name')) {
            if (this.parentElement && this.parentElement instanceof LitElement) {
                this.parentElement?.requestUpdate();
            }
        }
    }
    render() {
        return html `${this.isActive ? html ` <slot></slot>` : nothing}`;
    }
};
VlWizardPane = __decorate([
    customElement('vl-wizard-pane')
], VlWizardPane);
export { VlWizardPane };
