// @domg-wc/components/block/wizard/vl-wizard.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { BaseLitElement, registerWebComponents } from '@domg-wc/common';
import { resetStyle } from '@domg/govflanders-style/common';
import { wizardStyle } from '@domg/govflanders-style/component';
import { html } from 'lit';
import { customElement } from 'lit/decorators.js';
import { VlProgressIndicatorComponent } from '../progress-indicator';
import { VlWizardPane } from './vl-wizard-pane.component'; // onduidelijk waarom de vl declaratie en de util.js import nodig zijn, maar zonder falen de component.cy testen
let VlWizard = class VlWizard extends BaseLitElement {
    static get styles() {
        return [resetStyle, wizardStyle];
    }
    static get properties() {
        return {
            panes: { type: Array },
            activeStep: {
                type: Number,
                attribute: 'active-step',
                reflect: true,
            },
            hideLabels: { type: Boolean, attribute: 'hide-labels' },
            numeric: { type: Boolean, attribute: 'numeric' },
        };
    }
    constructor() {
        super();
        this.numeric = false;
        this.panes = [];
        this.activeStep = 1;
        this.hideLabels = false;
    }
    onSlotChange() {
        this.panes = [...this.querySelectorAll('vl-wizard-pane')];
    }
    updated() {
        this.panes.forEach((pane, index) => {
            pane.isActive = this.activeStep === index + 1;
        });
    }
    render() {
        return html `
            <section class="vl-wizard" wizard>
                <header class="vl-wizard__heading" role="none">
                    <slot name="title"></slot>
                    <slot name="header"></slot>
                </header>
                <vl-progress-indicator
                    active-step=${this.activeStep}
                    ?show-labels=${!this.hideLabels}
                    ?numeric=${this.numeric}
                    .steps=${this.panes.map((pane) => pane.name)}
                ></vl-progress-indicator>
                <div class="vl-wizard__panes">
                    <section class="vl-wizard__pane">
                        <slot @slotchange=${this.onSlotChange}></slot>
                    </section>
                </div>
            </section>
        `;
    }
};
(() => {
    registerWebComponents([VlProgressIndicatorComponent, VlWizardPane]);
})();
VlWizard = __decorate([
    customElement('vl-wizard'),
    __metadata("design:paramtypes", [])
], VlWizard);
export { VlWizard };
