// @domg-wc/components/block/proza-message/vl-proza-message-preloader.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var VlProzaMessagePreloader_1;
import { BaseHTMLElement, webComponent } from '@domg-wc/common';
import { ProzaRestClient } from './vl-proza-rest-client.util';
let VlProzaMessagePreloader = VlProzaMessagePreloader_1 = class VlProzaMessagePreloader extends BaseHTMLElement {
    static clearCache() {
        VlProzaMessagePreloader_1.__cache = {};
    }
    static get _observedAttributes() {
        return ['domain'];
    }
    constructor() {
        super();
        this.preload();
    }
    _domainChangedCallback() {
        this.preload();
    }
    get _domain() {
        return this.getAttribute('domain');
    }
    get _baseUrl() {
        return this.getAttribute('base-url');
    }
    preload() {
        if (this._domain) {
            VlProzaMessagePreloader_1.preload(this._domain, this._baseUrl);
        }
    }
    /**
     * @deprecated Gebruik het publieke `preload()`.
     */
    _preload() {
        return this.preload();
    }
    static preload(domain, baseUrl) {
        if (!VlProzaMessagePreloader_1.isPreloaded(domain)) {
            VlProzaMessagePreloader_1.__setPreloadedMessagesCacheForDomain(domain, ProzaRestClient.getMessages(domain, baseUrl));
        }
    }
    /**
     * @deprecated Gebruik het publieke `static preload()`.
     */
    static _preload(domain, baseUrl) {
        return VlProzaMessagePreloader_1.preload(domain, baseUrl);
    }
    /**
     * Geeft een Proza bericht terug.
     *
     * @param {String} domain - Het Proza domein waarin het Proza bericht zit.
     * @param {String} code - De code die het Proza bericht identificeert.
     * @return {Promise<string>} Resolved naar het Proza bericht indien teruggevonden en anders wordt de Promise rejected.
     */
    static getMessage(domain, code) {
        return VlProzaMessagePreloader_1._getMessages(domain).then((messages) => {
            const message = messages[code];
            if (message) {
                return message;
            }
            else {
                throw Error(`Bericht voor {domein: ${domain}, code: ${code}} niet gevonden`);
            }
        });
    }
    /**
     * Geeft de Proza codes terug op basis van een prefix.
     *
     * @param {String} domain - Het Proza domein waarin het Proza bericht zit.
     * @param {String} prefix - De prefix van de code die het Proza bericht
     * @param {String} [baseUrl] - De baseUrl waarvan de Proza berichten gefetched worden.
     * identificeert.
     * @return {Promise<[string]>} Resolved naar de Proza codes met de
     * opgegeven prefix
     */
    static async getProzaCodes(domain, prefix, baseUrl) {
        VlProzaMessagePreloader_1.preload(domain, baseUrl);
        const messages = await VlProzaMessagePreloader_1._getMessages(domain);
        return Object.keys(messages).filter((code) => code.startsWith(prefix));
    }
    /**
     * Geeft terug of het Proza domein al reeds preloaded werd.
     *
     * @param {String} domain - Het Proza domein.
     * @return {boolean}
     */
    static isPreloaded(domain) {
        return !!VlProzaMessagePreloader_1.__getPreloadedMessagesCacheForDomain(domain);
    }
    static _getMessages(domain) {
        if (VlProzaMessagePreloader_1.isPreloaded(domain)) {
            return VlProzaMessagePreloader_1.__getPreloadedMessagesCacheForDomain(domain);
        }
        else {
            return Promise.reject(new Error(`Berichten voor domein ${domain} zijn niet preloaded`));
        }
    }
    static get __domainCache() {
        if (!VlProzaMessagePreloader_1.__cache) {
            VlProzaMessagePreloader_1.__cache = {};
        }
        return VlProzaMessagePreloader_1.__cache;
    }
    static __getCacheForDomain(domain) {
        const cache = VlProzaMessagePreloader_1.__domainCache;
        if (!cache[domain]) {
            cache[domain] = {};
        }
        return cache[domain];
    }
    static __getPreloadedMessagesCacheForDomain(domain) {
        // @ts-expect-error: messages is of type unknown
        return VlProzaMessagePreloader_1.__getCacheForDomain(domain).messages;
    }
    static __setPreloadedMessagesCacheForDomain(domain, messages) {
        // @ts-expect-error: messages is of type unknown
        VlProzaMessagePreloader_1.__getCacheForDomain(domain).messages = messages;
    }
};
VlProzaMessagePreloader = VlProzaMessagePreloader_1 = __decorate([
    webComponent('vl-proza-message-preloader'),
    __metadata("design:paramtypes", [])
], VlProzaMessagePreloader);
export { VlProzaMessagePreloader };
