// @domg/govflanders-style/component/properties/vl-properties.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-properties__title {
  font-size: 2rem;
  font-weight: 500;
  margin-bottom: 2rem;
}
.vl-properties__list {
  display: flex;
  flex-wrap: wrap;
}
.vl-properties--full-width .vl-properties__list {
  display: grid;
  grid-template-columns: minmax(0, auto) 75%;
  grid-template-rows: 1fr;
  grid-gap: 0px 1rem;
  grid-template-areas: ". .";
}
@media screen and (max-width: 767px) {
  .vl-properties--full-width .vl-properties__list {
    grid-template-areas: ".";
    grid-template-columns: auto;
  }
}
.vl-properties__label, .vl-properties__data {
  font-size: 1.8rem;
  padding-bottom: 2rem;
  word-break: break-word;
}
.vl-properties__label--small, .vl-properties__data--small {
  padding-bottom: 0.5rem;
}
.vl-properties__label {
  flex: 0 0 25%;
  color: #687483;
  padding-right: 1rem;
  max-width: 17rem;
}
.vl-properties__label--has-input {
  padding-top: 0.7rem;
}
.vl-properties--full-width .vl-properties__label {
  flex: none;
  max-width: none;
  padding-right: 0;
}
.vl-properties__data {
  flex: 0 0 75%;
  flex-grow: 3;
}
.vl-properties--full-width .vl-properties__data {
  flex: none;
}
.vl-properties__column {
  display: inline-block;
  width: calc(50% - 1rem);
}
.vl-properties__column + .vl-properties__column {
  padding-right: 0;
  padding-left: 1rem;
}
.vl-properties__column--full {
  width: 100%;
  padding-left: 0 !important;
}
.vl-properties__column--full .vl-properties__label {
  flex: 0 0 calc(13% - 0.85rem);
}
.vl-properties__column--full .vl-properties__data {
  flex: 0 0 85%;
}
.vl-properties .vl-form__buttons {
  padding-top: 3rem;
}
.vl-properties--collapsed .vl-properties__title {
  font-size: 1.8rem;
}
.vl-properties--full-width .vl-properties--collapsed .vl-properties__list {
  grid-template-columns: 1fr;
  grid-template-rows: 1fr;
  grid-gap: 0px 0px;
  grid-template-areas: ".";
}
.vl-properties--collapsed .vl-properties__label, .vl-properties--collapsed .vl-properties__data {
  font-size: 1.6rem;
  padding-bottom: 0.5rem;
}
.vl-properties--collapsed .vl-properties__label--small, .vl-properties--collapsed .vl-properties__data--small {
  padding-bottom: 0.5rem;
}
.vl-properties--collapsed .vl-properties__label {
  flex: 0 0 100%;
  padding-right: 0;
}
.vl-properties--collapsed .vl-properties__data {
  flex: 0 0 100%;
  padding-bottom: 1.5rem;
}
.vl-properties--collapsed .vl-properties__column {
  width: 100%;
  padding-right: 0;
}
.vl-properties--collapsed .vl-properties__column + .vl-properties__column {
  padding-left: 0;
}
.vl-properties--collapsed .vl-properties__column--full .vl-properties__label {
  flex: 0 0 40%;
}
.vl-properties--collapsed .vl-properties__column--full .vl-properties__data {
  flex: 0 0 60%;
}
.vl-properties--bordered:before, .vl-properties--bordered:after {
  content: "";
  display: block;
  min-height: 6px;
  border: 0;
  background: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 6.04 5.99'%3E%3Cdefs%3E%3Cstyle%3E.cls-1%7Bfill:%23bec5cf;%7D%3C/style%3E%3C/defs%3E%3Ctitle%3EAsset 1%3C/title%3E%3Cg id='Layer_2' data-name='Layer 2'%3E%3Cg id='Layer_1-2' data-name='Layer 1'%3E%3Crect class='cls-1' x='1.01' y='3.99' width='1.01' height='1'/%3E%3Crect class='cls-1' y='4.99' width='1.01' height='1'/%3E%3Crect class='cls-1' x='3.02' y='2' width='1.01' height='1'/%3E%3Crect class='cls-1' x='2.01' y='2.99' width='1.01' height='1'/%3E%3Crect class='cls-1' x='5.04' width='1.01' height='1'/%3E%3Crect class='cls-1' x='4.03' y='1' width='1.01' height='1'/%3E%3C/g%3E%3C/g%3E%3C/svg%3E") repeat-x;
  background-size: 6px 6px;
}
.vl-properties--bordered:before {
  padding-bottom: 2rem;
}
.vl-properties--bordered:after {
  padding-top: 2rem;
}
@media screen and (max-width: 767px) {
  .vl-properties .vl-properties__title {
    font-size: 1.8rem;
  }
  .vl-properties--full-width .vl-properties .vl-properties__list {
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    grid-gap: 0px 0px;
    grid-template-areas: ".";
  }
  .vl-properties .vl-properties__label, .vl-properties .vl-properties__data {
    font-size: 1.6rem;
    padding-bottom: 0.5rem;
  }
  .vl-properties .vl-properties__label--small, .vl-properties .vl-properties__data--small {
    padding-bottom: 0.5rem;
  }
  .vl-properties .vl-properties__label {
    flex: 0 0 100%;
    padding-right: 0;
  }
  .vl-properties .vl-properties__data {
    flex: 0 0 100%;
    padding-bottom: 1.5rem;
  }
  .vl-properties .vl-properties__column {
    width: 100%;
    padding-right: 0;
  }
  .vl-properties .vl-properties__column + .vl-properties__column {
    padding-left: 0;
  }
  .vl-properties .vl-properties__column--full .vl-properties__label {
    flex: 0 0 40%;
  }
  .vl-properties .vl-properties__column--full .vl-properties__data {
    flex: 0 0 60%;
  }
}
`;
export default style;
