// @domg/govflanders-style/component/input-field/vl-input-field.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-input-field {
  display: inline-block;
  background: #fff;
  font-family: "Flanders Art Sans", sans-serif;
  font-size: 1.6rem;
  color: #333332;
  max-width: 100%;
  height: 3.5rem;
  line-height: 3.5rem;
  border-radius: 0.3rem;
  border: 0.1rem solid #8695a8;
  -webkit-appearance: none;
  -moz-appearance: textfield;
  padding: 0 1rem;
  transition: background-color 0.2s;
}
@media screen and (max-width: 767px) {
  .vl-input-field {
    font-size: 1.6rem;
  }
}
.vl-input-field:hover {
  border: 0.2rem solid rgba(0, 85, 204, 0.65);
  padding: 0 0.9rem;
}
.vl-input-field:hover.vl-input-field--error, .vl-input-field:hover.invalid.validated {
  border-color: #d2373c;
}
.vl-input-field:hover.vl-input-field--success, .vl-input-field:hover.valid.validated {
  border-color: #009e47;
}
.vl-input-field:hover.vl-input-field--small {
  padding: 0 0.7rem;
}
.vl-input-field:focus, .vl-input-field--focus {
  box-shadow: 0 0 0 2px #fff, 0 0 0 5px rgba(0, 85, 204, 0.65);
  outline: transparent solid 0.2rem;
  border: 0.1rem solid #8695a8;
  padding: 0 1rem;
}
@supports (outline-offset: 2px) {
  .vl-input-field:focus, .vl-input-field--focus {
    box-shadow: none;
    outline: 3px solid rgba(0, 85, 204, 0.65);
    outline-offset: 2px;
  }
}
.vl-input-field:focus.vl-input-field--error, .vl-input-field:focus.invalid.validated, .vl-input-field--focus.vl-input-field--error, .vl-input-field--focus.invalid.validated {
  border-color: #d2373c;
}
.vl-input-field:focus.vl-input-field--success, .vl-input-field:focus.valid.validated, .vl-input-field--focus.vl-input-field--success, .vl-input-field--focus.valid.validated {
  border-color: #009e47;
}
.vl-input-field:focus:hover, .vl-input-field--focus:hover {
  padding: 0 1rem;
}
.vl-input-field:focus.vl-input-field--small, .vl-input-field--focus.vl-input-field--small {
  padding: 0 0.8rem;
}
.vl-input-field::placeholder {
  color: #687483;
}
.vl-input-field::-webkit-search-cancel-button {
  -webkit-appearance: none;
}
.vl-input-field--block {
  display: block;
  width: 100%;
}
.vl-input-field--small {
  max-width: 6rem;
  padding: 0 0.8rem;
}
.vl-input-field--error, .vl-input-field.invalid.validated {
  border-color: #d2373c;
  background-color: #fbebec;
  display: inline;
}
.vl-input-field--error .vl-select, .vl-input-field.invalid.validated .vl-select {
  border-color: #d2373c;
  background-color: #fbebec;
}
.vl-input-field--error .js-vl-select[data-type*=select-one], .vl-input-field--error .js-vl-select[data-type*=select-multiple], .vl-input-field.invalid.validated .js-vl-select[data-type*=select-one], .vl-input-field.invalid.validated .js-vl-select[data-type*=select-multiple] {
  border-color: #d2373c;
  background-color: #fbebec;
}
.vl-input-field--error .js-vl-select[data-type*=select-one] .vl-select__inner, .vl-input-field--error .js-vl-select[data-type*=select-multiple] .vl-select__inner, .vl-input-field.invalid.validated .js-vl-select[data-type*=select-one] .vl-select__inner, .vl-input-field.invalid.validated .js-vl-select[data-type*=select-multiple] .vl-select__inner {
  border-color: #d2373c;
  background-color: #fbebec;
  background-color: transparent;
}
.vl-input-field--success, .vl-input-field.valid.validated {
  border-color: #009e47;
  background-color: #e6f5ed;
  display: inline;
}
.vl-input-field--success .vl-select, .vl-input-field.valid.validated .vl-select {
  border-color: #009e47;
  background-color: #e6f5ed;
}
.vl-input-field--success .js-vl-select[data-type*=select-one], .vl-input-field--success .js-vl-select[data-type*=select-multiple], .vl-input-field.valid.validated .js-vl-select[data-type*=select-one], .vl-input-field.valid.validated .js-vl-select[data-type*=select-multiple] {
  border-color: #009e47;
  background-color: #e6f5ed;
}
.vl-input-field--success .js-vl-select[data-type*=select-one] .vl-select__inner, .vl-input-field--success .js-vl-select[data-type*=select-multiple] .vl-select__inner, .vl-input-field.valid.validated .js-vl-select[data-type*=select-one] .vl-select__inner, .vl-input-field.valid.validated .js-vl-select[data-type*=select-multiple] .vl-select__inner {
  border-color: #009e47;
  background-color: #e6f5ed;
  background-color: transparent;
}
.vl-input-field--disabled, .vl-input-field[disabled] {
  background-color: #f3f5f6;
  border-color: #8695a8;
}
.vl-input-field--disabled:hover, .vl-input-field[disabled]:hover {
  border-width: 0.1rem;
  padding: 0 1rem;
}
.vl-input-field--inline {
  position: relative;
  display: block;
  max-width: 100%;
}
@media screen and (max-width: 767px) {
  .vl-input-field--inline {
    display: block;
    width: auto;
  }
}
.vl-input-field--inline .vl-input-field__submit {
  position: absolute;
  display: block;
  color: #fff;
  width: 4.3rem;
  height: 3.5rem;
  opacity: 0;
  padding: 0;
  margin-bottom: -2rem;
  transform: translateX(100%);
  top: 0;
  right: 0;
  z-index: -1;
}
.vl-input-field--inline .vl-input-field__submit:focus {
  transform: translateX(0%);
  transition: opacity 0.2s, transform 0.2s;
  opacity: 1;
  z-index: 1;
}
.vl-input-field--inline .vl-input-field__submit::before {
  position: absolute;
  font-size: 1.7rem;
  transform: translate(-50%, -50%);
  top: 50%;
  left: 50%;
}
.vl-input-field--inline .vl-input-field__input {
  display: block;
  width: 100%;
  text-align: left;
  padding-right: 5rem;
}
.vl-input-field--inline .vl-input-field__input:focus, .vl-input-field--inline .vl-input-field__input--focus {
  width: calc(100% - 5.2rem);
  padding-right: 0;
  transition: width 0.2s;
}
.vl-input-field--inline .vl-input-field__input:focus + .vl-input-field__submit, .vl-input-field--inline .vl-input-field__input--focus + .vl-input-field__submit {
  transform: translateX(0%);
  transition: opacity 0.2s, transform 0.2s;
  opacity: 1;
  z-index: 1;
}
`;
export default style;
