// @domg-wc/components/block/infotext/vl-infotext.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, formatNumber, throttle, webComponent } from '@domg-wc/common';
import { html, nothing } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import infotextStyle from './vl-infotext.css';
import { infotextDefaults } from './vl-infotext.defaults';
let VlInfotextComponent = class VlInfotextComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        // Attribute(s)
        this.badge = infotextDefaults.badge;
        this.href = infotextDefaults.href;
        this.external = infotextDefaults.external;
        // State
        this.value = '';
        // Variable(s)
        this.customStyleSheet = new CSSStyleSheet();
        this.valueObserver = null;
        this.throttledAdjustValue = throttle(() => this.adjustValue(), 100);
        this.getCustomFontSize = (textLength = 0) => {
            const shouldUseSmallerFontSize = !this.badge && window.innerWidth <= 767;
            const normalFontSize = shouldUseSmallerFontSize ? 2.4 : 4.8;
            const minimumFontSize = 1.6;
            const customFontSize = normalFontSize - textLength / 4;
            if (!textLength || textLength <= 4) {
                return `${normalFontSize}rem`;
            }
            if (customFontSize < minimumFontSize) {
                return `${minimumFontSize}rem`;
            }
            return `${customFontSize}rem`;
        };
    }
    static get styles() {
        return [infotextStyle];
    }
    static get properties() {
        return {
            badge: { type: Boolean },
            href: { type: String },
            external: { type: Boolean },
            value: { type: String, state: true },
        };
    }
    connectedCallback() {
        super.connectedCallback();
        this.adjustValue();
        this.valueObserver = new MutationObserver(() => this.adjustValue());
        this.getValueSlot() && this.valueObserver.observe(this.getValueSlot(), { subtree: true, childList: true });
        window.addEventListener('resize', this.throttledAdjustValue);
        this.shadowRoot.adoptedStyleSheets = [...this.shadowRoot.adoptedStyleSheets, this.customStyleSheet];
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.valueObserver?.disconnect();
        window.removeEventListener('resize', this.throttledAdjustValue);
    }
    render() {
        const classes = {
            'vl-infotext': true,
            'vl-infotext--badge': this.badge,
        };
        if (this.href) {
            const target = this.external ? '_blank' : nothing;
            return html `<a class=${classMap(classes)} href=${this.href} target=${target}>${this.renderContent()}</a>`;
        }
        return html ` <div class=${classMap(classes)}>${this.renderContent()}</div>`;
    }
    renderContent() {
        return html `
            <div class="vl-infotext__value">${this.value}</div>
            <div class="vl-infotext__text">
                <slot name="text"></slot>
            </div>
            <slot name="value" hidden></slot>
        `;
    }
    getValueSlot() {
        return this.querySelector('[slot="value"]');
    }
    adjustValue() {
        const valueSlot = this.getValueSlot();
        const formattedValue = this.getFormattedValue(valueSlot?.textContent);
        const customFontSize = this.getCustomFontSize(formattedValue.length);
        this.value = formattedValue;
        this.customStyleSheet.replaceSync(`.vl-infotext__value { font-size: ${customFontSize}; }`);
    }
    getFormattedValue(value) {
        if (!value || isNaN(Number(value))) {
            return value || '';
        }
        return formatNumber(Number(value));
    }
};
VlInfotextComponent = __decorate([
    webComponent('vl-infotext')
], VlInfotextComponent);
export { VlInfotextComponent };
