// @domg-wc/components/block/typography/vl-typography.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var VlTypography_1;
import { BaseHTMLElement, webComponent } from '@domg-wc/common';
import { baseStyle, elementStyle, resetStyle, typographyStyle as commonTypographyStyle, } from '@domg/govflanders-style/common';
import { titlesStyle, typographyStyle } from '@domg/govflanders-style/component';
import { vlTypographyFluxStyles } from './vl-typography.flux-css';
let VlTypography = VlTypography_1 = class VlTypography extends BaseHTMLElement {
    constructor() {
        const html = `
            <div id="content" class="vl-typography"></div>
        `;
        const styleSheets = [
            resetStyle.styleSheet,
            baseStyle.styleSheet,
            elementStyle.styleSheet,
            typographyStyle.styleSheet,
            commonTypographyStyle.styleSheet,
            vlTypographyFluxStyles.styleSheet,
            titlesStyle.styleSheet,
        ];
        super(html, styleSheets);
        this._observer = this.__observeSlotElements(() => this.__processSlotElements());
    }
    static get _observedAttributes() {
        return ['parameters'];
    }
    /**
     * Dit vormt een template met placeholders voor parameters om in een tekst waarin deze placeholders vervangen
     * zijn met hun waarden.
     *
     * @param {string} template De template waarin placeholders zitten in de vorm van "${parameter.parameterNaam}"
     * @param {object} params Een object met keys die de naam van de parameter voorstellen (bv. "parameterNaam") en hun waarden
     * @return {string} De resulterende tekst
     */
    static replaceTemplateParameters(template, params) {
        Object.keys(params).forEach((key) => (template = template.replaceAll('${parameter.' + key + '}', params[key])));
        template = template.replace(/\${parameter.\w+}/g, '');
        return template;
    }
    connectedCallback() {
        super.connectedCallback();
        this.__processSlotElements();
    }
    disconnectedCallback() {
        this._observer?.disconnect();
    }
    _parametersChangedCallback() {
        this.__processSlotElements();
    }
    __processSlotElements() {
        this.__clearChildren();
        const parameters = this.getAttribute('parameters') ? JSON.parse(this.getAttribute('parameters')) : {};
        const template = VlTypography_1.replaceTemplateParameters(this.innerHTML, parameters);
        this._element.insertAdjacentHTML('afterbegin', template);
    }
    __clearChildren() {
        while (this._element.hasChildNodes()) {
            this._element.firstChild.remove();
        }
    }
    __observeSlotElements(callback) {
        const node = this;
        const observer = new MutationObserver(callback);
        observer.observe(node, {
            attributes: true,
            childList: true,
            characterData: true,
            subtree: true,
        });
        return observer;
    }
};
VlTypography = VlTypography_1 = __decorate([
    webComponent('vl-typography'),
    __metadata("design:paramtypes", [])
], VlTypography);
export { VlTypography };
