// @domg/govflanders-style/component/info-tile/vl-info-tile.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

.vl-info-tile {
  display: block;
  position: relative;
  border: 1px solid #cbd2da;
  padding: 1.7rem;
}
@media screen and (max-width: 767px) {
  .vl-info-tile {
    padding: 1rem;
  }
}
.vl-info-tile--alt {
  background-color: #f7f9fc;
}
.vl-info-tile--clickable {
  text-decoration: none;
  cursor: pointer;
  text-align: left;
  appearance: none;
  border-radius: 0;
  color: var(--vl-theme-action-color, #05c);
}
.vl-info-tile--clickable:hover, .vl-info-tile--clickable:focus {
  background-color: #f7f9fc;
  color: var(--vl-theme-action-hover-color, #003bb0);
}
.vl-info-tile--clickable:hover .vl-info-tile__header__title, .vl-info-tile--clickable:focus .vl-info-tile__header__title {
  text-decoration: underline;
}
.vl-info-tile--clickable:hover.vl-info-tile--date .vl-info-tile__date, .vl-info-tile--clickable:focus.vl-info-tile--date .vl-info-tile__date {
  background: var(--vl-theme-action-hover-color, #003bb0);
}
.vl-info-tile--clickable.vl-info-tile--date .vl-info-tile__date {
  background: var(--vl-theme-action-hover-color, #003bb0);
}
.vl-info-tile--clickable:visited.vl-info-tile--date .vl-info-tile__date {
  background: #660599;
}
.vl-info-tile--m {
  padding: 2.5rem;
}
@media screen and (max-width: 767px) {
  .vl-info-tile--m {
    padding: 2rem;
  }
}
.vl-info-tile--l {
  padding: 4.5rem;
}
@media screen and (max-width: 767px) {
  .vl-info-tile--l {
    padding: 2rem;
  }
}
.vl-info-tile--center {
  text-align: center;
}
.vl-info-tile--center .vl-info-tile__header {
  display: block;
  margin-bottom: 1.6rem;
}
.vl-info-tile--center .vl-info-tile__header__wrapper {
  width: 100%;
}
.vl-info-tile--center .vl-info-tile__header__icon:not(.vl-badge) {
  color: #cbd2da;
  font-size: 3rem;
  line-height: 1;
  display: inline-block;
}
.vl-info-tile--center .vl-info-tile__footer {
  margin-top: 2rem;
}
.vl-info-tile--v-center {
  display: flex;
  flex-direction: column;
  justify-content: center;
}
.vl-info-tile.vl-js-equal-height {
  height: 100%;
}
.vl-info-tile--date {
  padding: 0;
  display: flex;
  align-items: stretch;
}
.vl-info-tile--date[href] .vl-info-tile__date {
  color: #fff;
}
.vl-info-tile--date .vl-info-tile__date {
  background-color: #f7f9fc;
  padding: 1.7rem;
}
@media screen and (max-width: 767px) {
  .vl-info-tile--date .vl-info-tile__date {
    padding: 1rem;
  }
}
.vl-info-tile--date .vl-info-tile__date__main {
  text-align: center;
  font-weight: bold;
}
.vl-info-tile--date .vl-info-tile__date__sub {
  text-align: center;
  font-size: 1.4rem;
}
.vl-info-tile--date .vl-info-tile__date__wrapper {
  padding: 1.7rem;
  flex: 1;
}
@media screen and (max-width: 767px) {
  .vl-info-tile--date .vl-info-tile__date__wrapper {
    padding: 1rem;
  }
}
.vl-info-tile--badge-bottom .vl-info-tile__content {
  display: flex;
}
.vl-info-tile__badges {
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  flex-shrink: 0;
  margin-left: 1.7rem;
}
.vl-info-tile__badges__badge:not(:first-of-type) {
  margin-top: 0.5rem;
}
.vl-info-tile--highlight {
  box-shadow: inset 0 0.5rem 0 0 var(--vl-theme-accent-color, var(--vl-theme-primary-color));
}
.vl-info-tile__header {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  flex: 1;
}
.vl-info-tile__header .vl-info-tile__badge__wrapper {
  margin-right: 1.4rem;
  flex-shrink: 0;
}
.vl-info-tile__header__title {
  font-weight: 500;
  line-height: 1.25;
  font-family: "Flanders Art Sans", sans-serif;
  font-size: 1.8rem;
}
@media screen and (max-width: 767px) {
  .vl-info-tile__header__title {
    font-size: 1.5rem;
  }
}
.vl-info-tile__header__subtitle {
  color: #687483;
  line-height: 1.3;
  font-family: "Flanders Art Sans", sans-serif;
  font-size: 1.4rem;
  margin-top: 0.3rem;
}
.vl-info-tile__content {
  font-size: 1.6rem;
  margin-top: 1rem;
  flex: 1;
  color: #333332;
}
@media screen and (max-width: 767px) {
  .vl-info-tile__content {
    font-size: 1.5rem;
  }
}
.vl-info-tile__content__wrapper {
  flex: 1;
}
`;
export default style;
