// @domg-wc/common/base/base.lit.element.js
import { LitElement } from 'lit';
import { GlobalStyles } from '../styles/global-styles';
export class BaseLitElement extends LitElement {
    constructor() {
        super();
        this.allowCustomCSS = true;
        this.customCSS = null;
        this.customCSSPrefix = null;
        GlobalStyles.getInstance().register();
    }
    static get properties() {
        return {
            customCSS: { type: String, attribute: 'custom-css', reflect: true },
        };
    }
    // We fixen hier de `getRootNode()` return type, omdat `getRootNode()` standaard `Node` retourneert 
    // en om te vermijden dat we overal het type moeten casten naar `Element` of `ShadowRoot`.
    getRootNode(options) {
        return super.getRootNode(options);
    }
    connectedCallback() {
        super.connectedCallback();
        this.addCustomCSS();
    }
    addCustomCSS() {
        this.customCSS = this.customCSSPrefix ? this.customCSSPrefix : this.customCSS;
        if (this.customCSS && !this.allowCustomCSS) {
            console.warn('Custom CSS is niet toegelaten voor dit component.');
            return;
        }
        if (this.customCSS && !this.shadowRoot) {
            console.warn('Dit component heeft geen shadow DOM om custom CSS aan toe te voegen.');
            return;
        }
        if (this.customCSS && this.shadowRoot) {
            const customStyleSheet = new CSSStyleSheet();
            customStyleSheet.replaceSync(this.customCSS);
            this.shadowRoot.adoptedStyleSheets = [...this.shadowRoot.adoptedStyleSheets, customStyleSheet];
        }
    }
}
