// @domg-wc/map/actions/mapaction.js
export class VlBaseMapAction {
    constructor(interactions) {
        if (!Array.isArray(interactions)) {
            interactions = [interactions];
        }
        this._interactions = [];
        interactions.forEach((interaction) => {
            this.addInteraction(interaction);
        });
    }
    get layer() {
        return this._layer;
    }
    set layer(layer) {
        this._layer = layer;
    }
    get interactions() {
        return this._interactions;
    }
    set interactions(interactions) {
        this._interactions = interactions;
    }
    addInteraction(interaction) {
        interaction.setActive(false);
        this._interactions.push(interaction);
    }
    activate() {
        this._interactions.forEach((interaction) => {
            interaction.setActive(true);
        });
    }
    deactivate() {
        this._interactions.forEach((interaction) => {
            interaction.setActive(false);
        });
    }
    getControl() {
        return this.map.getActionControlWithIdentifier(this.element.identifier);
    }
}
