// @domg-wc/components/form/form-label/vl-form-label.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, webComponent } from '@domg-wc/common';
import { resetStyle } from '@domg/govflanders-style/common';
import { html } from 'lit';
import { classMap } from 'lit/directives/class-map.js';
import { formLabelStyles } from './vl-form-label.css';
import { formLabelDefaults } from './vl-form-label.defaults';
let VlFormLabelComponent = class VlFormLabelComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        // Attributes
        this.for = formLabelDefaults.for;
        this.label = formLabelDefaults.label;
        this.block = formLabelDefaults.block;
        this.light = formLabelDefaults.light;
    }
    static get styles() {
        return [resetStyle, formLabelStyles];
    }
    static get properties() {
        return {
            for: { type: String },
            label: { type: String },
            block: { type: Boolean },
            light: { type: Boolean },
        };
    }
    connectedCallback() {
        super.connectedCallback();
        this.addEventListener('click', this.focusFormControl);
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        const formControl = this.getFormControl();
        formControl?.setAttribute('label', this.label);
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        const formControl = this.getFormControl();
        formControl?.removeAttribute('label');
        this.removeEventListener('click', this.focusFormControl);
    }
    render() {
        const classList = {
            'vl-form__label': true,
            'vl-form__label--block': this.block,
            'vl-form__label--light': this.light,
        };
        return html `
            <label for=${this.for} class=${classMap(classList)} part="label">
                ${this.label ? this.label : html ` <slot></slot>`}
            </label>
        `;
    }
    getFormControl() {
        const form = this.closest('form') || this.parentElement;
        return form?.querySelector(`[id="${this.for}"]`);
    }
    focusFormControl() {
        const formControl = this.getFormControl();
        formControl?.focus();
    }
};
VlFormLabelComponent = __decorate([
    webComponent('vl-form-label')
], VlFormLabelComponent);
export { VlFormLabelComponent };
