// @domg-wc/components/block/alert/vl-alert.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, findNodesForSlot } from '@domg-wc/common';
import { accessibilityStyle, markStyle, resetStyle } from '@domg/govflanders-style/common';
import { alertStyle } from '@domg/govflanders-style/component';
import { html } from 'lit';
import { customElement } from 'lit/decorators.js';
import { classMap } from 'lit/directives/class-map.js';
import { vlIconStyles } from '../../atom/icon-style/vl-icon-style.css';
import { VlAlertClosedEvent } from './vl-alert.model';
import { vlAlertFluxStyles } from './vl-alert.flux-css';
let VlAlert = class VlAlert extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.icon = '';
        this.title = '';
        this.size = '';
        this.message = '';
        this.naked = false;
        this.closable = false;
    }
    static get styles() {
        return [resetStyle, alertStyle, accessibilityStyle, markStyle, vlAlertFluxStyles, vlIconStyles];
    }
    static get properties() {
        return {
            icon: { type: String, attribute: 'icon' },
            title: { type: String, attribute: 'title' },
            closable: { type: Boolean, attribute: 'closable' },
            type: { type: String, attribute: 'type' },
            size: { type: String, attribute: 'size' },
            naked: { type: Boolean, attribute: 'naked' },
            message: { type: String, attribute: 'message' },
        };
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        this.processButtons();
    }
    render() {
        const classes = {
            'vl-alert': true,
            [`vl-alert--${this.type}`]: Boolean(this.type),
            'vl-alert--small': this.size === 'small',
            'vl-alert--naked': this.naked,
        };
        const markClass = this.naked ? `vl-u-mark--${this.type}` : '';
        const hideEmptyActionsSlot = findNodesForSlot(this, 'actions')?.length ? '' : 'vl-u-visually-hidden';
        return html `
            <div id="alert" class=${classMap(classes)} role="alert">
                ${this.icon &&
            html ` <div class="vl-alert__icon">
                    <span class="vl-icon vl-icon--${this.icon}"></span>
                </div>`}
                <div id="content" class="vl-alert__content">
                    <p id="title" class="vl-alert__title">
                        <slot class=${markClass} name="title">${this.title}</slot>
                    </p>
                    <div id="message" class="vl-alert__message">
                        <p class=${markClass}>${this.message}</p>
                        <slot id="message-slot"></slot>
                    </div>
                    <div id="actions" class="vl-alert__actions ${hideEmptyActionsSlot}">
                        <slot id="actions-slot" @slotchange=${this.requestUpdate} name="actions"></slot>
                    </div>
                </div>
                ${this.closable
            ? html `
                          <button
                              id="close"
                              class="vl-alert__close"
                              type="button"
                              aria-controls="alert"
                              aria-expanded="true"
                              @click=${this.removeAlert}
                          >
                              <span class="vl-icon vl-icon--cross" aria-hidden="true"></span>
                              <span class="vl-u-visually-hidden">Melding sluiten</span>
                          </button>
                      `
            : ''}
            </div>
        `;
    }
    removeAlert() {
        this.remove();
        this.dispatchEvent(new VlAlertClosedEvent());
    }
    processButtons() {
        const actionsSlotElement = this.renderRoot.querySelector('slot[name="actions"]');
        const buttonNodes = actionsSlotElement
            ?.assignedNodes()
            .filter((element) => element instanceof HTMLButtonElement);
        buttonNodes.forEach((node) => node.setAttribute('narrow', ''));
    }
};
VlAlert = __decorate([
    customElement('vl-alert')
], VlAlert);
export { VlAlert };
