// @domg-wc/map/components/action/click-action/vl-map-click-action.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import Overlay from 'ol/Overlay';
import { customElement } from 'lit/decorators.js';
import { VlMapClickActionPindrop } from './vl-map-click-action-pindrop';
import { VlMapClickedEvent } from './VlMapClickedEvent';
import { BaseLitElement } from '@domg-wc/common';
let VlMapClickAction = class VlMapClickAction extends BaseLitElement {
    constructor() {
        super();
        this.handleClick = (evt) => {
            if (!this.mapRef?.map)
                return;
            const doesOverlayAlreadyExist = this.mapRef.map.getOverlays().getArray().includes(this.overlay);
            this.overlay.setPosition(evt.coordinate);
            if (!doesOverlayAlreadyExist) {
                this.mapRef.map.addOverlay(this.overlay);
            }
            this.dispatchEvent(new VlMapClickedEvent(evt.coordinate, this.mapRef.map.getView().getResolution(), this.mapRef.map.getView().getProjection()));
        };
        this.overlay = new Overlay({
            element: new VlMapClickActionPindrop(),
            positioning: 'bottom-center',
            autoPan: { animation: { duration: 250 } },
        });
        this.mapRef = null;
    }
    connectedCallback() {
        super.connectedCallback();
        this.mapRef = this.closest('vl-map'); // Sla de map referentie op
        if (this.mapRef?.map) {
            this.mapRef.map.on('singleclick', this.handleClick);
        }
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        if (this.mapRef?.map) {
            this.mapRef.map.removeOverlay(this.overlay);
            this.mapRef.map.un('singleclick', this.handleClick);
        }
        this.mapRef = null; // Wis de map referentie
    }
};
VlMapClickAction = __decorate([
    customElement('vl-map-click-action'),
    __metadata("design:paramtypes", [])
], VlMapClickAction);
export { VlMapClickAction };
