// @domg/govflanders-style/component/link/vl-link.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-vi::before, .vl-vi::after {
  font-family: "vlaanderen-icon" !important;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-style: normal;
  font-variant: normal;
  font-weight: normal;
  text-decoration: none;
  text-transform: none;
  display: inline-block;
  vertical-align: middle;
}
.vl-vi.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
  vertical-align: middle;
}

.vl-vi-u-xs::before {
  font-size: 0.8rem;
}

.vl-vi-u-s::before {
  font-size: 1.3rem;
}

.vl-vi-u-m::before {
  font-size: 1.7rem;
}

.vl-vi-u-l::before {
  font-size: 2rem;
}

.vl-vi-u-xl::before {
  font-size: 2.2rem;
}

.vl-vi-u-90deg::before {
  display: inline-block;
  transform: rotate(90deg);
}

.vl-vi-u-180deg::before {
  display: inline-block;
  transform: rotate(180deg);
}

[type=submit].vl-link.vl-link--icon {
  padding-left: 1.8rem;
}

button.vl-link {
  cursor: default;
}

.vl-link {
  border-radius: 0;
  appearance: none;
  -webkit-appearance: none;
  border: 0;
  background-color: transparent;
  padding: 0;
  text-decoration: underline;
  display: inline-block;
  cursor: pointer;
  font-weight: inherit;
  text-align: left;
  font-family: inherit;
  font-size: inherit;
  color: var(--vl-theme-action-color, #05c);
  position: relative;
  word-break: break-word;
  line-height: inherit;
}
.vl-link:hover {
  text-decoration: none;
}
.vl-link:focus, .vl-link:active {
  text-decoration: none;
}
.vl-link:hover, .vl-link:focus, .vl-link:active {
  color: var(--vl-theme-action-hover-color, #003bb0);
  text-decoration: none;
}
.vl-link:focus, .vl-link:active {
  color: var(--vl-theme-action-active-color, #004099);
}
.vl-link:visited {
  color: #660599;
}
.vl-link:visited .vl-link__icon:not(.vl-link__icon--neutral) {
  color: #660599;
}
.vl-link--bold {
  font-weight: 500;
  text-decoration: none;
}
.vl-link--bold:hover, .vl-link--bold:focus, .vl-link--bold:active {
  text-decoration: underline;
}
.vl-link--small {
  font-size: 1.6rem;
}
@media screen and (max-width: 767px) {
  .vl-link--small {
    font-size: 1.5rem;
  }
}
.vl-link--large {
  font-size: 2rem;
}
@media screen and (max-width: 767px) {
  .vl-link--large {
    font-size: 1.8rem;
  }
}
.vl-link--block {
  display: block;
}
.vl-link--inline {
  display: inline;
}
.vl-link--flex {
  display: flex;
  align-items: center;
}
.vl-link--error {
  color: #d2373c;
}
.vl-link--error:hover, .vl-link--error:focus, .vl-link--error:active, .vl-link--error:visited {
  color: #aa2729;
}
.vl-link--neutral {
  color: #333332;
}
.vl-link--neutral:hover, .vl-link--neutral:focus, .vl-link--neutral:active, .vl-link--neutral:visited {
  color: #333332;
}

.vl-link__icon {
  position: relative;
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
  text-decoration: none;
  color: inherit;
  line-height: 1;
  margin-top: -0.1em;
  flex-shrink: 0;
}
.vl-link__icon::before, .vl-link__icon::after {
  display: inline-block;
}
.vl-link__icon--before {
  padding-right: 0.4rem;
}
.vl-link__icon--after {
  padding-left: 0.4rem;
}
.vl-link__icon--standalone {
  padding: 0;
  float: none;
  position: relative;
}
.vl-link__icon--light {
  color: #687483;
}
.vl-link__icon--neutral {
  color: #333332;
}
.vl-link__icon--borderless {
  overflow: visible;
}
.vl-link__icon--borderless::after {
  position: absolute;
  display: block;
  height: 0.1rem;
  left: 0;
  width: 100%;
  bottom: 0;
  margin-bottom: -0.3rem;
  background-color: #fff;
  content: "";
}
.vl-link__icon.vl-vi-external {
  font-size: 1em;
}
`;
export default style;
