// @domg/govflanders-style/component/toggle/vl-toggle.css.js
import { css } from 'lit';
const style = css `
:root {
  --vl-theme-primary-color: #ffe615;
  --vl-theme-primary-color-60: #fff073;
  --vl-theme-primary-color-70: #ffee5b;
  --vl-theme-primary-color-rgba-30: rgba(255, 230, 21, 0.3);
  --vl-theme-fg-color: #333332;
  --vl-theme-fg-color-60: #858584;
  --vl-theme-fg-color-70: #707070;
}

.vl-toggle {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  position: relative;
  overflow: hidden;
  width: 100%;
}
.vl-toggle--align-top {
  align-items: flex-start;
}
.vl-toggle--icon-right {
  flex-direction: row-reverse;
}
.vl-toggle--icon-right .vl-accordion__title {
  flex-grow: 1;
}
.vl-toggle > .vl-vi::before {
  transition: transform 0.2s;
}
.vl-toggle > .vl-vi.vl-vi-arrow-right-fat::before, .vl-toggle > .vl-vi.vl-vi-arrow-right-thin::before {
  transform: rotate(90deg);
}
.js-vl-accordion--open > .vl-toggle .vl-vi-plus::before,
.js-vl-accordion--open > .vl-toggle .vl-vi-minus::before {
  transform: rotate(0);
}
.js-vl-accordion--open > .vl-toggle > .vl-vi-arrow-right-fat::before, .js-vl-accordion--open > .vl-accordion__summary > .vl-toggle > .vl-vi-arrow-right-fat::before, .js-vl-toggle--open > .vl-toggle > .vl-vi-arrow-right-fat::before, [aria-expanded=true] > .vl-toggle > .vl-vi-arrow-right-fat::before,
.js-vl-accordion--open > .vl-toggle > .vl-vi-arrow-right-thin::before,
.js-vl-accordion--open > .vl-accordion__summary > .vl-toggle > .vl-vi-arrow-right-thin::before,
.js-vl-toggle--open > .vl-toggle > .vl-vi-arrow-right-thin::before,
[aria-expanded=true] > .vl-toggle > .vl-vi-arrow-right-thin::before {
  transform: rotate(-90deg);
}
.vl-toggle.js-vl-drawer-active-item .vl-vi-arrow-right-fat::before,
.vl-toggle.js-vl-drawer-active-item .vl-vi-arrow-right-thin::before, .vl-toggle.js-vl-dynamic-drawer-active-item .vl-vi-arrow-right-fat::before,
.vl-toggle.js-vl-dynamic-drawer-active-item .vl-vi-arrow-right-thin::before {
  transform: rotate(-90deg);
}

.js-vl-toggle {
  position: relative;
}
.js-vl-toggle--flex-reverse {
  display: flex;
  flex-direction: column-reverse;
}

summary {
  display: block;
}
summary::-webkit-details-marker {
  display: none;
}

.vl-toggle-wrapper--more-button {
  text-align: center;
  position: relative;
  overflow: hidden;
  transition: margin 0.2s;
}
.js-vl-accordion--open .vl-toggle-wrapper--more-button {
  margin-top: 3rem;
}
.vl-toggle-wrapper--more-button + .vl-accordion__content {
  display: block;
}

.vl-toggle--more-button {
  display: flex;
  align-items: center;
  margin: 0 auto;
  position: relative;
  overflow: visible;
  line-height: 2.3rem;
  width: initial;
}
.vl-toggle--more-button::before, .vl-toggle--more-button::after {
  display: block;
  position: absolute;
  content: "";
  top: 50%;
  margin-top: -1px;
  height: 1px;
  width: 100vw;
  background-color: #cbd2da;
}
.vl-toggle--more-button::before {
  right: calc(100% + 3px);
}
.vl-toggle--more-button::after {
  left: calc(100% + 3px);
}
.vl-toggle--more-button:hover, .vl-toggle--more-button:focus {
  color: #05c;
}
.vl-toggle--more-button .vl-toggle__icon:first-child {
  margin-right: 1rem;
}
.vl-toggle--more-button .vl-toggle__icon:last-child {
  margin-left: 1rem;
}
.vl-toggle--more-button .vl-toggle__icon::before {
  font-size: 1.2rem;
  color: currentColor;
  transition: transform 0.2s;
}
.js-vl-accordion--open .vl-toggle--more-button .vl-toggle__icon::before {
  transform: rotate(-180deg);
}
`;
export default style;
