// @domg-wc/components/block/properties/vl-properties.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement, onChildListChange, webComponent } from '@domg-wc/common';
import { vlLegacyStyles, vlResetStyles } from '@domg-wc/styles';
import { html } from 'lit';
import { buildProperties } from './vl-properties.builder';
import { labelWidthPercentage, propertiesStyles, sizeQueryStyles } from './vl-properties.css';
import { propertiesDefaults } from './vl-properties.defaults';
let VlPropertiesComponent = class VlPropertiesComponent extends BaseLitElement {
    constructor() {
        super(...arguments);
        this.attributeProps = propertiesDefaults.props;
        this.aggregatedProps = propertiesDefaults.props;
        this.mutationObserverList = [];
        this.labelWidth = propertiesDefaults.labelWidth;
        this.labelWidthSheet = new CSSStyleSheet();
    }
    static get styles() {
        return [vlResetStyles, vlLegacyStyles, propertiesStyles];
    }
    static get properties() {
        return {
            props: { type: Array },
            aggregatedProps: { type: Array, state: true },
            labelWidth: { type: Number, attribute: 'label-width' },
        };
    }
    set props(props) {
        this.attributeProps = props;
        this.buildInternalProperties();
    }
    get props() {
        return this.attributeProps;
    }
    connectedCallback() {
        super.connectedCallback();
        this.buildInternalProperties();
        this.observeLightPropertiesChange();
    }
    firstUpdated(_changedProperties) {
        super.firstUpdated(_changedProperties);
        if (this.shadowRoot) {
            // sizeQueryStyles (CSS voor media & container queries) moeten verwerkt worden na de CSS van
            // labelWidthSheet changes - anders wordt die CSS overschreven
            this.shadowRoot.adoptedStyleSheets = [
                ...this.shadowRoot.adoptedStyleSheets,
                this.labelWidthSheet,
                sizeQueryStyles.styleSheet,
            ];
        }
    }
    updated(changedProperties) {
        super.updated(changedProperties);
        if (changedProperties.has('labelWidth') && this.labelWidth) {
            this.labelWidthSheet.replace(labelWidthPercentage(this.labelWidth).toString());
        }
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        this.disconnectMutationObservers();
    }
    render() {
        return html ` <dl>${this.aggregatedProps.map((column) => this.renderColumn(column))}</dl> `;
    }
    renderColumn(column) {
        return column.class
            ? html ` <div class="${column.class}">${column.items.map((item) => this.renderItem(item))}</div>`
            : html ` ${column.items.map((item) => this.renderItem(item))}`;
    }
    renderItem(item) {
        return html ` <div class="item">${this.renderLabels(item.labels)} ${this.renderData(item.data)}</div>`;
    }
    renderLabels(labels) {
        return labels.map((label) => html ` <dt>${label}</dt>`);
    }
    renderData(itemData) {
        return itemData.map((data) => html ` <dd>${data}</dd>`);
    }
    buildInternalProperties() {
        this.aggregatedProps = [...this.attributeProps, ...buildProperties([...this.children])];
    }
    disconnectMutationObservers() {
        this.mutationObserverList.forEach((mutationObserver) => mutationObserver.disconnect());
        this.mutationObserverList = [];
    }
    observeLightPropertiesChange() {
        // verwijder alle bestaande mutation observers
        this.disconnectMutationObservers();
        // altijd de vl-properties zelf observeren
        this.mutationObserverList = [
            ...this.mutationObserverList,
            onChildListChange(this, (change) => {
                this.buildInternalProperties();
                // er kan een extra div toegevoegd zijn, deze heeft dan ook een mutation observer nodig
                this.observeLightPropertiesChange();
            }),
        ];
        // als de directe kinderen div's zijn deze ook observeren
        if (this.children.length > 0 && this.children[0].localName === 'div') {
            [...this.children].forEach((element) => (this.mutationObserverList = [
                ...this.mutationObserverList,
                onChildListChange(element, (change) => {
                    this.buildInternalProperties();
                }),
            ]));
        }
    }
};
VlPropertiesComponent = __decorate([
    webComponent('vl-properties')
], VlPropertiesComponent);
export { VlPropertiesComponent };
