// @domg-wc/map/components/loading-indicator/vl-map-loading-indicator.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { BaseLitElement } from '@domg-wc/common';
import { css } from 'lit';
import { customElement } from 'lit/decorators.js';
import { unByKey } from 'ol/Observable';
let VlMapLoadingIndicator = class VlMapLoadingIndicator extends BaseLitElement {
    connectedCallback() {
        super.connectedCallback();
        this.eventKeyLoadStart = this.map.on('loadstart', () => {
            this.toggleLoadingClass(true);
        });
        this.eventKeyLoadEnd = this.map.on('loadend', () => {
            this.toggleLoadingClass(false);
        });
    }
    disconnectedCallback() {
        super.disconnectedCallback();
        unByKey(this.eventKeyLoadStart);
        unByKey(this.eventKeyLoadEnd);
    }
    toggleLoadingClass(loading) {
        if (loading) {
            this.classList.add('loading');
        }
        else {
            this.classList.remove('loading');
        }
    }
    get map() {
        return this.closest('vl-map');
    }
};
VlMapLoadingIndicator.styles = css `
        @keyframes progress {
            0% {
                left: 0;
                width: 0;
            }
            50% {
                width: 100%;
            }
            100% {
                right: 0;
                width: 0;
            }
        }

        @keyframes reducedMotion {
            0% {
                opacity: 0.25;
            }
            50% {
                opacity: 1;
            }
            100% {
                opacity: 0.25;
            }
        }

        :host(.loading) {
            position: absolute;
            height: 3px;
            background: rgba(0, 85, 204, 0.6); /*komt overeen met blauw van de vl-button.*/
            animation: progress 2s infinite ease-in-out;
            animation-delay: 250ms;
            z-index: 2;
        }

        @media (prefers-reduced-motion: reduce) {
            :host(.loading) {
                width: 100%;
                animation: reducedMotion 2s infinite ease-in-out;
            }
        }
    `;
VlMapLoadingIndicator = __decorate([
    customElement('vl-map-loading-indicator')
], VlMapLoadingIndicator);
export { VlMapLoadingIndicator };
